<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Exploratory data analysis – katlas-raw</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.min.js" integrity="sha384-ZvpUoO/+PpLXR1lu4jmpXWu80pZlYUAfxl5NsBMWOEPSjUn/6Z/hRTt8+pR6L4N2" crossorigin="anonymous"></script><script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-000f0825b9c55ed21d14970d810c14a9.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="https://cdn.jsdelivr.net/npm/requirejs@2.3.6/require.min.js" integrity="sha384-c9c+LnTbwQ3aujuU7ULEPVvgLs+Fn6fJUvIGTsuu1ZcCf11fiEubah0ttpca4ntM sha384-6V1/AdqZRWk1KAlWbKBlGhN7VG4iE/yAZcO6NZPMF8od0vukrvr0tg4qY6NSrItx" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


<link rel="stylesheet" href="styles.css">
<meta property="og:title" content="Exploratory data analysis – katlas-raw">
<meta property="og:description" content="Raw code to generate figures in the paper of KATLAS">
<meta property="og:image" content="https://sky1ove.github.io/katlas-raw/02_EDA_fig1_files/figure-html/cell-8-output-1.png">
<meta property="og:site_name" content="katlas-raw">
<meta property="og:image:height" content="452">
<meta property="og:image:width" content="643">
<meta name="twitter:title" content="Exploratory data analysis – katlas-raw">
<meta name="twitter:description" content="Raw code to generate figures in the paper of KATLAS">
<meta name="twitter:image" content="https://sky1ove.github.io/katlas-raw/02_EDA_fig1_files/figure-html/cell-8-output-1.png">
<meta name="twitter:image-height" content="452">
<meta name="twitter:image-width" content="643">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="./index.html" class="navbar-brand navbar-brand-logo">
    </a>
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">katlas-raw</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./eda_fig1.html">Exploratory data analysis</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      </a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">katlas-raw</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./preprocess_dataset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preprocess kinase-substrate datasets</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./eda_fig1.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Exploratory data analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./site_promiscuity_fig1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Site promiscuity</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssm_hierarchical.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Hierarchical clustering of PSSM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05_PSSM_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">05_PSSM_example.html</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./modi_group.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Modi group</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scale_pspa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PSPA scaled data</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./background.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Background frequencies</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./specificity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Specificity comparison</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./sty_composition.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">STY composition</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./plot_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot heatmap and logo of CDDM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./plot_pspa_figure.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PSPA figures</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./13_CDDM_umap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CDDM umap</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./get_aa_relationship.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get AA relationship</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pspa_cddm_heatmap_compare.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Heatmap/ST/JSD/cosine comparison</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_no_psp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_psp_02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring_compare_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scoring evaluation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring_compare_psp_02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scoring evaluation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Reactome pathway for each PSSM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_plot_pspa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot reactome pathways: PSPA</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_plot_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot reactome pathways: CDDM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./motif_clustering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Kmeans motifs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./motif_plot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Motif plot</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./22_motif_web_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">22_motif_web_data.html</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./23_motif_kinase_similarity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cosine similarity with PSPA</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">kinase_domain</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_01_get_all_kd.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get all kinase domain sequences</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_02_pseudo_alignment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Alignment</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_03_feature_extract.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Extract KD features</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_04_prepare_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare data for training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_05_pspa_training_model_comparison_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Model comparison on T5 feature</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06a_pspa_training_feature_comparison_cnn_group.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on CNN model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06b_pspa_training_feature_comparison_cnn_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on CNN model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06c_pspa_training_feature_comparison_mlp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on MLP model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_07a_cddm_training_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_07b_pspa_training_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_08a_similarity_relationship.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KD similarity &amp; PSSM similarity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_08b_similarity_relationship_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KD similarity &amp; PSSM similarity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_09a_predict_pspa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Predict</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_09b_predict_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Predict</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_10_plot_umap_known_unk.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot UMAP of known &amp; unknown KD</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_11_plot_pspa_kd_t5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">kd_11_plot_pspa_kd_t5.html</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_12_compare_pred_psp_known.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Comopare the two</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup">Setup</a></li>
  <li><a href="#overlap-between-sources" id="toc-overlap-between-sources" class="nav-link" data-scroll-target="#overlap-between-sources">Overlap between sources</a></li>
  <li><a href="#overlap-of-phosphosites-with-human-phosphoproteome" id="toc-overlap-of-phosphosites-with-human-phosphoproteome" class="nav-link" data-scroll-target="#overlap-of-phosphosites-with-human-phosphoproteome">Overlap of phosphosites with human phosphoproteome</a></li>
  <li><a href="#stacked-plot-of-sty-compositions" id="toc-stacked-plot-of-sty-compositions" class="nav-link" data-scroll-target="#stacked-plot-of-sty-compositions">Stacked plot of S/T/Y compositions</a></li>
  <li><a href="#recovery-rate" id="toc-recovery-rate" class="nav-link" data-scroll-target="#recovery-rate">Recovery rate</a></li>
  <li><a href="#available-on-data" id="toc-available-on-data" class="nav-link" data-scroll-target="#available-on-data">Available on Data</a></li>
  <li><a href="#sites-distribution" id="toc-sites-distribution" class="nav-link" data-scroll-target="#sites-distribution">Sites distribution</a></li>
  <li><a href="#most-frequent-kinases" id="toc-most-frequent-kinases" class="nav-link" data-scroll-target="#most-frequent-kinases">Most frequent kinases</a>
  <ul class="collapse">
  <li><a href="#pie-chart" id="toc-pie-chart" class="nav-link" data-scroll-target="#pie-chart">Pie chart</a></li>
  <li><a href="#kinome-tree" id="toc-kinome-tree" class="nav-link" data-scroll-target="#kinome-tree">Kinome tree</a></li>
  </ul></li>
  <li><a href="#old" id="toc-old" class="nav-link" data-scroll-target="#old">Old</a></li>
  <li><a href="#co-occured-kinase" id="toc-co-occured-kinase" class="nav-link" data-scroll-target="#co-occured-kinase">Co-occured kinase</a></li>
  <li><a href="#old-1" id="toc-old-1" class="nav-link" data-scroll-target="#old-1">Old</a></li>
  <li><a href="#frequency-of-kinase" id="toc-frequency-of-kinase" class="nav-link" data-scroll-target="#frequency-of-kinase">Frequency of kinase</a></li>
  <li><a href="#kinase-count" id="toc-kinase-count" class="nav-link" data-scroll-target="#kinase-count">Kinase count</a></li>
  <li><a href="#source-confidence" id="toc-source-confidence" class="nav-link" data-scroll-target="#source-confidence">Source confidence</a></li>
  <li><a href="#self-phosphorylate" id="toc-self-phosphorylate" class="nav-link" data-scroll-target="#self-phosphorylate">Self-phosphorylate</a>
  <ul class="collapse">
  <li><a href="#nearby-co-occured-kinase" id="toc-nearby-co-occured-kinase" class="nav-link" data-scroll-target="#nearby-co-occured-kinase">Nearby co-occured kinase</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/sky1ove/katlas-raw/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div><div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="eda_fig1.html.md"><i class="bi bi-file-code"></i>CommonMark</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Exploratory data analysis</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<!-- WARNING: THIS FILE WAS AUTOGENERATED! DO NOT EDIT! -->
<section id="setup" class="level2">
<h2 class="anchored" data-anchor-id="setup">Setup</h2>
<div id="1d0bd96a-a8c7-4c6d-ba06-47440702743d" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib.colors <span class="im">import</span> LogNorm</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.core <span class="im">import</span> <span class="op">*</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="overlap-between-sources" class="level2">
<h2 class="anchored" data-anchor-id="overlap-between-sources">Overlap between sources</h2>
<div id="248e785b-9830-4f92-8088-3f8997057d02" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># direct concatenate of all source sequence</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>df<span class="op">=</span>pd.read_parquet(<span class="st">'raw/combine_source.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="20bb7b34-b027-43c9-a446-809bae3dbb1a" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># unique substrate site with source split by | </span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co"># df_group=pd.read_parquet('raw/combine_source_grouped.parquet')</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>df_group<span class="op">=</span>Data.get_ks_dataset()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="85c4de8e-b13b-4bff-82eb-4136b8ca4e13" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_overlap(df_concat,source_col<span class="op">=</span><span class="st">'source'</span>,id_col<span class="op">=</span><span class="st">'kin_sub_site'</span>,figsize<span class="op">=</span>(<span class="dv">7</span>,<span class="dv">5</span>),title<span class="op">=</span><span class="st">"Overlap of Unique KS Pairs Between Sources"</span>):</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>    source_ids <span class="op">=</span> df_concat.groupby(source_col)[id_col].<span class="bu">apply</span>(<span class="bu">set</span>).to_dict()</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>    sources <span class="op">=</span> <span class="bu">list</span>(source_ids.keys())</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>    overlap_matrix <span class="op">=</span> pd.DataFrame(index<span class="op">=</span>sources, columns<span class="op">=</span>sources, dtype<span class="op">=</span><span class="bu">int</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> src1 <span class="kw">in</span> sources:</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> src2 <span class="kw">in</span> sources:</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> src1 <span class="op">==</span> src2:</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>                overlap_matrix.loc[src1, src2] <span class="op">=</span> <span class="bu">len</span>(source_ids[src1])</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>            <span class="cf">else</span>:</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>                overlap_matrix.loc[src1, src2] <span class="op">=</span> <span class="bu">len</span>(source_ids[src1].intersection(source_ids[src2]))</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>    mask <span class="op">=</span> np.triu(np.ones_like(overlap_matrix, dtype<span class="op">=</span><span class="bu">bool</span>), k<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    overlap_matrix <span class="op">=</span> overlap_matrix.astype(<span class="bu">int</span>)</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>    vmin <span class="op">=</span> np.<span class="bu">min</span>(overlap_matrix[overlap_matrix <span class="op">&gt;</span> <span class="dv">0</span>])  <span class="co"># Smallest nonzero value</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>    vmax <span class="op">=</span> np.<span class="bu">max</span>(overlap_matrix)</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>figsize)</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>    sns.heatmap(overlap_matrix, </span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>                annot<span class="op">=</span><span class="va">True</span>, </span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>                mask<span class="op">=</span>mask, </span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>                fmt<span class="op">=</span><span class="st">","</span>, </span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>                <span class="co"># vmin=1, vmax=1,</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>                norm<span class="op">=</span>LogNorm(vmin<span class="op">=</span>vmin, vmax<span class="op">=</span>vmax),</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>                cmap<span class="op">=</span><span class="st">"Blues"</span>, </span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>                cbar<span class="op">=</span><span class="va">False</span>, </span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>                linewidths<span class="op">=</span><span class="dv">1</span>, </span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>                linecolor<span class="op">=</span><span class="st">'white'</span>)</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>    plt.title(title)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="8096399a-378e-49b8-aec8-fe5be7aa0cc2" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.plot <span class="im">import</span> <span class="op">*</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="2433ac3a-8f17-464e-9c8b-4e2604dbd54c" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>set_sns(<span class="dv">100</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="066447fd-5285-48b3-baf7-dc036e505e9d" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>plot_overlap(df)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>save_pdf(<span class="st">'../../fig/EDA_overlap.pdf'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="02_EDA_fig1_files/figure-html/cell-8-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div id="04a06392-f1cd-4a07-99ac-094679b1e255" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>df.site.<span class="bu">str</span>[<span class="dv">0</span>].value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>site
Y    92088
S    84341
T    35243
Name: count, dtype: int64</code></pre>
</div>
</div>
<div id="df1d4412-49db-4b89-a8aa-4469dce91825" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>df_st <span class="op">=</span> df[df.site.<span class="bu">str</span>[<span class="dv">0</span>]<span class="op">!=</span><span class="st">'Y'</span>]</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>df_y <span class="op">=</span> df[df.site.<span class="bu">str</span>[<span class="dv">0</span>]<span class="op">==</span><span class="st">'Y'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="b29f5a4f-0953-4355-8a09-ebec06dcc23f" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>plot_overlap(df_st,title<span class="op">=</span><span class="st">"Overlap of KS Pairs in S/T sites Between Sources"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>save_pdf(<span class="st">'../../fig/EDA_overlap_ST.pdf'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="02_EDA_fig1_files/figure-html/cell-11-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div id="b176df25-662c-464b-96b0-f2c987535ad7" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>plot_overlap(df_y,title<span class="op">=</span><span class="st">"Overlap of KS Pairs in Y sites Between Sources"</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>save_pdf(<span class="st">'../../fig/EDA_overlap_Y.pdf'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="02_EDA_fig1_files/figure-html/cell-12-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="overlap-of-phosphosites-with-human-phosphoproteome" class="level2">
<h2 class="anchored" data-anchor-id="overlap-of-phosphosites-with-human-phosphoproteome">Overlap of phosphosites with human phosphoproteome</h2>
<div id="259e424d-dbd4-4a0d-9e25-3235b0154987" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.data <span class="im">import</span> <span class="op">*</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="6408412e-6cae-446a-80ed-f4f2b77c5107" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>df<span class="op">=</span>pd.read_parquet(<span class="st">'raw/combine_source.parquet'</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>data <span class="op">=</span> pd.read_parquet(<span class="st">'raw/human_phosphoproteome.parquet'</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co"># data=Data.get_human_site()</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="825f409a-29d8-4e7b-b01b-e5473111835a" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>data <span class="op">=</span> data.rename(columns<span class="op">=</span>{<span class="st">'uniprot'</span>:<span class="st">'substrate_uniprot'</span>})</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="63acaec1-a932-4aa6-9cae-e78a1f7acd34" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> get_key(df):</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>    df <span class="op">=</span> df.copy()</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'original shape:'</span>,df.shape)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>    df[<span class="st">'sub_site'</span>]<span class="op">=</span>df[<span class="st">'substrate_uniprot'</span>]<span class="op">+</span><span class="st">'_'</span><span class="op">+</span>df[<span class="st">'site'</span>]</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    df <span class="op">=</span> df.drop_duplicates(subset<span class="op">=</span><span class="st">'sub_site'</span>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">'after removing duplicates'</span>,df.shape)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> df</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="97d76b7a-5423-4cef-bc45-398467bbed8e" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>data_site <span class="op">=</span> get_key(data)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>df_site<span class="op">=</span>get_key(df)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>original shape: (119955, 11)
after removing duplicates (119955, 11)
original shape: (211672, 7)
after removing duplicates (29811, 8)</code></pre>
</div>
</div>
<div id="dd48fa0e-60bd-449e-bf38-45512d59c6dd" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>data_site <span class="op">=</span> data_site[[<span class="st">'sub_site'</span>]].copy()</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>df_site <span class="op">=</span> df_site[[<span class="st">'sub_site'</span>]].copy()</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>data_site[<span class="st">'source'</span>]<span class="op">=</span><span class="st">'Human Phosphoprotome'</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>df_site[<span class="st">'source'</span>]<span class="op">=</span><span class="st">'KS Datasets'</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>df_concat <span class="op">=</span> pd.concat([data_site,df_site],ignore_index<span class="op">=</span><span class="va">True</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="5a13b209-5056-40f1-87d9-a35479fb7497" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>df_concat_group <span class="op">=</span> df_concat.groupby(<span class="st">'sub_site'</span>)[<span class="st">'source'</span>].agg(<span class="kw">lambda</span> x: <span class="st">'|'</span>.join(x)).reset_index()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="0f564270-1bee-4e07-a5b7-721a978f24f2" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>df_concat_group.source.value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>source
Human Phosphoprotome                102032
Human Phosphoprotome|KS Datasets     17923
KS Datasets                          11888
Name: count, dtype: int64</code></pre>
</div>
</div>
<div id="06afad63-578d-4f11-a7bc-4ad0550258db" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_overlap(df_concat,source_col='source',id_col='sub_site',figsize=(4,2),</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="co">#              title="Overlap of Unique Substrate Sites Between Datasets")</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Venn diagram:</p>
<div id="8f8235df-0521-43ff-baa1-3c47dad09858" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># pip install matplotlib_venn</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="686ef38e-ad88-4959-af62-49ba5af1aa88" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib_venn <span class="im">import</span> venn2</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>set_human <span class="op">=</span> <span class="bu">set</span>(data_site[<span class="st">'sub_site'</span>])</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>set_ks <span class="op">=</span> <span class="bu">set</span>(df_site[<span class="st">'sub_site'</span>])</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>plt.figure(figsize<span class="op">=</span>(<span class="dv">5</span>,<span class="dv">5</span>))</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>venn <span class="op">=</span> venn2([set_human, set_ks], </span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>             <span class="co"># set_labels=('Human Phosphoproteome', 'KS Datasets')</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>             set_labels<span class="op">=</span>(<span class="st">''</span>, <span class="st">''</span>)</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> label <span class="kw">in</span> venn.subset_labels:</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> label:  <span class="co"># Check if the label exists (not empty subset)</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>        label.set_text(<span class="ss">f"</span><span class="sc">{</span><span class="bu">int</span>(label.get_text())<span class="sc">:,}</span><span class="ss">"</span>)  <span class="co"># Add comma formatting</span></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>plt.gca().set_aspect(<span class="fl">0.8</span>)</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>plt.title(<span class="st">"Overlap of Unique Substrate Sites Between Datasets"</span>)</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>save_pdf(<span class="st">'fig/EDA_venn_ks_human_overlap.pdf'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="02_EDA_fig1_files/figure-html/cell-23-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="stacked-plot-of-sty-compositions" class="level2">
<h2 class="anchored" data-anchor-id="stacked-plot-of-sty-compositions">Stacked plot of S/T/Y compositions</h2>
<div id="34e3492e-cc97-486f-9657-2cb515dfc5ad" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>df<span class="op">=</span>pd.read_parquet(<span class="st">'raw/combine_source.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="b160710a-d24d-48a2-a511-dd964d14a080" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>df[<span class="st">'acceptor'</span>]<span class="op">=</span>df[<span class="st">'site'</span>].<span class="bu">str</span>[<span class="dv">0</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="df712ff1-b053-4b74-b4c4-a49b349a5761" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb28"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>df_sugi <span class="op">=</span> df[df.source<span class="op">==</span><span class="st">'Sugiyama'</span>].reset_index(drop<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>df_rest <span class="op">=</span> df[df.source<span class="op">!=</span><span class="st">'Sugiyama'</span>].reset_index(drop<span class="op">=</span><span class="va">True</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="1688abf9-73e2-4260-bd85-e570c833d5c3" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>df_sugi.acceptor.value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>acceptor
Y    84457
S    57728
T    26157
Name: count, dtype: int64</code></pre>
</div>
</div>
<div id="b8141cb7-6c45-4396-acf8-317d2ffbe0ad" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>df_rest.acceptor.value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>acceptor
S    26613
T     9086
Y     7631
Name: count, dtype: int64</code></pre>
</div>
</div>
<div id="18e35c08-a725-4df2-9ad4-3e3757118e9b" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.ticker <span class="im">as</span> mticker</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib <span class="im">import</span> pyplot <span class="im">as</span> plt</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="80be7cbf-ecad-43c4-b5ba-1d2e0def0dfe" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.plot <span class="im">import</span> <span class="op">*</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="62992f4c-5d05-4075-b33b-6fe327b2f04c" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb35"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_stacked(df,figsize<span class="op">=</span>(<span class="dv">6</span>,<span class="dv">4</span>)):</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>figsize)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>    sns.histplot(data<span class="op">=</span>df, </span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>                 x<span class="op">=</span><span class="st">'source'</span>, </span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>                 hue<span class="op">=</span><span class="st">'acceptor'</span>, </span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>                 multiple<span class="op">=</span><span class="st">'stack'</span>, </span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>                 discrete<span class="op">=</span><span class="va">True</span>, </span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>                 shrink<span class="op">=</span><span class="fl">0.8</span>, </span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>                 alpha<span class="op">=</span><span class="dv">1</span>,</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>                 palette<span class="op">=</span>get_color_dict([<span class="st">'S'</span>,<span class="st">'T'</span>,<span class="st">'Y'</span>],<span class="st">'tab20'</span>),hue_order<span class="op">=</span>[<span class="st">'S'</span>,<span class="st">'T'</span>,<span class="st">'Y'</span>])</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">''</span>)</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">'Number of KS Pairs'</span>)</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="st">'Total Number of Unique KS Pairs per Source'</span>)</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>    plt.gca().yaxis.set_major_formatter(mticker.FuncFormatter(<span class="kw">lambda</span> x, _: <span class="ss">f'</span><span class="sc">{</span><span class="bu">int</span>(x)<span class="sc">:,}</span><span class="ss">'</span>)) <span class="co"># show number with comma</span></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>    plt.xticks(rotation<span class="op">=</span><span class="dv">45</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="d599f897-3440-4587-93dc-427fc2d42a28" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>set_sns(<span class="dv">100</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="576c9d0d-2e33-40c0-9466-c9be577efdd0" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb37"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>plot_stacked(df_rest)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>save_pdf(<span class="st">'../../fig/EDA_stacked_KS_per_source.pdf'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="02_EDA_fig1_files/figure-html/cell-33-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div id="5f873e4d-35fa-4cf4-915c-e49490319d42" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb38"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_stacked2(df,figsize<span class="op">=</span>(<span class="dv">6</span>,<span class="dv">4</span>)):</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>figsize)</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>    sns.histplot(data<span class="op">=</span>df, </span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>                 x<span class="op">=</span><span class="st">'source'</span>, </span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>                 hue<span class="op">=</span><span class="st">'acceptor'</span>, </span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>                 multiple<span class="op">=</span><span class="st">'stack'</span>, </span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>                 discrete<span class="op">=</span><span class="va">True</span>, </span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>                 shrink<span class="op">=</span><span class="fl">0.8</span>, </span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>                 alpha<span class="op">=</span><span class="dv">1</span>,</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>                 palette<span class="op">=</span>get_color_dict([<span class="st">'S'</span>,<span class="st">'T'</span>,<span class="st">'Y'</span>],<span class="st">'tab20'</span>),hue_order<span class="op">=</span>[<span class="st">'S'</span>,<span class="st">'T'</span>,<span class="st">'Y'</span>])</span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">''</span>)</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">''</span>)</span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>    plt.xticks(rotation<span class="op">=</span><span class="dv">45</span>)</span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a>    plt.legend([],frameon<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a>    plt.gca().yaxis.set_major_formatter(mticker.FuncFormatter(<span class="kw">lambda</span> x, _: <span class="ss">f'</span><span class="sc">{</span><span class="bu">int</span>(x)<span class="sc">:,}</span><span class="ss">'</span>))</span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a>    plt.tick_params(axis<span class="op">=</span><span class="st">'y'</span>, labelright<span class="op">=</span><span class="va">True</span>, labelleft<span class="op">=</span><span class="va">False</span>, right<span class="op">=</span><span class="va">True</span>, left<span class="op">=</span><span class="va">False</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Uncomment below to plot the sugiyama site number:</p>
<div id="8f064c96-9859-443c-b7b2-837347343deb" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb39"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>plot_stacked2(df_sugi,(<span class="fl">0.7</span>,<span class="dv">4</span>))</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>save_pdf(<span class="st">'../../fig/EDA_stacked_KS_per_source_part2.pdf'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="02_EDA_fig1_files/figure-html/cell-35-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="recovery-rate" class="level2">
<h2 class="anchored" data-anchor-id="recovery-rate">Recovery rate</h2>
<p>Check <code>raw/recovery_rate.xlsx</code> for more info.</p>
<p>The data is collected by: - Filter only human first, get dataframe shape - After kinase &amp; substrate mapping, get the shape - After sites validation, get the shape</p>
</section>
<section id="available-on-data" class="level2">
<h2 class="anchored" data-anchor-id="available-on-data">Available on Data</h2>
<div id="7f837213-3ceb-434f-b4cc-d1d18bfefb5b" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb40"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.data <span class="im">import</span> <span class="op">*</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="022e4421-540e-4ef0-af3f-b4b59d2cf6f4" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb41"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> Data.get_ks_dataset()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="90c164da-b384-4163-9d4e-5c239d32b44c" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb42"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>df.head()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">kin_sub_site</th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">substrate_uniprot</th>
<th data-quarto-table-cell-role="th">site</th>
<th data-quarto-table-cell-role="th">source</th>
<th data-quarto-table-cell-role="th">substrate_genes</th>
<th data-quarto-table-cell-role="th">substrate_phosphoseq</th>
<th data-quarto-table-cell-role="th">position</th>
<th data-quarto-table-cell-role="th">site_seq</th>
<th data-quarto-table-cell-role="th">sub_site</th>
<th data-quarto-table-cell-role="th">substrate_sequence</th>
<th data-quarto-table-cell-role="th">kinase_on_tree</th>
<th data-quarto-table-cell-role="th">kinase_genes</th>
<th data-quarto-table-cell-role="th">kinase_group</th>
<th data-quarto-table-cell-role="th">kinase_family</th>
<th data-quarto-table-cell-role="th">kinase_pspa_big</th>
<th data-quarto-table-cell-role="th">kinase_pspa_small</th>
<th data-quarto-table-cell-role="th">kinase_coral_ID</th>
<th data-quarto-table-cell-role="th">num_kin</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">0</th>
<td>O00141_A4FU28_S140</td>
<td>O00141</td>
<td>A4FU28</td>
<td>S140</td>
<td>Sugiyama</td>
<td>CTAGE9</td>
<td>MEEPGATPQPYLGLVLEELGRVVAALPESMRPDENPYGFPSELVVC...</td>
<td>140</td>
<td>AAAEEARSLEATCEKLSRsNsELEDEILCLEKDLKEEKSKH</td>
<td>A4FU28_S140</td>
<td>MEEPGATPQPYLGLVLEELGRVVAALPESMRPDENPYGFPSELVVC...</td>
<td>1</td>
<td>SGK1 SGK</td>
<td>AGC</td>
<td>SGK</td>
<td>Basophilic</td>
<td>Akt/rock</td>
<td>SGK1</td>
<td>22</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1</th>
<td>O00141_O00141_S252</td>
<td>O00141</td>
<td>O00141</td>
<td>S252</td>
<td>Sugiyama</td>
<td>SGK1 SGK</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
<td>252</td>
<td>SQGHIVLTDFGLCKENIEHNsTtstFCGtPEyLAPEVLHKQ</td>
<td>O00141_S252</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
<td>1</td>
<td>SGK1 SGK</td>
<td>AGC</td>
<td>SGK</td>
<td>Basophilic</td>
<td>Akt/rock</td>
<td>SGK1</td>
<td>1</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">2</th>
<td>O00141_O00141_S255</td>
<td>O00141</td>
<td>O00141</td>
<td>S255</td>
<td>Sugiyama</td>
<td>SGK1 SGK</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
<td>255</td>
<td>HIVLTDFGLCKENIEHNsTtstFCGtPEyLAPEVLHKQPYD</td>
<td>O00141_S255</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
<td>1</td>
<td>SGK1 SGK</td>
<td>AGC</td>
<td>SGK</td>
<td>Basophilic</td>
<td>Akt/rock</td>
<td>SGK1</td>
<td>1</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">3</th>
<td>O00141_O00141_S397</td>
<td>O00141</td>
<td>O00141</td>
<td>S397</td>
<td>Sugiyama</td>
<td>SGK1 SGK</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
<td>397</td>
<td>sGPNDLRHFDPEFTEEPVPNsIGKsPDsVLVTAsVKEAAEA</td>
<td>O00141_S397</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
<td>1</td>
<td>SGK1 SGK</td>
<td>AGC</td>
<td>SGK</td>
<td>Basophilic</td>
<td>Akt/rock</td>
<td>SGK1</td>
<td>1</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">4</th>
<td>O00141_O00141_S404</td>
<td>O00141</td>
<td>O00141</td>
<td>S404</td>
<td>Sugiyama</td>
<td>SGK1 SGK</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
<td>404</td>
<td>HFDPEFTEEPVPNsIGKsPDsVLVTAsVKEAAEAFLGFsYA</td>
<td>O00141_S404</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
<td>1</td>
<td>SGK1 SGK</td>
<td>AGC</td>
<td>SGK</td>
<td>Basophilic</td>
<td>Akt/rock</td>
<td>SGK1</td>
<td>1</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div id="1334fdcf-083e-4613-a430-62b36f35507b" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb43"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>df.shape</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>(187066, 19)</code></pre>
</div>
</div>
</section>
<section id="sites-distribution" class="level2">
<h2 class="anchored" data-anchor-id="sites-distribution">Sites distribution</h2>
<div id="dd3a8689-fa19-40a1-9a23-2591bfc5a23b" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb45"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>df[<span class="st">'acceptor'</span>] <span class="op">=</span> df.site.<span class="bu">str</span>[<span class="dv">0</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="72fa37e1-cde9-4ebf-973e-639c0818aab7" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb46"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_pie(value_counts, <span class="co"># value counts</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>             hue_order<span class="op">=</span><span class="va">None</span>, <span class="co"># list of strings</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>             labeldistance<span class="op">=</span><span class="fl">0.8</span>,</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>             fontsize<span class="op">=</span><span class="dv">12</span>,</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>             fontcolor<span class="op">=</span><span class="st">'black'</span>,</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>             palette<span class="op">=</span><span class="st">'tab20'</span> </span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>            ):</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> hue_order <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>: value_counts <span class="op">=</span> value_counts.reindex(hue_order)</span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>    colors <span class="op">=</span> sns.color_palette(palette, n_colors<span class="op">=</span><span class="bu">len</span>(value_counts))</span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>    value_counts.plot.pie(</span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>        autopct<span class="op">=</span><span class="st">'</span><span class="sc">%1.1f%%</span><span class="st">'</span>,    <span class="co"># Show percentage inside slices</span></span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>        labeldistance<span class="op">=</span>labeldistance,    <span class="co"># Move labels closer to center</span></span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a>        textprops<span class="op">=</span>{<span class="st">'fontsize'</span>: fontsize, <span class="st">'color'</span>: fontcolor} ,</span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>        colors<span class="op">=</span>colors,</span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">''</span>)</span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'n=</span><span class="sc">{</span>value_counts<span class="sc">.</span><span class="bu">sum</span>()<span class="sc">:,}</span><span class="ss">'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="6e1a5748-17c0-4fa4-a3e7-776f5faf846d" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb47"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_pie(value_counts, </span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>             hue_order<span class="op">=</span><span class="va">None</span>,</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>             labeldistance<span class="op">=</span><span class="fl">0.8</span>,</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>             fontsize<span class="op">=</span><span class="dv">9</span>,</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>             fontcolor<span class="op">=</span><span class="st">'black'</span>,</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>             palette<span class="op">=</span><span class="st">'tab20'</span>):</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Sort value_counts by size for better pie layout</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>    value_counts_sorted <span class="op">=</span> value_counts.sort_values(ascending<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Map colors based on hue_order (if provided) or sorted labels</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>    labels <span class="op">=</span> value_counts_sorted.index.tolist()</span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> hue_order <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Assign fixed colors based on hue_order</span></span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>        palette_colors <span class="op">=</span> sns.color_palette(palette, n_colors<span class="op">=</span><span class="bu">len</span>(hue_order))</span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a>        color_map <span class="op">=</span> <span class="bu">dict</span>(<span class="bu">zip</span>(hue_order, palette_colors))</span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a>        colors <span class="op">=</span> [color_map.get(label, <span class="st">'grey'</span>) <span class="cf">for</span> label <span class="kw">in</span> labels]</span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a>        colors <span class="op">=</span> sns.color_palette(palette, n_colors<span class="op">=</span><span class="bu">len</span>(labels))</span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Plot pie</span></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a>    value_counts_sorted.plot.pie(</span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a>        autopct<span class="op">=</span><span class="st">'</span><span class="sc">%1.1f%%</span><span class="st">'</span>,</span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a>        labeldistance<span class="op">=</span>labeldistance,</span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a>        textprops<span class="op">=</span>{<span class="st">'fontsize'</span>: fontsize, <span class="st">'color'</span>: fontcolor},</span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a>        colors<span class="op">=</span>colors</span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">''</span>)</span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'n=</span><span class="sc">{</span>value_counts<span class="sc">.</span><span class="bu">sum</span>()<span class="sc">:,}</span><span class="ss">'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="f35187d8-e21f-41b0-8b05-88c4fa7934cd" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb48"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>cnt <span class="op">=</span> df[<span class="st">'acceptor'</span>].value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="37ed20b6-39ca-4ba3-a23d-02f1ab915085" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb49"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>plot_pie(cnt,[<span class="st">'S'</span>,<span class="st">'T'</span>,<span class="st">'Y'</span>])</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>save_pdf(<span class="st">'../../fig/EDA_pie_STY.pdf'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="02_EDA_fig1_files/figure-html/cell-44-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="most-frequent-kinases" class="level2">
<h2 class="anchored" data-anchor-id="most-frequent-kinases">Most frequent kinases</h2>
<section id="pie-chart" class="level3">
<h3 class="anchored" data-anchor-id="pie-chart">Pie chart</h3>
<div id="46de7ba6-a1fe-4a6d-ab41-0a8b08a916ee" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb50"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>cnt <span class="op">=</span> df.kinase_group.value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="85835002-58a1-4d36-a6e9-ccec00f027b5" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb51"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>info <span class="op">=</span> Data.get_kinase_info()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="94107a1e-f23e-433c-8826-cffd03f63517" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb52"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>info.modi_group.value_counts().index</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>Index(['TK', 'CAMK', 'Other', 'CMGC', 'AGC', 'STE', 'TKL', 'Atypical', 'CK1',
       'NEK', 'RGC'],
      dtype='object', name='modi_group')</code></pre>
</div>
</div>
<div id="28da8e56-977e-44a5-936c-70a39ca22333" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb54"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>group_color<span class="op">=</span>get_color_dict(</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>            [<span class="st">'CMGC'</span>,<span class="st">'AGC'</span>, <span class="co"># blue</span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>             <span class="st">'TK'</span>,<span class="st">'TKL'</span>, <span class="co"># orange</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>             <span class="st">'CAMK'</span>,<span class="st">'STE'</span>, <span class="co"># green</span></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>             <span class="st">'CK1'</span>, <span class="st">'NEK'</span>, <span class="co"># red</span></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>             <span class="st">'Atypical'</span>,<span class="st">'Other'</span>, <span class="co"># purple</span></span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>             <span class="st">'RGC'</span></span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>            ]</span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="beca5dbc-7aee-449b-89df-1171680e5360" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb55"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>plot_pie(cnt,hue_order<span class="op">=</span>group_color.keys(),</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>         fontsize<span class="op">=</span><span class="dv">9</span>,</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>         <span class="co"># labeldistance=1.1</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>save_pdf(<span class="st">'../../fig/EDA_pie_group.pdf'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="02_EDA_fig1_files/figure-html/cell-49-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="kinome-tree" class="level3">
<h3 class="anchored" data-anchor-id="kinome-tree">Kinome tree</h3>
<p>Coral:</p>
<div id="6702cd86-7801-4096-b782-00d0c711b3c6" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb56"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="6c88c275-1b93-4a35-a9d4-0e90a0a5d6d2" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb57"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>df<span class="op">=</span>Data.get_ks_dataset()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="b623a48b-4512-4812-88b9-48caeaf3e4f4" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb58"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>cnt <span class="op">=</span> df.groupby(<span class="st">'kinase_coral_ID'</span>)[<span class="st">'kin_sub_site'</span>].<span class="bu">apply</span>(<span class="kw">lambda</span> x: <span class="bu">len</span>(x.unique())).sort_values(ascending<span class="op">=</span><span class="va">False</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="a3c91edf-2085-4a21-8f7c-b4d7f13cbe69" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb59"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># cnt.apply(np.log10).round(2).to_csv('raw/kinome_tree_cnt.csv')</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Go to <a href="http://phanstiel-lab.med.unc.edu/CORAL/">Coral</a>. Follow the setting below:</p>
<ul>
<li>Node color –&gt; Color Scheme: quantitative</li>
<li>paste csv context (without heading) in “Kinases &amp; Value”</li>
<li>identifier: coralID</li>
<li>min:0 max:3.5</li>
<li>manual 2-color. Low: #E0E0E0, High: #FA6958</li>
</ul>
<p>To disable node label: advanced setting –&gt; labels font size to 0</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="02_EDA_fig1_files/figure-html/c2c1ccb6-21d8-4c0d-807b-a5e9ea6088ea-1-001eb23c-9ef6-4ef8-a42c-a0013062a402.png" class="img-fluid figure-img"></p>
<figcaption>image.png</figcaption>
</figure>
</div>
</section>
</section>
<section id="old" class="level2">
<h2 class="anchored" data-anchor-id="old">Old</h2>
</section>
<section id="co-occured-kinase" class="level2">
<h2 class="anchored" data-anchor-id="co-occured-kinase">Co-occured kinase</h2>
<div id="99d425e8-321f-4ec3-9d13-a561715effa2" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb60"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>df_group[<span class="st">'substrate_uniprot_site'</span>]<span class="op">=</span>df_group[<span class="st">'substrate_uniprot'</span>]<span class="op">+</span><span class="st">'_'</span><span class="op">+</span>df_group[<span class="st">'site'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="2c985ec1-ea4a-4b7e-b5f6-c97d1eb01f79" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb61"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> itertools <span class="im">import</span> combinations</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> collections <span class="im">import</span> Counter</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="2e700802-df9f-4a00-8609-4acc5e22c37c" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb62"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>df_group.groupby(<span class="st">'substrate_uniprot_site'</span>)[<span class="st">'kinase_uniprot'</span>].<span class="bu">apply</span>(<span class="bu">list</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>substrate_uniprot_site
A0A2R8Y4L2_S158                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      [P48730]
A0A2R8Y4L2_S22                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       [O14757, O95819, Q13557]
A0A2R8Y4L2_S6                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        [O15264, P06493, Q9NR20]
A0A2R8Y4L2_S95     [O00238, O14920, O15075, O43318, O75676, O94806, O96017, P05771-2, P05771, P15735, P17252, P23443, P24723, P24941, P36507, P36896, P36897, P41743, P42679, P49137, P49759, P49761, P50750, P51812, P51955, P51957, P52333, P57059, P78362, P80192, Q02156, Q02779, Q04759, Q04771, Q05513, Q06187, Q13131, Q13554, Q13555, Q13557, Q14012, Q14680, Q15139, Q15349, Q16512, Q16513, Q16566, Q16584, Q16644, Q16816, Q86UE8, Q8IU85, Q8IW41, Q8N568, Q8TD19, Q8TDX7, Q96GD4, Q9BYT3, Q9BZL6, Q9NYL2, Q9UKI8, Q9UPE1, Q9UQM7, Q9Y2U5, Q9Y4K4]
A0A2R8Y4L2_T25                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       [O14757, O95819, Q13557]
                                                                                                                                                                                                                                                                                      ...                                                                                                                                                                                                                                                                    
Q9Y6Y8_S894                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  [Q96PF2, Q9BXA7]
Q9Y6Y8_T893                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  [P27448, Q7KZI7, Q96L34, Q9P0L2]
Q9Y6Y8_Y935                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  [P43405, P52333]
Q9Y6Y9_Y131                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          [P07948]
Q9Y6Y9_Y22                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           [P07948]
Name: kinase_uniprot, Length: 29811, dtype: object</code></pre>
</div>
</div>
<div id="c046b11d-aeb9-4dd3-bbc3-54175fff749c" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb64"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Count co-occurring kinase pairs</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>co_occurrence <span class="op">=</span> Counter()</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> kinases <span class="kw">in</span> grouped:</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> pair <span class="kw">in</span> combinations(<span class="bu">sorted</span>(<span class="bu">set</span>(kinases)), <span class="dv">2</span>):  <span class="co"># remove duplicates and sort for consistency</span></span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>        co_occurrence[pair] <span class="op">+=</span> <span class="dv">1</span></span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a><span class="co"># print(co_occurrence)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="2a3d099c-cdd4-44b3-add4-064b565ae8e4" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb65"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>kinases <span class="op">=</span> <span class="bu">set</span>()</span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> pair <span class="kw">in</span> co_occurrence.keys():</span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>    kinases.update(pair)</span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>kinases <span class="op">=</span> <span class="bu">sorted</span>(kinases)</span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize the co-occurrence matrix with zeros</span></span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a>matrix <span class="op">=</span> pd.DataFrame(<span class="dv">0</span>, index<span class="op">=</span>kinases, columns<span class="op">=</span>kinases)</span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill in the counts (matrix is symmetric)</span></span>
<span id="cb65-10"><a href="#cb65-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (k1, k2), count <span class="kw">in</span> co_occurrence.items():</span>
<span id="cb65-11"><a href="#cb65-11" aria-hidden="true" tabindex="-1"></a>    matrix.loc[k1, k2] <span class="op">=</span> count</span>
<span id="cb65-12"><a href="#cb65-12" aria-hidden="true" tabindex="-1"></a>    matrix.loc[k2, k1] <span class="op">=</span> count</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="d1852680-474d-41a4-a979-5dd66addaf62" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb66"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>matrix[<span class="st">'P00533'</span>].sort_values()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>P31152      0
Q16659      0
Q16654      0
P37173      0
P35790      0
         ... 
P36888    478
P11362    480
P07949    480
P06213    481
Q15303    497
Name: P00533, Length: 433, dtype: int64</code></pre>
</div>
</div>
<div id="cdf00471-9059-4ca0-a180-8fab0891bbfc" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb68"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>counts <span class="op">=</span> df_group.groupby(<span class="st">'kinase_uniprot'</span>).agg({<span class="st">'substrate_uniprot_site'</span>:<span class="st">'size'</span>})</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Jaccard</p>
<div id="61d145c8-c0c3-4fa4-bf8b-00670573421c" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb69"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>counts_series <span class="op">=</span> counts[<span class="st">'substrate_uniprot_site'</span>]</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a copy for normalized values</span></span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a>norm_matrix <span class="op">=</span> matrix.copy().astype(<span class="bu">float</span>)</span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> k1 <span class="kw">in</span> matrix.index:</span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> k2 <span class="kw">in</span> matrix.columns:</span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a>        intersection <span class="op">=</span> matrix.loc[k1, k2]</span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a>        union <span class="op">=</span> counts_series[k1] <span class="op">+</span> counts_series[k2] <span class="op">-</span> intersection</span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a>        norm_matrix.loc[k1, k2] <span class="op">=</span> intersection <span class="op">/</span> union <span class="cf">if</span> union <span class="op">!=</span> <span class="dv">0</span> <span class="cf">else</span> <span class="dv">0</span></span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a>    norm_matrix.loc[k1, k1] <span class="op">=</span> <span class="dv">1</span> <span class="co"># diaganol to be 1</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="0c890593-273f-4863-920d-70ef79653ee6" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb70"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a boolean mask for the upper triangle (k=1 excludes the diagonal)</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>mask <span class="op">=</span> np.triu(np.ones(norm_matrix.shape), k<span class="op">=</span><span class="dv">1</span>).astype(<span class="bu">bool</span>)</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>norm_matrix <span class="op">=</span> norm_matrix.where(mask)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="798fde50-b3dc-4ebd-b2d0-adae225a879d" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb71"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>matrix[<span class="st">'P00533'</span>].sort_values(ascending<span class="op">=</span><span class="va">False</span>).head()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>Q15303    497
P06213    481
P07949    480
P11362    480
P36888    478
Name: P00533, dtype: int64</code></pre>
</div>
</div>
<div id="a3ea32c7-a95b-44f1-90ff-061cba9fea22" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb73"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>norm_matrix[<span class="st">'P00533'</span>].sort_values(ascending<span class="op">=</span><span class="va">False</span>).head()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>O60674    0.253078
O15146    0.213529
P00519    0.203051
O43318    0.122415
O75676    0.023234
Name: P00533, dtype: float64</code></pre>
</div>
</div>
<div id="6be815c4-c4c8-4080-a76f-671fa584bda5" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb75"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>kin_info <span class="op">=</span> Data.get_kinase_info()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="87cbce57-25a3-4987-8ab7-09f28ecc29de" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb76"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>st<span class="op">=</span>kin_info[kin_info.group<span class="op">!=</span><span class="st">'TK'</span>].uniprot.tolist()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="3e206ad8-ff65-4ff6-bbd9-1311fd93e33b" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb77"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>st_col <span class="op">=</span>norm_matrix.columns[norm_matrix.columns.isin(st)]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="cd45d026-63cb-4e58-91aa-23e6cd706045" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb78"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>st_matrix <span class="op">=</span> norm_matrix[st_col]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="6c290838-7477-444c-9142-1f015968b4bf" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb79"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> get_max_jaccard(matrix):   </span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>    a <span class="op">=</span> matrix.<span class="bu">max</span>()</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>    b <span class="op">=</span> matrix.idxmax()</span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>    matrix <span class="op">=</span> pd.concat([a, b], axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>    matrix.columns <span class="op">=</span> [<span class="st">'jaccard'</span>, <span class="st">'kinase_uniprot'</span>]</span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> matrix.sort_values(<span class="st">'jaccard'</span>, ascending<span class="op">=</span><span class="va">False</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="63e11378-6b07-4f6a-ac45-900cdecefc7f" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb80"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>get_max_jaccard(st_matrix)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>/tmp/ipykernel_169440/3714721923.py:3: FutureWarning: The behavior of DataFrame.idxmax with all-NA values, or any-NA and skipna=False, is deprecated. In a future version this will raise ValueError
  b = matrix.idxmax()</code></pre>
</div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">jaccard</th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">Q9NYV4</th>
<td>0.509804</td>
<td>Q14004</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">Q96QP1</th>
<td>0.500000</td>
<td>Q96L96</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">Q9Y6E0</th>
<td>0.465691</td>
<td>Q9P289</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">P28482</th>
<td>0.450737</td>
<td>P27361</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">P31751</th>
<td>0.437574</td>
<td>P31749</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">...</th>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">O76039</th>
<td>0.000000</td>
<td>O00141</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">Q2M2I8</th>
<td>0.000000</td>
<td>O00141</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">O60885</th>
<td>0.000000</td>
<td>O00141</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">O00311</th>
<td>0.000000</td>
<td>O00141</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">O00141</th>
<td>NaN</td>
<td>NaN</td>
</tr>
</tbody>
</table>

<p>323 rows × 2 columns</p>
</div>
</div>
</div>
<div id="83af9a6d-9039-4e2d-becd-28784b4e0c89" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb82"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>tyr<span class="op">=</span>kin_info[kin_info.group<span class="op">==</span><span class="st">'TK'</span>].uniprot.tolist()</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>tyr_col <span class="op">=</span>norm_matrix.columns[norm_matrix.columns.isin(tyr)]</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>tyr_matrix <span class="op">=</span> norm_matrix[tyr_col]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="3220f2b9-d6d0-4654-b6e0-e13c4603bd39" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb83"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>get_max_jaccard(tyr_matrix)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">jaccard</th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">P54762</th>
<td>0.554638</td>
<td>P54760</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">Q15375</th>
<td>0.539118</td>
<td>P21709</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">P54753</th>
<td>0.531496</td>
<td>P29320</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">Q06418</th>
<td>0.526057</td>
<td>P54762</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">P54760</th>
<td>0.521168</td>
<td>P54753</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">...</th>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">P04626</th>
<td>0.046957</td>
<td>O43318</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">O15146</th>
<td>0.044005</td>
<td>O00506</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">Q6J9G0</th>
<td>0.016393</td>
<td>O43293</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">Q6ZMQ8</th>
<td>0.004348</td>
<td>O96017</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">Q8IWU2</th>
<td>0.003676</td>
<td>Q13976</td>
</tr>
</tbody>
</table>

<p>83 rows × 2 columns</p>
</div>
</div>
</div>
<div id="42a84f4b-aad4-4e13-af95-079d9dae54d2" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb84"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>kin_info_uniprot <span class="op">=</span> pd.read_excel(<span class="st">'raw/uniprot_human_keyword_kinase.xlsx'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="65255b22-ca62-4e05-8b20-2432abafee14" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb85"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>kin_map <span class="op">=</span> kin_info_uniprot.set_index(<span class="st">'Entry'</span>)[<span class="st">'Entry Name'</span>].<span class="bu">str</span>.split(<span class="st">'_'</span>).<span class="bu">str</span>[<span class="dv">0</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="1f2583b3-fca7-4b8d-ac89-1f9ab57f1237" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb86"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>names <span class="op">=</span> norm_matrix.index.<span class="bu">map</span>(kin_map)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>todo: Use filter to only select kinase &gt;30 counts</p>
<div id="06080fcc-e2a5-405f-a236-5b3984bd90c0" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb87"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> scipy.cluster.hierarchy <span class="im">import</span> linkage, dendrogram</span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> sklearn.manifold <span class="im">import</span> MDS</span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> networkx <span class="im">as</span> nx</span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming norm_matrix is your pre-computed Jaccard similarity matrix</span></span>
<span id="cb87-10"><a href="#cb87-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-11"><a href="#cb87-11" aria-hidden="true" tabindex="-1"></a><span class="co"># ---------------------------</span></span>
<span id="cb87-12"><a href="#cb87-12" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. Clustering &amp; Heatmap</span></span>
<span id="cb87-13"><a href="#cb87-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-14"><a href="#cb87-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Alternatively, using hierarchical clustering with scipy:</span></span>
<span id="cb87-15"><a href="#cb87-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Z = linkage(norm_matrix, method='average')</span></span>
<span id="cb87-16"><a href="#cb87-16" aria-hidden="true" tabindex="-1"></a><span class="co"># plt.figure(figsize=(10, 5))</span></span>
<span id="cb87-17"><a href="#cb87-17" aria-hidden="true" tabindex="-1"></a><span class="co"># dendrogram(Z, labels=norm_matrix.index)</span></span>
<span id="cb87-18"><a href="#cb87-18" aria-hidden="true" tabindex="-1"></a><span class="co"># plt.title("Hierarchical Clustering Dendrogram")</span></span>
<span id="cb87-19"><a href="#cb87-19" aria-hidden="true" tabindex="-1"></a><span class="co"># plt.xlabel("Kinase")</span></span>
<span id="cb87-20"><a href="#cb87-20" aria-hidden="true" tabindex="-1"></a><span class="co"># plt.ylabel("Distance")</span></span>
<span id="cb87-21"><a href="#cb87-21" aria-hidden="true" tabindex="-1"></a><span class="co"># plt.show()</span></span>
<span id="cb87-22"><a href="#cb87-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-23"><a href="#cb87-23" aria-hidden="true" tabindex="-1"></a>Z <span class="op">=</span> linkage(norm_matrix, method<span class="op">=</span><span class="st">'average'</span>)</span>
<span id="cb87-24"><a href="#cb87-24" aria-hidden="true" tabindex="-1"></a>plt.figure(figsize<span class="op">=</span>(<span class="dv">4</span>,<span class="dv">50</span>))</span>
<span id="cb87-25"><a href="#cb87-25" aria-hidden="true" tabindex="-1"></a>dendrogram(Z,orientation<span class="op">=</span><span class="st">'left'</span>,leaf_font_size<span class="op">=</span><span class="dv">7</span>,labels<span class="op">=</span>names,color_threshold<span class="op">=</span><span class="fl">1.3</span>)</span>
<span id="cb87-26"><a href="#cb87-26" aria-hidden="true" tabindex="-1"></a>plt.title(<span class="st">'Hierarchical Clustering Dendrogram'</span>)</span>
<span id="cb87-27"><a href="#cb87-27" aria-hidden="true" tabindex="-1"></a>plt.ylabel(<span class="st">'Distance'</span>)</span>
<span id="cb87-28"><a href="#cb87-28" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="02_EDA_fig1_files/figure-html/cell-78-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="old-1" class="level2">
<h2 class="anchored" data-anchor-id="old-1">Old</h2>
<div id="e198ffd9-a6d8-4f57-a4de-6375a4b90554" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb88"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Network Analysis</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a><span class="co"># ---------------------------</span></span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Build a graph from norm_matrix (using a threshold to filter weak links)</span></span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>threshold <span class="op">=</span> <span class="fl">0.2</span>  <span class="co"># adjust threshold as needed</span></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>G <span class="op">=</span> nx.Graph()</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, kinase1 <span class="kw">in</span> <span class="bu">enumerate</span>(norm_matrix.index):</span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> j, kinase2 <span class="kw">in</span> <span class="bu">enumerate</span>(norm_matrix.columns):</span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> i <span class="op">&lt;</span> j <span class="kw">and</span> norm_matrix.loc[kinase1, kinase2] <span class="op">&gt;</span> threshold:</span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>            G.add_edge(kinase1, kinase2, weight<span class="op">=</span>norm_matrix.loc[kinase1, kinase2])</span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Draw the network</span></span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a>plt.figure(figsize<span class="op">=</span>(<span class="dv">10</span>, <span class="dv">10</span>))</span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a>pos <span class="op">=</span> nx.spring_layout(G, weight<span class="op">=</span><span class="st">'weight'</span>)</span>
<span id="cb88-14"><a href="#cb88-14" aria-hidden="true" tabindex="-1"></a>nx.draw(G, pos, with_labels<span class="op">=</span><span class="va">True</span>, node_size<span class="op">=</span><span class="dv">500</span>, font_size<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb88-15"><a href="#cb88-15" aria-hidden="true" tabindex="-1"></a>plt.title(<span class="st">"Kinase Similarity Network"</span>)</span>
<span id="cb88-16"><a href="#cb88-16" aria-hidden="true" tabindex="-1"></a>plt.show()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="02_EDA_fig1_files/figure-html/cell-79-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="frequency-of-kinase" class="level2">
<h2 class="anchored" data-anchor-id="frequency-of-kinase">Frequency of kinase</h2>
<p>Instead of showing PSSM, we can show the neigboring kinase</p>
</section>
<section id="kinase-count" class="level2">
<h2 class="anchored" data-anchor-id="kinase-count">Kinase count</h2>
<div id="a4be344d-1f8a-4c3b-8c56-06a2162ccda8" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb89"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>df_group[<span class="st">'substrate_uniprot_site'</span>]<span class="op">=</span>df_group[<span class="st">'substrate_uniprot'</span>]<span class="op">+</span>df_group[<span class="st">'site'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="fb3008f4-2411-43f0-94d2-203fe9d5fe57" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb90"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>counts <span class="op">=</span> df_group.groupby(<span class="st">'kinase_uniprot'</span>).agg({<span class="st">'substrate_uniprot_site'</span>:<span class="st">'size'</span>}).reset_index()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="a955ea38-ac0e-42ce-9350-a15bafa73268" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb91"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>counts</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">substrate_uniprot_site</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">0</th>
<td>O00141</td>
<td>387</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1</th>
<td>O00238</td>
<td>356</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">2</th>
<td>O00311</td>
<td>50</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">3</th>
<td>O00329</td>
<td>6</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">4</th>
<td>O00418</td>
<td>202</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">...</th>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">450</th>
<td>Q9Y572</td>
<td>10</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">451</th>
<td>Q9Y5S2</td>
<td>109</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">452</th>
<td>Q9Y6E0</td>
<td>846</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">453</th>
<td>Q9Y6M4</td>
<td>224</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">454</th>
<td>Q9Y6R4</td>
<td>15</td>
</tr>
</tbody>
</table>

<p>455 rows × 2 columns</p>
</div>
</div>
</div>
<div id="003d8e94-81d1-4044-8ba7-60dc799fe263" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb92"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>kin_info <span class="op">=</span> pd.read_excel(<span class="st">'raw/uniprot_human_keyword_kinase.xlsx'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="b00364df-5f37-4080-babd-15befd3c2fd5" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb93"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a>kin_info <span class="op">=</span> kin_info.rename(columns<span class="op">=</span>{<span class="st">'Entry'</span>:<span class="st">'kinase_uniprot'</span>})</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="ea4482e1-21ed-4857-9f0d-7fd6da6326ed" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb94"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>kin_info2 <span class="op">=</span> Data.get_kinase_info()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="d76faf6c-2479-4afc-81b2-c08fcaa498bc" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb95"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>(kin_info2.in_ST_paper<span class="op">==</span><span class="dv">1</span>).<span class="bu">sum</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>303</code></pre>
</div>
</div>
<div id="ea280523-54fa-4df6-97f9-f7b89c301768" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb97"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>(kin_info2.in_Tyr_paper<span class="op">==</span><span class="dv">1</span>).<span class="bu">sum</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>93</code></pre>
</div>
</div>
<div id="3ea432b3-0f0d-4902-b81b-43276fef8454" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb99"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>in_pspa<span class="op">=</span>(kin_info2.in_ST_paper<span class="op">==</span><span class="dv">1</span>)<span class="op">|</span>(kin_info2.in_Tyr_paper<span class="op">==</span><span class="dv">1</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="a3ee92b5-2225-4790-b0da-915bbdd0bc29" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb100"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>kin_info2[in_pspa]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">kinase</th>
<th data-quarto-table-cell-role="th">ID_coral</th>
<th data-quarto-table-cell-role="th">uniprot</th>
<th data-quarto-table-cell-role="th">ID_HGNC</th>
<th data-quarto-table-cell-role="th">group</th>
<th data-quarto-table-cell-role="th">family</th>
<th data-quarto-table-cell-role="th">subfamily_coral</th>
<th data-quarto-table-cell-role="th">subfamily</th>
<th data-quarto-table-cell-role="th">in_ST_paper</th>
<th data-quarto-table-cell-role="th">in_Tyr_paper</th>
<th data-quarto-table-cell-role="th">...</th>
<th data-quarto-table-cell-role="th">cytosol</th>
<th data-quarto-table-cell-role="th">cytoskeleton</th>
<th data-quarto-table-cell-role="th">plasma membrane</th>
<th data-quarto-table-cell-role="th">mitochondrion</th>
<th data-quarto-table-cell-role="th">Golgi apparatus</th>
<th data-quarto-table-cell-role="th">endoplasmic reticulum</th>
<th data-quarto-table-cell-role="th">vesicle</th>
<th data-quarto-table-cell-role="th">centrosome</th>
<th data-quarto-table-cell-role="th">aggresome</th>
<th data-quarto-table-cell-role="th">main_location</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">0</th>
<td>AAK1</td>
<td>AAK1</td>
<td>Q2M2I8</td>
<td>AAK1</td>
<td>Other</td>
<td>NAK</td>
<td>NaN</td>
<td>NAK</td>
<td>1</td>
<td>0</td>
<td>...</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1</th>
<td>ABL1</td>
<td>ABL1</td>
<td>P00519</td>
<td>ABL1</td>
<td>TK</td>
<td>Abl</td>
<td>NaN</td>
<td>Abl</td>
<td>0</td>
<td>1</td>
<td>...</td>
<td>6.0</td>
<td>NaN</td>
<td>4.0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>cytosol</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">2</th>
<td>ABL2</td>
<td>ABL2</td>
<td>P42684</td>
<td>ABL2</td>
<td>TK</td>
<td>Abl</td>
<td>NaN</td>
<td>Abl</td>
<td>0</td>
<td>1</td>
<td>...</td>
<td>4.0</td>
<td>6.0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>cytoskeleton</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">3</th>
<td>TNK2</td>
<td>ACK</td>
<td>Q07912</td>
<td>TNK2</td>
<td>TK</td>
<td>Ack</td>
<td>NaN</td>
<td>Ack</td>
<td>0</td>
<td>1</td>
<td>...</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>8.0</td>
<td>NaN</td>
<td>2.0</td>
<td>vesicle</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">4</th>
<td>ACVR2A</td>
<td>ACTR2</td>
<td>P27037</td>
<td>ACVR2A</td>
<td>TKL</td>
<td>STKR</td>
<td>STKR2</td>
<td>STKR2</td>
<td>1</td>
<td>0</td>
<td>...</td>
<td>5.0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>5.0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>cytosol</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">518</th>
<td>YSK1</td>
<td>YSK1</td>
<td>O00506</td>
<td>STK25</td>
<td>STE</td>
<td>STE20</td>
<td>YSK</td>
<td>YSK</td>
<td>1</td>
<td>0</td>
<td>...</td>
<td>6.0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>4.0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>cytosol</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">519</th>
<td>ZAK</td>
<td>ZAK</td>
<td>Q9NYL2</td>
<td>MAP3K20</td>
<td>TKL</td>
<td>MLK</td>
<td>ZAK</td>
<td>ZAK</td>
<td>1</td>
<td>0</td>
<td>...</td>
<td>5.0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>nucleus</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">520</th>
<td>ZAP70</td>
<td>ZAP70</td>
<td>P43403</td>
<td>ZAP70</td>
<td>TK</td>
<td>Syk</td>
<td>NaN</td>
<td>Syk</td>
<td>0</td>
<td>1</td>
<td>...</td>
<td>5.0</td>
<td>NaN</td>
<td>2.0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>cytosol</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">521</th>
<td>EEF2K</td>
<td>eEF2K</td>
<td>O00418</td>
<td>EEF2K</td>
<td>Atypical</td>
<td>Alpha</td>
<td>eEF2K</td>
<td>eEF2K</td>
<td>1</td>
<td>0</td>
<td>...</td>
<td>9.0</td>
<td>NaN</td>
<td>1.0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>cytosol</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">522</th>
<td>FAM20C</td>
<td>FAM20C</td>
<td>Q8IXL6</td>
<td>FAM20C</td>
<td>Atypical</td>
<td>FAM20C</td>
<td>NaN</td>
<td>FAM20C</td>
<td>1</td>
<td>0</td>
<td>...</td>
<td>2.0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>7.0</td>
<td>1.0</td>
<td>NaN</td>
<td>NaN</td>
<td>NaN</td>
<td>Golgi apparatus</td>
</tr>
</tbody>
</table>

<p>392 rows × 30 columns</p>
</div>
</div>
</div>
<div id="9729cf45-6502-4424-9db4-c1992582c55c" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb101"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>kin_info2[</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>Index(['kinase', 'ID_coral', 'uniprot', 'ID_HGNC', 'group', 'family',
       'subfamily_coral', 'subfamily', 'in_ST_paper', 'in_Tyr_paper',
       'in_cddm', 'pseudo', 'pspa_category_small', 'pspa_category_big',
       'cddm_big', 'cddm_small', 'length', 'human_uniprot_sequence',
       'kinasecom_domain', 'nucleus', 'cytosol', 'cytoskeleton',
       'plasma membrane', 'mitochondrion', 'Golgi apparatus',
       'endoplasmic reticulum', 'vesicle', 'centrosome', 'aggresome',
       'main_location'],
      dtype='object')</code></pre>
</div>
</div>
<div id="1c09cc4e-df96-47f7-8f4c-1cb2193b9abb" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb103"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a>df_group[df_group.kinase_uniprot<span class="op">==</span><span class="st">'P07948-2'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">kin_sub_site</th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">substrate_uniprot</th>
<th data-quarto-table-cell-role="th">site</th>
<th data-quarto-table-cell-role="th">source</th>
<th data-quarto-table-cell-role="th">substrate_genes</th>
<th data-quarto-table-cell-role="th">substrate_sequence</th>
<th data-quarto-table-cell-role="th">substrate_phosphoseq</th>
<th data-quarto-table-cell-role="th">substrate_uniprot_site</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">30312</th>
<td>P07948-2_A0MZ66_Y24</td>
<td>P07948-2</td>
<td>A0MZ66</td>
<td>Y24</td>
<td>Sugiyama</td>
<td>SHTN1 KIAA1598</td>
<td>MNSSDEEKQLQLITSLKEQAIGEYEDLRAENQKTKEKCDKIRQERD...</td>
<td>MNSSDEEKQLQLITSLKEQAIGEyEDLRAENQKTKEKCDKIRQERD...</td>
<td>A0MZ66Y24</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">30313</th>
<td>P07948-2_A1L429_Y10</td>
<td>P07948-2</td>
<td>A1L429</td>
<td>Y10</td>
<td>Sugiyama</td>
<td>GAGE12B; GAGE12C; GAGE12D; GAGE12E</td>
<td>MSWRGRSTYYWPRPRRYVQPPEMIGPMRPEQFSDEVEPATPEEGEP...</td>
<td>MSWRGRsTyyWPRPRRYVQPPEMIGPMRPEQFSDEVEPATPEEGEP...</td>
<td>A1L429Y10</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">30314</th>
<td>P07948-2_A1L429_Y9</td>
<td>P07948-2</td>
<td>A1L429</td>
<td>Y9</td>
<td>Sugiyama</td>
<td>GAGE12B; GAGE12C; GAGE12D; GAGE12E</td>
<td>MSWRGRSTYYWPRPRRYVQPPEMIGPMRPEQFSDEVEPATPEEGEP...</td>
<td>MSWRGRsTyyWPRPRRYVQPPEMIGPMRPEQFSDEVEPATPEEGEP...</td>
<td>A1L429Y9</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">30315</th>
<td>P07948-2_A1X283_Y855</td>
<td>P07948-2</td>
<td>A1X283</td>
<td>Y855</td>
<td>Sugiyama</td>
<td>SH3PXD2B FAD49 KIAA1295 TKS4</td>
<td>MPPRRSIVEVKVLDVQKRRVPNKHYVYIIRVTWSSGSTEAIYRRYS...</td>
<td>MPPRRSIVEVKVLDVQKRRVPNKHYVYIIRVTWSSGSTEAIYRRYS...</td>
<td>A1X283Y855</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">30316</th>
<td>P07948-2_A5A3E0_S939</td>
<td>P07948-2</td>
<td>A5A3E0</td>
<td>S939</td>
<td>Sugiyama</td>
<td>POTEF A26C1B</td>
<td>MVVEVDSMPAASSVKKPFGLRSKMGKWCCRCFPCCRESGKSNVGTS...</td>
<td>MVVEVDSMPAASSVKKPFGLRSKMGKWCCRCFPCCRESGKSNVGTS...</td>
<td>A5A3E0S939</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">31872</th>
<td>P07948-2_Q9Y5S9_Y54</td>
<td>P07948-2</td>
<td>Q9Y5S9</td>
<td>Y54</td>
<td>Sugiyama</td>
<td>RBM8A RBM8 HSPC114 MDS014</td>
<td>MADVLDLHEAGGEDFAMDEDGDESIHKLKEKAKKRKGRGFGSEEGS...</td>
<td>MADVLDLHEAGGEDFAMDEDGDESIHKLKEKAKKRKGRGFGSEEGS...</td>
<td>Q9Y5S9Y54</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">31873</th>
<td>P07948-2_Q9Y639_Y216</td>
<td>P07948-2</td>
<td>Q9Y639</td>
<td>Y216</td>
<td>Sugiyama</td>
<td>NPTN SDFR1 SDR1</td>
<td>MSGSSLPSALALSLLLVSGSLLPGPGAAQNAGFVKSPMSETKLTGD...</td>
<td>MSGSSLPSALALSLLLVSGSLLPGPGAAQNAGFVKSPMSETKLTGD...</td>
<td>Q9Y639Y216</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">31874</th>
<td>P07948-2_Q9Y639_Y220</td>
<td>P07948-2</td>
<td>Q9Y639</td>
<td>Y220</td>
<td>Sugiyama</td>
<td>NPTN SDFR1 SDR1</td>
<td>MSGSSLPSALALSLLLVSGSLLPGPGAAQNAGFVKSPMSETKLTGD...</td>
<td>MSGSSLPSALALSLLLVSGSLLPGPGAAQNAGFVKSPMSETKLTGD...</td>
<td>Q9Y639Y220</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">31875</th>
<td>P07948-2_Q9Y696_S245</td>
<td>P07948-2</td>
<td>Q9Y696</td>
<td>S245</td>
<td>Sugiyama</td>
<td>CLIC4</td>
<td>MALSMPLNGLKEEDKEPLIELFVKAGSDGESIGNCPFSQRLFMILW...</td>
<td>MALSMPLNGLKEEDKEPLIELFVKAGsDGEsIGNCPFSQRLFMILW...</td>
<td>Q9Y696S245</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">31876</th>
<td>P07948-2_Q9Y696_Y244</td>
<td>P07948-2</td>
<td>Q9Y696</td>
<td>Y244</td>
<td>Sugiyama</td>
<td>CLIC4</td>
<td>MALSMPLNGLKEEDKEPLIELFVKAGSDGESIGNCPFSQRLFMILW...</td>
<td>MALSMPLNGLKEEDKEPLIELFVKAGsDGEsIGNCPFSQRLFMILW...</td>
<td>Q9Y696Y244</td>
</tr>
</tbody>
</table>

<p>1565 rows × 9 columns</p>
</div>
</div>
</div>
<div id="aff9d996-7327-4f30-b314-bf96e67b2d76" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb104"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>counts</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">substrate_uniprot_site</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">0</th>
<td>O00141</td>
<td>387</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1</th>
<td>O00238</td>
<td>356</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">2</th>
<td>O00311</td>
<td>50</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">3</th>
<td>O00329</td>
<td>6</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">4</th>
<td>O00418</td>
<td>202</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">...</th>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">450</th>
<td>Q9Y572</td>
<td>10</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">451</th>
<td>Q9Y5S2</td>
<td>109</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">452</th>
<td>Q9Y6E0</td>
<td>846</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">453</th>
<td>Q9Y6M4</td>
<td>224</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">454</th>
<td>Q9Y6R4</td>
<td>15</td>
</tr>
</tbody>
</table>

<p>455 rows × 2 columns</p>
</div>
</div>
</div>
<div id="3cdaf8d3-1fb6-42e6-ad77-ac46634f0464" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb105"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>counts.merge(kin_info)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">substrate_uniprot_site</th>
<th data-quarto-table-cell-role="th">Entry Name</th>
<th data-quarto-table-cell-role="th">Protein names</th>
<th data-quarto-table-cell-role="th">Gene Names</th>
<th data-quarto-table-cell-role="th">uniprot_keyword_kinase</th>
<th data-quarto-table-cell-role="th">on_tree</th>
<th data-quarto-table-cell-role="th">Organism</th>
<th data-quarto-table-cell-role="th">Keywords</th>
<th data-quarto-table-cell-role="th">Sequence</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">0</th>
<td>O00141</td>
<td>387</td>
<td>SGK1_HUMAN</td>
<td>Serine/threonine-protein kinase Sgk1 (EC 2.7.1...</td>
<td>SGK1 SGK</td>
<td>1</td>
<td>1</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Alternative promoter usage;Altern...</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1</th>
<td>O00238</td>
<td>356</td>
<td>BMR1B_HUMAN</td>
<td>Bone morphogenetic protein receptor type-1B (B...</td>
<td>BMPR1B</td>
<td>1</td>
<td>1</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Alternative splicing;ATP-binding;...</td>
<td>MLLRSAGKLNVGTKKEDGESTAPTPRPKVLRCKCHHHCPEDSVNNI...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">2</th>
<td>O00311</td>
<td>50</td>
<td>CDC7_HUMAN</td>
<td>Cell division cycle 7-related protein kinase (...</td>
<td>CDC7 CDC7L1</td>
<td>1</td>
<td>1</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Alternative splicing;ATP-binding;...</td>
<td>MEASLGIQMDEPMAFSPQRDRFQAEGSLKKNEQNFKLAGVKKDIEK...</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">3</th>
<td>O00329</td>
<td>6</td>
<td>PK3CD_HUMAN</td>
<td>Phosphatidylinositol 4,5-bisphosphate 3-kinase...</td>
<td>PIK3CD</td>
<td>1</td>
<td>0</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Adaptive immunity;Alternative spl...</td>
<td>MPPGVDCPMEFWTKEENQSVVVDFLLPTGVYLNFPVSRNANLSTIK...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">4</th>
<td>O00418</td>
<td>202</td>
<td>EF2K_HUMAN</td>
<td>Eukaryotic elongation factor 2 kinase (eEF-2 k...</td>
<td>EEF2K</td>
<td>1</td>
<td>1</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Acetylation;ATP-binding;Calcium;C...</td>
<td>MADEDLIFRLEGVDGGQSPRAGHDGDSDGDSDDEEGYFICPITDDP...</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">447</th>
<td>Q9Y572</td>
<td>10</td>
<td>RIPK3_HUMAN</td>
<td>Receptor-interacting serine/threonine-protein ...</td>
<td>RIPK3 RIP3</td>
<td>1</td>
<td>1</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Alternative splicing;Apoptosis;AT...</td>
<td>MSCVKLWPSGAPAPLVSIEELENQELVGKGGFGTVFRAQHRKWGYD...</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">448</th>
<td>Q9Y5S2</td>
<td>109</td>
<td>MRCKB_HUMAN</td>
<td>Serine/threonine-protein kinase MRCK beta (EC ...</td>
<td>CDC42BPB KIAA1124</td>
<td>1</td>
<td>1</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;ATP-binding;Cell junction;Cell me...</td>
<td>MSAKVRLKKLEQLLLDGPWRNESALSVETLLDVLVCLYTECSHSAL...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">449</th>
<td>Q9Y6E0</td>
<td>846</td>
<td>STK24_HUMAN</td>
<td>Serine/threonine-protein kinase 24 (EC 2.7.11....</td>
<td>STK24 MST3 STK3</td>
<td>1</td>
<td>1</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Acetylation;Alternative splicing;...</td>
<td>MDSRAQLWGLALNKRRATLPHPGGSTNLKADPEELFTKLEKIGKGS...</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">450</th>
<td>Q9Y6M4</td>
<td>224</td>
<td>KC1G3_HUMAN</td>
<td>Casein kinase I isoform gamma-3 (CKI-gamma 3) ...</td>
<td>CSNK1G3</td>
<td>1</td>
<td>1</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Acetylation;Alternative splicing;...</td>
<td>MENKKKDKDKSDDRMARPSGRSGHNTRGTGSSSSGVLMVGPNFRVG...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">451</th>
<td>Q9Y6R4</td>
<td>15</td>
<td>M3K4_HUMAN</td>
<td>Mitogen-activated protein kinase kinase kinase...</td>
<td>MAP3K4 KIAA0213 MAPKKK4 MEKK4 MTK1</td>
<td>1</td>
<td>1</td>
<td>Homo sapiens (Human)</td>
<td>Alternative splicing;ATP-binding;Cytoplasm;Kin...</td>
<td>MREAAAALVPPPAFAVTPAAAMEEPPPPPPPPPPPPEPETESEPEC...</td>
</tr>
</tbody>
</table>

<p>452 rows × 10 columns</p>
</div>
</div>
</div>
</section>
<section id="source-confidence" class="level2">
<h2 class="anchored" data-anchor-id="source-confidence">Source confidence</h2>
<div id="5ca1973d-560e-4daf-8394-cfd04dee9fa5" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb106"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>df_group.source.<span class="bu">str</span>.split(<span class="st">'|'</span>).<span class="bu">str</span>.<span class="bu">len</span>().value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>source
1    174536
2      6123
3      3352
4      1312
5       923
6       770
7        50
Name: count, dtype: int64</code></pre>
</div>
</div>
</section>
<section id="self-phosphorylate" class="level2">
<h2 class="anchored" data-anchor-id="self-phosphorylate">Self-phosphorylate</h2>
<div id="180a836b-e8a5-4581-a788-2961fe70bc95" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb108"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>df_group[df_group[<span class="st">'kinase_uniprot'</span>]<span class="op">==</span>df_group[<span class="st">'substrate_uniprot'</span>]].substrate_genes.value_counts().head(<span class="dv">20</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>substrate_genes
LRRK2 PARK8                  57
TRPM7 CHAK1 LTRPC7           46
MARK2 EMK1                   35
BMX                          34
TTK MPS1 MPS1L1              34
NEK4 STK2                    33
DCLK2 DCAMKL2 DCDC3B DCK2    32
ERBB4 HER4                   32
ABL1 ABL JTK7                32
MARK1 KIAA1477 MARK          31
HIPK2                        31
RPS6KA2 MAPKAPK1C RSK3       29
BTK AGMX1 ATK BPK            29
STK4 KRS2 MST1               28
BUB1B BUBR1 MAD3L SSK1       28
PDGFRA PDGFR2 RHEPDGFRA      28
ITK EMT LYK                  28
RPS6KA6 RSK4                 28
PRKD2 PKD2 HSPC187           28
ABL2 ABLL ARG                27
Name: count, dtype: int64</code></pre>
</div>
</div>
<div id="4d5dc509-ab11-4d81-9876-cb80f02e1eaa" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb110"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>df_group.source</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>source
Sugiyama                     167633
EPSD|PSP                       3346
PSP                            2932
SIGNOR                         1984
SIGNOR|EPSD|PSP                1878
                              ...  
GPS6|SIGNOR|ELM|Sugiyama          1
ELM|iPTMNet|Sugiyama              1
ELM|iPTMNet|EPSD|Sugiyama         1
iPTMNet|PSP|Sugiyama              1
SIGNOR|ELM|EPSD|Sugiyama          1
Name: count, Length: 111, dtype: int64</code></pre>
</div>
</div>
<section id="nearby-co-occured-kinase" class="level3">
<h3 class="anchored" data-anchor-id="nearby-co-occured-kinase">Nearby co-occured kinase</h3>
<p>TODO</p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/sky1ove\.github\.io\/katlas-raw");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/sky1ove/katlas-raw/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></div></div></footer></body></html>