<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Motif plot – katlas-raw</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.min.js" integrity="sha384-ZvpUoO/+PpLXR1lu4jmpXWu80pZlYUAfxl5NsBMWOEPSjUn/6Z/hRTt8+pR6L4N2" crossorigin="anonymous"></script><script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-000f0825b9c55ed21d14970d810c14a9.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="https://cdn.jsdelivr.net/npm/requirejs@2.3.6/require.min.js" integrity="sha384-c9c+LnTbwQ3aujuU7ULEPVvgLs+Fn6fJUvIGTsuu1ZcCf11fiEubah0ttpca4ntM sha384-6V1/AdqZRWk1KAlWbKBlGhN7VG4iE/yAZcO6NZPMF8od0vukrvr0tg4qY6NSrItx" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>
<script src="https://cdn.jsdelivr.net/npm/@jupyter-widgets/html-manager@*/dist/embed-amd.js" crossorigin="anonymous"></script>


<link rel="stylesheet" href="styles.css">
<meta property="og:title" content="Motif plot – katlas-raw">
<meta property="og:image" content="https://sky1ove.github.io/katlas-raw/21_motif_plot_files/figure-html/cell-46-output-1.png">
<meta property="og:site_name" content="katlas-raw">
<meta property="og:image:height" content="143">
<meta property="og:image:width" content="1156">
<meta name="twitter:title" content="Motif plot – katlas-raw">
<meta name="twitter:image" content="https://sky1ove.github.io/katlas-raw/21_motif_plot_files/figure-html/cell-46-output-1.png">
<meta name="twitter:image-height" content="143">
<meta name="twitter:image-width" content="1156">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="./index.html" class="navbar-brand navbar-brand-logo">
    </a>
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">katlas-raw</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./motif_plot.html">Motif plot</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      </a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">katlas-raw</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./preprocess_dataset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preprocess kinase-substrate datasets</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./eda_fig1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Exploratory data analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./site_promiscuity_fig1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Site promiscuity</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssm_hierarchical.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Hierarchical clustering of PSSM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05_PSSM_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">05_PSSM_example.html</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./modi_group.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Modi group</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scale_pspa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PSPA scaled data</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./background.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Background frequencies</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./specificity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Specificity comparison</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./sty_composition.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">STY composition</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./plot_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot heatmap and logo of CDDM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./plot_pspa_figure.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PSPA figures</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./13_CDDM_umap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CDDM umap</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./get_aa_relationship.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get AA relationship</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pspa_cddm_heatmap_compare.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Heatmap/ST/JSD/cosine comparison</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_no_psp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_psp_02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring_compare_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scoring evaluation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring_compare_psp_02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scoring evaluation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Reactome pathway for each PSSM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_plot_pspa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot reactome pathways: PSPA</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_plot_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot reactome pathways: CDDM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./motif_clustering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Kmeans motifs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./motif_plot.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Motif plot</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./22_motif_web_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">22_motif_web_data.html</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./23_motif_kinase_similarity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cosine similarity with PSPA</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">kinase_domain</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_01_get_all_kd.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get all kinase domain sequences</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_02_pseudo_alignment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Alignment</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_03_feature_extract.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Extract KD features</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_04_prepare_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare data for training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_05_pspa_training_model_comparison_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Model comparison on T5 feature</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06a_pspa_training_feature_comparison_cnn_group.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on CNN model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06b_pspa_training_feature_comparison_cnn_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on CNN model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06c_pspa_training_feature_comparison_mlp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on MLP model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_07a_cddm_training_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_07b_pspa_training_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_08a_similarity_relationship.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KD similarity &amp; PSSM similarity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_08b_similarity_relationship_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KD similarity &amp; PSSM similarity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_09a_predict_pspa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Predict</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_09b_predict_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Predict</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_10_plot_umap_known_unk.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot UMAP of known &amp; unknown KD</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_11_plot_pspa_kd_t5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">kd_11_plot_pspa_kd_t5.html</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_12_compare_pred_psp_known.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Comopare the two</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#hierarchical-clustering-of-merged-pssms" id="toc-hierarchical-clustering-of-merged-pssms" class="nav-link active" data-scroll-target="#hierarchical-clustering-of-merged-pssms">Hierarchical clustering of merged pssms</a></li>
  <li><a href="#all-motif" id="toc-all-motif" class="nav-link" data-scroll-target="#all-motif">All motif</a></li>
  <li><a href="#pathway-for-each-motif" id="toc-pathway-for-each-motif" class="nav-link" data-scroll-target="#pathway-for-each-motif">Pathway for each motif</a></li>
  <li><a href="#special-motifs" id="toc-special-motifs" class="nav-link" data-scroll-target="#special-motifs">Special motifs</a>
  <ul class="collapse">
  <li><a href="#customized" id="toc-customized" class="nav-link" data-scroll-target="#customized">Customized</a></li>
  <li><a href="#most-common" id="toc-most-common" class="nav-link" data-scroll-target="#most-common">Most common</a></li>
  <li><a href="#most-common-y" id="toc-most-common-y" class="nav-link" data-scroll-target="#most-common-y">Most common y</a></li>
  <li><a href="#highest-ic-sum" id="toc-highest-ic-sum" class="nav-link" data-scroll-target="#highest-ic-sum">Highest IC sum</a></li>
  <li><a href="#c-terminal-motifs" id="toc-c-terminal-motifs" class="nav-link" data-scroll-target="#c-terminal-motifs">C-terminal motifs</a></li>
  <li><a href="#n-terminal-motifs" id="toc-n-terminal-motifs" class="nav-link" data-scroll-target="#n-terminal-motifs">N-Terminal motifs:</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/sky1ove/katlas-raw/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div><div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="motif_plot.html.md"><i class="bi bi-file-code"></i>CommonMark</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Motif plot</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<!-- WARNING: THIS FILE WAS AUTOGENERATED! DO NOT EDIT! -->
<div id="fb465500-76d8-44c4-b6a0-675bf95ece28" class="cell" data-execution_count="2">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="b7b29ff1-544c-4985-906c-1b91dec595b9" class="cell" data-execution_count="3">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>pssms <span class="op">=</span> pd.read_parquet(<span class="st">'out/all_site_pssms_filter.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="6d307392-a9fb-4d92-afd6-1be5c7c88001" class="cell" data-execution_count="4">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> pd.read_parquet(<span class="st">'raw/kmeans_site_long_cluster_new.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="cf6cb636-6c26-49bc-81e1-e5fd8a2ef6a6" class="cell" data-execution_count="9">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>pssms.shape</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="9">
<pre><code>(1169, 943)</code></pre>
</div>
</div>
<div id="f229b4b7-673e-4a29-8b0e-367b5255e1c9" class="cell" data-execution_count="10">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>df.shape</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="10">
<pre><code>(843252, 6)</code></pre>
</div>
</div>
<div id="53bb3751-ba91-4333-a8b8-d8fc35ab4deb" class="cell" data-execution_count="11">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>df.columns</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="11">
<pre><code>Index(['sub_site', 'site_seq', 'cluster_info', 'cluster', 'cluster_id',
       'cluster_new'],
      dtype='object')</code></pre>
</div>
</div>
<div id="3226ac11-e881-43e8-a64c-51cb9faf3c68" class="cell" data-execution_count="12">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>cols<span class="op">=</span>[<span class="st">'sub_site'</span>,<span class="st">'cluster_new'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="0e4d3617-992d-402f-ad74-23963ad2af6d" class="cell" data-execution_count="13">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> df[cols]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="060ee53f-a95e-4922-a5b4-f87d50dfbddb" class="cell" data-execution_count="14">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>info<span class="op">=</span>pd.read_parquet(<span class="st">'raw/kmeans_site.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="0afbcfeb-87d0-4b63-820c-365967d0430b" class="cell" data-execution_count="15">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>info <span class="op">=</span> info[[<span class="st">'substrate_uniprot'</span>, <span class="st">'substrate_genes'</span>, <span class="st">'site'</span>, <span class="st">'source'</span>,<span class="st">'sub_site'</span>,<span class="st">'site_seq'</span>]]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="e3ad5e8e-b5de-4f3c-820a-a0f6e7cc9f51" class="cell" data-execution_count="16">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> df.merge(info,on<span class="op">=</span><span class="st">'sub_site'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="2618357f-e9a1-4892-876c-1ab5aa73a219" class="cell" data-execution_count="17">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>df.columns</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="17">
<pre><code>Index(['sub_site', 'cluster_new', 'substrate_uniprot', 'substrate_genes',
       'site', 'source', 'site_seq'],
      dtype='object')</code></pre>
</div>
</div>
<div id="1c9f0d8d-4b12-4342-8191-7fef56cbf529" class="cell" data-execution_count="18">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>new_colname <span class="op">=</span> [<span class="st">'sub_site'</span>, <span class="st">'cluster'</span>, <span class="st">'uniprot'</span>, <span class="st">'genes'</span>,</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>       <span class="st">'site'</span>, <span class="st">'source'</span>, <span class="st">'site_seq'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="d7504c68-7a62-444a-b111-6fe778aca535" class="cell" data-execution_count="19">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>df.columns <span class="op">=</span> new_colname</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="df51a981-4cf6-4f0c-a68b-2b07f415b03d" class="cell" data-execution_count="20">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> df[df.cluster.isin(pssms.index)]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="457622c3-8627-48f8-ac4d-9799f2713b97" class="cell" data-execution_count="21">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>df.shape</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="21">
<pre><code>(751513, 7)</code></pre>
</div>
</div>
<div id="ca9d6ee1-f8fb-43c5-a61e-42b38bbf5c78" class="cell" data-execution_count="22">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> df.reset_index(drop<span class="op">=</span><span class="va">True</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="579f43d0-6160-43a7-b599-2afc94acd072" class="cell" data-execution_count="90">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>df.to_parquet(<span class="st">'out/web_motif_df.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="c7e786a1-a5b0-4a3f-a885-29e0ee3732ee" class="cell" data-execution_count="6">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> pd.read_parquet(<span class="st">'out/web_motif_df.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<section id="hierarchical-clustering-of-merged-pssms" class="level2">
<h2 class="anchored" data-anchor-id="hierarchical-clustering-of-merged-pssms">Hierarchical clustering of merged pssms</h2>
<div id="008732a4-2927-4579-a077-7a89542938cb" class="cell" data-execution_count="21">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.clustering <span class="im">import</span> <span class="op">*</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="5685c3bf-55a5-4fd1-9c15-ca12b3f3d0e6" class="cell" data-execution_count="25">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>Z <span class="op">=</span> get_Z(pssms)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<script type="application/vnd.jupyter.widget-view+json">
{"model_id":"fedbfa8f96014821af96fe10ad168774","version_major":2,"version_minor":0,"quarto_mimetype":"application/vnd.jupyter.widget-view+json"}
</script>
</div>
</div>
<div id="86b56756-4229-4815-9475-d22ddd2eb51e" class="cell" data-execution_count="23">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>count_map <span class="op">=</span> df.cluster.value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="680d0494-244c-44be-8571-258b7758992f" class="cell" data-execution_count="27">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb28"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>labels<span class="op">=</span> get_pssm_seq_labels(pssms,count_map <span class="op">=</span> count_map , thr<span class="op">=</span><span class="fl">0.3</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="b9ea9ddd-1849-40da-a4aa-2956449ad6d8" class="cell" data-execution_count="28">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>labels[:<span class="dv">4</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="28">
<pre><code>['29 (n=8,243): ....................s*P...................',
 '369 (n=6,453): ....................s*.s..................',
 '1054 (n=6,368): ....................t*P...................',
 '338 (n=6,182): .................R..s*....................']</code></pre>
</div>
</div>
<div id="a60f2baf-aa26-46c1-90df-744ec5d4718a" class="cell" data-execution_count="36">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.plot <span class="im">import</span> <span class="op">*</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib <span class="im">import</span> pyplot <span class="im">as</span> plt</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> scipy.cluster.hierarchy <span class="im">import</span> dendrogram</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="8a318b92-1a69-4ef7-a9d0-2accc481fc6d" class="cell" data-execution_count="41">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_dendrogram(Z,</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>                    color_thr<span class="op">=</span><span class="fl">0.07</span>,</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>                    dense<span class="op">=</span><span class="dv">7</span>, <span class="co"># the higher the more dense for each row</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>                    line_width<span class="op">=</span><span class="dv">1</span>,</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># title='Hierarchical Clustering Dendrogram',</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>                    scale<span class="op">=</span><span class="dv">1</span>,</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>                    <span class="op">**</span>kwargs):</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>    length <span class="op">=</span> (<span class="bu">len</span>(Z) <span class="op">+</span> <span class="dv">1</span>) <span class="op">//</span> dense</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">5</span><span class="op">*</span>scale,length<span class="op">*</span>scale))</span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">with</span> plt.rc_context({<span class="st">'lines.linewidth'</span>: line_width}):</span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>        dendrogram(</span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>            Z,</span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>            orientation<span class="op">=</span><span class="st">'left'</span>,</span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a>            leaf_font_size<span class="op">=</span><span class="dv">18</span>, <span class="co">#7</span></span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>            color_threshold<span class="op">=</span>color_thr,</span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a>            <span class="op">**</span>kwargs</span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>    ax <span class="op">=</span> plt.gca()</span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> spine <span class="kw">in</span> [<span class="st">'top'</span>, <span class="st">'right'</span>, <span class="st">'left'</span>, <span class="st">'bottom'</span>]:</span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>        ax.spines[spine].set_visible(<span class="va">False</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="4ba09c2e-43c2-4c70-97bb-e21e680e5755" class="cell" data-execution_count="43">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>plot_dendrogram(Z,labels<span class="op">=</span>labels,color_thr<span class="op">=</span><span class="fl">0.07</span>,scale<span class="op">=</span><span class="fl">2.5</span>)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="co"># save_pdf('out/dendrogram.pdf')</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>save_svg(<span class="st">'out/human_motif_dendrogram.svg'</span>)</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>plt.close()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="all-motif" class="level2">
<h2 class="anchored" data-anchor-id="all-motif">All motif</h2>
<div id="2bc02163-89ae-4f4d-90e1-ee0b3b0114e8" class="cell" data-execution_count="30">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.plot <span class="im">import</span> <span class="op">*</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.pssm <span class="im">import</span> <span class="op">*</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.utils <span class="im">import</span> <span class="op">*</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> tqdm <span class="im">import</span> tqdm</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> gc</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib <span class="im">import</span> pyplot <span class="im">as</span> plt</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="222313b5-a93a-45c3-92e3-3f53b90f2d6a" class="cell" data-execution_count="7">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb35"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>pssms <span class="op">=</span> pd.read_parquet(<span class="st">'out/all_site_pssms_filter.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="92f79801-e144-4e18-abd3-e5ebf14167ca" class="cell" data-execution_count="8">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> pd.read_parquet(<span class="st">'out/web_motif_df.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="be4341dc-9e2b-499f-a7a9-bbf2169751d7" class="cell" data-execution_count="10">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb37"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>cnt <span class="op">=</span> df.cluster.value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="cbef87de-aeb7-479c-b335-41f09079b426" class="cell" data-execution_count="19">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb38"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># cnt.index.get_loc(1198)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="19">
<pre><code>1146</code></pre>
</div>
</div>
<div id="0c46f3fb-3959-46cc-b0e9-2835c3f6edb3" class="cell" data-execution_count="28">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb40"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> save_show(path<span class="op">=</span><span class="va">None</span>, <span class="co"># image path, e.g., img.svg, if not None, will save, else plt.show()</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>              show_only<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>             ):</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>    <span class="co">"Show plot or save path"</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> show_only: plt.show()</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> path <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>: plt.savefig(path, bbox_inches<span class="op">=</span><span class="st">'tight'</span>, pad_inches<span class="op">=</span><span class="fl">0.05</span>,transparent<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>: plt.show()</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>    plt.close(<span class="st">'all'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="6c58478d-d326-4777-a84a-35598bc197f0" class="cell" data-execution_count="31">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb41"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> idx,row <span class="kw">in</span> tqdm(pssms.iterrows(),total<span class="op">=</span><span class="bu">len</span>(pssms)):</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>    plot_logo_heatmap(recover_pssm(row),title<span class="op">=</span><span class="ss">f'Motif #</span><span class="sc">{</span>idx<span class="sc">}</span><span class="ss"> (n=</span><span class="sc">{</span>cnt[idx]<span class="sc">}</span><span class="ss">)'</span>)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>    path<span class="op">=</span>prepare_path(<span class="ss">f'fig/motif/</span><span class="sc">{</span>idx<span class="sc">}</span><span class="ss">/pssm.svg'</span>)</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>    save_show(path,show_only<span class="op">=</span><span class="va">False</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>  2%|█▊                                                                                          | 23/1169 [01:07&lt;56:28,  2.96s/it]</code></pre>
</div>
</div>
</section>
<section id="pathway-for-each-motif" class="level2">
<h2 class="anchored" data-anchor-id="pathway-for-each-motif">Pathway for each motif</h2>
<div id="5dc41151-536c-4e32-9409-a58f3b37de03" class="cell" data-execution_count="34">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb43"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.pathway <span class="im">import</span> <span class="op">*</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.utils <span class="im">import</span> <span class="op">*</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="e982e296-89c7-45b8-b40b-80b424d51a5f" class="cell" data-execution_count="35">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb44"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>cnt <span class="op">=</span> df.cluster.value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="deb705e2-316c-489c-99f0-2300b9ee9f76" class="cell" data-execution_count="40">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb45"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># for idx in [545,</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="co">#             1058,</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="co">#             913,</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="co">#             597,</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a><span class="co">#             1231]:</span></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="co">#     df_motif = df[df.cluster==idx].copy()</span></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a><span class="co">#     gene_list = df_motif.genes.str.split(' ').str[0].dropna().drop_duplicates().tolist()</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a><span class="co">#     try:</span></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a><span class="co">#         df_path = get_reactome(gene_list,p_type='p')</span></span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a><span class="co">#         df_path = df_path[~df_path.name.str.contains('GTPase|SUMO',case=False,na=False)].copy()</span></span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a><span class="co">#         path = prepare_path(f'fig/motif/{idx}/path.parquet')</span></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a><span class="co">#         df_path.to_parquet(path)</span></span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a><span class="co">#     except Exception as e:</span></span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a><span class="co">#         print(f"Failed for {idx}: {e}")</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>Running pathway anlysis
Done
Failed for 1231: "None of [Index(['name', 'stId', 'entities.pValue'], dtype='object')] are in the [columns]"</code></pre>
</div>
</div>
<div id="7f366bfa-9f95-41fb-8e6e-1a0839cc94f9" class="cell" data-execution_count="36">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb47"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> idx <span class="kw">in</span> cnt.index:</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>    df_motif <span class="op">=</span> df[df.cluster<span class="op">==</span>idx].copy()</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>    gene_list <span class="op">=</span> df_motif.genes.<span class="bu">str</span>.split(<span class="st">' '</span>).<span class="bu">str</span>[<span class="dv">0</span>].dropna().drop_duplicates().tolist()</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">try</span>:</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>        df_path <span class="op">=</span> get_reactome(gene_list,p_type<span class="op">=</span><span class="st">'p'</span>)</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>        df_path <span class="op">=</span> df_path[<span class="op">~</span>df_path.name.<span class="bu">str</span>.contains(<span class="st">'GTPase|SUMO'</span>,case<span class="op">=</span><span class="va">False</span>,na<span class="op">=</span><span class="va">False</span>)].copy()</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>        path <span class="op">=</span> prepare_path(<span class="ss">f'fig/motif/</span><span class="sc">{</span>idx<span class="sc">}</span><span class="ss">/path.parquet'</span>)</span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>        df_path.to_parquet(path)</span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">except</span> <span class="pp">Exception</span> <span class="im">as</span> e:</span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>        <span class="bu">print</span>(<span class="ss">f"Failed for </span><span class="sc">{</span>idx<span class="sc">}</span><span class="ss">: </span><span class="sc">{</span>e<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
Failed for 545: cannot access local variable 'response' where it is not associated with a value
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
Failed for 1058: cannot access local variable 'response' where it is not associated with a value
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Failed for 913: ('Connection broken: IncompleteRead(6474 bytes read, 1718 more expected)', IncompleteRead(6474 bytes read, 1718 more expected))
Running pathway anlysis
('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
Failed for 597: cannot access local variable 'response' where it is not associated with a value
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Failed for 1231: "None of [Index(['name', 'stId', 'entities.pValue'], dtype='object')] are in the [columns]"
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done
Running pathway anlysis
Done</code></pre>
</div>
</div>
<div id="79c7004e-f464-46da-bded-b178941e26e1" class="cell" data-execution_count="44">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb49"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>pssms</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="44">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">-20P</th>
<th data-quarto-table-cell-role="th">-20G</th>
<th data-quarto-table-cell-role="th">-20A</th>
<th data-quarto-table-cell-role="th">-20C</th>
<th data-quarto-table-cell-role="th">-20S</th>
<th data-quarto-table-cell-role="th">-20T</th>
<th data-quarto-table-cell-role="th">-20V</th>
<th data-quarto-table-cell-role="th">-20I</th>
<th data-quarto-table-cell-role="th">-20L</th>
<th data-quarto-table-cell-role="th">-20M</th>
<th data-quarto-table-cell-role="th">...</th>
<th data-quarto-table-cell-role="th">20H</th>
<th data-quarto-table-cell-role="th">20K</th>
<th data-quarto-table-cell-role="th">20R</th>
<th data-quarto-table-cell-role="th">20Q</th>
<th data-quarto-table-cell-role="th">20N</th>
<th data-quarto-table-cell-role="th">20D</th>
<th data-quarto-table-cell-role="th">20E</th>
<th data-quarto-table-cell-role="th">20pS</th>
<th data-quarto-table-cell-role="th">20pT</th>
<th data-quarto-table-cell-role="th">20pY</th>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">cluster_new</th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">29</th>
<td>0.068293</td>
<td>0.056220</td>
<td>0.061707</td>
<td>0.018293</td>
<td>0.064512</td>
<td>0.043171</td>
<td>0.049878</td>
<td>0.039756</td>
<td>0.092317</td>
<td>0.021463</td>
<td>...</td>
<td>0.025225</td>
<td>0.067633</td>
<td>0.054838</td>
<td>0.047161</td>
<td>0.038874</td>
<td>0.058250</td>
<td>0.083841</td>
<td>0.028516</td>
<td>0.007677</td>
<td>0.002315</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">369</th>
<td>0.077786</td>
<td>0.067810</td>
<td>0.065004</td>
<td>0.012783</td>
<td>0.071551</td>
<td>0.039439</td>
<td>0.046610</td>
<td>0.033359</td>
<td>0.083087</td>
<td>0.015900</td>
<td>...</td>
<td>0.026542</td>
<td>0.064012</td>
<td>0.069945</td>
<td>0.046214</td>
<td>0.036222</td>
<td>0.049180</td>
<td>0.084309</td>
<td>0.052771</td>
<td>0.017018</td>
<td>0.002342</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">1054</th>
<td>0.088742</td>
<td>0.063003</td>
<td>0.074057</td>
<td>0.011843</td>
<td>0.064424</td>
<td>0.045476</td>
<td>0.055582</td>
<td>0.033633</td>
<td>0.075162</td>
<td>0.021159</td>
<td>...</td>
<td>0.022461</td>
<td>0.074185</td>
<td>0.059633</td>
<td>0.050775</td>
<td>0.027365</td>
<td>0.051882</td>
<td>0.071813</td>
<td>0.043499</td>
<td>0.017083</td>
<td>0.004587</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">338</th>
<td>0.076785</td>
<td>0.070604</td>
<td>0.071254</td>
<td>0.012201</td>
<td>0.072718</td>
<td>0.043273</td>
<td>0.052546</td>
<td>0.035627</td>
<td>0.088661</td>
<td>0.019847</td>
<td>...</td>
<td>0.022975</td>
<td>0.062082</td>
<td>0.069741</td>
<td>0.047091</td>
<td>0.034382</td>
<td>0.040574</td>
<td>0.074466</td>
<td>0.042529</td>
<td>0.010754</td>
<td>0.002933</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">427</th>
<td>0.076803</td>
<td>0.062933</td>
<td>0.069868</td>
<td>0.010749</td>
<td>0.071255</td>
<td>0.046637</td>
<td>0.047330</td>
<td>0.033807</td>
<td>0.076456</td>
<td>0.019071</td>
<td>...</td>
<td>0.023582</td>
<td>0.060864</td>
<td>0.063465</td>
<td>0.050460</td>
<td>0.035200</td>
<td>0.055141</td>
<td>0.081498</td>
<td>0.052540</td>
<td>0.013005</td>
<td>0.003468</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">539</th>
<td>0.022727</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.068182</td>
<td>0.000000</td>
<td>0.068182</td>
<td>0.295455</td>
<td>0.090909</td>
<td>0.022727</td>
<td>...</td>
<td>0.022727</td>
<td>0.090909</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.045455</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">131</th>
<td>0.000000</td>
<td>0.136364</td>
<td>0.045455</td>
<td>0.068182</td>
<td>0.022727</td>
<td>0.045455</td>
<td>0.022727</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>...</td>
<td>0.000000</td>
<td>0.068182</td>
<td>0.090909</td>
<td>0.000000</td>
<td>0.022727</td>
<td>0.272727</td>
<td>0.045455</td>
<td>0.045455</td>
<td>0.022727</td>
<td>0.000000</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">538</th>
<td>0.071429</td>
<td>0.000000</td>
<td>0.023810</td>
<td>0.000000</td>
<td>0.023810</td>
<td>0.047619</td>
<td>0.047619</td>
<td>0.023810</td>
<td>0.476190</td>
<td>0.000000</td>
<td>...</td>
<td>0.000000</td>
<td>0.071429</td>
<td>0.119048</td>
<td>0.095238</td>
<td>0.214286</td>
<td>0.023810</td>
<td>0.119048</td>
<td>0.047619</td>
<td>0.000000</td>
<td>0.023810</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1207</th>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>...</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.047619</td>
<td>0.119048</td>
<td>0.000000</td>
<td>0.047619</td>
<td>0.071429</td>
<td>0.047619</td>
<td>0.000000</td>
<td>0.023810</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">1224</th>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>...</td>
<td>0.023810</td>
<td>0.023810</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.023810</td>
<td>0.047619</td>
<td>0.166667</td>
<td>0.023810</td>
<td>0.000000</td>
<td>0.000000</td>
</tr>
</tbody>
</table>

<p>1169 rows × 943 columns</p>
</div>
</div>
</div>
</section>
<section id="special-motifs" class="level2">
<h2 class="anchored" data-anchor-id="special-motifs">Special motifs</h2>
<section id="customized" class="level3">
<h3 class="anchored" data-anchor-id="customized">Customized</h3>
<div id="bfe90a8f-1c86-44ad-8d40-91f4d5b19331" class="cell" data-execution_count="94">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb50"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.pssm <span class="im">import</span> <span class="op">*</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="0318c184-5638-43ab-bb2d-dd119f645f6d" class="cell" data-execution_count="101">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb51"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>plot_logos_idx(pssms,<span class="dv">412</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="21_motif_plot_files/figure-html/cell-46-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="most-common" class="level3">
<h3 class="anchored" data-anchor-id="most-common">Most common</h3>
<div id="e99d5009-bf0d-453b-81a6-d627140d67ca" class="cell" data-execution_count="102">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb52"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>cluster_cnt <span class="op">=</span> df.cluster.value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="51d5ca6a-bce3-4250-8b15-9a8c01c9567d" class="cell" data-scrolled="true" data-execution_count="103">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb53"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>plot_logos(pssms.head(<span class="dv">10</span>),count_dict<span class="op">=</span>cluster_cnt)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="21_motif_plot_files/figure-html/cell-48-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="most-common-y" class="level3">
<h3 class="anchored" data-anchor-id="most-common-y">Most common y</h3>
<div id="25977dc0-67a3-49c0-a525-389d74e5afac" class="cell" data-execution_count="104">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb54"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>pssms_y <span class="op">=</span> pssms[pssms[<span class="st">'0pY'</span>]<span class="op">&gt;</span><span class="fl">0.3</span>].copy()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="f1afa0e9-7e52-4ea8-9ee8-88d5c10fedc3" class="cell" data-execution_count="105">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb55"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>idxs_y <span class="op">=</span> pssms_y.index</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="0cd962df-20e9-471e-86cd-eb56d0374ae4" class="cell" data-execution_count="107">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb56"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="bu">len</span>(idxs_y)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="107">
<pre><code>19</code></pre>
</div>
</div>
<div id="6ed748d7-41f8-4392-b5b0-eabb3967e1f8" class="cell" data-execution_count="108">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb58"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>plot_logos(pssms.loc[idxs_y],count_dict<span class="op">=</span>cluster_cnt)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="21_motif_plot_files/figure-html/cell-52-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="highest-ic-sum" class="level3">
<h3 class="anchored" data-anchor-id="highest-ic-sum">Highest IC sum</h3>
<div id="4364f301-8c8b-410f-972f-85f24aa5cf72" class="cell" data-execution_count="109">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb59"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>ICs <span class="op">=</span> pssms.<span class="bu">apply</span>(<span class="kw">lambda</span> r: <span class="bu">sum</span>(get_IC_flat(r)) ,axis<span class="op">=</span><span class="dv">1</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="4f0d225c-1b36-49af-a4b8-aec92dab6ee4" class="cell" data-execution_count="110">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb60"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>idxs<span class="op">=</span>ICs.sort_values(ascending<span class="op">=</span><span class="va">False</span>).head(<span class="dv">20</span>).index</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>The first one is mostly Zinc finger protein</p>
<div id="e968aed0-9056-4b57-b0b1-69ab9c77a6a7" class="cell" data-execution_count="111">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb61"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>plot_logos(pssms.loc[idxs],count_dict<span class="op">=</span>cluster_cnt)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="21_motif_plot_files/figure-html/cell-55-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="c-terminal-motifs" class="level3">
<h3 class="anchored" data-anchor-id="c-terminal-motifs">C-terminal motifs</h3>
<div id="ed5243a1-3223-4c7b-8faf-3e042ecb175f" class="cell" data-execution_count="112">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb62"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>zeros_right <span class="op">=</span> pssms.<span class="bu">apply</span>(<span class="kw">lambda</span> r: (recover_pssm(r).loc[:,<span class="dv">1</span>:].<span class="bu">sum</span>()<span class="op">==</span><span class="dv">0</span>).<span class="bu">sum</span>() , axis<span class="op">=</span><span class="dv">1</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="e07b92da-7111-481f-a51d-3bce276f5a5b" class="cell" data-execution_count="113">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb63"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>zeros_right_idxs <span class="op">=</span> zeros_right[zeros_right<span class="op">&gt;</span><span class="dv">0</span>].index</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="66c89b27-e763-408c-a8be-624a4fc1049d" class="cell" data-execution_count="114">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb64"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>zeros_right_idxs</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="114">
<pre><code>Index([1150, 1164, 1160, 1134, 1161, 1137, 1140, 1159, 1163, 1155, 1141, 1158,
       1136, 1165],
      dtype='int64', name='cluster_new')</code></pre>
</div>
</div>
<div id="82a86ec4-ed1e-47a1-a966-fff02242a8f0" class="cell" data-execution_count="115">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb66"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>plot_logos(pssms.loc[zeros_right_idxs],count_dict<span class="op">=</span>cluster_cnt)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="21_motif_plot_files/figure-html/cell-59-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="n-terminal-motifs" class="level3">
<h3 class="anchored" data-anchor-id="n-terminal-motifs">N-Terminal motifs:</h3>
<div id="1d2cd9bd-3716-44a2-bc10-c8688591d22b" class="cell" data-execution_count="116">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb67"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>zeros_left <span class="op">=</span> pssms.<span class="bu">apply</span>(<span class="kw">lambda</span> r: (recover_pssm(r).loc[:,:<span class="dv">0</span>].<span class="bu">sum</span>()<span class="op">==</span><span class="dv">0</span>).<span class="bu">sum</span>() , axis<span class="op">=</span><span class="dv">1</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="8c493ab4-de1b-416e-b089-83df7f46213d" class="cell" data-execution_count="117">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb68"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>zeros_left_idxs <span class="op">=</span> zeros_left[zeros_left<span class="op">&gt;</span><span class="dv">0</span>].index</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="1be8364b-46d7-4f5b-a417-6ffaeb213513" class="cell" data-execution_count="118">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb69"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>idxs <span class="op">=</span> zeros_left_idxs[:<span class="dv">10</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="ffe6c8dd-35e3-4aca-9b39-6d637736a747" class="cell" data-execution_count="119">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb70"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co"># zeros = pssms.apply(lambda r: (recover_pssm(r).sum()==0).sum() , axis=1)</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="co"># idxs = zeros.sort_values(ascending=False).head(10).index</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="df158f80-26b3-4f37-9935-7e9895284677" class="cell" data-execution_count="120">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb71"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>plot_logos(pssms.loc[idxs],count_dict<span class="op">=</span>cluster_cnt)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="21_motif_plot_files/figure-html/cell-64-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>


</section>
</section>

</main> <!-- /main -->
<script type="application/vnd.jupyter.widget-state+json">
{"state":{},"version_major":2,"version_minor":0}
</script>
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/sky1ove\.github\.io\/katlas-raw");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/sky1ove/katlas-raw/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></div></div></footer></body></html>