<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Plot heatmap and logo of CDDM – katlas-raw</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.min.js" integrity="sha384-ZvpUoO/+PpLXR1lu4jmpXWu80pZlYUAfxl5NsBMWOEPSjUn/6Z/hRTt8+pR6L4N2" crossorigin="anonymous"></script><script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-000f0825b9c55ed21d14970d810c14a9.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="https://cdn.jsdelivr.net/npm/requirejs@2.3.6/require.min.js" integrity="sha384-c9c+LnTbwQ3aujuU7ULEPVvgLs+Fn6fJUvIGTsuu1ZcCf11fiEubah0ttpca4ntM sha384-6V1/AdqZRWk1KAlWbKBlGhN7VG4iE/yAZcO6NZPMF8od0vukrvr0tg4qY6NSrItx" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


<link rel="stylesheet" href="styles.css">
<meta property="og:title" content="Plot heatmap and logo of CDDM – katlas-raw">
<meta property="og:description" content="Raw code to generate figures in the paper of KATLAS">
<meta property="og:image" content="https://sky1ove.github.io/katlas-raw/11_plot_CDDM_files/figure-html/cell-59-output-1.png">
<meta property="og:site_name" content="katlas-raw">
<meta property="og:image:height" content="1490">
<meta property="og:image:width" content="1019">
<meta name="twitter:title" content="Plot heatmap and logo of CDDM – katlas-raw">
<meta name="twitter:description" content="Raw code to generate figures in the paper of KATLAS">
<meta name="twitter:image" content="https://sky1ove.github.io/katlas-raw/11_plot_CDDM_files/figure-html/cell-59-output-1.png">
<meta name="twitter:image-height" content="1490">
<meta name="twitter:image-width" content="1019">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="./index.html" class="navbar-brand navbar-brand-logo">
    </a>
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">katlas-raw</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./plot_cddm.html">Plot heatmap and logo of CDDM</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      </a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">katlas-raw</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./preprocess_dataset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preprocess kinase-substrate datasets</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./eda_fig1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Exploratory data analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./site_promiscuity_fig1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Site promiscuity</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssm_hierarchical.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Hierarchical clustering of PSSM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05_PSSM_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">05_PSSM_example.html</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./modi_group.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Modi group</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scale_pspa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PSPA scaled data</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./background.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Background frequencies</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./specificity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Specificity comparison</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./sty_composition.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">STY composition</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./plot_cddm.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Plot heatmap and logo of CDDM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./plot_pspa_figure.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PSPA figures</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./13_CDDM_umap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CDDM umap</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./get_aa_relationship.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get AA relationship</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pspa_cddm_heatmap_compare.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Heatmap/ST/JSD/cosine comparison</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_no_psp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_psp_02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring_compare_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scoring evaluation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring_compare_psp_02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scoring evaluation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Reactome pathway for each PSSM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_plot_pspa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot reactome pathways: PSPA</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_plot_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot reactome pathways: CDDM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./motif_clustering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Kmeans motifs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./motif_plot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Motif plot</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./22_motif_web_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">22_motif_web_data.html</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./23_motif_kinase_similarity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cosine similarity with PSPA</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">kinase_domain</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_01_get_all_kd.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get all kinase domain sequences</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_02_pseudo_alignment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Alignment</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_03_feature_extract.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Extract KD features</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_04_prepare_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare data for training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_05_pspa_training_model_comparison_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Model comparison on T5 feature</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06a_pspa_training_feature_comparison_cnn_group.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on CNN model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06b_pspa_training_feature_comparison_cnn_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on CNN model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06c_pspa_training_feature_comparison_mlp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on MLP model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_07a_cddm_training_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_07b_pspa_training_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_08a_similarity_relationship.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KD similarity &amp; PSSM similarity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_08b_similarity_relationship_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KD similarity &amp; PSSM similarity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_09a_predict_pspa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Predict</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_09b_predict_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Predict</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_10_plot_umap_known_unk.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot UMAP of known &amp; unknown KD</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_11_plot_pspa_kd_t5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">kd_11_plot_pspa_kd_t5.html</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_12_compare_pred_psp_known.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Comopare the two</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup">Setup</a></li>
  <li><a href="#logo-heatmap" id="toc-logo-heatmap" class="nav-link" data-scroll-target="#logo-heatmap">logo heatmap</a></li>
  <li><a href="#iterate" id="toc-iterate" class="nav-link" data-scroll-target="#iterate">Iterate</a></li>
  <li><a href="#load-data" id="toc-load-data" class="nav-link" data-scroll-target="#load-data">Load data</a></li>
  <li><a href="#generate-example-figures" id="toc-generate-example-figures" class="nav-link" data-scroll-target="#generate-example-figures">Generate example figures</a></li>
  <li><a href="#generate-all-figures" id="toc-generate-all-figures" class="nav-link" data-scroll-target="#generate-all-figures">Generate all figures</a></li>
  <li><a href="#combine-figures-for-pdf" id="toc-combine-figures-for-pdf" class="nav-link" data-scroll-target="#combine-figures-for-pdf">Combine figures for pdf</a></li>
  <li><a href="#get-pssm-data-of-cddm" id="toc-get-pssm-data-of-cddm" class="nav-link" data-scroll-target="#get-pssm-data-of-cddm">Get PSSM data of CDDM</a></li>
  <li><a href="#get-specialized-cddm-data-for-all-capital-substrates" id="toc-get-specialized-cddm-data-for-all-capital-substrates" class="nav-link" data-scroll-target="#get-specialized-cddm-data-for-all-capital-substrates">Get specialized CDDM data for all-capital substrates</a></li>
  <li><a href="#plot-other-kinases-mutated-lipid-kinase-isoforms" id="toc-plot-other-kinases-mutated-lipid-kinase-isoforms" class="nav-link" data-scroll-target="#plot-other-kinases-mutated-lipid-kinase-isoforms">Plot other kinases (mutated, lipid kinase, isoforms)</a></li>
  <li><a href="#combine-the-figures-for-pdf" id="toc-combine-the-figures-for-pdf" class="nav-link" data-scroll-target="#combine-the-figures-for-pdf">Combine the figures for pdf</a></li>
  <li><a href="#get-the-pssms-of-other-kinases" id="toc-get-the-pssms-of-other-kinases" class="nav-link" data-scroll-target="#get-the-pssms-of-other-kinases">Get the PSSMs of other kinases</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/sky1ove/katlas-raw/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div><div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="plot_cddm.html.md"><i class="bi bi-file-code"></i>CommonMark</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Plot heatmap and logo of CDDM</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<!-- WARNING: THIS FILE WAS AUTOGENERATED! DO NOT EDIT! -->
<section id="setup" class="level2">
<h2 class="anchored" data-anchor-id="setup">Setup</h2>
<div id="d095f068-88c6-4e79-a2da-da9bf147f10d" class="cell" data-execution_count="1">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.data <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.pssm <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.utils <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.plot <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.feature <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib <span class="im">import</span> pyplot <span class="im">as</span> plt</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> math</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> sklearn.cluster <span class="im">import</span> KMeans</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pathlib <span class="im">import</span> Path</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> tqdm <span class="im">import</span> tqdm</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="f8def88b-9bce-46cc-a985-714b7dcaaebf" class="cell" data-execution_count="37">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> Data.get_ks_dataset()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>CPU times: user 737 ms, sys: 289 ms, total: 1.03 s
Wall time: 6.98 s</code></pre>
</div>
</div>
<div id="52fbc670-1568-457d-b3f9-5dcc8e50a5a6" class="cell" data-execution_count="38">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>df.head()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="38">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">kin_sub_site</th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">substrate_uniprot</th>
<th data-quarto-table-cell-role="th">site</th>
<th data-quarto-table-cell-role="th">source</th>
<th data-quarto-table-cell-role="th">substrate_genes</th>
<th data-quarto-table-cell-role="th">substrate_phosphoseq</th>
<th data-quarto-table-cell-role="th">position</th>
<th data-quarto-table-cell-role="th">site_seq</th>
<th data-quarto-table-cell-role="th">sub_site</th>
<th data-quarto-table-cell-role="th">...</th>
<th data-quarto-table-cell-role="th">kinase_on_tree</th>
<th data-quarto-table-cell-role="th">kinase_genes</th>
<th data-quarto-table-cell-role="th">kinase_protein</th>
<th data-quarto-table-cell-role="th">kinase_group</th>
<th data-quarto-table-cell-role="th">kinase_family</th>
<th data-quarto-table-cell-role="th">kinase_subfamily</th>
<th data-quarto-table-cell-role="th">kinase_pspa_big</th>
<th data-quarto-table-cell-role="th">kinase_pspa_small</th>
<th data-quarto-table-cell-role="th">kinase_coral_ID</th>
<th data-quarto-table-cell-role="th">num_kin</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">0</th>
<td>O00141_A4FU28_S140</td>
<td>O00141</td>
<td>A4FU28</td>
<td>S140</td>
<td>Sugiyama</td>
<td>CTAGE9</td>
<td>MEEPGATPQPYLGLVLEELGRVVAALPESMRPDENPYGFPSELVVC...</td>
<td>140</td>
<td>AAAEEARSLEATCEKLSRsNsELEDEILCLEKDLKEEKSKH</td>
<td>A4FU28_S140</td>
<td>...</td>
<td>1</td>
<td>SGK1 SGK</td>
<td>SGK1</td>
<td>AGC</td>
<td>SGK</td>
<td>SGK</td>
<td>Basophilic</td>
<td>Akt/rock</td>
<td>SGK1</td>
<td>22</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1</th>
<td>O00141_O00141_S252</td>
<td>O00141</td>
<td>O00141</td>
<td>S252</td>
<td>Sugiyama</td>
<td>SGK1 SGK</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
<td>252</td>
<td>SQGHIVLTDFGLCKENIEHNsTtstFCGtPEyLAPEVLHKQ</td>
<td>O00141_S252</td>
<td>...</td>
<td>1</td>
<td>SGK1 SGK</td>
<td>SGK1</td>
<td>AGC</td>
<td>SGK</td>
<td>SGK</td>
<td>Basophilic</td>
<td>Akt/rock</td>
<td>SGK1</td>
<td>1</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">2</th>
<td>O00141_O00141_S255</td>
<td>O00141</td>
<td>O00141</td>
<td>S255</td>
<td>Sugiyama</td>
<td>SGK1 SGK</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
<td>255</td>
<td>HIVLTDFGLCKENIEHNsTtstFCGtPEyLAPEVLHKQPYD</td>
<td>O00141_S255</td>
<td>...</td>
<td>1</td>
<td>SGK1 SGK</td>
<td>SGK1</td>
<td>AGC</td>
<td>SGK</td>
<td>SGK</td>
<td>Basophilic</td>
<td>Akt/rock</td>
<td>SGK1</td>
<td>1</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">3</th>
<td>O00141_O00141_S397</td>
<td>O00141</td>
<td>O00141</td>
<td>S397</td>
<td>Sugiyama</td>
<td>SGK1 SGK</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
<td>397</td>
<td>sGPNDLRHFDPEFTEEPVPNsIGKsPDsVLVTAsVKEAAEA</td>
<td>O00141_S397</td>
<td>...</td>
<td>1</td>
<td>SGK1 SGK</td>
<td>SGK1</td>
<td>AGC</td>
<td>SGK</td>
<td>SGK</td>
<td>Basophilic</td>
<td>Akt/rock</td>
<td>SGK1</td>
<td>1</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">4</th>
<td>O00141_O00141_S404</td>
<td>O00141</td>
<td>O00141</td>
<td>S404</td>
<td>Sugiyama</td>
<td>SGK1 SGK</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
<td>404</td>
<td>HFDPEFTEEPVPNsIGKsPDsVLVTAsVKEAAEAFLGFsYA</td>
<td>O00141_S404</td>
<td>...</td>
<td>1</td>
<td>SGK1 SGK</td>
<td>SGK1</td>
<td>AGC</td>
<td>SGK</td>
<td>SGK</td>
<td>Basophilic</td>
<td>Akt/rock</td>
<td>SGK1</td>
<td>1</td>
</tr>
</tbody>
</table>

<p>5 rows × 21 columns</p>
</div>
</div>
</div>
<div id="92b03ce2-9fd7-470d-b044-ebb4d8131702" class="cell" data-execution_count="39">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>df.shape</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="39">
<pre><code>(187066, 21)</code></pre>
</div>
</div>
</section>
<section id="logo-heatmap" class="level2">
<h2 class="anchored" data-anchor-id="logo-heatmap">logo heatmap</h2>
<div id="fe8f1776-0fd8-4a26-a50e-e96efec07493" class="cell" data-execution_count="40">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>df[<span class="st">'kinase_uniprot_gene'</span>] <span class="op">=</span> df[<span class="st">'kinase_uniprot'</span>]<span class="op">+</span><span class="st">'_'</span><span class="op">+</span>df[<span class="st">'kinase_genes'</span>].<span class="bu">str</span>.split(<span class="st">' '</span>).<span class="bu">str</span>[<span class="dv">0</span>]</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>cnt <span class="op">=</span> df.kinase_uniprot_gene.value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="82b154cb-e575-4320-acf9-4d35c4ed98a0" class="cell" data-execution_count="41">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># cnt = df.kinase_coral_ID.value_counts()</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="fc74c2d5-f6d4-4ed0-98b8-e2278ef31acd" class="cell" data-execution_count="42">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> convert_source(x):</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> x <span class="op">==</span> <span class="st">"Sugiyama"</span>:</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>        <span class="cf">return</span> x</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> <span class="st">'Sugiyama'</span> <span class="kw">in</span> x <span class="kw">and</span> <span class="st">'|'</span> <span class="kw">in</span> x:</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>        <span class="cf">return</span> <span class="st">'Both'</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> <span class="st">'Sugiyama'</span> <span class="kw">not</span> <span class="kw">in</span> x:</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>        <span class="cf">return</span> <span class="st">'Non-Sugiyama'</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="53a48a7c-90d8-45e5-8315-8ae6533d3940" class="cell" data-execution_count="43">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>df[<span class="st">'source_combine'</span>] <span class="op">=</span> df.source.<span class="bu">apply</span>(convert_source)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="ec0fe53d-da1a-4305-9e09-c06048554231" class="cell" data-execution_count="44">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_hist_num_kin(df_k):</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>    <span class="co">"Plot histogram of num kin grouped by source_combine."</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>    g <span class="op">=</span> sns.displot(</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>        df_k,</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>        x<span class="op">=</span><span class="st">"num_kin"</span>,</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>        col<span class="op">=</span><span class="st">"source_combine"</span>,</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>        bins<span class="op">=</span><span class="dv">100</span>,</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>        col_wrap<span class="op">=</span><span class="dv">1</span>,</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>        height<span class="op">=</span><span class="fl">2.0</span>,</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>        aspect<span class="op">=</span><span class="dv">4</span>,</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>        facet_kws<span class="op">=</span>{<span class="st">'sharex'</span>: <span class="va">False</span>, <span class="st">'sharey'</span>: <span class="va">False</span>}</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>    g.set_axis_labels(<span class="st">"Number of Kinases per Site"</span>, <span class="st">"Count"</span>)</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Customize titles</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> ax, source <span class="kw">in</span> <span class="bu">zip</span>(g.axes.flatten(), g.col_names):</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>        count <span class="op">=</span> df_k[df_k[<span class="st">'source_combine'</span>] <span class="op">==</span> source].shape[<span class="dv">0</span>]</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>        ax.set_title(<span class="ss">f"</span><span class="sc">{</span>source<span class="sc">}</span><span class="ss"> (n=</span><span class="sc">{</span>count<span class="sc">:,}</span><span class="ss">)"</span>)</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>    g.figure.suptitle(<span class="st">"Histogram of # Kinases per Substrate Site"</span>)</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Adjust layout to make room for suptitle</span></span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>    plt.tight_layout()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="2e6ba4f0-f074-4127-bacf-eb1dc8e328c9" class="cell" data-execution_count="45">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_cnt_cddm(df_k):</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>    <span class="co">"Plot source combine counts via bar graph."</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>    source_cnt <span class="op">=</span> df_k.source_combine.value_counts()</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    plot_cnt(source_cnt)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="st">'# Substrate Sites per Source'</span>,pad<span class="op">=</span><span class="dv">20</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="f9b1c775-4b24-4fea-8696-c48345cf080c" class="cell" data-execution_count="46">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_cnt_cddm(df_k)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="c57ea215-0bc5-4064-8952-0cb3bbcaedd8" class="cell" data-execution_count="47">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_cnt_acceptor(df_k):</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>    <span class="co">"Plot site type via bar graph."</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>    acceptor_cnt <span class="op">=</span> df_k.acceptor.value_counts()</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>    plot_cnt(acceptor_cnt)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="st">'# Substrate Sites per Phospho-Acceptor Type'</span>,pad<span class="op">=</span><span class="dv">20</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="1cac1b7d-e389-4610-943c-fc78dc5bef4c" class="cell" data-execution_count="48">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_cnt_acceptor(df_k)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="d42005a7-3b94-4db8-a1c8-3c48420c6307" class="cell" data-execution_count="49">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set_sns()</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="cc3eb328-bfad-42a0-a006-966ea47d22f7" class="cell" data-execution_count="50">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># onehot = onehot_encode(df_k.site_seq)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="cb736ecf-ded3-4eb6-bd40-2bb23710a8fb" class="cell" data-execution_count="51">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> filter_range_columns(df,low<span class="op">=-</span><span class="dv">10</span>,high<span class="op">=</span><span class="dv">10</span>):</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>    positions <span class="op">=</span> df.columns.<span class="bu">str</span>[:<span class="op">-</span><span class="dv">1</span>].astype(<span class="bu">int</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>    mask <span class="op">=</span> (positions <span class="op">&gt;=</span> low) <span class="op">&amp;</span> (positions <span class="op">&lt;=</span> high)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> df.loc[:,mask]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="ba0e5843-2e27-4678-aa0e-25098557a946" class="cell" data-execution_count="52">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># positions = onehot.columns.str[:-1].astype(int)</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="co"># mask = (positions &gt;= -10) &amp; (positions &lt;= 10)</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="co"># onehot = onehot.loc[:,mask]</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="co"># df_k['cluster'] = kmeans(onehot,n=10,seed=42)</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="co"># pssms = get_cluster_pssms(df_k,'cluster',valid_thr=0.5)</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="co"># kmeans_cnt = df_k.cluster.value_counts()</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="9b8a415a-c9d5-4005-9810-0388c1ec8517" class="cell" data-execution_count="53">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_logos(pssms,kmeans_cnt)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="16257211-efc1-4d34-a8cf-3dd461117e08" class="cell" data-execution_count="54">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> get_onehot_add_cluster(df_k,n<span class="op">=</span><span class="dv">10</span>):</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>    df_k <span class="op">=</span> df_k.copy()</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>    onehot <span class="op">=</span> onehot_encode(df_k.site_seq)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>    onehot_10 <span class="op">=</span> filter_range_columns(onehot)</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>    df_k[<span class="st">'Cluster'</span>] <span class="op">=</span> kmeans(onehot_10,n<span class="op">=</span>n,seed<span class="op">=</span><span class="dv">42</span>)</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>    df_k <span class="op">=</span> df_k.reset_index(drop<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> df_k,onehot_10</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="6eba6443-7a4e-41df-83fe-65ea80aeae44" class="cell" data-execution_count="55">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># df_k,onehot_10 = get_onehot_add_cluster(df_k,n=10)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="bd487c28-7de0-471b-83da-9b0eaa531abf" class="cell" data-execution_count="56">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> get_kmeans_logos(df_k,cnt_thr<span class="op">=</span><span class="dv">10</span>):</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>    pssms <span class="op">=</span> get_cluster_pssms(df_k,<span class="st">'Cluster'</span>,valid_thr<span class="op">=</span><span class="dv">0</span>) <span class="co"># remove valid thr</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>    kmeans_cnt <span class="op">=</span> df_k[<span class="st">'Cluster'</span>].value_counts()</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># filter cluster with &gt;=10 counts</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>    valid_clusters <span class="op">=</span> kmeans_cnt[kmeans_cnt <span class="op">&gt;=</span> cnt_thr].index</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>    filtered_pssms <span class="op">=</span> pssms.loc[valid_clusters]</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">not</span> filtered_pssms.empty: plot_logos(filtered_pssms,kmeans_cnt)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="a8706655-13fd-475f-b101-34af68ce8409" class="cell" data-execution_count="57">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># df_k,onehot_10 = get_onehot_add_cluster(df_k,n=10)</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="co"># get_kmeans_logos(df_k)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="a80988ba-ffc2-4159-acce-4bd21e899d61" class="cell" data-execution_count="58">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_onehot(onehot_10,hue):</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>    plot_cluster(onehot_10,<span class="st">'pca'</span>,seed<span class="op">=</span><span class="dv">42</span>,complexity<span class="op">=</span><span class="dv">30</span>,hue<span class="op">=</span>hue,legend<span class="op">=</span><span class="va">True</span>,s<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'PCA of One-Hot Encoded</span><span class="ch">\n</span><span class="ss"> Substrate Site Sequences (n=</span><span class="sc">{</span><span class="bu">len</span>(onehot_10)<span class="sc">:,}</span><span class="ss">)'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="942aafb3-4e2c-42d1-899b-2347c88d0e26" class="cell" data-execution_count="59">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># df_k,onehot_10 = get_onehot_add_cluster(df_k,n=10)</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_onehot(onehot_10,df_k['Cluster'])</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="co"># get_kmeans_logos(df_k)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="307bd323-6e8a-496b-9d37-28164e6d3f83" class="cell" data-execution_count="60">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_cluster(onehot_10,'pca',seed=42,complexity=30,hue=df_k.cluster,legend=True,s=8)</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="co"># plt.title('PCA of One-Hot Encoded Sequences (-10 to +10')</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="e4855bcb-0330-4630-931f-f6ae644d0b79" class="cell" data-execution_count="61">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb28"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_logo_heatmap(pssm_df,title=f'{k} (n={len(df_k):,})',figsize=(17,10))</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="3571117c-e8aa-4b85-a4bd-90421ff51496" class="cell" data-execution_count="62">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># for site_type in ['S','T','Y']:</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     df_sty = df[df.kinase_uniprot_gene.upper()==site_type].copy()</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     pssm_sty = get_prob(df_sty,'site_seq')</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     plot_logo_heatmap(pssm_sty,title=f'{site_type} sites (n={len(df_sty):,})',figsize=(17,10))</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="co">#     save_show()</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="a47a8286-6445-4a77-8962-13f9542cf43d" class="cell" data-execution_count="63">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb30"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># pssm_LO = get_pssm_LO(pssm_sty,'S')</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="ecdfdac4-2fd3-4dd4-a4e5-56e135e9fbe2" class="cell" data-execution_count="64">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_logo_raw(pssm_LO,ytitle="Log-Odds Score (bits)")</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="5112b022-5fb1-4d20-a4ba-98b7700d50b6" class="cell" data-execution_count="65">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_logo_heatmap_enrich(pssm_LO)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="b10ee0cb-056d-44b0-ab2c-c6bd8799a08e" class="cell" data-execution_count="66">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># import logomaker</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="68f83861-f05e-4295-963f-6d59bcd6bca9" class="cell" data-execution_count="67">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># for site_type in ['S','T','Y']:</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     df_sty = df_k[df_k.acceptor.str.upper()==site_type].copy()</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     pssm_sty = get_prob(df_sty,'site_seq')</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     plot_logo_heatmap(pssm_sty,title=f'{site_type} sites (n={len(df_sty):,})',figsize=(17,10))</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="co">#     save_show()</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="co">#     break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="iterate" class="level2">
<h2 class="anchored" data-anchor-id="iterate">Iterate</h2>
<div id="89a80b54-83a8-4992-a666-371a83816457" class="cell" data-execution_count="68">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb35"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>df[<span class="st">'acceptor'</span>]<span class="op">=</span>df.site.<span class="bu">str</span>[<span class="dv">0</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="b5f8d05b-b64c-4263-b336-cae1ca66d4b2" class="cell" data-execution_count="69">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>SHOW<span class="op">=</span><span class="va">False</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="b9d24eb9-7e21-4e98-9870-e9091d993847" class="cell" data-execution_count="70">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb37"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>filter_cnt <span class="op">=</span> cnt[cnt<span class="op">&gt;=</span><span class="dv">40</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="030962b5-9a1a-4ba4-9a30-788430f2a38d" class="cell" data-execution_count="71">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb38"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="bu">list</span>(filter_cnt.index).index(<span class="st">'O43353_RIPK2'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="71">
<pre><code>301</code></pre>
</div>
</div>
<div id="d148af66-8549-4dfa-9dec-c16ece39ca8e" class="cell" data-execution_count="78">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb40"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> k <span class="kw">in</span> filter_cnt.index[<span class="dv">235</span>:]: <span class="cf">break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="4e167d53-ee50-427e-a3d0-e9bf0a82d300" class="cell" data-execution_count="79">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb41"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>k</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="79">
<pre><code>'P41743_PRKCI'</code></pre>
</div>
</div>
<div id="3ba678a2-01a4-46a0-b76a-9e1c86476cf0" class="cell" data-execution_count="76">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb43"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> prepare_path(path):</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>    <span class="co">"""Ensure the parent directory exists and return the full file path."""</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>    full_path <span class="op">=</span> Path(path).expanduser()</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>    full_path.parent.mkdir(parents<span class="op">=</span><span class="va">True</span>, exist_ok<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> full_path</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="209fcf83-a2e4-473e-a759-976379697e1f" class="cell" data-execution_count="81">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb44"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> save_show(path<span class="op">=</span><span class="va">None</span>, <span class="co"># image path, e.g., img.svg, if not None, will save, else plt.show()</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>              show_only<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>             ):</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>    <span class="co">"Show plot or save path"</span></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> show_only: plt.show()</span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> path <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>: </span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>        plt.savefig(path,</span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>            bbox_inches<span class="op">=</span><span class="st">'tight'</span>,</span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>            pad_inches<span class="op">=</span><span class="fl">0.05</span>,</span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>            transparent<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>            )  </span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>: plt.show()</span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a>    plt.close()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="890caa34-f72c-43ca-ba5c-7c182670a298" class="cell" data-execution_count="82">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb45"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib <span class="im">as</span> mpl</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib <span class="im">as</span> mpl</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>mpl.rcParams[<span class="st">'svg.fonttype'</span>] <span class="op">=</span> <span class="st">'none'</span>     <span class="co"># keep text as &lt;text&gt;, not paths</span></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>mpl.rcParams[<span class="st">'svg.image_inline'</span>] <span class="op">=</span> <span class="va">True</span>   <span class="co"># embed images inline (set False to link)</span></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>mpl.rcParams[<span class="st">'svg.hashsalt'</span>] <span class="op">=</span> <span class="st">''</span>         <span class="co"># avoid random IDs in defs</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>mpl.rcParams[<span class="st">'path.simplify'</span>] <span class="op">=</span> <span class="va">True</span></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>mpl.rcParams[<span class="st">'path.simplify_threshold'</span>] <span class="op">=</span> <span class="fl">0.01</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="ae244fe2-19d7-4738-bcec-b4a0d7a090cd" class="cell" data-execution_count="83">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb46"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> gc</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="d53394f4-e2e8-4cde-8ad8-c4d507f407af" class="cell" data-execution_count="84">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb47"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>filter_cnt <span class="op">=</span> cnt[cnt<span class="op">&gt;=</span><span class="dv">40</span>]</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i,k <span class="kw">in</span> <span class="bu">enumerate</span>(filter_cnt.index[<span class="dv">235</span>:]): </span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="co"># for k in ['P00519_ABL1']:</span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>    df_k <span class="op">=</span> df[df.kinase_uniprot_gene<span class="op">==</span>k].copy()</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># df_k = df_k[df_k.num_kin&gt;10]</span></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Freq PSSM</span></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>    pssm_df <span class="op">=</span> get_prob(df_k,<span class="st">'site_seq'</span>)</span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>    plot_logo_heatmap(pssm_df,title<span class="op">=</span><span class="ss">f'</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss"> (n=</span><span class="sc">{</span><span class="bu">len</span>(df_k)<span class="sc">:,}</span><span class="ss">)'</span>,figsize<span class="op">=</span>(<span class="dv">17</span>,<span class="dv">10</span>))</span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>    path1<span class="op">=</span>prepare_path(<span class="ss">f'fig/cddm/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/pssm_freq.svg'</span>)</span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>    save_show(path1,show_only<span class="op">=</span>SHOW)</span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Log odds PSSM</span></span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a>    pssm_LO <span class="op">=</span> get_pssm_LO(pssm_df,<span class="st">'STY'</span>)</span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a>    plot_logo_heatmap_LO(pssm_LO,title<span class="op">=</span><span class="ss">f'</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss"> (n=</span><span class="sc">{</span><span class="bu">len</span>(df_k)<span class="sc">:,}</span><span class="ss">)'</span>,figsize<span class="op">=</span>(<span class="dv">17</span>,<span class="dv">10</span>))</span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a>    path2<span class="op">=</span>prepare_path(<span class="ss">f'fig/cddm/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/pssm_LO.svg'</span>)</span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a>    save_show(path2,show_only<span class="op">=</span>SHOW)</span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plot S, T and Y motif</span></span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a>    sty_cnt <span class="op">=</span>df_k.acceptor.value_counts()</span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a>    acceptors <span class="op">=</span> sty_cnt[(sty_cnt<span class="op">/</span><span class="bu">len</span>(df_k) <span class="op">&gt;</span> <span class="fl">0.08</span>) <span class="op">&amp;</span> (sty_cnt<span class="op">&gt;=</span><span class="dv">10</span>)].index <span class="co"># Skip this site_type if it has less than 8% or less than 10 count</span></span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> site_type <span class="kw">in</span> acceptors:</span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a>        df_sty <span class="op">=</span> df_k[df_k.acceptor.<span class="bu">str</span>.upper()<span class="op">==</span>site_type].copy()</span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a>        <span class="co"># freq map</span></span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a>        pssm_sty <span class="op">=</span> get_prob(df_sty,<span class="st">'site_seq'</span>)</span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a>        plot_logo_heatmap(pssm_sty,title<span class="op">=</span><span class="ss">f'</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">: </span><span class="sc">{</span>site_type<span class="sc">}</span><span class="ss"> sites (n=</span><span class="sc">{</span><span class="bu">len</span>(df_sty)<span class="sc">:,}</span><span class="ss">)'</span>,figsize<span class="op">=</span>(<span class="dv">17</span>,<span class="dv">10</span>))</span>
<span id="cb47-30"><a href="#cb47-30" aria-hidden="true" tabindex="-1"></a>        path_3<span class="op">=</span>prepare_path(<span class="ss">f'fig/cddm/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/pssm_freq_</span><span class="sc">{</span>site_type<span class="sc">}</span><span class="ss">.svg'</span>)</span>
<span id="cb47-31"><a href="#cb47-31" aria-hidden="true" tabindex="-1"></a>        save_show(path_3,show_only<span class="op">=</span>SHOW)</span>
<span id="cb47-32"><a href="#cb47-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-33"><a href="#cb47-33" aria-hidden="true" tabindex="-1"></a>        <span class="co"># for log-odds</span></span>
<span id="cb47-34"><a href="#cb47-34" aria-hidden="true" tabindex="-1"></a>        pssm_LO <span class="op">=</span> get_pssm_LO(pssm_sty,site_type)</span>
<span id="cb47-35"><a href="#cb47-35" aria-hidden="true" tabindex="-1"></a>        plot_logo_heatmap_LO(pssm_LO,acceptor<span class="op">=</span>site_type,title<span class="op">=</span><span class="ss">f'</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">: </span><span class="sc">{</span>site_type<span class="sc">}</span><span class="ss"> sites (n=</span><span class="sc">{</span><span class="bu">len</span>(df_sty)<span class="sc">:,}</span><span class="ss">)'</span>,figsize<span class="op">=</span>(<span class="dv">17</span>,<span class="dv">10</span>))</span>
<span id="cb47-36"><a href="#cb47-36" aria-hidden="true" tabindex="-1"></a>        path_4<span class="op">=</span>prepare_path(<span class="ss">f'fig/cddm/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/pssm_LO_</span><span class="sc">{</span>site_type<span class="sc">}</span><span class="ss">.svg'</span>)</span>
<span id="cb47-37"><a href="#cb47-37" aria-hidden="true" tabindex="-1"></a>        save_show(path_4,show_only<span class="op">=</span>SHOW)</span>
<span id="cb47-38"><a href="#cb47-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-39"><a href="#cb47-39" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb47-40"><a href="#cb47-40" aria-hidden="true" tabindex="-1"></a>    path5<span class="op">=</span>prepare_path(<span class="ss">f'fig/cddm/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/bar_acceptor.svg'</span>)</span>
<span id="cb47-41"><a href="#cb47-41" aria-hidden="true" tabindex="-1"></a>    plot_cnt_acceptor(df_k)</span>
<span id="cb47-42"><a href="#cb47-42" aria-hidden="true" tabindex="-1"></a>    save_show(path5,show_only<span class="op">=</span>SHOW)</span>
<span id="cb47-43"><a href="#cb47-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-44"><a href="#cb47-44" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb47-45"><a href="#cb47-45" aria-hidden="true" tabindex="-1"></a>    <span class="co"># count of source</span></span>
<span id="cb47-46"><a href="#cb47-46" aria-hidden="true" tabindex="-1"></a>    path6<span class="op">=</span>prepare_path(<span class="ss">f'fig/cddm/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/bar_source.svg'</span>)</span>
<span id="cb47-47"><a href="#cb47-47" aria-hidden="true" tabindex="-1"></a>    plot_cnt_cddm(df_k)</span>
<span id="cb47-48"><a href="#cb47-48" aria-hidden="true" tabindex="-1"></a>    save_show(path6,show_only<span class="op">=</span>SHOW)</span>
<span id="cb47-49"><a href="#cb47-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-50"><a href="#cb47-50" aria-hidden="true" tabindex="-1"></a>    <span class="co"># histogram of num kin</span></span>
<span id="cb47-51"><a href="#cb47-51" aria-hidden="true" tabindex="-1"></a>    path7<span class="op">=</span>prepare_path(<span class="ss">f'fig/cddm/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/hist_num_kin.svg'</span>)</span>
<span id="cb47-52"><a href="#cb47-52" aria-hidden="true" tabindex="-1"></a>    plot_hist_num_kin(df_k)</span>
<span id="cb47-53"><a href="#cb47-53" aria-hidden="true" tabindex="-1"></a>    save_show(path7,show_only<span class="op">=</span>SHOW)</span>
<span id="cb47-54"><a href="#cb47-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-55"><a href="#cb47-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-56"><a href="#cb47-56" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb47-57"><a href="#cb47-57" aria-hidden="true" tabindex="-1"></a>    <span class="co"># onehot of sequences</span></span>
<span id="cb47-58"><a href="#cb47-58" aria-hidden="true" tabindex="-1"></a>    <span class="co"># filter out noise acceptor for stratification</span></span>
<span id="cb47-59"><a href="#cb47-59" aria-hidden="true" tabindex="-1"></a>    df_k <span class="op">=</span> df_k[df_k.acceptor.isin(acceptors)]</span>
<span id="cb47-60"><a href="#cb47-60" aria-hidden="true" tabindex="-1"></a>    df_k,onehot_10 <span class="op">=</span> get_onehot_add_cluster(df_k,n<span class="op">=</span><span class="dv">10</span>)</span>
<span id="cb47-61"><a href="#cb47-61" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb47-62"><a href="#cb47-62" aria-hidden="true" tabindex="-1"></a>    path8<span class="op">=</span>prepare_path(<span class="ss">f'fig/cddm/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/pca_onehot.svg'</span>)</span>
<span id="cb47-63"><a href="#cb47-63" aria-hidden="true" tabindex="-1"></a>    plot_onehot(onehot_10,df_k.Cluster)</span>
<span id="cb47-64"><a href="#cb47-64" aria-hidden="true" tabindex="-1"></a>    save_show(path8,show_only<span class="op">=</span>SHOW)</span>
<span id="cb47-65"><a href="#cb47-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-66"><a href="#cb47-66" aria-hidden="true" tabindex="-1"></a>    path9<span class="op">=</span>prepare_path(<span class="ss">f'fig/cddm/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/motif_kmeans.svg'</span>)</span>
<span id="cb47-67"><a href="#cb47-67" aria-hidden="true" tabindex="-1"></a>    get_kmeans_logos(df_k)</span>
<span id="cb47-68"><a href="#cb47-68" aria-hidden="true" tabindex="-1"></a>    save_show(path9,show_only<span class="op">=</span>SHOW)</span>
<span id="cb47-69"><a href="#cb47-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-70"><a href="#cb47-70" aria-hidden="true" tabindex="-1"></a>    path00<span class="op">=</span>prepare_path(<span class="ss">f'fig/cddm/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/df_k.parquet'</span>)</span>
<span id="cb47-71"><a href="#cb47-71" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb47-72"><a href="#cb47-72" aria-hidden="true" tabindex="-1"></a>    df_k.to_parquet(path00)</span>
<span id="cb47-73"><a href="#cb47-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-74"><a href="#cb47-74" aria-hidden="true" tabindex="-1"></a>    plt.close()</span>
<span id="cb47-75"><a href="#cb47-75" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> i <span class="op">%</span> <span class="dv">10</span> <span class="op">==</span> <span class="dv">0</span>:</span>
<span id="cb47-76"><a href="#cb47-76" aria-hidden="true" tabindex="-1"></a>        gc.collect()</span>
<span id="cb47-77"><a href="#cb47-77" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb47-78"><a href="#cb47-78" aria-hidden="true" tabindex="-1"></a>    <span class="co"># break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 73.75it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 146.48it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 92.02it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 144.13it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 106.27it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 84.45it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 116.55it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 56.13it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 83.22it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 80.70it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 82.79it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 218.11it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 76.83it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 95.90it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 70.55it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 67.97it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 77.00it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 98.59it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 90.30it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 92.44it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 70.34it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 116.14it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 70.21it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 77.78it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 87.13it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 151.65it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 111.28it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 220.84it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 76.75it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 72.21it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 72.33it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 123.73it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 93.41it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 71.76it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 78.07it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 129.75it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 79.42it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 76.79it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 129.33it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 128.24it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 132.44it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 109.60it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 133.85it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 84.67it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 102.86it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 124.24it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 224.12it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 145.26it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 90.85it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 210.07it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 162.78it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 111.04it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 133.06it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 162.31it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 232.12it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 182.80it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 228.90it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 345.97it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 138.72it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 134.44it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 162.07it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 220.88it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 297.18it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 217.93it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 330.48it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 330.39it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 227.04it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 593.80it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 205.43it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 237.42it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 235.05it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 328.17it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 606.48it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 332.83it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 330.93it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 234.44it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 226.36it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 632.95it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 609.24it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 641.18it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 168445.94it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 609.52it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 649.26it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 662.40it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 651.09it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 487.47it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 143640.55it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 103819.41it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 576.35it/s]
100%|███████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 41651.48it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 576.70it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 130257.89it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 133576.56it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 107271.20it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 326.52it/s]
100%|███████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 79588.31it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 132731.14it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 144134.16it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 104077.02it/s]
100%|███████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 93414.34it/s]</code></pre>
</div>
</div>
<div id="fff5d495-243d-436c-9167-f8f964c97cfe" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb49"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>genes <span class="op">=</span> <span class="bu">set</span>(df_k.substrate_genes.<span class="bu">str</span>.split(<span class="st">' '</span>).<span class="bu">str</span>[<span class="dv">0</span>])</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="84782af0-8816-4fb3-ad01-d01db26fa3ef" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb50"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>path_ref <span class="op">=</span> pd.read_excel(<span class="st">'raw/idmapping_kinase_info_2025_05_27.xlsx'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="0c917041-ab42-4252-8b8b-7639343c059a" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb51"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>k</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>'P00519_ABL1'</code></pre>
</div>
</div>
<div id="774a3935-fb78-4603-a9d5-efa9874feadc" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb53"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>path_ref[<span class="st">'uniprot_gene'</span>] <span class="op">=</span> path_ref.uniprot<span class="op">+</span><span class="st">'_'</span><span class="op">+</span>path_ref[<span class="st">'Gene Names (primary)'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="951ffcb6-2cf4-411e-8a7e-edd7e36d50de" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb54"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>idx_path <span class="op">=</span> path_ref[path_ref.uniprot_gene<span class="op">==</span>k].Reactome.<span class="bu">str</span>.split(<span class="st">';'</span>).iloc[<span class="dv">0</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="b01d6f93-c335-4e56-b2e7-417df7708be8" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb55"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>path_df_raw</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">stId</th>
<th data-quarto-table-cell-role="th">dbId</th>
<th data-quarto-table-cell-role="th">name</th>
<th data-quarto-table-cell-role="th">llp</th>
<th data-quarto-table-cell-role="th">inDisease</th>
<th data-quarto-table-cell-role="th">species.dbId</th>
<th data-quarto-table-cell-role="th">species.taxId</th>
<th data-quarto-table-cell-role="th">species.name</th>
<th data-quarto-table-cell-role="th">entities.resource</th>
<th data-quarto-table-cell-role="th">entities.total</th>
<th data-quarto-table-cell-role="th">entities.found</th>
<th data-quarto-table-cell-role="th">entities.ratio</th>
<th data-quarto-table-cell-role="th">entities.pValue</th>
<th data-quarto-table-cell-role="th">entities.fdr</th>
<th data-quarto-table-cell-role="th">entities.exp</th>
<th data-quarto-table-cell-role="th">reactions.resource</th>
<th data-quarto-table-cell-role="th">reactions.total</th>
<th data-quarto-table-cell-role="th">reactions.found</th>
<th data-quarto-table-cell-role="th">reactions.ratio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">0</th>
<td>R-HSA-72649</td>
<td>72649</td>
<td>Translation initiation complex formation</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>62</td>
<td>39</td>
<td>0.003861</td>
<td>1.110223e-16</td>
<td>4.884981e-15</td>
<td>[]</td>
<td>TOTAL</td>
<td>2</td>
<td>2</td>
<td>0.000129</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1</th>
<td>R-HSA-72695</td>
<td>72695</td>
<td>Formation of the ternary complex, and subseque...</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>54</td>
<td>33</td>
<td>0.003362</td>
<td>1.110223e-16</td>
<td>4.884981e-15</td>
<td>[]</td>
<td>TOTAL</td>
<td>3</td>
<td>3</td>
<td>0.000193</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">2</th>
<td>R-HSA-72702</td>
<td>72702</td>
<td>Ribosomal scanning and start codon recognition</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>64</td>
<td>39</td>
<td>0.003985</td>
<td>1.110223e-16</td>
<td>4.884981e-15</td>
<td>[]</td>
<td>TOTAL</td>
<td>2</td>
<td>2</td>
<td>0.000129</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">3</th>
<td>R-HSA-72662</td>
<td>72662</td>
<td>Activation of the mRNA upon binding of the cap...</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>66</td>
<td>39</td>
<td>0.004110</td>
<td>1.110223e-16</td>
<td>4.884981e-15</td>
<td>[]</td>
<td>TOTAL</td>
<td>6</td>
<td>6</td>
<td>0.000386</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">4</th>
<td>R-HSA-72706</td>
<td>72706</td>
<td>GTP hydrolysis and joining of the 60S ribosoma...</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>120</td>
<td>66</td>
<td>0.007472</td>
<td>1.110223e-16</td>
<td>4.884981e-15</td>
<td>[]</td>
<td>TOTAL</td>
<td>3</td>
<td>3</td>
<td>0.000193</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">1754</th>
<td>R-HSA-198933</td>
<td>198933</td>
<td>Immunoregulatory interactions between a Lympho...</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>249</td>
<td>1</td>
<td>0.015504</td>
<td>1.000000e+00</td>
<td>1.000000e+00</td>
<td>[]</td>
<td>TOTAL</td>
<td>44</td>
<td>1</td>
<td>0.002829</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1755</th>
<td>R-HSA-373076</td>
<td>373076</td>
<td>Class A/1 (Rhodopsin-like receptors)</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>415</td>
<td>2</td>
<td>0.025841</td>
<td>1.000000e+00</td>
<td>1.000000e+00</td>
<td>[]</td>
<td>TOTAL</td>
<td>187</td>
<td>3</td>
<td>0.012023</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">1756</th>
<td>R-HSA-425407</td>
<td>425407</td>
<td>SLC-mediated transmembrane transport</td>
<td>False</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>424</td>
<td>2</td>
<td>0.026401</td>
<td>1.000000e+00</td>
<td>1.000000e+00</td>
<td>[]</td>
<td>TOTAL</td>
<td>200</td>
<td>2</td>
<td>0.012858</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1757</th>
<td>R-HSA-500792</td>
<td>500792</td>
<td>GPCR ligand binding</td>
<td>False</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>610</td>
<td>4</td>
<td>0.037983</td>
<td>1.000000e+00</td>
<td>1.000000e+00</td>
<td>[]</td>
<td>TOTAL</td>
<td>219</td>
<td>4</td>
<td>0.014080</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">1758</th>
<td>R-HSA-9709957</td>
<td>9709957</td>
<td>Sensory Perception</td>
<td>False</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>1258</td>
<td>10</td>
<td>0.078331</td>
<td>1.000000e+00</td>
<td>1.000000e+00</td>
<td>[]</td>
<td>TOTAL</td>
<td>138</td>
<td>2</td>
<td>0.008872</td>
</tr>
</tbody>
</table>

<p>1759 rows × 19 columns</p>
</div>
</div>
</div>
<div id="44dbe6f1-0d6f-47d3-b95e-b74e0e44e4f1" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb56"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>ref_paths <span class="op">=</span> path_df_raw[path_df_raw.stId.isin(idx_path)]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="c126bd5a-98a9-4d80-895c-6d75178fbbc5" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb57"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>ref_paths.name</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>80      RUNX1 regulates transcription of genes involve...
129                  RHO GTPases Activate WASPs and WAVEs
157     Recruitment and ATM-mediated phosphorylation o...
306     MLL4 and MLL3 complexes regulate expression of...
423     Factors involved in megakaryocyte development ...
424                      Cyclin D associated events in G1
473     Turbulent (oscillatory, disturbed) flow shear ...
485     Regulation of actin dynamics for phagocytic cu...
526             HDR through Single Strand Annealing (SSA)
629                          FCGR3A-mediated phagocytosis
662                    Role of ABL in ROBO-SLIT signaling
922                                            Myogenesis
1193           RUNX2 regulates osteoblast differentiation
Name: name, dtype: object</code></pre>
</div>
</div>
<div id="5ab54d1f-cf5a-451a-a938-fa0119435c4e" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb59"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>idx_path</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>['R-HSA-2029482',
 'R-HSA-428890',
 'R-HSA-525793',
 'R-HSA-5663213',
 'R-HSA-5685938',
 'R-HSA-5693565',
 'R-HSA-69231',
 'R-HSA-8939236',
 'R-HSA-8940973',
 'R-HSA-9664422',
 'R-HSA-983231',
 'R-HSA-9841922',
 'R-HSA-9860927',
 '']</code></pre>
</div>
</div>
<div id="271ea332-89a6-4688-bc29-72f236208834" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb61"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>path_df <span class="op">=</span> get_reactome(genes)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="75b54503-b008-4899-96ef-f945d2986fea" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb62"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>path_df</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">name</th>
<th data-quarto-table-cell-role="th">fdr</th>
<th data-quarto-table-cell-role="th">-log10_fdr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">0</th>
<td>Translation initiation complex formation</td>
<td>4.884981e-15</td>
<td>14.311</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1</th>
<td>Formation of the ternary complex, and subseque...</td>
<td>4.884981e-15</td>
<td>14.311</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">2</th>
<td>Ribosomal scanning and start codon recognition</td>
<td>4.884981e-15</td>
<td>14.311</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">3</th>
<td>Activation of the mRNA upon binding of the cap...</td>
<td>4.884981e-15</td>
<td>14.311</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">4</th>
<td>GTP hydrolysis and joining of the 60S ribosoma...</td>
<td>4.884981e-15</td>
<td>14.311</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">...</th>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">1754</th>
<td>Immunoregulatory interactions between a Lympho...</td>
<td>1.000000e+00</td>
<td>0.000</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1755</th>
<td>Class A/1 (Rhodopsin-like receptors)</td>
<td>1.000000e+00</td>
<td>0.000</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">1756</th>
<td>SLC-mediated transmembrane transport</td>
<td>1.000000e+00</td>
<td>0.000</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1757</th>
<td>GPCR ligand binding</td>
<td>1.000000e+00</td>
<td>0.000</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">1758</th>
<td>Sensory Perception</td>
<td>1.000000e+00</td>
<td>-0.000</td>
</tr>
</tbody>
</table>

<p>1759 rows × 3 columns</p>
</div>
</div>
</div>
<div id="57ef743c-2bc3-4a19-8801-7518891abaf9" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb63"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>ref_paths.name.shape</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>(13,)</code></pre>
</div>
</div>
<div id="05b66341-e8f9-4535-a2ee-b2ed07ea37ec" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb65"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>plot_path(path_df,top_n<span class="op">=</span><span class="dv">50</span>,path_list<span class="op">=</span>ref_paths.name)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="11_plot_CDDM_files/figure-html/cell-59-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div id="4e6bfb67-4754-4385-822a-4ab39513f64f" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb66"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>path_df_raw[path_df_raw.stId.isin(idx_path)]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="9e9feead-8257-46f4-a83b-ab2282372c84" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb67"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> get_reactome(gene_list,</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>                 col<span class="op">=</span><span class="st">'entities.fdr'</span>, <span class="co"># column of p value or fdr (e.g., entities.pValue)</span></span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>                 ref_list<span class="op">=</span><span class="va">None</span>, <span class="co"># list of reactome idx</span></span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>                ):</span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>    <span class="co">"Reactome pathway analysis for a given gene set; returns formated output in dataframe with additional -log10(p)"</span></span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>    out <span class="op">=</span> get_reactome_raw(gene_list).copy()</span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a>    col_rename <span class="op">=</span> col.split(<span class="st">'.'</span>)[<span class="dv">1</span>]</span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a>    out <span class="op">=</span> out[[<span class="st">'stId'</span>,<span class="st">'name'</span>,col]].rename(columns<span class="op">=</span>{col:col_rename,<span class="st">'stId'</span>:<span class="st">'ID'</span>})</span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a>    out[<span class="st">'significant'</span>]<span class="op">=</span>(out[col_rename]<span class="op">&lt;=</span><span class="fl">0.05</span>).astype(<span class="bu">int</span>)</span>
<span id="cb67-10"><a href="#cb67-10" aria-hidden="true" tabindex="-1"></a>    out[<span class="ss">f'-log10(</span><span class="sc">{</span>col_rename<span class="sc">}</span><span class="ss">)'</span>] <span class="op">=</span> <span class="op">-</span>np.log10(out[col_rename]).<span class="bu">round</span>(<span class="dv">3</span>)</span>
<span id="cb67-11"><a href="#cb67-11" aria-hidden="true" tabindex="-1"></a>    out[<span class="ss">f'rank'</span>]<span class="op">=</span>out[col_rename].rank().astype(<span class="bu">int</span>)</span>
<span id="cb67-12"><a href="#cb67-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> ref_list: out[<span class="st">'in_ref'</span>]<span class="op">=</span>out.ID.isin(ref_list).astype(<span class="bu">int</span>)</span>
<span id="cb67-13"><a href="#cb67-13" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb67-14"><a href="#cb67-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> out</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="3bae4d60-ddd5-47a6-837d-4e8727011361" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb68"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>out <span class="op">=</span> get_reactome(genes,ref_list<span class="op">=</span>idx_path)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="f926eab0-1f76-4c0a-916a-549bcf4cae39" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb69"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>out[out.in_ref<span class="op">==</span><span class="dv">1</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">ID</th>
<th data-quarto-table-cell-role="th">name</th>
<th data-quarto-table-cell-role="th">fdr</th>
<th data-quarto-table-cell-role="th">significant</th>
<th data-quarto-table-cell-role="th">-log10(fdr)</th>
<th data-quarto-table-cell-role="th">rank</th>
<th data-quarto-table-cell-role="th">in_ref</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">80</th>
<td>R-HSA-8939236</td>
<td>RUNX1 regulates transcription of genes involve...</td>
<td>1.253766e-08</td>
<td>1</td>
<td>7.902</td>
<td>81</td>
<td>1</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">129</th>
<td>R-HSA-5663213</td>
<td>RHO GTPases Activate WASPs and WAVEs</td>
<td>3.404711e-06</td>
<td>1</td>
<td>5.468</td>
<td>130</td>
<td>1</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">157</th>
<td>R-HSA-5693565</td>
<td>Recruitment and ATM-mediated phosphorylation o...</td>
<td>1.482829e-05</td>
<td>1</td>
<td>4.829</td>
<td>156</td>
<td>1</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">306</th>
<td>R-HSA-9841922</td>
<td>MLL4 and MLL3 complexes regulate expression of...</td>
<td>6.713774e-04</td>
<td>1</td>
<td>3.173</td>
<td>304</td>
<td>1</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">423</th>
<td>R-HSA-983231</td>
<td>Factors involved in megakaryocyte development ...</td>
<td>4.163072e-03</td>
<td>1</td>
<td>2.381</td>
<td>424</td>
<td>1</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">424</th>
<td>R-HSA-69231</td>
<td>Cyclin D associated events in G1</td>
<td>4.191536e-03</td>
<td>1</td>
<td>2.378</td>
<td>426</td>
<td>1</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">473</th>
<td>R-HSA-9860927</td>
<td>Turbulent (oscillatory, disturbed) flow shear ...</td>
<td>7.981719e-03</td>
<td>1</td>
<td>2.098</td>
<td>474</td>
<td>1</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">485</th>
<td>R-HSA-2029482</td>
<td>Regulation of actin dynamics for phagocytic cu...</td>
<td>9.889877e-03</td>
<td>1</td>
<td>2.005</td>
<td>486</td>
<td>1</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">526</th>
<td>R-HSA-5685938</td>
<td>HDR through Single Strand Annealing (SSA)</td>
<td>1.734879e-02</td>
<td>1</td>
<td>1.761</td>
<td>527</td>
<td>1</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">629</th>
<td>R-HSA-9664422</td>
<td>FCGR3A-mediated phagocytosis</td>
<td>4.756334e-02</td>
<td>1</td>
<td>1.323</td>
<td>630</td>
<td>1</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">662</th>
<td>R-HSA-428890</td>
<td>Role of ABL in ROBO-SLIT signaling</td>
<td>6.281753e-02</td>
<td>0</td>
<td>1.202</td>
<td>663</td>
<td>1</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">922</th>
<td>R-HSA-525793</td>
<td>Myogenesis</td>
<td>1.742057e-01</td>
<td>0</td>
<td>0.759</td>
<td>923</td>
<td>1</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">1193</th>
<td>R-HSA-8940973</td>
<td>RUNX2 regulates osteoblast differentiation</td>
<td>4.049378e-01</td>
<td>0</td>
<td>0.393</td>
<td>1194</td>
<td>1</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div id="a68b6b2a-250b-4a67-8f97-0772ea4aa8b2" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb70"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>out.fdr.rank()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>0         22.0
1         22.0
2         22.0
3         22.0
4         22.0
         ...  
1754    1755.0
1755    1756.0
1756    1757.0
1757    1758.0
1758    1759.0
Name: fdr, Length: 1759, dtype: float64</code></pre>
</div>
</div>
<div id="af9f9e83-f679-401a-b598-0e7c65c7cffe" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb72"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_path(df,col<span class="op">=</span><span class="st">'-log10(fdr)'</span>, top_n<span class="op">=</span><span class="dv">10</span>,max_label_length<span class="op">=</span><span class="dv">80</span>):</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>    <span class="co">"Plot the bar graph of pathways from get_reactome function."</span></span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Extract the data and reverse it</span></span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>    data <span class="op">=</span> df.head(top_n).set_index(<span class="st">'name'</span>)[col].iloc[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Truncate labels if they are too long</span></span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a>    truncated_labels <span class="op">=</span> [label[:max_label_length] <span class="op">+</span> <span class="st">'...'</span> <span class="cf">if</span> <span class="bu">len</span>(label) <span class="op">&gt;</span> max_label_length <span class="cf">else</span> label <span class="cf">for</span> label <span class="kw">in</span> data.index]</span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a>    data.index <span class="op">=</span> truncated_labels</span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate the required width: base width + additional width for the longest label</span></span>
<span id="cb72-12"><a href="#cb72-12" aria-hidden="true" tabindex="-1"></a>    base_width <span class="op">=</span> <span class="dv">2</span></span>
<span id="cb72-13"><a href="#cb72-13" aria-hidden="true" tabindex="-1"></a>    max_label_length <span class="op">=</span> <span class="bu">max</span>(data.index, key<span class="op">=</span><span class="bu">len</span>)</span>
<span id="cb72-14"><a href="#cb72-14" aria-hidden="true" tabindex="-1"></a>    additional_width <span class="op">=</span> <span class="bu">len</span>(max_label_length) <span class="op">*</span> <span class="fl">0.1</span>  <span class="co"># Adjust scaling factor as needed</span></span>
<span id="cb72-15"><a href="#cb72-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb72-16"><a href="#cb72-16" aria-hidden="true" tabindex="-1"></a>    figsize <span class="op">=</span> (base_width <span class="op">+</span> additional_width, <span class="dv">3</span><span class="op">*</span>top_n<span class="op">/</span><span class="dv">10</span>)  <span class="co"># Adjust height as necessary</span></span>
<span id="cb72-17"><a href="#cb72-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-18"><a href="#cb72-18" aria-hidden="true" tabindex="-1"></a>    data.plot.barh(figsize<span class="op">=</span>figsize)</span>
<span id="cb72-19"><a href="#cb72-19" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">''</span>)</span>
<span id="cb72-20"><a href="#cb72-20" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(col)</span>
<span id="cb72-21"><a href="#cb72-21" aria-hidden="true" tabindex="-1"></a>    plt.tight_layout()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="49715283-b82c-4bd1-95c3-a85956273cc3" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb73"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_path(df, col<span class="op">=</span><span class="st">'-log10(fdr)'</span>, top_n<span class="op">=</span><span class="dv">10</span>, max_label_length<span class="op">=</span><span class="dv">80</span>, path_list<span class="op">=</span><span class="va">None</span>):</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>    <span class="co">"""</span></span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a><span class="co">    Plot the bar graph of pathways from get_reactome function.</span></span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a><span class="co">    Highlights pathways in path_list with a different color (dark red).</span></span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a><span class="co">    """</span></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Extract and reverse data</span></span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a>    data <span class="op">=</span> df.head(top_n).set_index(<span class="st">'name'</span>)[col].iloc[::<span class="op">-</span><span class="dv">1</span>]</span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save original full names to match against path_list</span></span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a>    full_names <span class="op">=</span> data.index.tolist()</span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Truncate labels if too long</span></span>
<span id="cb73-15"><a href="#cb73-15" aria-hidden="true" tabindex="-1"></a>    truncated_labels <span class="op">=</span> [label[:max_label_length] <span class="op">+</span> <span class="st">'...'</span> <span class="cf">if</span> <span class="bu">len</span>(label) <span class="op">&gt;</span> max_label_length <span class="cf">else</span> label <span class="cf">for</span> label <span class="kw">in</span> full_names]</span>
<span id="cb73-16"><a href="#cb73-16" aria-hidden="true" tabindex="-1"></a>    data.index <span class="op">=</span> truncated_labels</span>
<span id="cb73-17"><a href="#cb73-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-18"><a href="#cb73-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Determine colors</span></span>
<span id="cb73-19"><a href="#cb73-19" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> path_list <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb73-20"><a href="#cb73-20" aria-hidden="true" tabindex="-1"></a>        path_set <span class="op">=</span> <span class="bu">set</span>(path_list)</span>
<span id="cb73-21"><a href="#cb73-21" aria-hidden="true" tabindex="-1"></a>        colors <span class="op">=</span> [<span class="st">'darkred'</span> <span class="cf">if</span> name <span class="kw">in</span> path_set <span class="cf">else</span> <span class="st">'darkblue'</span> <span class="cf">for</span> name <span class="kw">in</span> full_names]</span>
<span id="cb73-22"><a href="#cb73-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb73-23"><a href="#cb73-23" aria-hidden="true" tabindex="-1"></a>        colors <span class="op">=</span> <span class="st">'darkblue'</span></span>
<span id="cb73-24"><a href="#cb73-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-25"><a href="#cb73-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate dynamic figure width</span></span>
<span id="cb73-26"><a href="#cb73-26" aria-hidden="true" tabindex="-1"></a>    base_width <span class="op">=</span> <span class="dv">2</span></span>
<span id="cb73-27"><a href="#cb73-27" aria-hidden="true" tabindex="-1"></a>    max_label <span class="op">=</span> <span class="bu">max</span>(data.index, key<span class="op">=</span><span class="bu">len</span>)</span>
<span id="cb73-28"><a href="#cb73-28" aria-hidden="true" tabindex="-1"></a>    additional_width <span class="op">=</span> <span class="bu">len</span>(max_label) <span class="op">*</span> <span class="fl">0.1</span></span>
<span id="cb73-29"><a href="#cb73-29" aria-hidden="true" tabindex="-1"></a>    figsize <span class="op">=</span> (base_width <span class="op">+</span> additional_width, <span class="dv">3</span> <span class="op">*</span> top_n <span class="op">/</span> <span class="dv">10</span>)</span>
<span id="cb73-30"><a href="#cb73-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-31"><a href="#cb73-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Plot</span></span>
<span id="cb73-32"><a href="#cb73-32" aria-hidden="true" tabindex="-1"></a>    ax <span class="op">=</span> data.plot.barh(figsize<span class="op">=</span>figsize, color<span class="op">=</span>colors)</span>
<span id="cb73-33"><a href="#cb73-33" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">''</span>)</span>
<span id="cb73-34"><a href="#cb73-34" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(col)</span>
<span id="cb73-35"><a href="#cb73-35" aria-hidden="true" tabindex="-1"></a>    plt.tight_layout()</span>
<span id="cb73-36"><a href="#cb73-36" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> ax</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="b64f1374-e83c-45dd-9d89-9a3bf01b7f6e" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb74"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>path_fdr <span class="op">=</span> path_df_raw[path_df_raw[<span class="st">'entities.fdr'</span>]<span class="op">&lt;</span><span class="fl">0.05</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="df6dee81-5ed6-401f-9ed5-96a03580e14d" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb75"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>path_fdr</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">stId</th>
<th data-quarto-table-cell-role="th">dbId</th>
<th data-quarto-table-cell-role="th">name</th>
<th data-quarto-table-cell-role="th">llp</th>
<th data-quarto-table-cell-role="th">inDisease</th>
<th data-quarto-table-cell-role="th">species.dbId</th>
<th data-quarto-table-cell-role="th">species.taxId</th>
<th data-quarto-table-cell-role="th">species.name</th>
<th data-quarto-table-cell-role="th">entities.resource</th>
<th data-quarto-table-cell-role="th">entities.total</th>
<th data-quarto-table-cell-role="th">entities.found</th>
<th data-quarto-table-cell-role="th">entities.ratio</th>
<th data-quarto-table-cell-role="th">entities.pValue</th>
<th data-quarto-table-cell-role="th">entities.fdr</th>
<th data-quarto-table-cell-role="th">entities.exp</th>
<th data-quarto-table-cell-role="th">reactions.resource</th>
<th data-quarto-table-cell-role="th">reactions.total</th>
<th data-quarto-table-cell-role="th">reactions.found</th>
<th data-quarto-table-cell-role="th">reactions.ratio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">0</th>
<td>R-HSA-72649</td>
<td>72649</td>
<td>Translation initiation complex formation</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>62</td>
<td>39</td>
<td>0.003861</td>
<td>1.110223e-16</td>
<td>4.884981e-15</td>
<td>[]</td>
<td>TOTAL</td>
<td>2</td>
<td>2</td>
<td>0.000129</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1</th>
<td>R-HSA-72695</td>
<td>72695</td>
<td>Formation of the ternary complex, and subseque...</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>54</td>
<td>33</td>
<td>0.003362</td>
<td>1.110223e-16</td>
<td>4.884981e-15</td>
<td>[]</td>
<td>TOTAL</td>
<td>3</td>
<td>3</td>
<td>0.000193</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">2</th>
<td>R-HSA-72702</td>
<td>72702</td>
<td>Ribosomal scanning and start codon recognition</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>64</td>
<td>39</td>
<td>0.003985</td>
<td>1.110223e-16</td>
<td>4.884981e-15</td>
<td>[]</td>
<td>TOTAL</td>
<td>2</td>
<td>2</td>
<td>0.000129</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">3</th>
<td>R-HSA-72662</td>
<td>72662</td>
<td>Activation of the mRNA upon binding of the cap...</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>66</td>
<td>39</td>
<td>0.004110</td>
<td>1.110223e-16</td>
<td>4.884981e-15</td>
<td>[]</td>
<td>TOTAL</td>
<td>6</td>
<td>6</td>
<td>0.000386</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">4</th>
<td>R-HSA-72706</td>
<td>72706</td>
<td>GTP hydrolysis and joining of the 60S ribosoma...</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>120</td>
<td>66</td>
<td>0.007472</td>
<td>1.110223e-16</td>
<td>4.884981e-15</td>
<td>[]</td>
<td>TOTAL</td>
<td>3</td>
<td>3</td>
<td>0.000193</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">631</th>
<td>R-HSA-8866427</td>
<td>8866427</td>
<td>VLDLR internalisation and degradation</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>16</td>
<td>4</td>
<td>0.000996</td>
<td>2.450345e-02</td>
<td>4.900690e-02</td>
<td>[]</td>
<td>TOTAL</td>
<td>4</td>
<td>4</td>
<td>0.000257</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">632</th>
<td>R-HSA-8984722</td>
<td>8984722</td>
<td>Interleukin-35 Signalling</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>16</td>
<td>4</td>
<td>0.000996</td>
<td>2.450345e-02</td>
<td>4.900690e-02</td>
<td>[]</td>
<td>TOTAL</td>
<td>26</td>
<td>25</td>
<td>0.001672</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">633</th>
<td>R-HSA-1679131</td>
<td>1679131</td>
<td>Trafficking and processing of endosomal TLR</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>16</td>
<td>4</td>
<td>0.000996</td>
<td>2.450345e-02</td>
<td>4.900690e-02</td>
<td>[]</td>
<td>TOTAL</td>
<td>7</td>
<td>4</td>
<td>0.000450</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">634</th>
<td>R-HSA-879415</td>
<td>879415</td>
<td>Advanced glycosylation endproduct receptor sig...</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>16</td>
<td>4</td>
<td>0.000996</td>
<td>2.450345e-02</td>
<td>4.900690e-02</td>
<td>[]</td>
<td>TOTAL</td>
<td>4</td>
<td>2</td>
<td>0.000257</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">635</th>
<td>R-HSA-2173789</td>
<td>2173789</td>
<td>TGF-beta receptor signaling activates SMADs</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>51</td>
<td>8</td>
<td>0.003176</td>
<td>2.483726e-02</td>
<td>4.967452e-02</td>
<td>[]</td>
<td>TOTAL</td>
<td>44</td>
<td>26</td>
<td>0.002829</td>
</tr>
</tbody>
</table>

<p>636 rows × 19 columns</p>
</div>
</div>
</div>
<div id="84f43ee6-8374-4eb8-a3e0-46334152784c" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb76"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>plot_path??</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div class="ansi-escaped-output">
<pre><span class="ansi-red-fg">Signature:</span> plot_path<span class="ansi-blue-fg">(</span>react_out<span class="ansi-blue-fg">,</span> top_n<span class="ansi-blue-fg">=</span><span class="ansi-cyan-fg">10</span><span class="ansi-blue-fg">,</span> max_label_length<span class="ansi-blue-fg">=</span><span class="ansi-cyan-fg">80</span><span class="ansi-blue-fg">)</span>
<span class="ansi-red-fg">Source:</span>   
<span class="ansi-green-fg">def</span> plot_path<span class="ansi-blue-fg">(</span>react_out<span class="ansi-blue-fg">,</span> top_n<span class="ansi-blue-fg">=</span><span class="ansi-cyan-fg">10</span><span class="ansi-blue-fg">,</span>max_label_length<span class="ansi-blue-fg">=</span><span class="ansi-cyan-fg">80</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">:</span>
    <span class="ansi-blue-fg">"Plot the bar graph of pathways from get_reactome function."</span>
    
    <span class="ansi-red-fg"># Extract the data and reverse it</span>
    data <span class="ansi-blue-fg">=</span> react_out<span class="ansi-blue-fg">.</span>head<span class="ansi-blue-fg">(</span>top_n<span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">.</span>set_index<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">'name'</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">'-log10_pValue'</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">.</span>iloc<span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">:</span><span class="ansi-blue-fg">:</span><span class="ansi-blue-fg">-</span><span class="ansi-cyan-fg">1</span><span class="ansi-blue-fg">]</span>
    
    <span class="ansi-red-fg"># Truncate labels if they are too long</span>
    truncated_labels <span class="ansi-blue-fg">=</span> <span class="ansi-blue-fg">[</span>label<span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">:</span>max_label_length<span class="ansi-blue-fg">]</span> <span class="ansi-blue-fg">+</span> <span class="ansi-blue-fg">'...'</span> <span class="ansi-green-fg">if</span> len<span class="ansi-blue-fg">(</span>label<span class="ansi-blue-fg">)</span> <span class="ansi-blue-fg">&gt;</span> max_label_length <span class="ansi-green-fg">else</span> label <span class="ansi-green-fg">for</span> label <span class="ansi-green-fg">in</span> data<span class="ansi-blue-fg">.</span>index<span class="ansi-blue-fg">]</span>
    data<span class="ansi-blue-fg">.</span>index <span class="ansi-blue-fg">=</span> truncated_labels
    <span class="ansi-red-fg"># Calculate the required width: base width + additional width for the longest label</span>
    base_width <span class="ansi-blue-fg">=</span> <span class="ansi-cyan-fg">2</span>
    max_label_length <span class="ansi-blue-fg">=</span> max<span class="ansi-blue-fg">(</span>data<span class="ansi-blue-fg">.</span>index<span class="ansi-blue-fg">,</span> key<span class="ansi-blue-fg">=</span>len<span class="ansi-blue-fg">)</span>
    additional_width <span class="ansi-blue-fg">=</span> len<span class="ansi-blue-fg">(</span>max_label_length<span class="ansi-blue-fg">)</span> <span class="ansi-blue-fg">*</span> <span class="ansi-cyan-fg">0.1</span>  <span class="ansi-red-fg"># Adjust scaling factor as needed</span>
    
    figsize <span class="ansi-blue-fg">=</span> <span class="ansi-blue-fg">(</span>base_width <span class="ansi-blue-fg">+</span> additional_width<span class="ansi-blue-fg">,</span> <span class="ansi-cyan-fg">3</span><span class="ansi-blue-fg">*</span>top_n<span class="ansi-blue-fg">/</span><span class="ansi-cyan-fg">10</span><span class="ansi-blue-fg">)</span>  <span class="ansi-red-fg"># Adjust height as necessary</span>
    data<span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">.</span>barh<span class="ansi-blue-fg">(</span>figsize<span class="ansi-blue-fg">=</span>figsize<span class="ansi-blue-fg">)</span>
    plt<span class="ansi-blue-fg">.</span>ylabel<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">''</span><span class="ansi-blue-fg">)</span>
    plt<span class="ansi-blue-fg">.</span>xlabel<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">'-log10(p)'</span><span class="ansi-blue-fg">)</span>
    plt<span class="ansi-blue-fg">.</span>tight_layout<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-red-fg">File:</span>      ~/katlas/katlas/utils.py
<span class="ansi-red-fg">Type:</span>      function</pre>
</div>
</div>
</div>
<div id="199d8b77-0fcc-404b-86dc-09735c605160" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb77"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>path_fdr.loc[path_fdr[<span class="st">'entities.ratio'</span>].sort_values(ascending<span class="op">=</span><span class="va">False</span>).index]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">stId</th>
<th data-quarto-table-cell-role="th">dbId</th>
<th data-quarto-table-cell-role="th">name</th>
<th data-quarto-table-cell-role="th">llp</th>
<th data-quarto-table-cell-role="th">inDisease</th>
<th data-quarto-table-cell-role="th">species.dbId</th>
<th data-quarto-table-cell-role="th">species.taxId</th>
<th data-quarto-table-cell-role="th">species.name</th>
<th data-quarto-table-cell-role="th">entities.resource</th>
<th data-quarto-table-cell-role="th">entities.total</th>
<th data-quarto-table-cell-role="th">entities.found</th>
<th data-quarto-table-cell-role="th">entities.ratio</th>
<th data-quarto-table-cell-role="th">entities.pValue</th>
<th data-quarto-table-cell-role="th">entities.fdr</th>
<th data-quarto-table-cell-role="th">entities.exp</th>
<th data-quarto-table-cell-role="th">reactions.resource</th>
<th data-quarto-table-cell-role="th">reactions.total</th>
<th data-quarto-table-cell-role="th">reactions.found</th>
<th data-quarto-table-cell-role="th">reactions.ratio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">180</th>
<td>R-HSA-162582</td>
<td>162582</td>
<td>Signal Transduction</td>
<td>False</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>3049</td>
<td>267</td>
<td>0.189851</td>
<td>3.562203e-06</td>
<td>3.562203e-05</td>
<td>[]</td>
<td>TOTAL</td>
<td>2584</td>
<td>1258</td>
<td>0.166131</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">48</th>
<td>R-HSA-1643685</td>
<td>1643685</td>
<td>Disease</td>
<td>False</td>
<td>True</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>2851</td>
<td>299</td>
<td>0.177522</td>
<td>1.554312e-15</td>
<td>6.061818e-14</td>
<td>[]</td>
<td>TOTAL</td>
<td>2011</td>
<td>716</td>
<td>0.129292</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">94</th>
<td>R-HSA-168256</td>
<td>168256</td>
<td>Immune System</td>
<td>False</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>2664</td>
<td>256</td>
<td>0.165878</td>
<td>2.664359e-09</td>
<td>5.328718e-08</td>
<td>[]</td>
<td>TOTAL</td>
<td>1733</td>
<td>791</td>
<td>0.111418</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">32</th>
<td>R-HSA-392499</td>
<td>392499</td>
<td>Metabolism of proteins</td>
<td>False</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>2417</td>
<td>268</td>
<td>0.150498</td>
<td>1.110223e-16</td>
<td>4.884981e-15</td>
<td>[]</td>
<td>TOTAL</td>
<td>909</td>
<td>428</td>
<td>0.058442</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">188</th>
<td>R-HSA-74160</td>
<td>74160</td>
<td>Gene expression (Transcription)</td>
<td>False</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>1990</td>
<td>186</td>
<td>0.123910</td>
<td>4.133596e-06</td>
<td>3.895872e-05</td>
<td>[]</td>
<td>TOTAL</td>
<td>1140</td>
<td>486</td>
<td>0.073293</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">...</th>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">443</th>
<td>R-HSA-9818025</td>
<td>9818025</td>
<td>NFE2L2 regulating TCA cycle genes</td>
<td>True</td>
<td>False</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>7</td>
<td>4</td>
<td>0.000436</td>
<td>1.440323e-03</td>
<td>5.761294e-03</td>
<td>[]</td>
<td>TOTAL</td>
<td>4</td>
<td>4</td>
<td>0.000257</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">474</th>
<td>R-HSA-9649913</td>
<td>9649913</td>
<td>RAS GTPase cycle mutants</td>
<td>False</td>
<td>True</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>4</td>
<td>3</td>
<td>0.000249</td>
<td>2.704293e-03</td>
<td>8.112878e-03</td>
<td>[]</td>
<td>TOTAL</td>
<td>2</td>
<td>2</td>
<td>0.000129</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">475</th>
<td>R-HSA-9753510</td>
<td>9753510</td>
<td>Signaling by RAS GAP mutants</td>
<td>True</td>
<td>True</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>4</td>
<td>3</td>
<td>0.000249</td>
<td>2.704293e-03</td>
<td>8.112878e-03</td>
<td>[]</td>
<td>TOTAL</td>
<td>1</td>
<td>1</td>
<td>0.000064</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">476</th>
<td>R-HSA-9753512</td>
<td>9753512</td>
<td>Signaling by RAS GTPase mutants</td>
<td>True</td>
<td>True</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>4</td>
<td>3</td>
<td>0.000249</td>
<td>2.704293e-03</td>
<td>8.112878e-03</td>
<td>[]</td>
<td>TOTAL</td>
<td>1</td>
<td>1</td>
<td>0.000064</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">610</th>
<td>R-HSA-9636667</td>
<td>9636667</td>
<td>Manipulation of host energy metabolism</td>
<td>True</td>
<td>True</td>
<td>48887</td>
<td>9606</td>
<td>Homo sapiens</td>
<td>TOTAL</td>
<td>3</td>
<td>2</td>
<td>0.000187</td>
<td>1.804436e-02</td>
<td>4.155485e-02</td>
<td>[]</td>
<td>TOTAL</td>
<td>2</td>
<td>2</td>
<td>0.000129</td>
</tr>
</tbody>
</table>

<p>636 rows × 19 columns</p>
</div>
</div>
</div>
<div id="d9b2cf85-4bcb-4edb-b9f1-f7858bfb08ae" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb78"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>path_df <span class="op">=</span> get_reactome(genes)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="2c8bc36e-eba5-4b1a-943d-742abcbaff1b" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb79"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>plot_path??</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div class="ansi-escaped-output">
<pre><span class="ansi-red-fg">Signature:</span> plot_path<span class="ansi-blue-fg">(</span>react_out<span class="ansi-blue-fg">,</span> top_n<span class="ansi-blue-fg">=</span><span class="ansi-cyan-fg">10</span><span class="ansi-blue-fg">,</span> max_label_length<span class="ansi-blue-fg">=</span><span class="ansi-cyan-fg">80</span><span class="ansi-blue-fg">)</span>
<span class="ansi-red-fg">Source:</span>   
<span class="ansi-green-fg">def</span> plot_path<span class="ansi-blue-fg">(</span>react_out<span class="ansi-blue-fg">,</span> top_n<span class="ansi-blue-fg">=</span><span class="ansi-cyan-fg">10</span><span class="ansi-blue-fg">,</span>max_label_length<span class="ansi-blue-fg">=</span><span class="ansi-cyan-fg">80</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">:</span>
    <span class="ansi-blue-fg">"Plot the bar graph of pathways from get_reactome function."</span>
    
    <span class="ansi-red-fg"># Extract the data and reverse it</span>
    data <span class="ansi-blue-fg">=</span> react_out<span class="ansi-blue-fg">.</span>head<span class="ansi-blue-fg">(</span>top_n<span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">.</span>set_index<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">'name'</span><span class="ansi-blue-fg">)</span><span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">'-log10_pValue'</span><span class="ansi-blue-fg">]</span><span class="ansi-blue-fg">.</span>iloc<span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">:</span><span class="ansi-blue-fg">:</span><span class="ansi-blue-fg">-</span><span class="ansi-cyan-fg">1</span><span class="ansi-blue-fg">]</span>
    
    <span class="ansi-red-fg"># Truncate labels if they are too long</span>
    truncated_labels <span class="ansi-blue-fg">=</span> <span class="ansi-blue-fg">[</span>label<span class="ansi-blue-fg">[</span><span class="ansi-blue-fg">:</span>max_label_length<span class="ansi-blue-fg">]</span> <span class="ansi-blue-fg">+</span> <span class="ansi-blue-fg">'...'</span> <span class="ansi-green-fg">if</span> len<span class="ansi-blue-fg">(</span>label<span class="ansi-blue-fg">)</span> <span class="ansi-blue-fg">&gt;</span> max_label_length <span class="ansi-green-fg">else</span> label <span class="ansi-green-fg">for</span> label <span class="ansi-green-fg">in</span> data<span class="ansi-blue-fg">.</span>index<span class="ansi-blue-fg">]</span>
    data<span class="ansi-blue-fg">.</span>index <span class="ansi-blue-fg">=</span> truncated_labels
    <span class="ansi-red-fg"># Calculate the required width: base width + additional width for the longest label</span>
    base_width <span class="ansi-blue-fg">=</span> <span class="ansi-cyan-fg">2</span>
    max_label_length <span class="ansi-blue-fg">=</span> max<span class="ansi-blue-fg">(</span>data<span class="ansi-blue-fg">.</span>index<span class="ansi-blue-fg">,</span> key<span class="ansi-blue-fg">=</span>len<span class="ansi-blue-fg">)</span>
    additional_width <span class="ansi-blue-fg">=</span> len<span class="ansi-blue-fg">(</span>max_label_length<span class="ansi-blue-fg">)</span> <span class="ansi-blue-fg">*</span> <span class="ansi-cyan-fg">0.1</span>  <span class="ansi-red-fg"># Adjust scaling factor as needed</span>
    
    figsize <span class="ansi-blue-fg">=</span> <span class="ansi-blue-fg">(</span>base_width <span class="ansi-blue-fg">+</span> additional_width<span class="ansi-blue-fg">,</span> <span class="ansi-cyan-fg">3</span><span class="ansi-blue-fg">*</span>top_n<span class="ansi-blue-fg">/</span><span class="ansi-cyan-fg">10</span><span class="ansi-blue-fg">)</span>  <span class="ansi-red-fg"># Adjust height as necessary</span>
    data<span class="ansi-blue-fg">.</span>plot<span class="ansi-blue-fg">.</span>barh<span class="ansi-blue-fg">(</span>figsize<span class="ansi-blue-fg">=</span>figsize<span class="ansi-blue-fg">)</span>
    plt<span class="ansi-blue-fg">.</span>ylabel<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">''</span><span class="ansi-blue-fg">)</span>
    plt<span class="ansi-blue-fg">.</span>xlabel<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">'-log10(p)'</span><span class="ansi-blue-fg">)</span>
    plt<span class="ansi-blue-fg">.</span>tight_layout<span class="ansi-blue-fg">(</span><span class="ansi-blue-fg">)</span>
<span class="ansi-red-fg">File:</span>      ~/katlas/katlas/utils.py
<span class="ansi-red-fg">Type:</span>      function</pre>
</div>
</div>
</div>
<div id="499e65e7-dff1-4ba2-8ebd-af4aabf710dd" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb80"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>path_df</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">name</th>
<th data-quarto-table-cell-role="th">pValue</th>
<th data-quarto-table-cell-role="th">-log10_pValue</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">0</th>
<td>Translation initiation complex formation</td>
<td>1.110223e-16</td>
<td>15.955</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1</th>
<td>Formation of the ternary complex, and subseque...</td>
<td>1.110223e-16</td>
<td>15.955</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">2</th>
<td>Ribosomal scanning and start codon recognition</td>
<td>1.110223e-16</td>
<td>15.955</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">3</th>
<td>Activation of the mRNA upon binding of the cap...</td>
<td>1.110223e-16</td>
<td>15.955</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">4</th>
<td>GTP hydrolysis and joining of the 60S ribosoma...</td>
<td>1.110223e-16</td>
<td>15.955</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">...</th>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">1754</th>
<td>Immunoregulatory interactions between a Lympho...</td>
<td>1.000000e+00</td>
<td>0.000</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1755</th>
<td>Class A/1 (Rhodopsin-like receptors)</td>
<td>1.000000e+00</td>
<td>0.000</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">1756</th>
<td>SLC-mediated transmembrane transport</td>
<td>1.000000e+00</td>
<td>0.000</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1757</th>
<td>GPCR ligand binding</td>
<td>1.000000e+00</td>
<td>0.000</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">1758</th>
<td>Sensory Perception</td>
<td>1.000000e+00</td>
<td>-0.000</td>
</tr>
</tbody>
</table>

<p>1759 rows × 3 columns</p>
</div>
</div>
</div>
<div id="9eca2b4e-a35e-4d87-927d-d2cf8891f8ae" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb81"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>plot_path(path_df,<span class="dv">100</span>)</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>plt.title(<span class="st">'AKT1'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>Text(0.5, 1.0, 'AKT1')</code></pre>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="11_plot_CDDM_files/figure-html/cell-74-output-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div id="1544396d-1fd5-43f5-9518-4553bcf1a8ff" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb83"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>genes</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="e4c17118-c355-445f-9408-c3b0f5db47a5" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb84"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.core <span class="im">import</span> <span class="op">*</span></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.plot <span class="im">import</span> <span class="op">*</span></span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> scipy.stats <span class="im">import</span> spearmanr, pearsonr</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> os</span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> PIL <span class="im">import</span> Image</span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> tqdm <span class="im">import</span> tqdm</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="8b5ecd77-7ea7-41cf-a45a-5dd92d550000" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb85"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_count(df_k,title):</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get value counts</span></span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>    source_counts <span class="op">=</span> df_k.source.replace({<span class="st">'pplus'</span>:<span class="st">'PP'</span>,<span class="st">'large_scale'</span>:<span class="st">'LS'</span>}).value_counts()</span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">7</span>,<span class="dv">1</span>))</span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>    source_counts.plot(kind<span class="op">=</span><span class="st">'barh'</span>, stacked<span class="op">=</span><span class="va">True</span>, color<span class="op">=</span>[<span class="st">'darkred'</span>, <span class="st">'darkblue'</span>])</span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Annotate with the actual values</span></span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> index, value <span class="kw">in</span> <span class="bu">enumerate</span>(source_counts):</span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a>        plt.text(value, index, <span class="bu">str</span>(value),fontsize<span class="op">=</span><span class="dv">10</span>,rotation<span class="op">=-</span><span class="dv">90</span>, va<span class="op">=</span><span class="st">'center'</span>)</span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">'Count'</span>)</span>
<span id="cb85-12"><a href="#cb85-12" aria-hidden="true" tabindex="-1"></a>    plt.title(title)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="6bbc2b30-d8f4-4617-bf47-64043b7386d1" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb86"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>sns.<span class="bu">set</span>(rc<span class="op">=</span>{<span class="st">"figure.dpi"</span>:<span class="dv">200</span>, <span class="st">'savefig.dpi'</span>:<span class="dv">200</span>})</span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>sns.set_context(<span class="st">'notebook'</span>)</span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>sns.set_style(<span class="st">"ticks"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="load-data" class="level2">
<h2 class="anchored" data-anchor-id="load-data">Load data</h2>
<div id="1ae60c7d-fb8f-4283-aa1c-e7a5333953a1" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb87"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> Data.get_ks_dataset()</span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>df[<span class="st">'SUB'</span>] <span class="op">=</span> df.substrate.<span class="bu">str</span>.upper()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="52e09485-eba3-4c90-b08c-00b896bbf554" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb88"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>info <span class="op">=</span> Data.get_kinase_info().query(<span class="st">'pseudo=="0"'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="d3341fea-7998-41cf-8cb4-86e3316c2b76" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb89"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co"># It only contains kinase on the tree</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>cnt <span class="op">=</span> df.kinase_paper.value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="4387895c-6d14-4679-b272-155740cd6fa1" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb90"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>ST <span class="op">=</span> info[info.group<span class="op">!=</span><span class="st">"TK"</span>].kinase</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="d7a7b03e-b583-49a8-9529-04ba963e0c9f" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb91"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>df[df.kinase_paper.isin(ST)].kinase_paper.value_counts()[<span class="dv">10</span>:<span class="dv">20</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>NEK6     950
PLK1     943
CK2A1    919
P38D     907
DYRK2    907
HGK      902
TTBK1    896
MST3     890
MST1     884
IKKE     880
Name: kinase_paper, dtype: int64</code></pre>
</div>
</div>
<div id="125a1ea0-8baa-4137-91cb-9290f4f5204b" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb93"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a>cnt <span class="op">=</span> cnt[cnt<span class="op">&gt;</span><span class="dv">100</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="generate-example-figures" class="level2">
<h2 class="anchored" data-anchor-id="generate-example-figures">Generate example figures</h2>
<div id="e154bcb2-ea28-4a73-a8d5-f7b34b492d47" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb94"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_heatmap2(matrix, title, figsize<span class="op">=</span>(<span class="dv">6</span>,<span class="dv">10</span>), label_size<span class="op">=</span><span class="dv">20</span>):</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>figsize)</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>    sns.heatmap(matrix, cmap<span class="op">=</span><span class="st">'binary'</span>, annot<span class="op">=</span><span class="va">False</span>,cbar<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>    plt.title(title,fontsize<span class="op">=</span>label_size)</span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Set the font size for the tick labels</span></span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>    plt.xticks(fontsize<span class="op">=</span>label_size)</span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>    plt.yticks(fontsize<span class="op">=</span>label_size)</span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">''</span>)</span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a>    plt.ylabel(<span class="st">''</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="d395e3c8-6148-41fa-8979-d7c57b39b475" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb95"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>kinase_list <span class="op">=</span> [<span class="st">'SRC'</span>,<span class="st">'ABL1'</span>,<span class="st">'ERK2'</span>,<span class="st">'PKACA'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="ace2cb69-70e0-4ada-833c-768227313da8" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb96"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>sns.<span class="bu">set</span>(rc<span class="op">=</span>{<span class="st">"figure.dpi"</span>:<span class="dv">200</span>, <span class="st">'savefig.dpi'</span>:<span class="dv">200</span>})</span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>sns.set_context(<span class="st">'notebook'</span>)</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>sns.set_style(<span class="st">"ticks"</span>)</span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> k <span class="kw">in</span> kinase_list:</span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a>    df_k <span class="op">=</span> df.query(<span class="ss">f'kinase=="</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">"'</span>)</span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a>    df_k <span class="op">=</span> df_k.drop_duplicates(subset<span class="op">=</span><span class="st">'SUB'</span>).reset_index()</span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>    paper,full <span class="op">=</span> get_freq(df_k)</span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a>    plot_heatmap2(full.drop(columns<span class="op">=</span>[<span class="dv">0</span>]),<span class="ss">f'</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">'</span>,figsize<span class="op">=</span>(<span class="dv">6</span>,<span class="dv">10</span>))</span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a>    plt.close()</span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a>    <span class="cf">break</span></span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-17"><a href="#cb96-17" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if you want to generate and save all of figures, uncomment below</span></span>
<span id="cb96-18"><a href="#cb96-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plt.savefig(f'fig/{k}.png',bbox_inches='tight', pad_inches=0.3)</span></span>
<span id="cb96-19"><a href="#cb96-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plt.close()</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="11_plot_CDDM_files/figure-html/cell-94-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="generate-all-figures" class="level2">
<h2 class="anchored" data-anchor-id="generate-all-figures">Generate all figures</h2>
<p>Uncomment plt.savefig to save figures</p>
<div id="7af400b5-63fc-479d-9a61-6964297cc61b" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb97"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>sns.<span class="bu">set</span>(rc<span class="op">=</span>{<span class="st">"figure.dpi"</span>:<span class="dv">200</span>, <span class="st">'savefig.dpi'</span>:<span class="dv">200</span>})</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>sns.set_context(<span class="st">'notebook'</span>)</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>sns.set_style(<span class="st">"ticks"</span>)</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> k <span class="kw">in</span> tqdm(cnt.index,total<span class="op">=</span><span class="bu">len</span>(cnt)):</span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a>    df_k <span class="op">=</span> df.query(<span class="ss">f'kinase=="</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">"'</span>)</span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a>    plot_count(df_k,k)</span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plt.savefig(f'fig/count/{k}.png',bbox_inches='tight', pad_inches=0.1)</span></span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a>    plt.show() <span class="co"># if visualize in jupyter notebook, uncheck the savefig</span></span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a>    plt.close()</span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a>    df_k <span class="op">=</span> df_k.drop_duplicates(subset<span class="op">=</span><span class="st">'SUB'</span>).reset_index()</span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb97-17"><a href="#cb97-17" aria-hidden="true" tabindex="-1"></a>    paper,full <span class="op">=</span> get_freq(df_k)</span>
<span id="cb97-18"><a href="#cb97-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-19"><a href="#cb97-19" aria-hidden="true" tabindex="-1"></a>    get_logo2(full, k)</span>
<span id="cb97-20"><a href="#cb97-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plt.savefig(f'fig/logo/{k}.png',bbox_inches='tight', pad_inches=0.3)</span></span>
<span id="cb97-21"><a href="#cb97-21" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb97-22"><a href="#cb97-22" aria-hidden="true" tabindex="-1"></a>    plt.close()</span>
<span id="cb97-23"><a href="#cb97-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-24"><a href="#cb97-24" aria-hidden="true" tabindex="-1"></a>    plot_heatmap(full.drop(columns<span class="op">=</span>[<span class="dv">0</span>]),<span class="ss">f'</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss"> (n=</span><span class="sc">{</span><span class="bu">len</span>(df_k)<span class="sc">}</span><span class="ss">)'</span>,figsize<span class="op">=</span>(<span class="fl">7.5</span>,<span class="dv">10</span>))</span>
<span id="cb97-25"><a href="#cb97-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plt.savefig(f'fig/heatmap/{k}.png',bbox_inches='tight', pad_inches=0)</span></span>
<span id="cb97-26"><a href="#cb97-26" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb97-27"><a href="#cb97-27" aria-hidden="true" tabindex="-1"></a>    plt.close()</span>
<span id="cb97-28"><a href="#cb97-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># break</span></span>
<span id="cb97-29"><a href="#cb97-29" aria-hidden="true" tabindex="-1"></a>    <span class="cf">break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>  0%|          | 0/289 [00:00&lt;?, ?it/s]</code></pre>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="11_plot_CDDM_files/figure-html/cell-95-output-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="11_plot_CDDM_files/figure-html/cell-95-output-3.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="11_plot_CDDM_files/figure-html/cell-95-output-4.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  0%|          | 0/289 [00:02&lt;?, ?it/s]</code></pre>
</div>
</div>
</section>
<section id="combine-figures-for-pdf" class="level2">
<h2 class="anchored" data-anchor-id="combine-figures-for-pdf">Combine figures for pdf</h2>
<div id="ba6246db-34a1-4489-9cae-c4b4447112dc" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb100"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> combine_images_vertically(image_paths, output_path):</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>    images <span class="op">=</span> [Image.<span class="bu">open</span>(image_path).convert(<span class="st">'RGBA'</span>) <span class="cf">for</span> image_path <span class="kw">in</span> image_paths]</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>    total_width <span class="op">=</span> <span class="bu">max</span>(image.width <span class="cf">for</span> image <span class="kw">in</span> images)</span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>    total_height <span class="op">=</span> <span class="bu">sum</span>(image.height <span class="cf">for</span> image <span class="kw">in</span> images)</span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a>    combined_image <span class="op">=</span> Image.new(<span class="st">'RGBA'</span>, (total_width, total_height))</span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>    y_offset <span class="op">=</span> <span class="dv">0</span></span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> image <span class="kw">in</span> images:</span>
<span id="cb100-11"><a href="#cb100-11" aria-hidden="true" tabindex="-1"></a>        combined_image.paste(image, (<span class="dv">0</span>, y_offset), image)</span>
<span id="cb100-12"><a href="#cb100-12" aria-hidden="true" tabindex="-1"></a>        y_offset <span class="op">+=</span> image.height</span>
<span id="cb100-13"><a href="#cb100-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-14"><a href="#cb100-14" aria-hidden="true" tabindex="-1"></a>    combined_image.save(output_path)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Uncomment below to run</p>
<div id="6f9b1d0a-4661-40ca-8527-19174f3bb108" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb101"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="co"># folders = ["fig/count", "fig/logo", "fig/heatmap"]</span></span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for k in tqdm(cnt.index,total=len(cnt)):</span></span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     filename = f"{k}.png"</span></span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     image_paths = [os.path.join(folder, filename) for folder in folders]</span></span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a><span class="co">#     output_path = f"fig/combine/{k}.png"</span></span>
<span id="cb101-6"><a href="#cb101-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb101-7"><a href="#cb101-7" aria-hidden="true" tabindex="-1"></a><span class="co">#     combine_images_vertically(image_paths, output_path)</span></span>
<span id="cb101-8"><a href="#cb101-8" aria-hidden="true" tabindex="-1"></a><span class="co">#     # break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="get-pssm-data-of-cddm" class="level2">
<h2 class="anchored" data-anchor-id="get-pssm-data-of-cddm">Get PSSM data of CDDM</h2>
<div id="8fb0d8c3-2706-477a-8ab4-ac1b6ab4d0e3" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb102"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i,k <span class="kw">in</span> <span class="bu">enumerate</span>(cnt.index):</span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>    df_k <span class="op">=</span> df.query(<span class="ss">f'kinase=="</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">"'</span>)</span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>    df_k <span class="op">=</span> df_k.drop_duplicates(subset<span class="op">=</span><span class="st">'SUB'</span>).reset_index()</span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>    paper,full <span class="op">=</span> get_freq(df_k)</span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-8"><a href="#cb102-8" aria-hidden="true" tabindex="-1"></a>    melt <span class="op">=</span> full.drop(columns <span class="op">=</span> [<span class="dv">0</span>]).reset_index().melt(id_vars<span class="op">=</span>[<span class="st">'aa'</span>], value_name<span class="op">=</span>k, var_name<span class="op">=</span><span class="st">'Position'</span>)</span>
<span id="cb102-9"><a href="#cb102-9" aria-hidden="true" tabindex="-1"></a>    melt[<span class="st">'substrate'</span>]<span class="op">=</span>melt[<span class="st">'Position'</span>].astype(<span class="bu">str</span>)<span class="op">+</span> melt[<span class="st">'aa'</span>]</span>
<span id="cb102-10"><a href="#cb102-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-11"><a href="#cb102-11" aria-hidden="true" tabindex="-1"></a>    position_0 <span class="op">=</span> full[<span class="dv">0</span>][[<span class="st">'s'</span>,<span class="st">'t'</span>,<span class="st">'y'</span>]].reset_index().rename(columns<span class="op">=</span>{<span class="dv">0</span>:k})</span>
<span id="cb102-12"><a href="#cb102-12" aria-hidden="true" tabindex="-1"></a>    position_0[<span class="st">'substrate'</span>] <span class="op">=</span> <span class="st">'0'</span><span class="op">+</span>position_0[<span class="st">'aa'</span>]</span>
<span id="cb102-13"><a href="#cb102-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-14"><a href="#cb102-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> i <span class="op">==</span><span class="dv">0</span>:</span>
<span id="cb102-15"><a href="#cb102-15" aria-hidden="true" tabindex="-1"></a>        first <span class="op">=</span> pd.concat([melt,position_0])[[<span class="st">'substrate'</span>,k]].set_index(<span class="st">'substrate'</span>)</span>
<span id="cb102-16"><a href="#cb102-16" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb102-17"><a href="#cb102-17" aria-hidden="true" tabindex="-1"></a>        k <span class="op">=</span> pd.concat([melt,position_0])[[<span class="st">'substrate'</span>,k]].set_index(<span class="st">'substrate'</span>)</span>
<span id="cb102-18"><a href="#cb102-18" aria-hidden="true" tabindex="-1"></a>        data <span class="op">=</span> pd.concat([first,k],axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb102-19"><a href="#cb102-19" aria-hidden="true" tabindex="-1"></a>        first <span class="op">=</span> data.copy()</span>
<span id="cb102-20"><a href="#cb102-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb102-21"><a href="#cb102-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="b4cc0903-ac54-48ff-929e-c21da4d0fb9c" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb103"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a>data <span class="op">=</span> data.T</span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>data.index <span class="op">=</span> data.index.rename(<span class="st">'kinase'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>To save</p>
<div id="970af65d-251b-4a11-818c-52688c96dd4e" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb104"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># data.to_csv('supp/CDDM.csv')</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a><span class="co"># data.to_parquet('ks_main.parquet')</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="get-specialized-cddm-data-for-all-capital-substrates" class="level2">
<h2 class="anchored" data-anchor-id="get-specialized-cddm-data-for-all-capital-substrates">Get specialized CDDM data for all-capital substrates</h2>
<blockquote class="blockquote">
<p>combine s,t,y to S,T,Y</p>
</blockquote>
<div id="3781c6de-5e9f-4cb7-943c-dd6a5968b93a" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb105"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># List of suffixes</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>suffixes <span class="op">=</span> [<span class="st">'S'</span>, <span class="st">'T'</span>, <span class="st">'Y'</span>]</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> suffix <span class="kw">in</span> suffixes:</span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="op">-</span><span class="dv">7</span>, <span class="dv">8</span>):  <span class="co"># looping from -7 to 7</span></span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> i <span class="op">==</span> <span class="dv">0</span>:  <span class="co"># Skip 0</span></span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a>            <span class="cf">continue</span></span>
<span id="cb105-8"><a href="#cb105-8" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb105-9"><a href="#cb105-9" aria-hidden="true" tabindex="-1"></a>        upper_col <span class="op">=</span> <span class="ss">f"</span><span class="sc">{</span>i<span class="sc">}{</span>suffix<span class="sc">}</span><span class="ss">"</span>  <span class="co"># e.g., -7S</span></span>
<span id="cb105-10"><a href="#cb105-10" aria-hidden="true" tabindex="-1"></a>        lower_col <span class="op">=</span> <span class="ss">f"</span><span class="sc">{</span>i<span class="sc">}{</span>suffix<span class="sc">.</span>lower()<span class="sc">}</span><span class="ss">"</span>  <span class="co"># e.g., -7s</span></span>
<span id="cb105-11"><a href="#cb105-11" aria-hidden="true" tabindex="-1"></a>        data[upper_col] <span class="op">=</span> data[upper_col] <span class="op">+</span> data[lower_col]</span>
<span id="cb105-12"><a href="#cb105-12" aria-hidden="true" tabindex="-1"></a>        data.drop(lower_col, axis<span class="op">=</span><span class="dv">1</span>,inplace<span class="op">=</span><span class="va">True</span>)  <span class="co"># Drop the lowercase column after combining</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="1142743a-dbc4-40d9-824c-e4e49413ef81" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb106"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>data.columns[data.columns.<span class="bu">str</span>.contains(<span class="st">'S'</span>)]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>Index(['-7S', '-6S', '-5S', '-4S', '-3S', '-2S', '-1S', '1S', '2S', '3S', '4S',
       '5S', '6S', '7S'],
      dtype='object', name='substrate')</code></pre>
</div>
</div>
<div id="93952a08-7d61-469f-9ff6-7ea264f1aac5" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb108"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="co"># make sure the "s" in positions other than 0 is deleted from the columns</span></span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>data.columns[data.columns.<span class="bu">str</span>.contains(<span class="st">'s'</span>)]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>Index(['0s'], dtype='object', name='substrate')</code></pre>
</div>
</div>
<div id="32af5193-1e6b-4e05-a0a8-defbc368a5c3" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb110"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure very position's sum is 1</span></span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>data.loc[:,data.columns.<span class="bu">str</span>.contains(<span class="st">'-7'</span>)].<span class="bu">sum</span>(<span class="dv">1</span>).sort_values()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<pre><code>kinase
DDR2      1.0
NEK11     1.0
MSK1      1.0
TEK       1.0
NIM1      1.0
         ... 
CAMK2G    1.0
PKG2      1.0
MELK      1.0
NEK1      1.0
TLK2      1.0
Length: 289, dtype: float64</code></pre>
</div>
</div>
<div id="563ae01d-d346-41e5-bb9c-8e70206b4a09" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb112"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a>data <span class="op">=</span> data.rename(columns<span class="op">=</span>{<span class="st">'0s'</span>:<span class="st">'0S'</span>,<span class="st">'0t'</span>:<span class="st">'0T'</span>,<span class="st">'0y'</span>:<span class="st">'0Y'</span>})</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="f29b5755-bc88-4866-94fd-08c993436ebf" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb113"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>data.index <span class="op">=</span> data.index.rename(<span class="st">'kinase'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>To save</p>
<div id="b066fc36-e1be-4117-80eb-e0a72e903e0b" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb114"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="co"># data.to_parquet('ks_main_upper.parquet')</span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a><span class="co"># data.to_csv('supp/CDDM_upper.csv')</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="plot-other-kinases-mutated-lipid-kinase-isoforms" class="level2">
<h2 class="anchored" data-anchor-id="plot-other-kinases-mutated-lipid-kinase-isoforms">Plot other kinases (mutated, lipid kinase, isoforms)</h2>
<blockquote class="blockquote">
<p>kinases not on kinome tree</p>
</blockquote>
<div id="4b7a2708-aca5-408e-8ddf-ee668fef402c" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb115"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>cnt_other <span class="op">=</span> df.query(<span class="st">'on_tree==0'</span>).kinase.value_counts()</span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a>cnt_other <span class="op">=</span> cnt_other[cnt_other<span class="op">&gt;</span><span class="dv">100</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="07b311e1-415c-4bb6-8fbf-9ab96d462439" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb116"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>others <span class="op">=</span> cnt_other.index.tolist()<span class="op">+</span>[<span class="st">'LYN'</span>,<span class="st">'ABL1'</span>,<span class="st">'RET'</span>,<span class="st">'FGFR3'</span>,<span class="st">'PDGFRA'</span>,<span class="st">'ALK'</span>,</span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>                          <span class="st">'EGFR'</span>,<span class="st">'KIT'</span>,<span class="st">'MET'</span>,<span class="st">'PKCB'</span>,<span class="st">'BRAF'</span>,<span class="st">'PKG1'</span>] <span class="co"># BRAF is less than 100</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Uncheck savefig to save figures</p>
<div id="367b0bb9-d3cb-4dfe-9ee1-7ab37890d08d" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb117"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> k <span class="kw">in</span> tqdm(others,total<span class="op">=</span><span class="bu">len</span>(others)):</span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>    df_k <span class="op">=</span> df.query(<span class="ss">f'kinase=="</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">"'</span>)</span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a>    plot_count(df_k,k)</span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plt.savefig(f'fig_others/count/{k.replace("/", "_")}.png',bbox_inches='tight', pad_inches=0.1)</span></span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a>    plt.show() <span class="co"># if visualize in jupyter notebook, uncheck the savefig</span></span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a>    plt.close()</span>
<span id="cb117-8"><a href="#cb117-8" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb117-9"><a href="#cb117-9" aria-hidden="true" tabindex="-1"></a>    df_k <span class="op">=</span> df_k.drop_duplicates(subset<span class="op">=</span><span class="st">'SUB'</span>).reset_index()</span>
<span id="cb117-10"><a href="#cb117-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb117-11"><a href="#cb117-11" aria-hidden="true" tabindex="-1"></a>    paper,full <span class="op">=</span> get_freq(df_k)</span>
<span id="cb117-12"><a href="#cb117-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb117-13"><a href="#cb117-13" aria-hidden="true" tabindex="-1"></a>    get_logo2(full,k)</span>
<span id="cb117-14"><a href="#cb117-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plt.savefig(f'fig_others/logo/{k.replace("/", "_")}.png',bbox_inches='tight', pad_inches=0.3)</span></span>
<span id="cb117-15"><a href="#cb117-15" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb117-16"><a href="#cb117-16" aria-hidden="true" tabindex="-1"></a>    plt.close()</span>
<span id="cb117-17"><a href="#cb117-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb117-18"><a href="#cb117-18" aria-hidden="true" tabindex="-1"></a>    plot_heatmap(full.drop(columns<span class="op">=</span>[<span class="dv">0</span>]),<span class="ss">f'</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss"> (n=</span><span class="sc">{</span><span class="bu">len</span>(df_k)<span class="sc">}</span><span class="ss">)'</span>,figsize<span class="op">=</span>(<span class="fl">7.5</span>,<span class="dv">10</span>))</span>
<span id="cb117-19"><a href="#cb117-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plt.savefig(f'fig_others/heatmap/{k.replace("/", "_")}.png',bbox_inches='tight', pad_inches=0)</span></span>
<span id="cb117-20"><a href="#cb117-20" aria-hidden="true" tabindex="-1"></a>    plt.show()</span>
<span id="cb117-21"><a href="#cb117-21" aria-hidden="true" tabindex="-1"></a>    plt.close()</span>
<span id="cb117-22"><a href="#cb117-22" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb117-23"><a href="#cb117-23" aria-hidden="true" tabindex="-1"></a>    <span class="cf">break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>  0%|          | 0/36 [00:00&lt;?, ?it/s]</code></pre>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="11_plot_CDDM_files/figure-html/cell-110-output-2.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="11_plot_CDDM_files/figure-html/cell-110-output-3.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="11_plot_CDDM_files/figure-html/cell-110-output-4.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  0%|          | 0/36 [00:01&lt;?, ?it/s]</code></pre>
</div>
</div>
</section>
<section id="combine-the-figures-for-pdf" class="level2">
<h2 class="anchored" data-anchor-id="combine-the-figures-for-pdf">Combine the figures for pdf</h2>
<p>Uncomment below to run</p>
<div id="cd184b0b-54ac-407b-84e4-ae0b8e52c069" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb120"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="co"># folders = ["fig_others/count", "fig_others/logo", "fig_others/heatmap"]</span></span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a><span class="co"># for k in tqdm(others,total = len(others)):</span></span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     k = k.replace("/", "_")</span></span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb120-5"><a href="#cb120-5" aria-hidden="true" tabindex="-1"></a><span class="co">#     filename = f"{k}.png"</span></span>
<span id="cb120-6"><a href="#cb120-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb120-7"><a href="#cb120-7" aria-hidden="true" tabindex="-1"></a><span class="co">#     image_paths = [os.path.join(folder, filename) for folder in folders]</span></span>
<span id="cb120-8"><a href="#cb120-8" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb120-9"><a href="#cb120-9" aria-hidden="true" tabindex="-1"></a><span class="co">#     output_path = f"fig_others/combine/{k}.png"</span></span>
<span id="cb120-10"><a href="#cb120-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb120-11"><a href="#cb120-11" aria-hidden="true" tabindex="-1"></a><span class="co">#     combine_images_vertically(image_paths, output_path)</span></span>
<span id="cb120-12"><a href="#cb120-12" aria-hidden="true" tabindex="-1"></a><span class="co">#     # break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="get-the-pssms-of-other-kinases" class="level2">
<h2 class="anchored" data-anchor-id="get-the-pssms-of-other-kinases">Get the PSSMs of other kinases</h2>
<div id="eb81a8fc-c719-4058-a9e3-a4ad638abd64" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb121"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i,k <span class="kw">in</span> <span class="bu">enumerate</span>(others):</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>    df_k <span class="op">=</span> df.query(<span class="ss">f'kinase=="</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">"'</span>)</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>    df_k <span class="op">=</span> df_k.drop_duplicates(subset<span class="op">=</span><span class="st">'SUB'</span>).reset_index()</span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a>    paper,full <span class="op">=</span> get_freq(df_k)</span>
<span id="cb121-6"><a href="#cb121-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb121-7"><a href="#cb121-7" aria-hidden="true" tabindex="-1"></a>    melt <span class="op">=</span> full.drop(columns <span class="op">=</span> [<span class="dv">0</span>]).reset_index().melt(id_vars<span class="op">=</span>[<span class="st">'aa'</span>], value_name<span class="op">=</span>k, var_name<span class="op">=</span><span class="st">'Position'</span>)</span>
<span id="cb121-8"><a href="#cb121-8" aria-hidden="true" tabindex="-1"></a>    melt[<span class="st">'substrate'</span>]<span class="op">=</span>melt[<span class="st">'Position'</span>].astype(<span class="bu">str</span>)<span class="op">+</span> melt[<span class="st">'aa'</span>]</span>
<span id="cb121-9"><a href="#cb121-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-10"><a href="#cb121-10" aria-hidden="true" tabindex="-1"></a>    position_0 <span class="op">=</span> full[<span class="dv">0</span>][[<span class="st">'s'</span>,<span class="st">'t'</span>,<span class="st">'y'</span>]].reset_index().rename(columns<span class="op">=</span>{<span class="dv">0</span>:k})</span>
<span id="cb121-11"><a href="#cb121-11" aria-hidden="true" tabindex="-1"></a>    position_0[<span class="st">'substrate'</span>] <span class="op">=</span> <span class="st">'0'</span><span class="op">+</span>position_0[<span class="st">'aa'</span>]</span>
<span id="cb121-12"><a href="#cb121-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-13"><a href="#cb121-13" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> i <span class="op">==</span><span class="dv">0</span>:</span>
<span id="cb121-14"><a href="#cb121-14" aria-hidden="true" tabindex="-1"></a>        first <span class="op">=</span> pd.concat([melt,position_0])[[<span class="st">'substrate'</span>,k]].set_index(<span class="st">'substrate'</span>)</span>
<span id="cb121-15"><a href="#cb121-15" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>:</span>
<span id="cb121-16"><a href="#cb121-16" aria-hidden="true" tabindex="-1"></a>        k <span class="op">=</span> pd.concat([melt,position_0])[[<span class="st">'substrate'</span>,k]].set_index(<span class="st">'substrate'</span>)</span>
<span id="cb121-17"><a href="#cb121-17" aria-hidden="true" tabindex="-1"></a>        data <span class="op">=</span> pd.concat([first,k],axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb121-18"><a href="#cb121-18" aria-hidden="true" tabindex="-1"></a>        first <span class="op">=</span> data.copy()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="19a79f4f-af7a-4840-9c75-1a2555ba1a35" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb122"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>data <span class="op">=</span> data.T</span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a>data.index <span class="op">=</span> data.index.rename(<span class="st">'kinase'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>To save:</p>
<div id="aea04a21-c319-4ad3-83dc-165fc8b2d035" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb123"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="co"># data.to_csv('supp/CDDM_others.csv')</span></span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a><span class="co"># data.to_parquet('ks_others.parquet')</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/sky1ove\.github\.io\/katlas-raw");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/sky1ove/katlas-raw/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></div></div></footer></body></html>