# Preprocess kinase-substrate datasets


<!-- WARNING: THIS FILE WAS AUTOGENERATED! DO NOT EDIT! -->

## Motif

Poll \[Sugiyama & Douglass\]:
https://biosignaling.biomedcentral.com/articles/10.1186/s12964-023-01436-2

Poll web motif:
https://esbl.nhlbi.nih.gov/Databases/Kinase_Logos/KinaseTree.html;
https://esbl.nhlbi.nih.gov/Databases/Kinase_Logos/GeneSymbol.html

Sugiyama: https://pmc.ncbi.nlm.nih.gov/articles/PMC6642169/

Douglass:
https://journals.physiology.org/doi/full/10.1152/ajpcell.00166.2012

RegPhos: Supplementary Table S4: motifs grouped by kinase family

PhosphoNetwork, logos of 300 motifs:
https://pmc.ncbi.nlm.nih.gov/articles/PMC3658267/; web app:
https://phosphonetworks.org/

GPS: https://gps.biocuckoo.cn/links.php#l1

## Review of database

Wilson, AACR cancer research:
https://aacrjournals.org/cancerres/article/78/1/15/625062/New-Perspectives-Opportunities-and-Challenges-in

GPS 6.0, List of database: https://gps.biocuckoo.cn/links.php

paper: https://academic.oup.com/nar/article/51/W1/W243/7157529

phospho.ELM links: http://phospho.elm.eu.org/links.html

GPS, https://gps.biocuckoo.cn/links.php#l1

## Set up

``` python
from katlas.data import *
import pandas as pd
```

``` python
def map_substrate(idmapping_fname, ori_df, sub_col,remove_nonhuman=True):
    "Map uniprot id with uniprot info"

    substrate_id = pd.read_excel(idmapping_fname)
    substrate_id = substrate_id.drop_duplicates('From')
    substrate_id = substrate_id[['From', 'Entry', 'Gene Names', 'Sequence','Organism']]
    ori_df = ori_df.copy()

    # prevent name conflict
    if sub_col == 'substrate_uniprot':
        sub_col = 'substrate_uniprot_tmp'
        ori_df = ori_df.rename(columns={'substrate_uniprot':'substrate_uniprot_tmp'})

    substrate_id.columns = [sub_col, 'substrate_uniprot', 'substrate_genes', 'substrate_sequence','substrate_species']

    print('Shape before processing', ori_df.shape)

    ori_df = ori_df.merge(substrate_id, on=sub_col)

    print('Species counts:',ori_df.substrate_species.value_counts())

    if remove_nonhuman:
        print('Removing non-human substrates')
        ori_df = ori_df[ori_df.substrate_species=='Homo sapiens (Human)']

    ori_df = ori_df.drop(columns=[sub_col])  # Drop temp column after merging
    ori_df = ori_df.dropna(subset=['substrate_sequence'])
    print('Shape after processing', ori_df.shape)

    return ori_df.reset_index(drop=True)
```

``` python
# kinase_uniprot=pd.read_excel('raw/uniprot_human_keyword_kinase.xlsx',engine="openpyxl")
kinase_uniprot = Data.get_kinase_uniprot()
```

``` python
kinase_uniprot.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Entry</th>
<th data-quarto-table-cell-role="th">Entry Name</th>
<th data-quarto-table-cell-role="th">Protein names</th>
<th data-quarto-table-cell-role="th">Gene Names</th>
<th data-quarto-table-cell-role="th">uniprot_keyword_kinase</th>
<th data-quarto-table-cell-role="th">on_tree</th>
<th data-quarto-table-cell-role="th">Organism</th>
<th data-quarto-table-cell-role="th">Keywords</th>
<th data-quarto-table-cell-role="th">Sequence</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>A2RU49</td>
<td>HYKK_HUMAN</td>
<td>Hydroxylysine kinase (5-hydroxy-L-lysine kinas...</td>
<td>HYKK AGPHD1</td>
<td>1</td>
<td>0</td>
<td>Homo sapiens (Human)</td>
<td>Alternative splicing;Cytoplasm;Kinase;Proteomi...</td>
<td>MSSGNYQQSEALSKPTFSEEQASALVESVFGLKVSKVRPLPSYDDQ...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>A4D2B8</td>
<td>PM2P1_HUMAN</td>
<td>Putative postmeiotic segregation increased 2-l...</td>
<td>PMS2P1 PMS2L1 PMS2L13 PMS2L6 PMS2L8 PMS3 PMS8 ...</td>
<td>1</td>
<td>0</td>
<td>Homo sapiens (Human)</td>
<td>Alternative splicing;Kinase;Reference proteome...</td>
<td>MVTMCGGHRPENFLHQVLTEFGEELAGEGKSEVGGGAPRSYLQVAS...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>O00141</td>
<td>SGK1_HUMAN</td>
<td>Serine/threonine-protein kinase Sgk1 (EC 2.7.1...</td>
<td>SGK1 SGK</td>
<td>1</td>
<td>1</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Alternative promoter usage;Altern...</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>O00142</td>
<td>KITM_HUMAN</td>
<td>Thymidine kinase 2, mitochondrial (EC 2.7.1.21...</td>
<td>TK2</td>
<td>1</td>
<td>0</td>
<td>Homo sapiens (Human)</td>
<td>Alternative splicing;ATP-binding;Direct protei...</td>
<td>MLLWPLRGWAARALRCFGPGSRGSPASGPGPRRVQRRAWPPDKEQE...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>O00238</td>
<td>BMR1B_HUMAN</td>
<td>Bone morphogenetic protein receptor type-1B (B...</td>
<td>BMPR1B</td>
<td>1</td>
<td>1</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Alternative splicing;ATP-binding;...</td>
<td>MLLRSAGKLNVGTKKEDGESTAPTPRPKVLRCKCHHHCPEDSVNNI...</td>
</tr>
</tbody>
</table>

</div>

## Sugiyama dataset

### Data

Go to https://www.nature.com/articles/s41598-019-46385-4#Sec21, download
table S2.

Modify the header (2 lines to 1 line) so that it can be read by pandas.

``` python
df = pd.read_csv('raw/Large_scale_S2.csv').iloc[:,:-2]
```

``` python
df.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Type</th>
<th data-quarto-table-cell-role="th">Kinase</th>
<th data-quarto-table-cell-role="th">Substrate_uniprot</th>
<th data-quarto-table-cell-role="th">Position</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>TK</td>
<td>ABL1</td>
<td>1433B_HUMAN</td>
<td>S212</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>TK</td>
<td>ABL1</td>
<td>1433B_HUMAN</td>
<td>Y151</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>TK</td>
<td>ABL1</td>
<td>1433B_HUMAN</td>
<td>Y21</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>TK</td>
<td>ABL1</td>
<td>1433B_HUMAN</td>
<td>Y50</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>TK</td>
<td>ABL1</td>
<td>1433E_HUMAN</td>
<td>Y152</td>
</tr>
</tbody>
</table>

</div>

``` python
df.shape
```

    (198536, 4)

### Kinase mapping

``` python
# pd.DataFrame(df.Kinase.unique())[0].str.split('/').str[0].to_csv('raw/sugiyama_kinase_name.csv')
```

Map each kinase name with uniprot ID

``` python
kinase_id = pd.read_csv('raw/LS_info2.csv').iloc[:,:3]
```

``` python
df = df.merge(kinase_id)
```

``` python
df
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Type</th>
<th data-quarto-table-cell-role="th">Kinase</th>
<th data-quarto-table-cell-role="th">Substrate_uniprot</th>
<th data-quarto-table-cell-role="th">Position</th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">kinase_paper</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>TK</td>
<td>ABL1</td>
<td>1433B_HUMAN</td>
<td>S212</td>
<td>P00519</td>
<td>ABL1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>TK</td>
<td>ABL1</td>
<td>1433B_HUMAN</td>
<td>Y151</td>
<td>P00519</td>
<td>ABL1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>TK</td>
<td>ABL1</td>
<td>1433B_HUMAN</td>
<td>Y21</td>
<td>P00519</td>
<td>ABL1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>TK</td>
<td>ABL1</td>
<td>1433B_HUMAN</td>
<td>Y50</td>
<td>P00519</td>
<td>ABL1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>TK</td>
<td>ABL1</td>
<td>1433E_HUMAN</td>
<td>Y152</td>
<td>P00519</td>
<td>ABL1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">198531</td>
<td>LK</td>
<td>SPHK2</td>
<td>TICN3_HUMAN</td>
<td>T118</td>
<td>Q9NRA0</td>
<td>NaN</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">198532</td>
<td>LK</td>
<td>SPHK2</td>
<td>TPM4_HUMAN</td>
<td>T241</td>
<td>Q9NRA0</td>
<td>NaN</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">198533</td>
<td>LK</td>
<td>SPHK2</td>
<td>ULK3_HUMAN</td>
<td>S305</td>
<td>Q9NRA0</td>
<td>NaN</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">198534</td>
<td>LK</td>
<td>SPHK2</td>
<td>ZRAB2_HUMAN</td>
<td>S165</td>
<td>Q9NRA0</td>
<td>NaN</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">198535</td>
<td>LK</td>
<td>SPHK2</td>
<td>ZRAB2_HUMAN</td>
<td>S181</td>
<td>Q9NRA0</td>
<td>NaN</td>
</tr>
</tbody>
</table>

<p>198536 rows × 6 columns</p>
</div>

### Substrate mapping

``` python
# pd.DataFrame(df.Substrate_uniprot.unique()).to_csv('raw/sugiyama_uniprot_id.csv')
```

ID mapping of `Substrate_uniprot` to uniprot ID in
https://www.uniprot.org/id-mapping

3,753 IDs were mapped to 3,753 results

375 ID were not mapped

``` python
df = map_substrate('raw/idmapping_2025_03_02.xlsx',df,'Substrate_uniprot')
```

    /home/sky1ove/miniconda3/lib/python3.13/site-packages/logomaker/../openpyxl/styles/stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
      warn("Workbook contains no default style, apply openpyxl's default")

    Shape before processing (198536, 6)
    Species counts: substrate_species
    Homo sapiens (Human)    187129
    Name: count, dtype: int64
    Removing non-human substrates
    Shape after processing (187129, 9)

``` python
# lipid kinase
df[df.Type=='LK'].Kinase.unique()
```

    array(['PIK3C3', 'PIK3CB/PIK3R1', 'PIK3CD/PIK3R1', 'PIK3CG/PIK3R5',
           'PIK3CG', 'PIK4CA', 'PIP4K2B', 'PIP5K1C', 'SPHK1', 'SPHK2'],
          dtype=object)

``` python
df.isna().sum()
```

    Type                      0
    Kinase                    0
    Position                  0
    kinase_uniprot        16591
    kinase_paper          18891
    substrate_uniprot         0
    substrate_genes          11
    substrate_sequence        0
    substrate_species         0
    dtype: int64

``` python
df.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Type</th>
<th data-quarto-table-cell-role="th">Kinase</th>
<th data-quarto-table-cell-role="th">Position</th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">kinase_paper</th>
<th data-quarto-table-cell-role="th">substrate_uniprot</th>
<th data-quarto-table-cell-role="th">substrate_genes</th>
<th data-quarto-table-cell-role="th">substrate_sequence</th>
<th data-quarto-table-cell-role="th">substrate_species</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>TK</td>
<td>ABL1</td>
<td>S212</td>
<td>P00519</td>
<td>ABL1</td>
<td>P31946</td>
<td>YWHAB</td>
<td>MTMDKSELVQKAKLAEQAERYDDMAAAMKAVTEQGHELSNEERNLL...</td>
<td>Homo sapiens (Human)</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>TK</td>
<td>ABL1</td>
<td>Y151</td>
<td>P00519</td>
<td>ABL1</td>
<td>P31946</td>
<td>YWHAB</td>
<td>MTMDKSELVQKAKLAEQAERYDDMAAAMKAVTEQGHELSNEERNLL...</td>
<td>Homo sapiens (Human)</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>TK</td>
<td>ABL1</td>
<td>Y21</td>
<td>P00519</td>
<td>ABL1</td>
<td>P31946</td>
<td>YWHAB</td>
<td>MTMDKSELVQKAKLAEQAERYDDMAAAMKAVTEQGHELSNEERNLL...</td>
<td>Homo sapiens (Human)</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>TK</td>
<td>ABL1</td>
<td>Y50</td>
<td>P00519</td>
<td>ABL1</td>
<td>P31946</td>
<td>YWHAB</td>
<td>MTMDKSELVQKAKLAEQAERYDDMAAAMKAVTEQGHELSNEERNLL...</td>
<td>Homo sapiens (Human)</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>TK</td>
<td>ABL1</td>
<td>Y152</td>
<td>P00519</td>
<td>ABL1</td>
<td>P62258</td>
<td>YWHAE</td>
<td>MDDREDLVYQAKLAEQAERYDEMVESMKKVAGMDVELTVEERNLLS...</td>
<td>Homo sapiens (Human)</td>
</tr>
</tbody>
</table>

</div>

``` python
df = df.rename(columns = {'Position':'site'})
```

``` python
df.shape
```

    (187129, 9)

### Validate site

``` python
df['site_match'] =validate_site_df(df,'site','substrate_sequence')
```

``` python
df.site_match.value_counts()
```

    site_match
    1    184948
    0      2181
    Name: count, dtype: int64

``` python
df=df[df.site_match==1]
```

``` python
df.shape
```

    (184948, 10)

### Save

``` python
df['source']='Sugiyama'
```

``` python
# df.to_excel('raw/sugiyama.xlsx',index=False)
```

## PhosphoSitePlus (PSP)

### Data

PSP paper: https://academic.oup.com/nar/article/43/D1/D512/2439467

Go to https://www.phosphosite.org/staticDownloads, download
`Kinase_Substrate_Dataset.txt`

``` python
import pandas as pd
```

``` python
psp = pd.read_csv('raw/Kinase_Substrate_Dataset_final.csv')
```

``` python
psp.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">GENE</th>
<th data-quarto-table-cell-role="th">Kinase</th>
<th data-quarto-table-cell-role="th">KIN_ACC_ID</th>
<th data-quarto-table-cell-role="th">KIN_ORGANISM</th>
<th data-quarto-table-cell-role="th">SUBSTRATE</th>
<th data-quarto-table-cell-role="th">SUB_GENE_ID</th>
<th data-quarto-table-cell-role="th">SUB_ACC_ID</th>
<th data-quarto-table-cell-role="th">SUB_GENE</th>
<th data-quarto-table-cell-role="th">SUB_ORGANISM</th>
<th data-quarto-table-cell-role="th">SUB_MOD_RSD</th>
<th data-quarto-table-cell-role="th">SITE_GRP_ID</th>
<th data-quarto-table-cell-role="th">substrate</th>
<th data-quarto-table-cell-role="th">DOMAIN</th>
<th data-quarto-table-cell-role="th">IN_VIVO_RXN</th>
<th data-quarto-table-cell-role="th">IN_VITRO_RXN</th>
<th data-quarto-table-cell-role="th">CST_CAT#</th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">kinase_paper</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>Dyrk2</td>
<td>DYRK2</td>
<td>Q5U4C9</td>
<td>mouse</td>
<td>NDEL1</td>
<td>83431.0</td>
<td>Q9ERR1</td>
<td>Ndel1</td>
<td>mouse</td>
<td>S336</td>
<td>1869686801</td>
<td>LGSsRPSsAPGMLPL</td>
<td>NaN</td>
<td></td>
<td>X</td>
<td>NaN</td>
<td>Q92630</td>
<td>DYRK2</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>Q92630</td>
<td>human</td>
<td>GLI2</td>
<td>14633.0</td>
<td>Q0VGT2</td>
<td>Gli2</td>
<td>mouse</td>
<td>S385</td>
<td>3339001</td>
<td>AEGLRPAsPLGLTQE</td>
<td>NaN</td>
<td></td>
<td>X</td>
<td>NaN</td>
<td>Q92630</td>
<td>DYRK2</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>Q92630</td>
<td>human</td>
<td>SIAH2</td>
<td>6478.0</td>
<td>O43255</td>
<td>SIAH2</td>
<td>human</td>
<td>S68</td>
<td>5393502</td>
<td>GGGAGPVsPQHHELT</td>
<td>NaN</td>
<td>X</td>
<td></td>
<td>NaN</td>
<td>Q92630</td>
<td>DYRK2</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>Q92630</td>
<td>human</td>
<td>CARHSP1</td>
<td>23589.0</td>
<td>Q9Y2V2</td>
<td>CARHSP1</td>
<td>human</td>
<td>S41</td>
<td>455702</td>
<td>LRGNVVPsPLPtRRt</td>
<td>NaN</td>
<td></td>
<td>X</td>
<td>NaN</td>
<td>Q92630</td>
<td>DYRK2</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>Q92630</td>
<td>human</td>
<td>Doublecortin iso2</td>
<td>1641.0</td>
<td>O43602-2</td>
<td>DCX</td>
<td>human</td>
<td>S306</td>
<td>454122</td>
<td>GPMRRSKsPADSANG</td>
<td>NaN</td>
<td>X</td>
<td></td>
<td>NaN</td>
<td>Q92630</td>
<td>DYRK2</td>
</tr>
</tbody>
</table>

</div>

``` python
psp =psp[psp.KIN_ORGANISM=='human'].reset_index(drop=True) 

psp =psp[psp.SUB_ORGANISM=='human'].reset_index(drop=True)
```

``` python
psp.shape
```

    (14081, 18)

``` python
psp = psp[['KIN_ACC_ID','kinase_paper','GENE','SUB_ACC_ID','SUB_GENE','SUB_MOD_RSD','substrate']]
```

``` python
psp.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">KIN_ACC_ID</th>
<th data-quarto-table-cell-role="th">kinase_paper</th>
<th data-quarto-table-cell-role="th">GENE</th>
<th data-quarto-table-cell-role="th">SUB_ACC_ID</th>
<th data-quarto-table-cell-role="th">SUB_GENE</th>
<th data-quarto-table-cell-role="th">SUB_MOD_RSD</th>
<th data-quarto-table-cell-role="th">substrate</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>O43255</td>
<td>SIAH2</td>
<td>S68</td>
<td>GGGAGPVsPQHHELT</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>Q9Y2V2</td>
<td>CARHSP1</td>
<td>S41</td>
<td>LRGNVVPsPLPtRRt</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>O43602-2</td>
<td>DCX</td>
<td>S306</td>
<td>GPMRRSKsPADSANG</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>P30304</td>
<td>CDC25A</td>
<td>S283</td>
<td>PErsQEEsPPGSTKr</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>O43255</td>
<td>SIAH2</td>
<td>T119</td>
<td>PTCRGALtPSIRNLA</td>
</tr>
</tbody>
</table>

</div>

### Substrate mapping

``` python
# psp.SUB_ACC_ID.drop_duplicates().to_csv('raw/psp_substrate_id.csv')
```

4,441 IDs were mapped to 4,446 results

19 ID were not mapped: NP_001184222 NP_001100737 AAA40678 HSBO22
NP_776683 NP_579829 P18433-2 NP_001076191 NP_001099740 NP_001005762
NP_001178533 XP_008773743 NP_001104263 ABR15760 AAB24204 AAB24205
BAA34185.2 NP_001103022 AAC50053

IDs with multiple results: P62991, P62988

``` python
psp = map_substrate('raw/idmapping_2025_03_02_psp.xlsx',psp,'SUB_ACC_ID')
```

    /home/sky1ove/miniconda3/lib/python3.13/site-packages/openpyxl/styles/stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
      warn("Workbook contains no default style, apply openpyxl's default")

    Shape before processing (14081, 7)
    Species counts: substrate_species
    Homo sapiens (Human)    14069
    Name: count, dtype: int64
    Removing non-human substrates
    Shape after processing (14069, 10)

``` python
psp.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">KIN_ACC_ID</th>
<th data-quarto-table-cell-role="th">kinase_paper</th>
<th data-quarto-table-cell-role="th">GENE</th>
<th data-quarto-table-cell-role="th">SUB_GENE</th>
<th data-quarto-table-cell-role="th">SUB_MOD_RSD</th>
<th data-quarto-table-cell-role="th">substrate</th>
<th data-quarto-table-cell-role="th">substrate_uniprot</th>
<th data-quarto-table-cell-role="th">substrate_genes</th>
<th data-quarto-table-cell-role="th">substrate_sequence</th>
<th data-quarto-table-cell-role="th">substrate_species</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>SIAH2</td>
<td>S68</td>
<td>GGGAGPVsPQHHELT</td>
<td>O43255</td>
<td>SIAH2</td>
<td>MSRPSSTGPSANKPCSKQPPPQPQHTPSPAAPPAAATISAAGPGSS...</td>
<td>Homo sapiens (Human)</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>CARHSP1</td>
<td>S41</td>
<td>LRGNVVPsPLPtRRt</td>
<td>Q9Y2V2</td>
<td>CARHSP1</td>
<td>MSSEPPPPPQPPTHQASVGLLDTPRSRERSPSPLRGNVVPSPLPTR...</td>
<td>Homo sapiens (Human)</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>DCX</td>
<td>S306</td>
<td>GPMRRSKsPADSANG</td>
<td>O43602</td>
<td>DCX DBCN LISX</td>
<td>MELDFGHFDERDKTSRNMRGSRMNGLPSPTHSAHCSFYRTRTLQAL...</td>
<td>Homo sapiens (Human)</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>CDC25A</td>
<td>S283</td>
<td>PErsQEEsPPGSTKr</td>
<td>P30304</td>
<td>CDC25A</td>
<td>MELGPEPPHRRRLLFACSPPPASQPVVKALFGASAAGGLSPVTNLT...</td>
<td>Homo sapiens (Human)</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>SIAH2</td>
<td>T119</td>
<td>PTCRGALtPSIRNLA</td>
<td>O43255</td>
<td>SIAH2</td>
<td>MSRPSSTGPSANKPCSKQPPPQPQHTPSPAAPPAAATISAAGPGSS...</td>
<td>Homo sapiens (Human)</td>
</tr>
</tbody>
</table>

</div>

``` python
psp.shape
```

    (14069, 10)

### Kinase mapping

``` python
# psp.KIN_ACC_ID.drop_duplicates().to_csv('raw/psp_kin_id.csv')
```

Mapp KIN_ACC_ID to uniprot id

440 IDs were mapped to 440 results

1 ID was not mapped:

AAA58698

``` python
kinase_id = pd.read_excel('raw/idmapping_2025_03_02_psp_kinase.xlsx')[['From','Entry','Gene Names']]
```

    /home/sky1ove/miniconda3/lib/python3.13/site-packages/openpyxl/styles/stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
      warn("Workbook contains no default style, apply openpyxl's default")

``` python
kinase_id.columns = ['KIN_ACC_ID','kinase_uniprot','kinase_genes']
```

``` python
kinase_id[kinase_id.KIN_ACC_ID=="A9UF07"]
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">KIN_ACC_ID</th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">kinase_genes</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">124</td>
<td>A9UF07</td>
<td>A9UF07</td>
<td>BCR/ABL fusion</td>
</tr>
</tbody>
</table>

</div>

``` python
psp['KIN_ACC_ID'] = psp['KIN_ACC_ID'].replace('A9UF07','P00519')
```

``` python
psp = psp.merge(kinase_id)
```

``` python
psp = psp.rename(columns={'SUB_MOD_RSD':'site'})
```

``` python
psp = psp[['kinase_uniprot','kinase_genes','kinase_paper','substrate_uniprot','substrate_genes','site','substrate_sequence','substrate','substrate_species']]
```

``` python
psp.isna().sum()
```

    kinase_uniprot          0
    kinase_genes            0
    kinase_paper          134
    substrate_uniprot       0
    substrate_genes         0
    site                    0
    substrate_sequence      0
    substrate               0
    substrate_species       0
    dtype: int64

``` python
psp = psp.dropna(subset='substrate_sequence').reset_index(drop=True)
```

### Check if kinase-uniprot belongs to kinase class

Not mapped:

``` python
from katlas.data import *
```

``` python
kinase_uniprot=Data.get_kinase_uniprot()
```

``` python
psp[~psp.kinase_uniprot.isin(kinase_uniprot.Entry)].kinase_genes.value_counts()
```

    kinase_genes
    CSNK2B CK2N G5A                13
    PRKAG2                          9
    PRKAB1 AMPK                     9
    GTF2F1 RAP74                    4
    BLVRA BLVR BVR                  3
    TGM2                            2
    RET/PTC2                        2
    PHKA1 PHKA                      1
    HSPA5 GRP78                     1
    ENPP3 PDNP3                     1
    PIK3R1 GRB1                     1
    CERT1 CERT COL4A3BP STARD11     1
    JMJD6 KIAA0585 PSR PTDSR        1
    Name: count, dtype: int64

CSNK2B belongs to CK2, PRKAB1, PRKAG2 belongs to AMPK subunit; as their
count is not many, we can simply ignore them

``` python
psp = psp[psp.kinase_uniprot.isin(kinase_uniprot.Entry)].copy()
```

``` python
psp.shape
```

    (14016, 9)

### Validate site

``` python
from katlas.utils import *
```

Check if it match with substrate sequence

``` python
psp['site_seq']=extract_site_seq(psp,seq_col='substrate_sequence',site_col='site',n=7)
```

    100%|███████████████████████████████████████████████████████████████████████| 14016/14016 [00:00<00:00, 26427.27it/s]

``` python
(psp['site_seq']==psp['substrate'].str.upper()).value_counts()
```

    True     13156
    False      860
    Name: count, dtype: int64

``` python
unmatch = psp[~(psp['site_seq']==psp['substrate'].str.upper())]
```

``` python
unmatch.kinase_genes.value_counts().head(10)
```

    kinase_genes
    PRKACA PKACA                                  40
    MAPK1 ERK2 PRKM1 PRKM2                        40
    SRC SRC1                                      39
    CDK1 CDC2 CDC28A CDKN1 P34CDC2                38
    GSK3B                                         37
    MAPK3 ERK1 PRKM3                              35
    AURKB AIK2 AIM1 AIRK2 ARK2 STK1 STK12 STK5    33
    PRKCA PKCA PRKACA                             33
    PRKCB PKCB PRKCB1                             25
    CDK5 CDKN5 PSSALRE                            25
    Name: count, dtype: int64

We’ll drop the rows with unmatch substrate sequence

``` python
psp = psp[psp['site_seq']==psp['substrate'].str.upper()]
```

Drop sites with center aa not in S,T or Y

``` python
psp['site_seq'].str[7].value_counts()
```

    site_seq
    S    8303
    T    2812
    Y    2029
    H       6
    K       4
    R       2
    Name: count, dtype: int64

``` python
psp = psp[psp['site_seq'].str[7].str.upper().isin(list('STY'))]
```

Drop site with unmatched residue and position

``` python
psp['site_match'] =validate_site_df(psp,'site','substrate_sequence')
```

``` python
psp['site_match'].value_counts()
```

    site_match
    1    13144
    Name: count, dtype: int64

``` python
psp.shape
```

    (13144, 11)

### Save

``` python
psp['source']='PSP'
```

``` python
psp.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">kinase_genes</th>
<th data-quarto-table-cell-role="th">kinase_paper</th>
<th data-quarto-table-cell-role="th">substrate_uniprot</th>
<th data-quarto-table-cell-role="th">substrate_genes</th>
<th data-quarto-table-cell-role="th">site</th>
<th data-quarto-table-cell-role="th">substrate_sequence</th>
<th data-quarto-table-cell-role="th">substrate</th>
<th data-quarto-table-cell-role="th">substrate_species</th>
<th data-quarto-table-cell-role="th">site_seq</th>
<th data-quarto-table-cell-role="th">site_match</th>
<th data-quarto-table-cell-role="th">source</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>O43255</td>
<td>SIAH2</td>
<td>S68</td>
<td>MSRPSSTGPSANKPCSKQPPPQPQHTPSPAAPPAAATISAAGPGSS...</td>
<td>GGGAGPVsPQHHELT</td>
<td>Homo sapiens (Human)</td>
<td>GGGAGPVSPQHHELT</td>
<td>1</td>
<td>PSP</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>Q9Y2V2</td>
<td>CARHSP1</td>
<td>S41</td>
<td>MSSEPPPPPQPPTHQASVGLLDTPRSRERSPSPLRGNVVPSPLPTR...</td>
<td>LRGNVVPsPLPtRRt</td>
<td>Homo sapiens (Human)</td>
<td>LRGNVVPSPLPTRRT</td>
<td>1</td>
<td>PSP</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>P30304</td>
<td>CDC25A</td>
<td>S283</td>
<td>MELGPEPPHRRRLLFACSPPPASQPVVKALFGASAAGGLSPVTNLT...</td>
<td>PErsQEEsPPGSTKr</td>
<td>Homo sapiens (Human)</td>
<td>PERSQEESPPGSTKR</td>
<td>1</td>
<td>PSP</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>O43255</td>
<td>SIAH2</td>
<td>T119</td>
<td>MSRPSSTGPSANKPCSKQPPPQPQHTPSPAAPPAAATISAAGPGSS...</td>
<td>PTCRGALtPSIRNLA</td>
<td>Homo sapiens (Human)</td>
<td>PTCRGALTPSIRNLA</td>
<td>1</td>
<td>PSP</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">5</td>
<td>Q92630</td>
<td>DYRK2</td>
<td>DYRK2</td>
<td>O75449</td>
<td>KATNA1</td>
<td>S42</td>
<td>MSLLMISENVKLAREYALLGNYDSAMVYYQGVLDQMNKYLYSVKDT...</td>
<td>QMNKYLYsVkDTYLQ</td>
<td>Homo sapiens (Human)</td>
<td>QMNKYLYSVKDTYLQ</td>
<td>1</td>
<td>PSP</td>
</tr>
</tbody>
</table>

</div>

``` python
# psp.to_excel('raw/psp_human.xlsx',index=False)
```

## KiNET (ESPD + iPTMNet)

> an integration of PSP, iPTMNet, EPSD

### Data

KiNET paper: https://www.nature.com/articles/s41540-024-00442-5

KiNET web: https://kinet.kinametrix.com/

iPTMnet paper: https://pmc.ncbi.nlm.nih.gov/articles/PMC5753337/;

iPTMnet web: https://research.bioinformatics.udel.edu/iptmnet/ ; can’t
directly download through Download

EPSD paper:
https://academic.oup.com/bib/article/22/1/298/5686325?login=false

EPSD web: https://epsd.biocuckoo.cn/Download.php

Go to https://kinet.kinametrix.com/, and click Download the full KiNet
interaction dataset

``` python
df = pd.read_csv('raw/ksi_source_full_dataset.csv')
```

``` python
df = df.dropna(subset='Kinase')
```

``` python
df['Source Database'].value_counts()
```

    Source Database
    PhosphoSitePlus    13135
    EPSD               10442
    iPTMNet             3846
    Name: count, dtype: int64

As we already included PSP, we will drop it

``` python
df = df[df['Source Database']!='PhosphoSitePlus']
```

``` python
df.shape
```

    (14288, 8)

### Kinase mapping

``` python
df.Kinase.isin(kinase_uniprot.Entry).value_counts()
```

    Kinase
    True     14283
    False        5
    Name: count, dtype: int64

``` python
df = df[df.Kinase.isin(kinase_uniprot.Entry)]
```

### Substrate mapping

``` python
# df.Substrate.drop_duplicates().to_csv('raw/KiNet_substrate_id.csv')
```

Map the id to uniprot seq

``` python
df = map_substrate('raw/idmapping_2025_03_02_KiNET_substrate.xlsx',df,'Substrate')
```

    /home/sky1ove/miniconda3/lib/python3.13/site-packages/logomaker/../openpyxl/styles/stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
      warn("Workbook contains no default style, apply openpyxl's default")

    Shape before processing (14283, 8)
    Species counts: substrate_species
    Homo sapiens (Human)    14283
    Name: count, dtype: int64
    Removing non-human substrates
    Shape after processing (14283, 11)

``` python
# substrate_id = pd.read_excel('raw/idmapping_2025_03_02_KiNET_substrate.xlsx')[['From','Entry','Gene Names','Sequence']]

# substrate_id.columns = ['Substrate','substrate_uniprot','substrate_genes','substrate_sequence']

# df = df.merge(substrate_id)
```

``` python
df = df[['Kinase','Kinase Name','substrate_uniprot','substrate_genes','Site','Source Database','Evidence','substrate_sequence']]
```

``` python
df.columns
```

    Index(['Kinase', 'Kinase Name', 'substrate_uniprot', 'substrate_genes', 'Site',
           'Source Database', 'Evidence', 'substrate_sequence'],
          dtype='object')

``` python
df.columns = ['kinase_uniprot', 'Kinase Name', 'substrate_uniprot', 'substrate_genes', 'site',

       'source', 'evidence', 'substrate_sequence']
```

``` python
df.shape
```

    (14283, 8)

### Validate site

We drop site with site residue info unmatch with its site position info
in the protein sequence

``` python
df['site_match'] =validate_site_df(df,'site','substrate_sequence')
```

``` python
df.site_match.value_counts()
```

    site_match
    1    13786
    0      497
    Name: count, dtype: int64

``` python
df[df.site_match==0].source.value_counts()
```

    source
    iPTMNet    368
    EPSD       129
    Name: count, dtype: int64

``` python
df[df.site_match==1].source.value_counts()
```

    source
    EPSD       10308
    iPTMNet     3478
    Name: count, dtype: int64

``` python
df=df[df.site_match==1]
```

``` python
df.shape
```

    (13786, 9)

### Save

``` python
df.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">Kinase Name</th>
<th data-quarto-table-cell-role="th">substrate_uniprot</th>
<th data-quarto-table-cell-role="th">substrate_genes</th>
<th data-quarto-table-cell-role="th">site</th>
<th data-quarto-table-cell-role="th">source</th>
<th data-quarto-table-cell-role="th">evidence</th>
<th data-quarto-table-cell-role="th">substrate_sequence</th>
<th data-quarto-table-cell-role="th">site_match</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>O00141</td>
<td>SGK1</td>
<td>O00213</td>
<td>APBB1 FE65 RIR</td>
<td>S610</td>
<td>EPSD</td>
<td>Unspecified experimental method</td>
<td>MSVPSSLSQSAINANSHGGPALSLPLPLHAAHNQLLNAKLQATAVG...</td>
<td>1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>O00141</td>
<td>SGK1</td>
<td>O14920</td>
<td>IKBKB IKKB</td>
<td>S181</td>
<td>EPSD</td>
<td>Unspecified experimental method</td>
<td>MSWSPSLTTQTCGAWEMKERLGTGGFGNVIRWHNQETGEQIAIKQC...</td>
<td>1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>O00141</td>
<td>SGK1</td>
<td>O15111</td>
<td>CHUK IKKA TCF16</td>
<td>S180</td>
<td>EPSD</td>
<td>Unspecified experimental method</td>
<td>MERPPGLRPGAGGPWEMRERLGTGGFGNVCLYQHRELDLKIAIKSC...</td>
<td>1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>O00141</td>
<td>SGK1</td>
<td>O43524</td>
<td>FOXO3 FKHRL1 FOXO3A</td>
<td>T32</td>
<td>EPSD</td>
<td>Unspecified experimental method</td>
<td>MAEAPASPAPLSPLEVELDPEFEPQSRPRSCTWPLQRPELQASPAK...</td>
<td>1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>O00141</td>
<td>SGK1</td>
<td>O43524</td>
<td>FOXO3 FKHRL1 FOXO3A</td>
<td>T32</td>
<td>iPTMNet</td>
<td>Text mining</td>
<td>MAEAPASPAPLSPLEVELDPEFEPQSRPRSCTWPLQRPELQASPAK...</td>
<td>1</td>
</tr>
</tbody>
</table>

</div>

``` python
# df.to_excel('raw/KiNet.xlsx',index=False)

# df[df['source'] =='EPSD'].to_excel('raw/EPSD.xlsx',index=False)

# df[df['source'] =='iPTMNet'].to_excel('raw/iPTMNet.xlsx',index=False)
```

## Phospho.ELM

### Data

paper: https://pmc.ncbi.nlm.nih.gov/articles/PMC3013696/

Go to http://phospho.elm.eu.org/dataset.html, fill info–\>accept, then
download the dataset

Open dump in excel, filter human and non-blank from kinases

``` python
elm=pd.read_csv('raw/phosphoELM.csv')
```

``` python
elm.shape
```

    (3599, 6)

``` python
elm.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">substrate_uniprot</th>
<th data-quarto-table-cell-role="th">position</th>
<th data-quarto-table-cell-role="th">acceptor</th>
<th data-quarto-table-cell-role="th">kinase</th>
<th data-quarto-table-cell-role="th">LTP_HTP</th>
<th data-quarto-table-cell-role="th">species</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>O14543</td>
<td>204</td>
<td>Y</td>
<td>Lck</td>
<td>LTP</td>
<td>Homo sapiens</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>O14543</td>
<td>221</td>
<td>Y</td>
<td>Lck</td>
<td>LTP</td>
<td>Homo sapiens</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>O14746</td>
<td>824</td>
<td>S</td>
<td>PKB_group</td>
<td>LTP</td>
<td>Homo sapiens</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>O14746</td>
<td>227</td>
<td>S</td>
<td>PKB_group</td>
<td>LTP</td>
<td>Homo sapiens</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>O14746</td>
<td>707</td>
<td>Y</td>
<td>SRC</td>
<td>LTP</td>
<td>Homo sapiens</td>
</tr>
</tbody>
</table>

</div>

``` python
elm.kinase = elm.kinase.str.upper()
```

### Kinase mapping

``` python
# kinase_id = pd.read_excel('raw/uniprot_human_keyword_kinase.xlsx')
kinase_id = Data.get_kinase_uniprot()

kinase_id['Gene Names'] = kinase_id['Gene Names'].str.split(' ')

kinase_id = kinase_id.explode('Gene Names')
```

``` python
gene2uniprot = kinase_id.set_index('Gene Names')['Entry'].to_dict()
```

``` python
dup_name = set(kinase_id[kinase_id['Gene Names'].duplicated(keep=False)]['Gene Names'])
```

``` python
dup_name.remove('PAK1')

dup_name.remove('PASK')
```

``` python
gene2uniprot['PAK1'] = 'Q13153'

gene2uniprot['PASK']='Q96RG2'
```

``` python
elm = elm[~elm.kinase.isin(dup_name)].copy()
```

``` python
elm[~elm.kinase.isin(gene2uniprot.keys())].kinase.unique()
```

    array(['PKB_GROUP', 'IKK_GROUP', 'AURORA B', 'PDK-1', 'PKC_THETA',
           'SGK_GROUP', 'IKK_BETA', 'CDK_GROUP', 'CAM-KII_ALPHA', 'PKC_GROUP',
           'PKC_ALPHA', 'CK2_GROUP', 'CK2_ALPHA', 'GSK-3_GROUP',
           'GSK-3_ALPHA', 'GSK-3_BETA', 'PKA_GROUP', 'RSK_GROUP', 'PKB_BETA',
           'MAPK_GROUP', 'PAK_GROUP', 'DNA-PK', 'JNK_GROUP', 'CK1_GROUP',
           'AURORA A', 'P70S6KB', 'PKG/CGK_GROUP', 'PKC_DELTA', 'PDGFR_BETA',
           'CAM-KII_GROUP', 'GRK-2', 'GRK-5', 'GRK_GROUP', 'CK1_ALPHA',
           'PKC_EPSILON', 'ROCK_GROUP', 'PKG1/CGK-I', 'PDGFR_GROUP',
           'P70S6K_GROUP', 'P70S6K', 'PHK_GROUP', 'PKC_BETA', 'DMPK_GROUP',
           'CK1_DELTA', 'CK2_BETA', 'RSK-3', 'RSK-2', 'CAM-KIV', 'RSK-1',
           'IKK_ALPHA', 'CK1_EPSILON', 'MAP2K_GROUP', 'PIM-1', 'EG3 KINASE',
           'GRK-4', 'PKC_ZETA', 'PDKC', 'AMPK_GROUP', 'FGFR_GROUP',
           'JAK_GROUP', 'MAP3K_GROUP', 'RSK-5', 'IKK_EPSILON', 'CAM-KI_GROUP',
           'SRC_GROUP', 'PKA_ALPHA', 'PKC_ETA', 'PDK-2', 'GRK-6',
           'TITIN KINASE', 'CAM-KI_ALPHA', 'GRK-1', 'PDGFR_ALPHA', 'CCDPK',
           'MARK_GROUP', 'P38_GROUP', 'GRK-3', 'DAPK_GROUP', 'MRCKA',
           'PKG2/CGK-II', 'PKC_GAMMA', 'PKC_IOTA', 'CAM-KK_ALPHA'],
          dtype=object)

map them to genes

``` python
ids = pd.read_csv('raw/elm_kinase_id.csv')
```

``` python
ids = ids.set_index('kinase')['kinase_gene'].to_dict()
```

``` python
elm['kinase_genes'] = elm.kinase.map(ids).fillna(elm.kinase)
```

``` python
# for kinase group, we only consider the first two items

elm['kinase_genes'] = elm.kinase_genes.str.split(' ').str[:2]
```

``` python
elm = elm.explode('kinase_genes')
```

``` python
elm[elm.kinase_genes.isin(dup_name)].kinase_genes.unique()
```

    array(['PRKACA'], dtype=object)

``` python
dup_name.remove('PRKACA')

gene2uniprot['PRKACA']= 'P17612'
```

``` python
elm[elm.kinase_genes.isin(dup_name)]
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">substrate_uniprot</th>
<th data-quarto-table-cell-role="th">position</th>
<th data-quarto-table-cell-role="th">acceptor</th>
<th data-quarto-table-cell-role="th">kinase</th>
<th data-quarto-table-cell-role="th">LTP_HTP</th>
<th data-quarto-table-cell-role="th">species</th>
<th data-quarto-table-cell-role="th">kinase_genes</th>
</tr>
</thead>
<tbody>
</tbody>
</table>

</div>

unmapped:

``` python
elm[elm.kinase_genes.map(gene2uniprot).isna()].kinase_genes.unique()
```

    array(['PHKA1', 'PHKA2', 'CSNK2B', 'CDPK1', 'CDPK2'], dtype=object)

``` python
elm['kinase_uniprot'] = elm.kinase_genes.map(gene2uniprot)
```

``` python
elm = elm.dropna(subset='kinase_uniprot')
```

### Substrate mapping

``` python
# elm.substrate_uniprot.drop_duplicates().to_csv('raw/elm_substrate_id.csv')
```

916 IDs were mapped to 919 results

10 ID were not mapped: ENSP00000328213 ENSP00000343690 ENSP00000352232
ENSP00000347528 ENSP00000357298 ENSP00000248996 ENSP00000357225
ENSP00000261937 ENSP00000248419 ENSP00000267569

Found two of the unmapped are kinases:

``` python
ensp = {'ENSP00000328213':'P06239','ENSP00000261937':'P35916'}
```

``` python
elm.substrate_uniprot = elm.substrate_uniprot.map(ensp).fillna(elm.substrate_uniprot)
```

``` python
elm = map_substrate('raw/idmapping_2025_03_12_elm.xlsx',elm,'substrate_uniprot')
```

    Shape before processing (4699, 8)
    Species counts: substrate_species
    Homo sapiens (Human)    4675
    Name: count, dtype: int64
    Removing non-human substrates
    Shape after processing (4675, 11)

    /home/sky1ove/miniconda3/lib/python3.13/site-packages/logomaker/../openpyxl/styles/stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
      warn("Workbook contains no default style, apply openpyxl's default")

``` python
elm.shape
```

    (4675, 11)

### Validate site

``` python
elm.acceptor.value_counts()
```

    acceptor
    S    3018
    T     908
    Y     749
    Name: count, dtype: int64

``` python
elm['site']=elm['acceptor']+elm['position'].astype(str)
```

``` python
elm['site_match'] =validate_site_df(elm,'site','substrate_sequence')
```

``` python
elm.site_match.value_counts()
```

    site_match
    1    4519
    0     156
    Name: count, dtype: int64

``` python
elm = elm[elm.site_match==1]
```

``` python
elm.shape
```

    (4519, 13)

### Save

``` python
elm.columns
```

    Index(['position', 'acceptor', 'kinase', 'LTP_HTP', 'species', 'kinase_genes',
           'kinase_uniprot', 'substrate_uniprot', 'substrate_genes',
           'substrate_sequence', 'substrate_species', 'site', 'site_match'],
          dtype='object')

``` python
col = ['kinase','kinase_uniprot', 'kinase_genes',

       'substrate_uniprot', 'substrate_genes','site','LTP_HTP', 'species', 

       'substrate_sequence',]
```

``` python
elm = elm[col]
```

``` python
elm['source']='ELM'
```

``` python
# elm.to_excel('raw/ELM.xlsx',index=False)
```

## Signor

### Data

paper: https://academic.oup.com/nar/article/51/D1/D631/6761728

web: https://signor.uniroma2.it/

Go to web link, Downloads –\> Latest Release download (Jan 2025)

Open in excel, filter mechanism to be phosphorylation

``` python
import pandas as pd
from katlas.data import *
```

``` python
sig = pd.read_excel('raw/signor_phosphorylation.xlsx')
```

``` python
sig.shape
```

    (12973, 27)

### Kinase mapping

TYPEA column contains complex proteins, we need to filter out the kinase
name

``` python
comp = sig[sig.TYPEA=='complex'].copy()
```

``` python
# comp.ENTITYA.drop_duplicates().to_csv('raw/sig_complex.csv')
```

``` python
comp_id = pd.read_csv('raw/sig_complex_label.csv')
```

``` python
comp_id = comp_id.set_index('ENTITYA')['kinase_gene']
```

``` python
comp['kinase_gene'] = comp.ENTITYA.map(comp_id)
```

``` python
comp = comp.dropna(subset='kinase_gene')
```

``` python
comp.shape
```

    (742, 28)

Fusion proteins

``` python
fus_id = {'BCR-ABL':'ABL1','EML4-ALK':'ALK'}
```

``` python
fus = sig[sig.TYPEA=='fusion protein'].copy()
```

``` python
fus['kinase_gene'] = fus.ENTITYA.map(fus_id)
```

For Protein family, we’ll make it apply to first two family members

``` python
fam = sig[sig.TYPEA=='proteinfamily'].copy()
```

``` python
# fam.ENTITYA.drop_duplicates().to_csv('raw/sig_fam.csv')
```

``` python
fam_id = pd.read_csv('raw/sig_fam_label.csv')
```

``` python
fam_id.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">ENTITYA</th>
<th data-quarto-table-cell-role="th">kinase_gene</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>ERK1/2</td>
<td>MAPK3 MAPK1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>AKT</td>
<td>AKT1 AKT2 AKT3</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>RPS6K</td>
<td>RPS6KB1 RPS6KB2</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>p38</td>
<td>MAPK14 MAPK11 MAPK12 MAPK13</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>JNK</td>
<td>MAPK8 MAPK9 MAPK10</td>
</tr>
</tbody>
</table>

</div>

``` python
fam_id = fam_id.set_index('ENTITYA')['kinase_gene']
```

``` python
fam['kinase_gene'] = fam.ENTITYA.map(fam_id)

fam = fam.dropna(subset='kinase_gene')
```

``` python
fam['kinase_gene'] = fam.kinase_gene.str.split(' ').str[:2]
```

``` python
fam = fam.explode('kinase_gene')
```

Protein:

``` python
pro = sig[sig.TYPEA=='protein'].copy()
```

for consistency:

``` python
pro['kinase_gene'] = pro['ENTITYA']
```

combine:

``` python
df = pd.concat([pro,comp,fus,fam])
```

Mapping:

``` python
# kinase_id = pd.read_excel('raw/uniprot_human_keyword_kinase.xlsx')
kinase_id =Data.get_kinase_uniprot()
```

``` python
kinase_id['Gene Names'] = kinase_id['Gene Names'].str.split(' ')
```

``` python
kinase_id = kinase_id.explode('Gene Names')
```

Some kinases share same gene names:

``` python
# kinase_id[kinase_id['Gene Names'].duplicated(keep=False)].sort_values('Gene Names').head()
```

We’ll drop them to prevent confusion.

``` python
dup_name = set(kinase_id[kinase_id['Gene Names'].duplicated(keep=False)]['Gene Names'])
```

``` python
df.kinase_gene.isin(dup_name).sum()
```

    np.int64(674)

``` python
df = df[~df.kinase_gene.isin(dup_name)].copy()
```

``` python
gene2uniprot = kinase_id.set_index('Gene Names')['Entry'].to_dict()
```

``` python
df.shape
```

    (13013, 28)

unmapped:

``` python
df[df.kinase_gene.str.upper().map(gene2uniprot).isna()].kinase_gene.unique()
```

    array(['HRAS', 'IL4R', 'CSNK2B', 'TLR4', 'IL6R', 'THAP12', 'PRKAR2A',
           'RHOA', 'AREG', 'RET/PTC2', 'PCSK7', 'CCNC', 'TGM2', 'SLC12A1',
           'RALA', 'IL6ST', 'CRK', 'SMAD9', 'PLCG1', 'ELOC', 'PPP2CA', 'CCR5',
           'GHR', 'CAD', 'SMO', 'FRS2', 'LEPR', 'BLVRA', 'TAB1', 'PRKAB1',
           'PIAS4', 'IFNAR1', 'HSP90AA1', 'PTPRJ', 'PRKAG2', 'CSN1S1',
           'GTF2F1', 'VASP', 'KRAS', 'SLC12A3', 'PRPF4B', 'CDKN2A', 'IL1R1',
           'PRKAR2B', 'IL15RA', 'KRT1', 'SMAD1', 'GTF2H1', 'CCR2', 'IL5RA',
           'PLCG2', 'SLC12A2', 'GTF2H2', 'TMIGD2', 'IKBKG', 'BORA', 'MNAT1',
           'PHKA1', 'BGLF4', 'RIN1', 'DLG1', 'CDK5RAP2', 'CCR1', 'IL10RA',
           'SMAD5', 'CCN4'], dtype=object)

``` python
df['kinase_uniprot'] = df.kinase_gene.str.upper().map(gene2uniprot)
```

``` python
df = df.dropna(subset='kinase_uniprot')
```

### Substrate mapping

``` python
df = df.dropna(subset='IDB')
```

We can’t trace the specific whole protein sequence based on SIGNOR ID,
so we filter them out

``` python
df = df[~df.IDB.str.contains('SIGNOR')]
```

``` python
# df.IDB.drop_duplicates().to_csv('raw/sig_substrate_id.csv')
```

2,298 IDs were mapped to 2,298 results

1 ID was not mapped: CHEBI:15721

Sequences are all from human

``` python
df = map_substrate('raw/idmapping_2025_03_12_signor.xlsx',df,'IDB')
```

    /home/sky1ove/miniconda3/lib/python3.13/site-packages/openpyxl/styles/stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
      warn("Workbook contains no default style, apply openpyxl's default")

    Shape before processing (12354, 29)
    Species counts: substrate_species
    Homo sapiens (Human)    12353
    Name: count, dtype: int64
    Removing non-human substrates
    Shape after processing (12353, 32)

``` python
df.shape
```

    (12353, 32)

### Validate site

``` python
df.RESIDUE = df.RESIDUE.str.split(';')
```

``` python
df = df.explode('RESIDUE')
```

``` python
df['acceptor'] = df['RESIDUE'].str[:3]
```

``` python
df.acceptor = df.acceptor.map({'Ser':'S','Thr':'T','Tyr':'Y'})
```

``` python
df['position'] = df['RESIDUE'].str[3:]
```

``` python
df['site']=df['acceptor']+df['position']
```

``` python
df = df.dropna(subset='site')
```

``` python
from katlas.utils import *
```

``` python
df['site_match'] =validate_site_df(df,'site','substrate_sequence')
```

``` python
df.site_match.value_counts()
```

    site_match
    1    11212
    0      118
    Name: count, dtype: int64

``` python
df = df[df.site_match==1]
```

``` python
df.shape
```

    (11212, 36)

### Save

``` python
df.columns
```

    Index(['ENTITYA', 'TYPEA', 'IDA', 'DATABASEA', 'ENTITYB', 'TYPEB', 'DATABASEB',
           'EFFECT', 'MECHANISM', 'RESIDUE', 'SEQUENCE', 'TAX_ID', 'CELL_DATA',
           'TISSUE_DATA', 'MODULATOR_COMPLEX', 'TARGET_COMPLEX', 'MODIFICATIONA',
           'MODASEQ', 'MODIFICATIONB', 'MODBSEQ', 'PMID', 'DIRECT', 'NOTES',
           'ANNOTATOR', 'SENTENCE', 'SIGNOR_ID', 'kinase_gene', 'kinase_uniprot',
           'substrate_uniprot', 'substrate_genes', 'substrate_sequence',
           'substrate_species', 'acceptor', 'position', 'site', 'site_match'],
          dtype='object')

``` python
col = ['kinase_uniprot', 'kinase_gene', 'ENTITYA', 'TYPEA', 

       'substrate_uniprot', 'substrate_genes','site', 'substrate_sequence']
```

``` python
df = df[col].copy()
```

``` python
df.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">kinase_gene</th>
<th data-quarto-table-cell-role="th">ENTITYA</th>
<th data-quarto-table-cell-role="th">TYPEA</th>
<th data-quarto-table-cell-role="th">substrate_uniprot</th>
<th data-quarto-table-cell-role="th">substrate_genes</th>
<th data-quarto-table-cell-role="th">site</th>
<th data-quarto-table-cell-role="th">substrate_sequence</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>P68400</td>
<td>CSNK2A1</td>
<td>CSNK2A1</td>
<td>protein</td>
<td>P05455</td>
<td>SSB</td>
<td>S366</td>
<td>MAENGDNEKMAALEAKICHQIEYYFGDFNLPRDKFLKEQIKLDEGW...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>P28482</td>
<td>MAPK1</td>
<td>MAPK1</td>
<td>protein</td>
<td>P43354</td>
<td>NR4A2 NOT NURR1 TINUR</td>
<td>S126</td>
<td>MPCVQAQYGSSPQGASPASQSYSYHSSGEYSSDFLTPEFVKFSMDL...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>P11362</td>
<td>FGFR1</td>
<td>FGFR1</td>
<td>protein</td>
<td>P56945</td>
<td>BCAR1 CAS CASS1 CRKAS</td>
<td>Y128</td>
<td>MNHLNVLAKALYDNVAESPDELSFRKGDIMTVLEQDTQGLDGWWLC...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>P27361</td>
<td>MAPK3</td>
<td>MAPK3</td>
<td>protein</td>
<td>P41182</td>
<td>BCL6 BCL5 LAZ3 ZBTB27 ZNF51</td>
<td>S343</td>
<td>MASPADSCIQFTRHASDVLLNLNRLRSRDILTDVVIVVSREQFRAH...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>Q16539</td>
<td>MAPK14</td>
<td>MAPK14</td>
<td>protein</td>
<td>Q02078</td>
<td>MEF2A MEF2</td>
<td>S408</td>
<td>MGRKKIQITRIMDERNRQVTFTKRKFGLMKKAYELSVLCDCEIALI...</td>
</tr>
</tbody>
</table>

</div>

``` python
df.shape
```

    (11212, 8)

``` python
df['source']="SIGNOR"
```

``` python
# df.to_excel('raw/signor.xlsx',index=False)
```

## GPS 6.0

### Data

paper: https://academic.oup.com/nar/article/51/W1/W243/7157529#409532969

web: https://gps.biocuckoo.cn/index.php

Go to paper link, download supplementary data - zip file, get Table S5

``` python
import pandas as pd
```

``` python
gps = pd.read_csv('raw/GPS6_tableS5.csv')
```

``` python
gps = gps[gps.source!="PhosphositePlus"]
```

``` python
gps = gps[gps.species=="Homo sapiens"]
```

``` python
gps.shape
```

    (6087, 6)

``` python
gps = gps[~gps.gene.str.contains('family')]
```

### Kinase mapping

``` python
from katlas.data import *
```

``` python
# kinase_id = pd.read_excel('raw/uniprot_human_keyword_kinase.xlsx')
kinase_id = Data.get_kinase_uniprot()
```

``` python
kinase_id['Gene Names'] = kinase_id['Gene Names'].str.split(' ')
```

``` python
kinase_id = kinase_id.explode('Gene Names')
```

Some kinases share same gene names:

``` python
kinase_id[kinase_id['Gene Names'].duplicated(keep=False)].sort_values('Gene Names').head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Entry</th>
<th data-quarto-table-cell-role="th">Entry Name</th>
<th data-quarto-table-cell-role="th">Protein names</th>
<th data-quarto-table-cell-role="th">Gene Names</th>
<th data-quarto-table-cell-role="th">uniprot_keyword_kinase</th>
<th data-quarto-table-cell-role="th">on_tree</th>
<th data-quarto-table-cell-role="th">Organism</th>
<th data-quarto-table-cell-role="th">Keywords</th>
<th data-quarto-table-cell-role="th">Sequence</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">586</td>
<td>Q9UIJ7</td>
<td>KAD3_HUMAN</td>
<td>GTP:AMP phosphotransferase AK3, mitochondrial ...</td>
<td>AK3</td>
<td>1</td>
<td>0</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Acetylation;Alternative splicing;...</td>
<td>MGASARLLRAVIMGAPGSGKGTVSSRITTHFELKHLSSGDLLRDNM...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">152</td>
<td>P27144</td>
<td>KAD4_HUMAN</td>
<td>Adenylate kinase 4, mitochondrial (EC 2.7.4.4)...</td>
<td>AK3</td>
<td>1</td>
<td>0</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Acetylation;ATP-binding;GTP-bindi...</td>
<td>MASKLLRAVILGPPGSGKGTVCQRIAQNFGLQHLSSGHFLRENIKA...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">586</td>
<td>Q9UIJ7</td>
<td>KAD3_HUMAN</td>
<td>GTP:AMP phosphotransferase AK3, mitochondrial ...</td>
<td>AK3L1</td>
<td>1</td>
<td>0</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Acetylation;Alternative splicing;...</td>
<td>MGASARLLRAVIMGAPGSGKGTVSSRITTHFELKHLSSGDLLRDNM...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">152</td>
<td>P27144</td>
<td>KAD4_HUMAN</td>
<td>Adenylate kinase 4, mitochondrial (EC 2.7.4.4)...</td>
<td>AK3L1</td>
<td>1</td>
<td>0</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Acetylation;ATP-binding;GTP-bindi...</td>
<td>MASKLLRAVILGPPGSGKGTVCQRIAQNFGLQHLSSGHFLRENIKA...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">609</td>
<td>Q9Y3D8</td>
<td>KAD6_HUMAN</td>
<td>Adenylate kinase isoenzyme 6 (AK6) (EC 2.7.4.3...</td>
<td>AK6</td>
<td>1</td>
<td>0</td>
<td>Homo sapiens (Human)</td>
<td>3D-structure;Alternative splicing;ATP-binding;...</td>
<td>MLLPNILLTGTPGVGKTTLGKELASKSGLKYINVGDLAREEQLYDG...</td>
</tr>
</tbody>
</table>

</div>

We’ll drop them to prevent confusion.

``` python
dup_name = set(kinase_id[kinase_id['Gene Names'].duplicated(keep=False)]['Gene Names'])
```

``` python
gps.gene.isin(dup_name).sum()
```

    np.int64(76)

``` python
gps.shape
```

    (4260, 6)

``` python
gps = gps[~gps.gene.isin(dup_name)].copy()
```

``` python
gene2uniprot = kinase_id.set_index('Gene Names')['Entry'].to_dict()
```

``` python
gps.shape
```

    (4184, 6)

unmapped:

``` python
gps[gps.gene.str.upper().map(gene2uniprot).isna()].gene.unique()
```

    array(['DMPK1', 'MRCKa', 'YPKA', 'PKACb', 'PKCt', 'PKCh', 'PKCi', 'PKCz',
           'PKG1', 'PKG2', 'PRKG1 Isoform Alpha', 'p70S6K', 'p70S6Kb',
           'CCDPK', 'AMPKa1', 'PKD3', 'CK1d', 'CK1e', 'CK1g1', 'CSNK2B',
           'p38b', 'p38d', 'p38g', 'AurB', 'AurC', 'TP53RK ', 'STLK3',
           'PAK3 Isoform 2', 'TAO1', 'ACK'], dtype=object)

``` python
gps['kinase_uniprot'] = gps.gene.str.upper().map(gene2uniprot)
```

``` python
gps = gps.dropna(subset='kinase_uniprot')
```

### Substrate mapping

``` python
# gps.uniprot.drop_duplicates().to_csv('raw/GPS_substrate_id.csv')
```

1,203 IDs were mapped to 1,208 results

1,202 active entries and 1 obsolete entry are found

``` python
gps.shape
```

    (3750, 7)

``` python
gps = map_substrate('raw/idmapping_2025_03_12_GPS.xlsx',gps,'uniprot')
```

    Shape before processing (3750, 7)
    Species counts: substrate_species
    Homo sapiens (Human)                                                                 3614
    Rattus norvegicus (Rat)                                                                30
    Mus musculus (Mouse)                                                                   19
    Pongo abelii (Sumatran orangutan) (Pongo pygmaeus abelii)                              10
    Pan troglodytes (Chimpanzee)                                                            6
    Sus scrofa (Pig)                                                                        6
    Bos taurus (Bovine)                                                                     5
    Oryctolagus cuniculus (Rabbit)                                                          4
    Macaca fascicularis (Crab-eating macaque) (Cynomolgus monkey)                           4
    Mesocricetus auratus (Golden hamster)                                                   4
    Tupaia belangeri (Common tree shrew) (Tupaia glis belangeri)                            2
    Canis lupus familiaris (Dog) (Canis familiaris)                                         2
    Phodopus roborovskii (Roborovski's desert hamster) (Cricetulus roborovskii)             2
    Macaca mulatta (Rhesus macaque)                                                         2
    Xenopus tropicalis (Western clawed frog) (Silurana tropicalis)                          2
    Hylobates lar (Lar gibbon) (White-handed gibbon)                                        1
    Cricetulus griseus (Chinese hamster) (Cricetulus barabensis griseus)                    1
    Xenopus laevis (African clawed frog)                                                    1
    Galeopterus variegatus (Malayan flying lemur) (Cynocephalus variegatus)                 1
    Nannospalax galili (Northern Israeli blind subterranean mole rat) (Spalax galili)       1
    Spermophilus citellus (European suslik) (Citellus citellus)                             1
    Marmota monax (Woodchuck)                                                               1
    Pan paniscus (Pygmy chimpanzee) (Bonobo)                                                1
    Danio rerio (Zebrafish) (Brachydanio rerio)                                             1
    Hepatitis delta virus genotype II (isolate 7/18/83) (HDV)                               1
    Human T-cell leukemia virus 1 (strain Japan ATK-1 subtype A) (HTLV-1)                   1
    Macaca fuscata fuscata (Japanese macaque)                                               1
    Human T-cell leukemia virus 1 (isolate Caribbea HS-35 subtype A) (HTLV-1)               1
    Human T-cell leukemia virus 1 (isolate Melanesia mel5 subtype C) (HTLV-1)               1
    Ovis aries (Sheep)                                                                      1
    Human immunodeficiency virus type 1 group M subtype B (isolate NY5) (HIV-1)             1
    Human immunodeficiency virus type 2 subtype A (isolate BEN) (HIV-2)                     1
    Human immunodeficiency virus type 1 group M subtype B (isolate YU-2) (HIV-1)            1
    Human immunodeficiency virus type 1 group M subtype B (isolate WMJ22) (HIV-1)           1
    Cavia porcellus (Guinea pig)                                                            1
    Human immunodeficiency virus type 1 group M subtype A (isolate MAL) (HIV-1)             1
    Human immunodeficiency virus type 1 group M subtype F2 (isolate MP257) (HIV-1)          1
    Human immunodeficiency virus type 1 group M subtype B (isolate LW123) (HIV-1)           1
    Human immunodeficiency virus type 1 group M subtype B (isolate JRCSF) (HIV-1)           1
    Human immunodeficiency virus type 1 group M subtype B (isolate JH32) (HIV-1)            1
    Human immunodeficiency virus type 1 group M subtype B (isolate HXB2) (HIV-1)            1
    Human immunodeficiency virus type 1 group M subtype C (isolate ETH2220) (HIV-1)         1
    Human immunodeficiency virus type 1 group M subtype D (isolate ELI) (HIV-1)             1
    Human immunodeficiency virus type 1 group M subtype B (isolate CDC-451) (HIV-1)         1
    Human immunodeficiency virus type 1 group O (isolate ANT70) (HIV-1)                     1
    Human immunodeficiency virus type 1 group M subtype A (isolate U455) (HIV-1)            1
    Human immunodeficiency virus type 2 subtype A (isolate KR) (HIV-2)                      1
    Phodopus campbelli (Campbell's dwarf Russian hamster)                                   1
    Human immunodeficiency virus type 2 subtype A (isolate ROD) (HIV-2)                     1
    Human immunodeficiency virus type 1 group O (isolate MVP5180) (HIV-1)                   1
    Felis catus (Cat) (Felis silvestris catus)                                              1
    Name: count, dtype: int64
    Removing non-human substrates
    Shape after processing (3614, 10)

    /home/sky1ove/miniconda3/lib/python3.13/site-packages/openpyxl/styles/stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
      warn("Workbook contains no default style, apply openpyxl's default")

``` python
gps.shape
```

    (3614, 10)

### Validate site

``` python
from katlas.utils import *
```

``` python
gps['site']=gps['code'] + gps['position'].astype(int).astype(str)
```

``` python
gps['site_match'] =validate_site_df(gps,'site','substrate_sequence')
```

``` python
gps['site_match'].value_counts()
```

    site_match
    1    3435
    0     179
    Name: count, dtype: int64

``` python
gps=gps[gps.site_match==1]
```

``` python
gps.shape
```

    (3435, 12)

### Save

``` python
gps.columns
```

    Index(['position', 'code', 'gene', 'species', 'source', 'kinase_uniprot',
           'substrate_uniprot', 'substrate_genes', 'substrate_sequence',
           'substrate_species', 'site', 'site_match'],
          dtype='object')

``` python
gps = gps.rename(columns={'source':'GPS_source'})
```

``` python
gps['source']='GPS6'
```

``` python
col = ['kinase_uniprot','gene', 

       'substrate_uniprot', 'substrate_genes', 'site','substrate_sequence', 'GPS_source','source']
```

``` python
gps = gps[col].copy()
```

``` python
gps.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">gene</th>
<th data-quarto-table-cell-role="th">substrate_uniprot</th>
<th data-quarto-table-cell-role="th">substrate_genes</th>
<th data-quarto-table-cell-role="th">site</th>
<th data-quarto-table-cell-role="th">substrate_sequence</th>
<th data-quarto-table-cell-role="th">GPS_source</th>
<th data-quarto-table-cell-role="th">source</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>P31749</td>
<td>AKT1</td>
<td>Q9Y261</td>
<td>FOXA2 HNF3B TCF3B</td>
<td>T156</td>
<td>MLGAVKMEGHEPSDWSSYYAEPEGYSSVSNMNAGLGMNGMNTYMSM...</td>
<td>14500912</td>
<td>GPS6</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>P31749</td>
<td>AKT1</td>
<td>P49760</td>
<td>CLK2</td>
<td>T127</td>
<td>MPHPRRYHSSERGSRGSYREHYRSRKHKRRRSRSWSSSSDRTRRRR...</td>
<td>UniProt</td>
<td>GPS6</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>P31749</td>
<td>AKT1</td>
<td>P49815</td>
<td>TSC2 TSC4</td>
<td>T1462</td>
<td>MAKPTSKDSGLKEKFKILLGLGTPRPNPRSAEGKQTEFIITAEILR...</td>
<td>12150915;15342917;12172553;UniProt</td>
<td>GPS6</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>P31749</td>
<td>AKT1</td>
<td>P46527</td>
<td>CDKN1B KIP1 p27</td>
<td>T187</td>
<td>MSNVRVSNGSPSLERMDARQAEHPKPSACRNLFGPVDHEELTRDLE...</td>
<td>12042314;9192873;9399644;9311993;9388487;17254...</td>
<td>GPS6</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>P31749</td>
<td>AKT1</td>
<td>O15111</td>
<td>CHUK IKKA TCF16</td>
<td>T23</td>
<td>MERPPGLRPGAGGPWEMRERLGTGGFGNVCLYQHRELDLKIAIKSC...</td>
<td>10485710;UniProt</td>
<td>GPS6</td>
</tr>
</tbody>
</table>

</div>

``` python
# gps.to_excel('raw/GPS6.xlsx',index=False)
```

## Other datasets

### Douglass

Paper:
https://journals.physiology.org/doi/full/10.1152/ajpcell.00166.2012

Data is not available to download, but available upon request to authors

### RegPhos 2.0

Paper:
https://academic.oup.com/database/article/doi/10.1093/database/bau034/2634150

Go to the paper link, scroll down to Supplementary data, download the
zip file, docx file, table S4 is motif

### Phosida

http://www.phosida.com/ is not accessible

Phosida paper: https://pmc.ncbi.nlm.nih.gov/articles/PMC2258193/

### PhosphoNetwork

2013 bioinformatics:
https://pmc.ncbi.nlm.nih.gov/articles/PMC3866559/#btt627-T1

2013 Mol Syst Biol.: https://pmc.ncbi.nlm.nih.gov/articles/PMC3658267/

web app: https://www.phosphonetworks.org/

The web app provides kinase-substrate pairs, but does not provides
specific info of phosphorylation site.

It provides logo and pssm though in Download –\> Motif Matrix, which can
be used to compare the PSSM result

### BioGRID

Go to : https://downloads.thebiogrid.org/BioGRID,
Current-Release–\>BIOGRID-PTMS-4.4.242.ptm.zip

There are two files in after extraction, however, they don’t contain
kinase-substrate relationship.

## Combine all

``` python
import pandas as pd
from pathlib import Path
```

``` python
paths = [
    'raw/GPS6.xlsx',
    'raw/signor.xlsx',
    'raw/ELM.xlsx',
    'raw/iPTMNet.xlsx','raw/EPSD.xlsx',# 'raw/KiNet.xlsx',
    'raw/psp_human.xlsx',
    'raw/sugiyama.xlsx'
]
```

``` python
dfs = [pd.read_excel(path) for path in paths]
```

``` python
[print(df.source.value_counts()) for df in dfs]
```

    source
    GPS6    3435
    Name: count, dtype: int64
    source
    SIGNOR    11212
    Name: count, dtype: int64
    source
    ELM    4519
    Name: count, dtype: int64
    source
    iPTMNet    3478
    Name: count, dtype: int64
    source
    EPSD    10308
    Name: count, dtype: int64
    source
    PSP    13144
    Name: count, dtype: int64
    source
    Sugiyama    184948
    Name: count, dtype: int64

    [None, None, None, None, None, None, None]

For each df, we need to drop duplicates of
kinase(uniprot)-substrate(uniprot)-site

``` python
def get_key(df):
    df = df.copy()
    print('original shape:',df.shape)
    df['kin_sub_site']=df['kinase_uniprot']+'_'+df['substrate_uniprot']+'_'+df['site']
    df = df.drop_duplicates(subset='kin_sub_site')
    print('after removing duplicates',df.shape)
    return df
```

``` python
dfs=[get_key(df) for df in dfs]
```

    original shape: (3435, 8)
    after removing duplicates (3326, 9)
    original shape: (11212, 9)
    after removing duplicates (9320, 10)
    original shape: (4519, 10)
    after removing duplicates (3807, 11)
    original shape: (3478, 9)
    after removing duplicates (3478, 10)
    original shape: (10308, 9)
    after removing duplicates (10308, 10)
    original shape: (13144, 13)
    after removing duplicates (13091, 14)
    original shape: (184948, 11)
    after removing duplicates (168342, 12)

``` python
for path,df in zip(paths,dfs):
    print(Path(path).stem, df.shape)
```

    GPS6 (3326, 9)
    signor (9320, 10)
    ELM (3807, 11)
    iPTMNet (3478, 10)
    EPSD (10308, 10)
    psp_human (13091, 14)
    sugiyama (168342, 12)

``` python
common_cols = ['kinase_uniprot','substrate_uniprot','site','kin_sub_site','source','substrate_genes','substrate_sequence']
```

``` python
df_all = pd.concat(dfs,ignore_index=True)
df_all = df_all[common_cols].copy()
```

``` python
df_all.source.value_counts()
```

    source
    Sugiyama    168342
    PSP          13091
    EPSD         10308
    SIGNOR        9320
    ELM           3807
    iPTMNet       3478
    GPS6          3326
    Name: count, dtype: int64

``` python
# df_all.to_parquet('raw/combine_source.parquet')
```

``` python
df_grouped = df_all.groupby("kin_sub_site").agg({
    "kinase_uniprot": "first",
    "substrate_uniprot": "first",
    "site": "first",
    "source": '|'.join,  # Concatenate sources with '|'
    "substrate_genes": "first",
    "substrate_sequence": "first"
}).reset_index()
```

``` python
df_grouped.shape
```

    (187066, 7)

``` python
df_grouped.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">kin_sub_site</th>
<th data-quarto-table-cell-role="th">kinase_uniprot</th>
<th data-quarto-table-cell-role="th">substrate_uniprot</th>
<th data-quarto-table-cell-role="th">site</th>
<th data-quarto-table-cell-role="th">source</th>
<th data-quarto-table-cell-role="th">substrate_genes</th>
<th data-quarto-table-cell-role="th">substrate_sequence</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>O00141_A4FU28_S140</td>
<td>O00141</td>
<td>A4FU28</td>
<td>S140</td>
<td>Sugiyama</td>
<td>CTAGE9</td>
<td>MEEPGATPQPYLGLVLEELGRVVAALPESMRPDENPYGFPSELVVC...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>O00141_O00141_S252</td>
<td>O00141</td>
<td>O00141</td>
<td>S252</td>
<td>Sugiyama</td>
<td>SGK1 SGK</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>O00141_O00141_S255</td>
<td>O00141</td>
<td>O00141</td>
<td>S255</td>
<td>Sugiyama</td>
<td>SGK1 SGK</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>O00141_O00141_S397</td>
<td>O00141</td>
<td>O00141</td>
<td>S397</td>
<td>Sugiyama</td>
<td>SGK1 SGK</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>O00141_O00141_S404</td>
<td>O00141</td>
<td>O00141</td>
<td>S404</td>
<td>Sugiyama</td>
<td>SGK1 SGK</td>
<td>MTVKTEAAKGTLTYSRMRGMVAILIAFMKQRRMGLNDFIQKIANNS...</td>
</tr>
</tbody>
</table>

</div>

``` python
# df_grouped.to_parquet('raw/combine_source_grouped.parquet')
```

## Human phosphoproteome

``` python
from katlas.core import *
import pandas as pd
```

### Data

``` python
human = Data.get_combine_site_psp_ochoa()
```

``` python
human.shape
```

    (121419, 8)

### Substrate mapping

``` python
# human.uniprot.drop_duplicates().to_csv('raw/human_phosphoproteome_uniprot.csv')
```

11,243 IDs were mapped to 11,241 results

5 ID were not mapped: AAA58698 P18433-2 AAC50053 AAA60149 NP_001184222

11,242 active entries and 1 obsolete entry are found

``` python
human = map_substrate('raw/idmapping_2025_03_20_human_phosphoproteome.xlsx',human,'uniprot')
```

    /home/sky1ove/miniconda3/lib/python3.13/site-packages/openpyxl/styles/stylesheet.py:237: UserWarning: Workbook contains no default style, apply openpyxl's default
      warn("Workbook contains no default style, apply openpyxl's default")

    Shape before processing (121419, 8)
    Species counts: substrate_species
    Homo sapiens (Human)    121332
    Name: count, dtype: int64
    Removing non-human substrates
    Shape after processing (121332, 11)

### Validate site

``` python
human.columns
```

    Index(['gene', 'site', 'site_seq', 'source', 'AM_pathogenicity', 'CDDM_upper',
           'CDDM_max_score', 'substrate_uniprot', 'substrate_genes',
           'substrate_sequence', 'substrate_species'],
          dtype='object')

``` python
human['site_match'] =validate_site_df(human,'site','substrate_sequence')
```

``` python
human['site_match'].value_counts()
```

    site_match
    1    120084
    0      1248
    Name: count, dtype: int64

``` python
human=human[human.site_match==1].copy()
```

``` python
human.shape
```

    (120084, 12)

### Remove duplicates

``` python
human['sub_site'] = human['substrate_uniprot']+'_'+human['site']
```

``` python
human = human.drop_duplicates(subset='sub_site')
```

``` python
human.shape
```

    (119955, 13)

### Save

``` python
human.columns
```

    Index(['gene', 'site', 'site_seq', 'source', 'AM_pathogenicity', 'CDDM_upper',
           'CDDM_max_score', 'substrate_uniprot', 'substrate_genes',
           'substrate_sequence', 'substrate_species', 'site_match', 'sub_site'],
          dtype='object')

``` python
cols = ['substrate_uniprot', 'substrate_genes',

        'site', 'source', 'AM_pathogenicity', 

       'substrate_sequence', 'substrate_species', 'sub_site']
```

``` python
human = human[cols]
```

``` python
# human.to_parquet('raw/human_phosphoproteome.parquet')
```

## Phosphorylate sequence

### Combine human phosphoproteome and KS dataset site info

``` python
human = pd.read_parquet('raw/human_phosphoproteome.parquet')
```

``` python
cols = ['substrate_uniprot','site','substrate_sequence']
```

``` python
df_grouped = pd.read_parquet('raw/combine_source_grouped.parquet')
```

``` python
human = human[cols]

df_grouped = df_grouped[cols]
```

``` python
comb = pd.concat([human,df_grouped])
```

``` python
comb['sub_site'] = comb['substrate_uniprot']+'_'+comb['site']
```

``` python
comb = comb.drop_duplicates('sub_site')
```

### Phosphorylate sequence

``` python
seq = phosphorylate_seq_df(comb)
```

``` python
seq.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">substrate_uniprot</th>
<th data-quarto-table-cell-role="th">site</th>
<th data-quarto-table-cell-role="th">substrate_sequence</th>
<th data-quarto-table-cell-role="th">phosphoseq</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>A0A024R4G9</td>
<td>[S20]</td>
<td>MTVLEAVLEIQAITGSRLLSMVPGPARPPGSCWDPTQCTRTWLLSH...</td>
<td>MTVLEAVLEIQAITGSRLLsMVPGPARPPGSCWDPTQCTRTWLLSH...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>A0A075B6Q4</td>
<td>[S24, S35, S57, S68, S71, S72]</td>
<td>MDIQKSENEDDSEWEDVDDEKGDSNDDYDSAGLLSDEDCMSVPGKT...</td>
<td>MDIQKSENEDDSEWEDVDDEKGDsNDDYDSAGLLsDEDCMSVPGKT...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>A0A075B6T3</td>
<td>[S24, S26]</td>
<td>XLKRAYRGLEEVQWCLEQLLTSPSPS</td>
<td>XLKRAYRGLEEVQWCLEQLLTSPsPs</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>A0A075B759</td>
<td>[T68]</td>
<td>MVNSVVFFEITRDGKPLGRISIKLFADKIPKTAENFRALSTGEKGF...</td>
<td>MVNSVVFFEITRDGKPLGRISIKLFADKIPKTAENFRALSTGEKGF...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>A0A087WTJ2</td>
<td>[T8, Y14, Y213, T215, S221, S421, S424]</td>
<td>MGGRKMATDEENVYGLEENAQSRQESTRRLILVGRTGAGKSATGNS...</td>
<td>MGGRKMAtDEENVyGLEENAQSRQESTRRLILVGRTGAGKSATGNS...</td>
</tr>
</tbody>
</table>

</div>

``` python
# seq.to_csv('raw/phosphoseq_map.csv',index=False)
```

### Map to df

``` python
human = pd.read_parquet('raw/human_phosphoproteome.parquet')

df_grouped = pd.read_parquet('raw/combine_source_grouped.parquet')
```

``` python
seq_map = seq.set_index('substrate_uniprot')['phosphoseq']
```

``` python
human['substrate_phosphoseq'] = human.substrate_uniprot.map(seq_map)

df_grouped['substrate_phosphoseq'] = df_grouped.substrate_uniprot.map(seq_map)
```

``` python
human['substrate_phosphoseq'].isna().sum()
```

    np.int64(0)

``` python
df_grouped['substrate_phosphoseq'].isna().sum()
```

    np.int64(0)

``` python
# human.to_parquet('raw/human_phosphoproteome.parquet')

# df_grouped.to_parquet('raw/combine_source_grouped.parquet')
```

## Extract site sequence

``` python
human['position'] = human['site'].str[1:].astype(int)

df_grouped['position']=df_grouped['site'].str[1:].astype(int)
```

``` python
extract_site_seq?
```

    Signature:
    extract_site_seq(
        df: pandas.core.frame.DataFrame,
        seq_col: str,
        site_col: str,
        n=7,
    )
    Docstring: Extract -n to +n site sequence from protein sequence
    File:      /mnt/f/git/kkatlas/katlas/katlas/utils.py
    Type:      function

``` python
human['site_seq'] = extract_site_seq(human,
                                  seq_col='substrate_phosphoseq',
                                  site_col='site',n=20)
```

    100%|█████████████████████████████████████████████████████████████████████| 119955/119955 [00:04<00:00, 25764.66it/s]

``` python
df_grouped['site_seq'] = extract_site_seq(df_grouped,
                                  seq_col='substrate_phosphoseq',
                                  site_col='site',n=20)
```

    100%|█████████████████████████████████████████████████████████████████████| 187066/187066 [00:07<00:00, 25809.29it/s]

``` python
df_grouped['sub_site'] = df_grouped['substrate_uniprot']+'_'+ df_grouped['site']

human['sub_site'] = human['substrate_uniprot']+'_'+ human['site']
```

``` python
# human.to_parquet('raw/human_phosphoproteome.parquet')

# df_grouped.to_parquet('raw/combine_source_grouped.parquet')
```

## Add kinase info

``` python
df=df_grouped.copy()
```

``` python
# Remove pseudokinase duplicates by UniProt ID, keep only one entry per kinase
info = Data.get_kinase_info().sort_values('kinase').drop_duplicates('uniprot')

# Pre-extract UniProt ID without isoform for matching
df['uniprot_clean'] = df['kinase_uniprot'].str.split('-').str[0]

info_indexed = info.set_index('uniprot')
group_map = info_indexed['group']
family_map = info_indexed['family']
pspa_small_map = info_indexed['pspa_category_small']
pspa_big_map = info_indexed['pspa_category_big']

df['kinase_on_tree'] = df['uniprot_clean'].isin(info['uniprot']).astype(int)

kinase_gene_map = Data.get_kinase_uniprot().set_index('Entry')['Gene Names']
df['kinase_genes'] = df['uniprot_clean'].map(kinase_gene_map)

df['kinase_group'] = df['uniprot_clean'].map(group_map)
df['kinase_family'] = df['uniprot_clean'].map(family_map)
df['kinase_pspa_big'] = df['uniprot_clean'].map(pspa_big_map)
df['kinase_pspa_small'] = df['uniprot_clean'].map(pspa_small_map)

df.drop(columns='uniprot_clean', inplace=True)
```

The above code has been added into `Data` class when loading the ks
dataset
