<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Plot reactome pathways: CDDM – katlas-raw</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.min.js" integrity="sha384-ZvpUoO/+PpLXR1lu4jmpXWu80pZlYUAfxl5NsBMWOEPSjUn/6Z/hRTt8+pR6L4N2" crossorigin="anonymous"></script><script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-000f0825b9c55ed21d14970d810c14a9.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="https://cdn.jsdelivr.net/npm/requirejs@2.3.6/require.min.js" integrity="sha384-c9c+LnTbwQ3aujuU7ULEPVvgLs+Fn6fJUvIGTsuu1ZcCf11fiEubah0ttpca4ntM sha384-6V1/AdqZRWk1KAlWbKBlGhN7VG4iE/yAZcO6NZPMF8od0vukrvr0tg4qY6NSrItx" crossorigin="anonymous"></script>

<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


<link rel="stylesheet" href="styles.css">
<meta property="og:title" content="Plot reactome pathways: CDDM – katlas-raw">
<meta property="og:image" content="https://sky1ove.github.io/katlas-raw/19c_reactome_path_plot_cddm_files/figure-html/cell-13-output-1.png">
<meta property="og:site_name" content="katlas-raw">
<meta property="og:image:height" content="220">
<meta property="og:image:width" content="549">
<meta name="twitter:title" content="Plot reactome pathways: CDDM – katlas-raw">
<meta name="twitter:image" content="https://sky1ove.github.io/katlas-raw/19c_reactome_path_plot_cddm_files/figure-html/cell-13-output-1.png">
<meta name="twitter:image-height" content="220">
<meta name="twitter:image-width" content="549">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="./index.html" class="navbar-brand navbar-brand-logo">
    </a>
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">katlas-raw</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./reactome_path_plot_cddm.html">Plot reactome pathways: CDDM</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      </a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">katlas-raw</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./preprocess_dataset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preprocess kinase-substrate datasets</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./eda_fig1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Exploratory data analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./site_promiscuity_fig1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Site promiscuity</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssm_hierarchical.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Hierarchical clustering of PSSM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05_PSSM_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">05_PSSM_example.html</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./modi_group.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Modi group</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scale_pspa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PSPA scaled data</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./background.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Background frequencies</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./specificity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Specificity comparison</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./sty_composition.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">STY composition</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./plot_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot heatmap and logo of CDDM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./plot_pspa_figure.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PSPA figures</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./13_CDDM_umap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CDDM umap</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./get_aa_relationship.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get AA relationship</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pspa_cddm_heatmap_compare.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Heatmap/ST/JSD/cosine comparison</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_no_psp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_psp_02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring_compare_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scoring evaluation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring_compare_psp_02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scoring evaluation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Reactome pathway for each PSSM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_plot_pspa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot reactome pathways: PSPA</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_plot_cddm.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Plot reactome pathways: CDDM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./motif_clustering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Kmeans motifs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./motif_plot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Motif plot</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./22_motif_web_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">22_motif_web_data.html</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./23_motif_kinase_similarity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cosine similarity with PSPA</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">kinase_domain</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_01_get_all_kd.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get all kinase domain sequences</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_02_pseudo_alignment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Alignment</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_03_feature_extract.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Extract KD features</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_04_prepare_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare data for training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_05_pspa_training_model_comparison_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Model comparison on T5 feature</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06a_pspa_training_feature_comparison_cnn_group.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on CNN model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06b_pspa_training_feature_comparison_cnn_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on CNN model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06c_pspa_training_feature_comparison_mlp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on MLP model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_07a_cddm_training_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_07b_pspa_training_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_08a_similarity_relationship.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KD similarity &amp; PSSM similarity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_08b_similarity_relationship_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KD similarity &amp; PSSM similarity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_09a_predict_pspa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Predict</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_09b_predict_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Predict</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_10_plot_umap_known_unk.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot UMAP of known &amp; unknown KD</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_11_plot_pspa_kd_t5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">kd_11_plot_pspa_kd_t5.html</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_12_compare_pred_psp_known.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Comopare the two</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#data" id="toc-data" class="nav-link active" data-scroll-target="#data">Data</a></li>
  <li><a href="#site-pssms" id="toc-site-pssms" class="nav-link" data-scroll-target="#site-pssms">Site pssms</a></li>
  <li><a href="#plot-distribution" id="toc-plot-distribution" class="nav-link" data-scroll-target="#plot-distribution">Plot distribution</a></li>
  <li><a href="#plot-reactome-pathway" id="toc-plot-reactome-pathway" class="nav-link" data-scroll-target="#plot-reactome-pathway">Plot reactome pathway</a></li>
  <li><a href="#get-accuracy-for-overlap-pathways" id="toc-get-accuracy-for-overlap-pathways" class="nav-link" data-scroll-target="#get-accuracy-for-overlap-pathways">Get accuracy for overlap pathways</a></li>
  <li><a href="#analysis" id="toc-analysis" class="nav-link" data-scroll-target="#analysis">Analysis</a>
  <ul class="collapse">
  <li><a href="#lowest" id="toc-lowest" class="nav-link" data-scroll-target="#lowest">Lowest</a></li>
  </ul></li>
  <li><a href="#compare-with-pspa" id="toc-compare-with-pspa" class="nav-link" data-scroll-target="#compare-with-pspa">Compare with PSPA</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/sky1ove/katlas-raw/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div><div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="reactome_path_plot_cddm.html.md"><i class="bi bi-file-code"></i>CommonMark</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Plot reactome pathways: CDDM</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<!-- WARNING: THIS FILE WAS AUTOGENERATED! DO NOT EDIT! -->
<div id="622fd9c3-c77c-4f40-9cb4-2ccaa1154c7b" class="cell" data-execution_count="1">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.pssm <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.score <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.data <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.plot <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.pathway <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.utils <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd,seaborn <span class="im">as</span> sns</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib <span class="im">import</span> pyplot <span class="im">as</span> plt</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> tqdm <span class="im">import</span> tqdm</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> time, random</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<section id="data" class="level2">
<h2 class="anchored" data-anchor-id="data">Data</h2>
<div id="70a9e22d-2c0e-4e98-ad0b-756abe87a330" class="cell" data-execution_count="2">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>cddm_info<span class="op">=</span>pd.read_parquet(<span class="st">'raw/cddm_path_info.parquet'</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>df<span class="op">=</span>pd.read_parquet(<span class="st">'raw/cddm_score_df.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="cf767a2d-4347-4542-b1c8-1883e069107d" class="cell" data-execution_count="3">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>thr_cols <span class="op">=</span> [<span class="st">'S_thr'</span>,<span class="st">'T_thr'</span>,<span class="st">'Y_thr'</span>]</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>gene_cols <span class="op">=</span> [<span class="st">'S_genes'</span>,<span class="st">'T_genes'</span>,<span class="st">'Y_genes'</span>]</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>cnt_cols <span class="op">=</span> [<span class="st">'S_cnt'</span>,<span class="st">'T_cnt'</span>,<span class="st">'Y_cnt'</span>]</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>site_cols <span class="op">=</span> [<span class="st">'S_sites'</span>,<span class="st">'T_sites'</span>,<span class="st">'Y_sites'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="site-pssms" class="level2">
<h2 class="anchored" data-anchor-id="site-pssms">Site pssms</h2>
<div id="4cd09680-dfe8-43cf-a26a-addb1321cf91" class="cell" data-execution_count="6">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>human_site<span class="op">=</span> Data.get_human_site()</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>seq_map <span class="op">=</span> human_site.set_index(<span class="st">'sub_site'</span>)[<span class="st">'site_seq'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="5e10bc92-f15b-47af-994e-7191391e034a" class="cell" data-execution_count="7">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>series<span class="op">=</span>cddm_info.iloc[<span class="dv">0</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="77609ca2-4f48-489e-9594-d03899d7171f" class="cell" data-execution_count="8">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> get_site_pssm(series):</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>    pssm_dict <span class="op">=</span> {}</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>    count_dict<span class="op">=</span>{}</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> acceptor, sites <span class="kw">in</span> <span class="bu">zip</span>([<span class="st">'S'</span>,<span class="st">'T'</span>,<span class="st">'Y'</span>],series[site_cols]):</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>        sites<span class="op">=</span><span class="bu">list</span>(sites)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> sites:</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>            pssm <span class="op">=</span> get_prob(seq_map[sites])</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>            key <span class="op">=</span> series.name<span class="op">+</span><span class="st">'_'</span><span class="op">+</span>acceptor</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>            pssm_dict[key] <span class="op">=</span> pd.Series(flatten_pssm(pssm))</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>            count_dict[key]<span class="op">=</span><span class="bu">len</span>(sites)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> pssm_dict,count_dict</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="2d7467db-68bc-4346-ad2b-b641ce588ea3" class="cell" data-execution_count="9">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>pssm_out <span class="op">=</span> {}</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>count_out <span class="op">=</span> {}</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i,r <span class="kw">in</span> cddm_info.iterrows():</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>    pssms,counts <span class="op">=</span> get_site_pssm(r)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    pssm_out.update(pssms)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    count_out.update(counts)</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="acc9df07-3efb-4897-a040-27475a2a6b83" class="cell" data-execution_count="10">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>site_pssms <span class="op">=</span> pd.DataFrame(pssm_out).T</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="7355971c-a0b5-408c-9930-68244a6a85d8" class="cell" data-execution_count="19">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> k <span class="kw">in</span> tqdm(cddm_info.index,total<span class="op">=</span><span class="bu">len</span>(cddm_info)):</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>    k_pssms <span class="op">=</span> site_pssms[site_pssms.index.<span class="bu">str</span>.contains(<span class="ss">f'</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">_'</span>)]</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>    plot_logos(k_pssms,count_dict<span class="op">=</span>count_out)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>    path <span class="op">=</span> prepare_path(<span class="ss">f'fig/cddm/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/path/logo.svg'</span>)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>    save_show(path,show_only<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>    plt.close(<span class="st">'all'</span>)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>100%|█████████████████████████████████████████████████████████| 328/328 [43:05&lt;00:00,  7.88s/it]</code></pre>
</div>
</div>
<div id="002b724d-deb2-46b0-b32b-2475f980b26b" class="cell" data-execution_count="20">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>site_pssms.to_parquet(<span class="st">'raw/cddm_site_pssm.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="plot-distribution" class="level2">
<h2 class="anchored" data-anchor-id="plot-distribution">Plot distribution</h2>
<div id="645360c6-879a-4b7c-9765-f261ebffdef4" class="cell" data-execution_count="21">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_hist(df,</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>              kinase,</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>              sty_thr<span class="op">=</span><span class="va">None</span>, <span class="co"># dict of thresholds per acceptor</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>              hue<span class="op">=</span><span class="st">'acceptor'</span>):</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    <span class="co">"Plot histogram of a column (kinase). "</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">6</span>,<span class="dv">2</span>))</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>    hist_params <span class="op">=</span> {<span class="st">'element'</span>:<span class="st">'poly'</span>,</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>              <span class="st">'edgecolor'</span>: <span class="va">None</span>,</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>              <span class="st">'alpha'</span>:<span class="fl">0.5</span>,</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>              <span class="st">'bins'</span>:<span class="dv">100</span>,</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>              <span class="st">'kde'</span>:<span class="va">True</span>,</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>              <span class="st">'palette'</span>:sty_color}</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>    sns.histplot(data<span class="op">=</span>df,x<span class="op">=</span>kinase,hue<span class="op">=</span>hue,<span class="op">**</span>hist_params)</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">''</span>)</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'</span><span class="sc">{</span>kinase<span class="sc">}</span><span class="ss"> score distribution'</span>)</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> sty_thr:</span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> acceptor,thr <span class="kw">in</span> sty_thr.items():</span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> thr <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>                plt.axvline(thr,color<span class="op">=</span>sty_color[acceptor])</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="44a07cb4-c6d4-4c0d-b0c2-56d476f8a6bb" class="cell" data-execution_count="23">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>plot_hist(df,<span class="st">'AKT1'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="19c_reactome_path_plot_cddm_files/figure-html/cell-13-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div id="b7aaf657-3795-4e2f-896c-12031cb2dacd" class="cell" data-execution_count="24">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> k,r <span class="kw">in</span> tqdm(cddm_info.iterrows(),total<span class="op">=</span><span class="bu">len</span>(cddm_info)):</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>    sty_thr<span class="op">=</span>r[thr_cols].rename(index<span class="op">=</span>{<span class="st">'S_thr'</span>:<span class="st">'S'</span>,<span class="st">'T_thr'</span>:<span class="st">'T'</span>,<span class="st">'Y_thr'</span>:<span class="st">'Y'</span>})[<span class="kw">lambda</span> x: x<span class="op">!=</span><span class="dv">0</span>].to_dict()</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>    plot_hist(df,k,sty_thr<span class="op">=</span>sty_thr)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>    save_show(<span class="ss">f'fig/cddm/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/path/hist.svg'</span>,show_only<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>plt.close(<span class="st">'all'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>100%|█████████████████████████████████████████████████████████| 328/328 [03:14&lt;00:00,  1.68it/s]</code></pre>
</div>
</div>
</section>
<section id="plot-reactome-pathway" class="level2">
<h2 class="anchored" data-anchor-id="plot-reactome-pathway">Plot reactome pathway</h2>
<div id="cfbed19d-89ab-432d-a203-ab2ede422a0f" class="cell" data-execution_count="5">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>info <span class="op">=</span> Data.get_kinase_info()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="e8c1347f-8e65-4bd5-97a6-eb17426bcd53" class="cell" data-execution_count="6">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>info <span class="op">=</span> info[info.pseudo<span class="op">==</span><span class="st">'0'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="8d635bdb-8a9a-4196-a06a-16a96f0d781b" class="cell" data-execution_count="7">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>info.uniprot.duplicated().<span class="bu">sum</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="7">
<pre><code>np.int64(0)</code></pre>
</div>
</div>
<div id="7c054e5d-b415-4e78-87bd-bc006c42f715" class="cell" data-execution_count="8">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>kinase_map <span class="op">=</span> info.set_index(<span class="st">'kinase'</span>)[<span class="st">'uniprot'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="28b98855-6f9a-4e80-bf26-afa0ba1e406e" class="cell" data-execution_count="9">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> get_path(series):</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>    out <span class="op">=</span> get_reactome(series[<span class="st">'genes'</span>],p_type<span class="op">=</span><span class="st">'p'</span>)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>    kinase<span class="op">=</span>series.name.split(<span class="st">'_'</span>)[<span class="dv">0</span>] <span class="co"># for those with _TRY, need to remove that</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>    uniprot <span class="op">=</span> kinase_map[kinase]</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>    out <span class="op">=</span> add_reactome_ref(out,uniprot)</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>    exclude<span class="op">=</span>[<span class="st">'GTPase'</span>,<span class="st">'SUMO'</span>]</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>    out <span class="op">=</span> out[<span class="op">~</span>out.name.<span class="bu">str</span>.contains(<span class="st">'|'</span>.join(exclude))]</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> out</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="061cd930-e2dd-4524-9081-a29ff7ddaf40" class="cell" data-execution_count="10">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>show_only<span class="op">=</span><span class="va">False</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="e316f1cb-61d1-47f8-b2ad-a21c70735cfb" class="cell" data-execution_count="37">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>cddm_info.index.get_loc(<span class="st">'YSK1'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="37">
<pre><code>96</code></pre>
</div>
</div>
<div id="8f5fa4ea-8f1b-45d9-81ab-587a46fc4453" class="cell" data-execution_count="14">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="bu">len</span>(cddm_info)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="14">
<pre><code>328</code></pre>
</div>
</div>
<div id="ddb34ba3-9681-4377-a8e0-f8367326cc57" class="cell" data-quarto-private-1="{&quot;key&quot;:&quot;jupyter&quot;,&quot;value&quot;:{&quot;outputs_hidden&quot;:true}}" data-execution_count="13">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i,(k,r) <span class="kw">in</span> <span class="bu">enumerate</span>(cddm_info.iterrows()):</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(i,<span class="st">':'</span>,k)</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>    save_dir <span class="op">=</span> <span class="ss">f'fig/cddm/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/path'</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>    path_df <span class="op">=</span> get_path(r)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>    path_df.to_parquet(<span class="ss">f'</span><span class="sc">{</span>save_dir<span class="sc">}</span><span class="ss">/path.parquet'</span>)</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>    plot_path(path_df,p_type<span class="op">=</span><span class="st">'p'</span>,ref_col<span class="op">=</span><span class="st">'ref_path'</span>,top_n<span class="op">=</span><span class="dv">15</span>,max_label_length<span class="op">=</span><span class="dv">55</span>)</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="st">'Reactome pathways of substrates based on scores'</span>)</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>    save_show(<span class="ss">f'</span><span class="sc">{</span>save_dir<span class="sc">}</span><span class="ss">/path.svg'</span>,show_only<span class="op">=</span>show_only)</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> path_df.ref_path.<span class="bu">sum</span>()<span class="op">!=</span><span class="dv">0</span>:</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>        accuracy <span class="op">=</span> get_overlap(path_df, ref_col<span class="op">=</span><span class="st">'ref_path'</span>,plot<span class="op">=</span><span class="va">True</span>,p_type<span class="op">=</span><span class="st">'p'</span>,figsize<span class="op">=</span>(<span class="dv">5</span>,<span class="dv">3</span>))</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>        plt.title(<span class="ss">f'Distribution of –log₁₀(p) for reference pathways'</span>)</span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>        save_show(<span class="ss">f'</span><span class="sc">{</span>save_dir<span class="sc">}</span><span class="ss">/overlap.svg'</span>,show_only<span class="op">=</span>show_only)</span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> path_df.ref_path_lowest.<span class="bu">sum</span>()<span class="op">!=</span><span class="dv">0</span>:</span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>        accuracy_lo <span class="op">=</span> get_overlap(path_df, ref_col<span class="op">=</span><span class="st">'ref_path_lowest'</span>,plot<span class="op">=</span><span class="va">True</span>,p_type<span class="op">=</span><span class="st">'p'</span>,figsize<span class="op">=</span>(<span class="dv">5</span>,<span class="dv">3</span>))</span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>        plt.title(<span class="ss">f'Distribution of –log₁₀(p) for reference pathways </span><span class="ch">\n</span><span class="ss">(lowest-level)'</span>)</span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>        save_show(<span class="ss">f'</span><span class="sc">{</span>save_dir<span class="sc">}</span><span class="ss">/overlap_lo.svg'</span>,show_only<span class="op">=</span>show_only)</span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>    plt.close(<span class="st">'all'</span>)</span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># break</span></span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ✅ polite delay to avoid API throttling</span></span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a>    delay <span class="op">=</span> random.uniform(<span class="fl">1.5</span>, <span class="dv">4</span>)</span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Sleeping </span><span class="sc">{</span>delay<span class="sc">:.2f}</span><span class="ss">s before next kinase..."</span>)</span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>    time.sleep(delay)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>0 : SRC
Running pathway anlysis
Done
Sleeping 3.98s before next kinase...
1 : EPHA3
Running pathway anlysis
Done
Sleeping 1.77s before next kinase...
2 : FES
Running pathway anlysis
Done
Sleeping 2.18s before next kinase...
3 : NTRK3
Running pathway anlysis
Done
Sleeping 3.29s before next kinase...
4 : ALK
Running pathway anlysis
Done
Sleeping 3.96s before next kinase...
5 : ABL1
Running pathway anlysis
Done
Sleeping 2.73s before next kinase...
6 : FLT3
Running pathway anlysis
Done
Sleeping 2.95s before next kinase...
7 : EPHA8
Running pathway anlysis
Done
Sleeping 1.94s before next kinase...
8 : EPHB2
Running pathway anlysis
Done
Sleeping 3.69s before next kinase...
9 : EPHB1
Running pathway anlysis
Done
Sleeping 2.95s before next kinase...
10 : RET
Running pathway anlysis
Done
Sleeping 2.36s before next kinase...
11 : FYN
Running pathway anlysis
Done
Sleeping 3.14s before next kinase...
12 : FER
Running pathway anlysis
Done
Sleeping 1.84s before next kinase...
13 : INSR
Running pathway anlysis
Done
Sleeping 3.89s before next kinase...
14 : MUSK
Running pathway anlysis
Done
Sleeping 2.47s before next kinase...
15 : EPHB3
Running pathway anlysis
Done
Sleeping 1.56s before next kinase...
16 : EPHB4
Running pathway anlysis
Done
Sleeping 3.63s before next kinase...
17 : NTRK1
Running pathway anlysis
Done
Sleeping 3.03s before next kinase...
18 : TYRO3
Running pathway anlysis
Done
Sleeping 3.11s before next kinase...
19 : ROS1
Running pathway anlysis
Done
Sleeping 3.30s before next kinase...
20 : MET
Running pathway anlysis
Done
Sleeping 2.40s before next kinase...
21 : SYK
Running pathway anlysis
Done
Sleeping 2.55s before next kinase...
22 : TNK1
Running pathway anlysis
Done
Sleeping 2.43s before next kinase...
23 : FLT4
Running pathway anlysis
Done
Sleeping 1.53s before next kinase...
24 : YES1
Running pathway anlysis
Done
Sleeping 2.50s before next kinase...
25 : BMX
Running pathway anlysis
Done
Sleeping 3.36s before next kinase...
26 : ERK2
Running pathway anlysis
Done
Sleeping 2.63s before next kinase...
27 : PKACA
Running pathway anlysis
Done
Sleeping 3.91s before next kinase...
28 : LTK
Running pathway anlysis
Done
Sleeping 2.90s before next kinase...
29 : FGFR1
Running pathway anlysis
Done
Sleeping 3.38s before next kinase...
30 : FRK
Running pathway anlysis
Done
Sleeping 2.72s before next kinase...
31 : EPHA5
Running pathway anlysis
Done
Sleeping 2.92s before next kinase...
32 : TEK
Running pathway anlysis
Done
Sleeping 3.88s before next kinase...
33 : LCK
Running pathway anlysis
Done
Sleeping 1.94s before next kinase...
34 : CDK1
Running pathway anlysis
Done
Sleeping 1.60s before next kinase...
35 : LYN
Running pathway anlysis
Done
Sleeping 2.90s before next kinase...
36 : MST1R
Running pathway anlysis
Done
Sleeping 3.27s before next kinase...
37 : IKKB
Running pathway anlysis
Done
Sleeping 3.81s before next kinase...
38 : FGFR2
Running pathway anlysis
Done
Sleeping 3.32s before next kinase...
39 : CDK2
Running pathway anlysis
Done
Sleeping 2.22s before next kinase...
40 : FGR
Running pathway anlysis
Done
Sleeping 3.88s before next kinase...
41 : SRMS
Running pathway anlysis
Done
Sleeping 1.81s before next kinase...
42 : JAK3
Running pathway anlysis
Done
Sleeping 1.75s before next kinase...
43 : TEC
Running pathway anlysis
Done
Sleeping 2.32s before next kinase...
44 : PDGFRA
Running pathway anlysis
Done
Sleeping 1.66s before next kinase...
45 : EPHA1
Running pathway anlysis
Done
Sleeping 3.39s before next kinase...
46 : ERK1
Running pathway anlysis
Done
Sleeping 2.55s before next kinase...
47 : IGF1R
Running pathway anlysis
Done
Sleeping 2.01s before next kinase...
48 : EPHA2
Running pathway anlysis
Done
Sleeping 1.53s before next kinase...
49 : AXL
Running pathway anlysis
Done
Sleeping 2.36s before next kinase...
50 : INSRR
Running pathway anlysis
Done
Sleeping 3.09s before next kinase...
51 : PKCA
Running pathway anlysis
Done
Sleeping 2.38s before next kinase...
52 : BTK
Running pathway anlysis
Done
Sleeping 2.88s before next kinase...
53 : BLK
Running pathway anlysis
Done
Sleeping 3.56s before next kinase...
54 : MERTK
Running pathway anlysis
Done
Sleeping 1.66s before next kinase...
55 : ERBB4
Running pathway anlysis
Done
Sleeping 3.64s before next kinase...
56 : EPHA7
Running pathway anlysis
Done
Sleeping 1.72s before next kinase...
57 : JAK2
Running pathway anlysis
Done
Sleeping 2.83s before next kinase...
58 : HCK
Running pathway anlysis
Done
Sleeping 2.63s before next kinase...
59 : PTK6
Running pathway anlysis
Done
Sleeping 2.23s before next kinase...
60 : PDGFRB
Running pathway anlysis
Done
Sleeping 3.30s before next kinase...
61 : MLK1
Running pathway anlysis
Done
Sleeping 3.62s before next kinase...
62 : JNK1
Running pathway anlysis
Done
Sleeping 3.77s before next kinase...
63 : ABL2
Running pathway anlysis
Done
Sleeping 3.98s before next kinase...
64 : P38A
Running pathway anlysis
Done
Sleeping 3.98s before next kinase...
65 : KDR
Running pathway anlysis
Done
Sleeping 3.05s before next kinase...
66 : NEK6
Running pathway anlysis
Done
Sleeping 3.35s before next kinase...
67 : EPHA6
Running pathway anlysis
Done
Sleeping 3.38s before next kinase...
68 : PLK1
Running pathway anlysis
Done
Sleeping 3.46s before next kinase...
69 : PTK2B
Running pathway anlysis
Done
Sleeping 3.93s before next kinase...
70 : HGK
Running pathway anlysis
Done
Sleeping 2.89s before next kinase...
71 : MST1
Running pathway anlysis
Done
Sleeping 1.87s before next kinase...
72 : CK2A1
Running pathway anlysis
Done
Sleeping 3.12s before next kinase...
73 : MST3
Running pathway anlysis
Done
Sleeping 3.50s before next kinase...
74 : TTBK1
Running pathway anlysis
Done
Sleeping 3.65s before next kinase...
75 : NTRK2
Running pathway anlysis
Done
Sleeping 3.24s before next kinase...
76 : IKKE
Running pathway anlysis
Done
Sleeping 2.24s before next kinase...
77 : DDR2
Running pathway anlysis
Done
Sleeping 3.92s before next kinase...
78 : MST2
Running pathway anlysis
Done
Sleeping 1.59s before next kinase...
79 : NEK2
Running pathway anlysis
Done
Sleeping 1.64s before next kinase...
80 : P38D
Running pathway anlysis
Done
Sleeping 3.94s before next kinase...
81 : EGFR
Running pathway anlysis
Done
Sleeping 1.71s before next kinase...
82 : NEK7
Running pathway anlysis
Done
Sleeping 2.05s before next kinase...
83 : DYRK2
Running pathway anlysis
Done
Sleeping 3.52s before next kinase...
84 : MINK
Running pathway anlysis
Done
Sleeping 2.54s before next kinase...
85 : TXK
Running pathway anlysis
Done
Sleeping 3.38s before next kinase...
86 : CSF1R
Running pathway anlysis
Done
Sleeping 2.40s before next kinase...
87 : NEK1
Running pathway anlysis
Done
Sleeping 3.38s before next kinase...
88 : ZAP70
Running pathway anlysis
Done
Sleeping 1.60s before next kinase...
89 : MST4
Running pathway anlysis
Done
Sleeping 2.67s before next kinase...
90 : RSK3
Running pathway anlysis
Done
Sleeping 2.14s before next kinase...
91 : KHS1
Running pathway anlysis
Done
Sleeping 2.08s before next kinase...
92 : CHK1
Running pathway anlysis
Done
Sleeping 2.58s before next kinase...
93 : QIK
Running pathway anlysis
Done
Sleeping 2.27s before next kinase...
94 : CK1D
Running pathway anlysis
Done
Sleeping 2.55s before next kinase...
95 : FGFR3
Running pathway anlysis
Done
Sleeping 1.61s before next kinase...
96 : YSK1
Running pathway anlysis
Done
Sleeping 1.86s before next kinase...
97 : KIT
Running pathway anlysis
Done
Sleeping 3.24s before next kinase...
98 : P38G
Running pathway anlysis
Done
Sleeping 1.86s before next kinase...
99 : TNIK
Running pathway anlysis
Done
Sleeping 3.95s before next kinase...
100 : NEK4
Running pathway anlysis
Done
Sleeping 2.65s before next kinase...
101 : ITK
Running pathway anlysis
Done
Sleeping 2.22s before next kinase...
102 : CDK5
Running pathway anlysis
Done
Sleeping 1.59s before next kinase...
103 : PKACB
Running pathway anlysis
Done
Sleeping 3.71s before next kinase...
104 : GSK3B
Running pathway anlysis
Done
Sleeping 3.77s before next kinase...
105 : AKT1
Running pathway anlysis
Done
Sleeping 2.62s before next kinase...
106 : AURB
Running pathway anlysis
Done
Sleeping 2.63s before next kinase...
107 : TAO2
Running pathway anlysis
Done
Sleeping 1.69s before next kinase...
108 : MAPKAPK3
Running pathway anlysis
Done
Sleeping 1.84s before next kinase...
109 : NIM1
Running pathway anlysis
Done
Sleeping 3.43s before next kinase...
110 : FGFR4
Running pathway anlysis
Done
Sleeping 2.84s before next kinase...
111 : AURA
Running pathway anlysis
Done
Sleeping 3.15s before next kinase...
112 : FLT1
Running pathway anlysis
Done
Sleeping 2.50s before next kinase...
113 : PAK3
Running pathway anlysis
Done
Sleeping 2.53s before next kinase...
114 : EPHA4
Running pathway anlysis
Done
Sleeping 1.83s before next kinase...
115 : MLK3
Running pathway anlysis
Done
Sleeping 2.17s before next kinase...
116 : CAMK2A
Running pathway anlysis
Done
Sleeping 2.36s before next kinase...
117 : PIM1
Running pathway anlysis
Done
Sleeping 2.78s before next kinase...
118 : AURC
Running pathway anlysis
Done
Sleeping 2.92s before next kinase...
119 : CK2A2
Running pathway anlysis
Done
Sleeping 2.36s before next kinase...
120 : CAMK2G
Running pathway anlysis
Done
Sleeping 2.12s before next kinase...
121 : MSK1
Running pathway anlysis
Done
Sleeping 1.56s before next kinase...
122 : PLK3
Running pathway anlysis
Done
Sleeping 2.92s before next kinase...
123 : MARK4
Running pathway anlysis
Done
Sleeping 3.63s before next kinase...
124 : NEK9
Running pathway anlysis
Done
Sleeping 2.95s before next kinase...
125 : MEKK1
Running pathway anlysis
Done
Sleeping 1.64s before next kinase...
126 : P70S6K
Running pathway anlysis
Done
Sleeping 2.51s before next kinase...
127 : IKKA
Running pathway anlysis
Done
Sleeping 3.15s before next kinase...
128 : ZAK
Running pathway anlysis
Done
Sleeping 2.71s before next kinase...
129 : GCK
Running pathway anlysis
Done
Sleeping 3.23s before next kinase...
130 : PAK2
Running pathway anlysis
Done
Sleeping 2.40s before next kinase...
131 : P38B
Running pathway anlysis
Done
Sleeping 3.01s before next kinase...
132 : MARK2
Running pathway anlysis
Done
Sleeping 3.34s before next kinase...
133 : MLK2
Running pathway anlysis
Done
Sleeping 2.19s before next kinase...
134 : PAK1
Running pathway anlysis
Done
Sleeping 3.37s before next kinase...
135 : PKCB
Running pathway anlysis
Done
Sleeping 1.73s before next kinase...
136 : SGK3
Running pathway anlysis
Done
Sleeping 1.74s before next kinase...
137 : JNK2
Running pathway anlysis
Done
Sleeping 3.29s before next kinase...
138 : CAMK2D
Running pathway anlysis
Done
Sleeping 1.84s before next kinase...
139 : TAK1
Running pathway anlysis
Done
Sleeping 1.83s before next kinase...
140 : PHKG2
Running pathway anlysis
Done
Sleeping 2.46s before next kinase...
141 : ERK5
Running pathway anlysis
Done
Sleeping 3.75s before next kinase...
142 : AKT2
Running pathway anlysis
Done
Sleeping 1.57s before next kinase...
143 : TBK1
Running pathway anlysis
Done
Sleeping 2.26s before next kinase...
144 : DYRK3
Running pathway anlysis
Done
Sleeping 1.89s before next kinase...
145 : LIMK1
Running pathway anlysis
Done
Sleeping 1.93s before next kinase...
146 : PIM3
Running pathway anlysis
Done
Sleeping 3.99s before next kinase...
147 : P90RSK
Running pathway anlysis
Done
Sleeping 3.55s before next kinase...
148 : NLK
Running pathway anlysis
Done
Sleeping 2.52s before next kinase...
149 : MAPKAPK5
Running pathway anlysis
Done
Sleeping 1.73s before next kinase...
150 : TSSK1
Running pathway anlysis
Done
Sleeping 2.12s before next kinase...
151 : TLK1
Running pathway anlysis
Done
Sleeping 3.08s before next kinase...
152 : DCAMKL2
Running pathway anlysis
Done
Sleeping 3.20s before next kinase...
153 : SIK
Running pathway anlysis
Done
Sleeping 2.24s before next kinase...
154 : MATK
Running pathway anlysis
Done
Sleeping 3.98s before next kinase...
155 : CAMK2B
Running pathway anlysis
Done
Sleeping 3.81s before next kinase...
156 : MAPKAPK2
Running pathway anlysis
Done
Sleeping 1.51s before next kinase...
157 : ALK4
Running pathway anlysis
Done
Sleeping 3.21s before next kinase...
158 : JNK3
Running pathway anlysis
Done
Sleeping 3.60s before next kinase...
159 : IRAK1
Running pathway anlysis
Done
Sleeping 3.96s before next kinase...
160 : AMPKA1
Running pathway anlysis
Done
Sleeping 1.83s before next kinase...
161 : TNK2
Running pathway anlysis
Done
Sleeping 3.85s before next kinase...
162 : DSTYK
Running pathway anlysis
Done
Sleeping 2.99s before next kinase...
163 : PRKD3
Running pathway anlysis
Done
Sleeping 3.16s before next kinase...
164 : IRAK4
Running pathway anlysis
Done
Sleeping 2.21s before next kinase...
165 : MNK2
Running pathway anlysis
Done
Sleeping 2.77s before next kinase...
166 : MARK1
Running pathway anlysis
Done
Sleeping 2.51s before next kinase...
167 : RSK4
Running pathway anlysis
Done
Sleeping 2.29s before next kinase...
168 : CDK9
Running pathway anlysis
Done
Sleeping 2.41s before next kinase...
169 : TAO1
Running pathway anlysis
Done
Sleeping 2.63s before next kinase...
170 : CK1E
Running pathway anlysis
Done
Sleeping 2.23s before next kinase...
171 : PKCT
Running pathway anlysis
Done
Sleeping 3.84s before next kinase...
172 : SGK1
Running pathway anlysis
Done
Sleeping 2.06s before next kinase...
173 : TSSK2
Running pathway anlysis
Done
Sleeping 2.85s before next kinase...
174 : TLK2
Running pathway anlysis
Done
Sleeping 3.16s before next kinase...
175 : PKG2
Running pathway anlysis
Done
Sleeping 3.40s before next kinase...
176 : DYRK4
Running pathway anlysis
Done
Sleeping 2.46s before next kinase...
177 : MARK3
Running pathway anlysis
Done
Sleeping 3.80s before next kinase...
178 : RSK2
Running pathway anlysis
Done
Sleeping 3.94s before next kinase...
179 : DAPK1
Running pathway anlysis
Done
Sleeping 2.39s before next kinase...
180 : PRKD1
Running pathway anlysis
Done
Sleeping 1.72s before next kinase...
181 : PKN1
Running pathway anlysis
Done
Sleeping 2.21s before next kinase...
182 : ULK3
Running pathway anlysis
Done
Sleeping 2.13s before next kinase...
183 : GRK2
Running pathway anlysis
Done
Sleeping 2.33s before next kinase...
184 : BMPR1B
Running pathway anlysis
Done
Sleeping 1.76s before next kinase...
185 : AMPKA2
Running pathway anlysis
Done
Sleeping 1.72s before next kinase...
186 : CSK
Running pathway anlysis
Done
Sleeping 2.94s before next kinase...
187 : NEK11
Running pathway anlysis
Done
Sleeping 2.23s before next kinase...
188 : PTK2
Running pathway anlysis
Done
Sleeping 2.27s before next kinase...
189 : ATM
Running pathway anlysis
Done
Sleeping 3.24s before next kinase...
190 : ROCK1
Running pathway anlysis
Done
Sleeping 2.44s before next kinase...
191 : LATS2
Running pathway anlysis
Done
Sleeping 1.74s before next kinase...
192 : DYRK1A
Running pathway anlysis
Done
Sleeping 1.66s before next kinase...
193 : CK1G2
Running pathway anlysis
Done
Sleeping 3.13s before next kinase...
194 : GRK5
Running pathway anlysis
Done
Sleeping 1.84s before next kinase...
195 : PKCZ
Running pathway anlysis
Done
Sleeping 3.92s before next kinase...
196 : CK1A
Running pathway anlysis
Done
Sleeping 2.19s before next kinase...
197 : ALK2
Running pathway anlysis
Done
Sleeping 3.81s before next kinase...
198 : ROCK2
Running pathway anlysis
Done
Sleeping 3.17s before next kinase...
199 : CAMKK1
Running pathway anlysis
Done
Sleeping 3.09s before next kinase...
200 : WNK3
Running pathway anlysis
Done
Sleeping 2.84s before next kinase...
201 : TGFBR1
Running pathway anlysis
Done
Sleeping 1.63s before next kinase...
202 : PKCG
Running pathway anlysis
Done
Sleeping 3.68s before next kinase...
203 : JAK1
Running pathway anlysis
Done
Sleeping 3.40s before next kinase...
204 : CDK6
Running pathway anlysis
Done
Sleeping 2.73s before next kinase...
205 : TYK2
Running pathway anlysis
Done
Sleeping 2.05s before next kinase...
206 : MSK2
Running pathway anlysis
Done
Sleeping 2.37s before next kinase...
207 : PKCD
Running pathway anlysis
Done
Sleeping 3.88s before next kinase...
208 : PKG1
Running pathway anlysis
Done
Sleeping 3.47s before next kinase...
209 : GRK7
Running pathway anlysis
Done
Sleeping 2.30s before next kinase...
210 : NUAK2
Running pathway anlysis
Done
Sleeping 1.71s before next kinase...
211 : AKT3
Running pathway anlysis
Done
Sleeping 2.86s before next kinase...
212 : WNK1
Running pathway anlysis
Done
Sleeping 3.41s before next kinase...
213 : LKB1
Running pathway anlysis
Done
Sleeping 3.17s before next kinase...
214 : PBK
Running pathway anlysis
Done
Sleeping 2.27s before next kinase...
215 : SKMLCK
Running pathway anlysis
Done
Sleeping 2.63s before next kinase...
216 : WNK2
Running pathway anlysis
Done
Sleeping 1.95s before next kinase...
217 : SGK2
Running pathway anlysis
Done
Sleeping 2.16s before next kinase...
218 : PRKD2
Running pathway anlysis
Done
Sleeping 1.97s before next kinase...
219 : PLK4
Running pathway anlysis
Done
Sleeping 3.44s before next kinase...
220 : CAMK4
Running pathway anlysis
Done
Sleeping 2.99s before next kinase...
221 : LATS1
Running pathway anlysis
Done
Sleeping 3.05s before next kinase...
222 : ICK
Running pathway anlysis
Done
Sleeping 3.25s before next kinase...
223 : CK1G1
Running pathway anlysis
Done
Sleeping 2.92s before next kinase...
224 : CDK7
Running pathway anlysis
Done
Sleeping 3.19s before next kinase...
225 : PKCH
Running pathway anlysis
Done
Sleeping 2.85s before next kinase...
226 : WEE1
Running pathway anlysis
Done
Sleeping 2.81s before next kinase...
227 : CHK2
Running pathway anlysis
Done
Sleeping 2.06s before next kinase...
228 : CDK3
Running pathway anlysis
Done
Sleeping 2.75s before next kinase...
229 : CK1G3
Running pathway anlysis
Done
Sleeping 3.98s before next kinase...
230 : SLK
Running pathway anlysis
Done
Sleeping 1.70s before next kinase...
231 : MEK2
Running pathway anlysis
Done
Sleeping 1.55s before next kinase...
232 : PKCI
Running pathway anlysis
Done
Sleeping 1.85s before next kinase...
233 : PRKX
Running pathway anlysis
Done
Sleeping 1.90s before next kinase...
234 : PKCE
Running pathway anlysis
Done
Sleeping 2.39s before next kinase...
235 : CAMKK2
Running pathway anlysis
Done
Sleeping 1.77s before next kinase...
236 : MTOR
Running pathway anlysis
Done
Sleeping 2.74s before next kinase...
237 : DCAMKL1
Running pathway anlysis
Done
Sleeping 3.26s before next kinase...
238 : EEF2K
Running pathway anlysis
Done
Sleeping 1.87s before next kinase...
239 : MELK
Running pathway anlysis
Done
Sleeping 3.00s before next kinase...
240 : HIPK2
Running pathway anlysis
Done
Sleeping 2.25s before next kinase...
241 : CDK4
Running pathway anlysis
Done
Sleeping 3.08s before next kinase...
242 : GSK3A
Running pathway anlysis
Done
Sleeping 2.70s before next kinase...
243 : NDR1
Running pathway anlysis
Done
Sleeping 1.67s before next kinase...
244 : PIM2
Running pathway anlysis
Done
Sleeping 3.44s before next kinase...
245 : P70S6KB
Running pathway anlysis
Done
Sleeping 2.46s before next kinase...
246 : MEK1
Running pathway anlysis
Done
Sleeping 2.71s before next kinase...
247 : NUAK1
Running pathway anlysis
Done
Sleeping 2.65s before next kinase...
248 : TESK1
Running pathway anlysis
Done
Sleeping 2.11s before next kinase...
249 : MOK
Running pathway anlysis
Done
Sleeping 2.11s before next kinase...
250 : OSR1
Running pathway anlysis
Done
Sleeping 1.96s before next kinase...
251 : PAK6
Running pathway anlysis
Done
Sleeping 2.83s before next kinase...
252 : NDR2
Running pathway anlysis
Done
Sleeping 2.80s before next kinase...
253 : CLK1
Running pathway anlysis
Done
Sleeping 1.70s before next kinase...
254 : ERK7
Running pathway anlysis
Done
Sleeping 3.15s before next kinase...
255 : COT
Running pathway anlysis
Done
Sleeping 2.41s before next kinase...
256 : PHKG1
Running pathway anlysis
Done
Sleeping 2.69s before next kinase...
257 : ACVR2A
Running pathway anlysis
Done
Sleeping 1.62s before next kinase...
258 : DNAPK
Running pathway anlysis
Done
Sleeping 3.98s before next kinase...
259 : HIPK4
Running pathway anlysis
Done
Sleeping 1.86s before next kinase...
260 : MEKK2
Running pathway anlysis
Done
Sleeping 3.13s before next kinase...
261 : MAP2K4
Running pathway anlysis
Done
Sleeping 3.36s before next kinase...
262 : PERK
Running pathway anlysis
Done
Sleeping 2.42s before next kinase...
263 : MOS
Running pathway anlysis
Done
Sleeping 3.51s before next kinase...
264 : PAK5
Running pathway anlysis
Done
Sleeping 2.23s before next kinase...
265 : LIMK2
Running pathway anlysis
Done
Sleeping 3.76s before next kinase...
266 : LRRK2
Running pathway anlysis
Done
Sleeping 2.08s before next kinase...
267 : BRSK2
Running pathway anlysis
Done
Sleeping 2.61s before next kinase...
268 : PAK4
Running pathway anlysis
Done
Sleeping 3.44s before next kinase...
269 : HASPIN
Running pathway anlysis
Done
Sleeping 2.04s before next kinase...
270 : ATR
Running pathway anlysis
Done
Sleeping 3.19s before next kinase...
271 : CAMK1D
Running pathway anlysis
Done
Sleeping 1.99s before next kinase...
272 : STLK3
Running pathway anlysis
Done
Sleeping 1.70s before next kinase...
273 : CLK3
Running pathway anlysis
Done
Sleeping 2.34s before next kinase...
274 : ASK1
Running pathway anlysis
Done
Sleeping 3.90s before next kinase...
275 : CAMK1A
Running pathway anlysis
Done
Sleeping 2.40s before next kinase...
276 : HIPK1
Running pathway anlysis
Done
Sleeping 1.76s before next kinase...
277 : PKN2
Running pathway anlysis
Done
Sleeping 3.38s before next kinase...
278 : MEK5
Running pathway anlysis
Done
Sleeping 1.68s before next kinase...
279 : MRCKB
Running pathway anlysis
Done
Sleeping 2.98s before next kinase...
280 : MAP2K7
Running pathway anlysis
Done
Sleeping 1.83s before next kinase...
281 : MPSK1
Running pathway anlysis
Done
Sleeping 1.82s before next kinase...
282 : CDK8
Running pathway anlysis
Done
Sleeping 1.83s before next kinase...
283 : BUB1
Running pathway anlysis
Done
Sleeping 2.51s before next kinase...
284 : MEKK3
Running pathway anlysis
Done
Sleeping 3.58s before next kinase...
285 : PDHK1
Running pathway anlysis
Done
Sleeping 1.98s before next kinase...
286 : HIPK3
Running pathway anlysis
Done
Sleeping 2.48s before next kinase...
287 : MAP2K3
Running pathway anlysis
Done
Sleeping 2.21s before next kinase...
288 : TTK
Running pathway anlysis
Done
Sleeping 2.92s before next kinase...
289 : GRK1
Running pathway anlysis
Done
Sleeping 2.51s before next kinase...
290 : PLK2
Running pathway anlysis
Done
Sleeping 3.79s before next kinase...
291 : STK33
Running pathway anlysis
Done
Sleeping 3.36s before next kinase...
292 : HRI
Running pathway anlysis
Done
Sleeping 3.23s before next kinase...
293 : MAP2K6
Running pathway anlysis
Done
Sleeping 3.20s before next kinase...
294 : BMPR2
Running pathway anlysis
Done
Sleeping 3.90s before next kinase...
295 : RIPK2
Running pathway anlysis
Done
Sleeping 3.93s before next kinase...
296 : DYRK1B
Running pathway anlysis
Done
Sleeping 2.54s before next kinase...
297 : LOK
Running pathway anlysis
Done
Sleeping 3.39s before next kinase...
298 : DDR1
Running pathway anlysis
Done
Sleeping 2.49s before next kinase...
299 : PASK
Running pathway anlysis
Done
Sleeping 3.63s before next kinase...
300 : ULK1
Running pathway anlysis
Done
Sleeping 3.31s before next kinase...
301 : PDK1
Running pathway anlysis
Done
Sleeping 3.06s before next kinase...
302 : PKACG
Running pathway anlysis
Done
Sleeping 1.64s before next kinase...
303 : BMPR1A
Running pathway anlysis
Done
Sleeping 3.76s before next kinase...
304 : TNNI3K
Running pathway anlysis
Done
Sleeping 2.95s before next kinase...
305 : BRSK1
Running pathway anlysis
Done
Sleeping 2.36s before next kinase...
306 : PDHK3
Running pathway anlysis
Done
Sleeping 2.71s before next kinase...
307 : ERBB2
Running pathway anlysis
Done
Sleeping 1.91s before next kinase...
308 : CLK2
Running pathway anlysis
Done
Sleeping 2.07s before next kinase...
309 : SRPK3
Running pathway anlysis
Done
Sleeping 3.05s before next kinase...
310 : DAPK3
Running pathway anlysis
Done
Sleeping 3.62s before next kinase...
311 : ROR2
Running pathway anlysis
Done
Sleeping 1.66s before next kinase...
312 : CHAK1
Running pathway anlysis
Done
Sleeping 1.68s before next kinase...
313 : PKR
Running pathway anlysis
Done
Sleeping 2.78s before next kinase...
314 : NIK
Running pathway anlysis
Done
Sleeping 3.63s before next kinase...
315 : GCN2
Running pathway anlysis
Done
Sleeping 2.24s before next kinase...
316 : SRPK2
Running pathway anlysis
Done
Sleeping 1.80s before next kinase...
317 : VRK2
Running pathway anlysis
Done
Sleeping 2.59s before next kinase...
318 : VRK1
Running pathway anlysis
Done
Sleeping 2.52s before next kinase...
319 : PKMYT1
Running pathway anlysis
Done
Sleeping 2.25s before next kinase...
320 : GRK3
Running pathway anlysis
Done
Sleeping 3.80s before next kinase...
321 : CAMK1B
Running pathway anlysis
Done
Sleeping 2.92s before next kinase...
322 : CDC7
Running pathway anlysis
Done
Sleeping 3.70s before next kinase...
323 : SMMLCK
Running pathway anlysis
Done
Sleeping 3.90s before next kinase...
324 : ROR1
Running pathway anlysis
Done
Sleeping 2.98s before next kinase...
325 : GAK
Running pathway anlysis
Done
Sleeping 2.75s before next kinase...
326 : MAST2
Running pathway anlysis
Done
Sleeping 2.68s before next kinase...
327 : BRAF
Running pathway anlysis
Done
Sleeping 2.51s before next kinase...</code></pre>
</div>
</div>
</section>
<section id="get-accuracy-for-overlap-pathways" class="level2">
<h2 class="anchored" data-anchor-id="get-accuracy-for-overlap-pathways">Get accuracy for overlap pathways</h2>
<div id="f23e74f7-9214-4696-ab05-9173819b9fb1" class="cell" data-execution_count="16">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>acc_dict, num_dict <span class="op">=</span> {}, {}</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>acc_lo_dict, num_lo_dict <span class="op">=</span> {}, {}</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> k,r <span class="kw">in</span> tqdm(cddm_info.iterrows(),total<span class="op">=</span><span class="bu">len</span>(cddm_info)):</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>    save_dir <span class="op">=</span> <span class="ss">f'fig/cddm/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/path'</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>    path_df<span class="op">=</span>pd.read_parquet(<span class="ss">f'</span><span class="sc">{</span>save_dir<span class="sc">}</span><span class="ss">/path.parquet'</span>)</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> path_df.ref_path.<span class="bu">sum</span>()<span class="op">!=</span><span class="dv">0</span>:</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>        accuracy <span class="op">=</span> get_overlap(path_df, ref_col<span class="op">=</span><span class="st">'ref_path'</span>,plot<span class="op">=</span><span class="va">False</span>,p_type<span class="op">=</span><span class="st">'p'</span>)</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>        acc_dict[k] <span class="op">=</span> accuracy</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>        num_dict[k] <span class="op">=</span>path_df.ref_path.<span class="bu">sum</span>()</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> path_df.ref_path_lowest.<span class="bu">sum</span>()<span class="op">!=</span><span class="dv">0</span>:</span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>        accuracy_lo <span class="op">=</span> get_overlap(path_df, ref_col<span class="op">=</span><span class="st">'ref_path_lowest'</span>,plot<span class="op">=</span><span class="va">False</span>,p_type<span class="op">=</span><span class="st">'p'</span>)</span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>        acc_lo_dict[k] <span class="op">=</span> accuracy_lo</span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>        num_lo_dict[k] <span class="op">=</span> path_df.ref_path_lowest.<span class="bu">sum</span>()</span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>100%|███████████████████████████████████████████████████████████████████████████████████████████| 328/328 [00:01&lt;00:00, 214.60it/s]</code></pre>
</div>
</div>
<div id="360b7a10-6287-41e0-8187-080978d08405" class="cell" data-execution_count="17">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>cddm_info[<span class="st">'accuracy'</span>] <span class="op">=</span> acc_dict</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>cddm_info[<span class="st">'path_num'</span>] <span class="op">=</span> num_dict</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>cddm_info[<span class="st">'accuracy_lowest'</span>] <span class="op">=</span> acc_lo_dict</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>cddm_info[<span class="st">'path_lowest_num'</span>] <span class="op">=</span> num_lo_dict</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="da901a88-7c3d-470c-971b-6d82f889f35b" class="cell" data-execution_count="18">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>cddm_info[<span class="st">'path_num'</span>] <span class="op">=</span> cddm_info[<span class="st">'path_num'</span>].fillna(<span class="dv">0</span>).astype(<span class="bu">int</span>)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>cddm_info[<span class="st">'path_lowest_num'</span>] <span class="op">=</span> cddm_info[<span class="st">'path_lowest_num'</span>].fillna(<span class="dv">0</span>).astype(<span class="bu">int</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="e3aa7122-4409-48dc-bb75-9db9280fbdeb" class="cell" data-execution_count="20">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>cddm_info.to_parquet(<span class="st">'raw/cddm_path_info.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="analysis" class="level2">
<h2 class="anchored" data-anchor-id="analysis">Analysis</h2>
<div id="37516382-5e4c-4a44-af32-fb99fe98c2aa" class="cell" data-execution_count="7">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_path_hist(df,</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>              colname,</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>              <span class="op">**</span>kwargs,</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>              ):</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>    <span class="co">"Plot histogram of a column (kinase). "</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">6</span>,<span class="dv">2</span>))</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>    hist_params <span class="op">=</span> {<span class="st">'element'</span>:<span class="st">'poly'</span>,</span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>              <span class="st">'edgecolor'</span>: <span class="va">None</span>,</span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>              <span class="st">'alpha'</span>:<span class="fl">0.5</span>,</span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>              <span class="st">'bins'</span>:<span class="dv">100</span>,</span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a>              <span class="st">'kde'</span>:<span class="va">True</span>,</span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a>              }</span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>    sns.histplot(data<span class="op">=</span>df,x<span class="op">=</span>colname,<span class="op">**</span>kwargs, <span class="op">**</span>hist_params)</span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">''</span>)</span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Accuracy score distribution'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="991a16f8-fd1b-4a6f-ae10-ad1f8880198c" class="cell" data-execution_count="40">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb35"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>q75, q50 <span class="op">=</span> cddm_info.path_num[<span class="kw">lambda</span> x: x<span class="op">!=</span><span class="dv">0</span>].quantile([<span class="fl">0.75</span>,<span class="fl">0.5</span>])</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>q75, q50</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="40">
<pre><code>(37.5, 14.0)</code></pre>
</div>
</div>
<div id="e9263d33-2334-4b44-865e-e8efa8675347" class="cell" data-execution_count="41">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb37"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> categorize_path_num(x):</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> x <span class="op">&gt;=</span> q75: <span class="cf">return</span> <span class="ss">f'well known (≥</span><span class="sc">{</span>q75<span class="sc">:.0f}</span><span class="ss">)'</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> x <span class="op">&lt;=</span> q50: <span class="cf">return</span> <span class="ss">f'less known (≤</span><span class="sc">{</span>q50<span class="sc">:.0f}</span><span class="ss">)'</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>: <span class="cf">return</span> <span class="ss">f'middle (</span><span class="sc">{</span>q50<span class="sc">:.0f}</span><span class="ss">–</span><span class="sc">{</span>q75<span class="sc">:.0f}</span><span class="ss">)'</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="31dea136-6d63-4fea-94f1-dce49aaed503" class="cell" data-execution_count="42">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb38"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>cddm_info[<span class="st">'category'</span>] <span class="op">=</span> cddm_info[<span class="st">'path_num'</span>].<span class="bu">apply</span>(categorize_path_num)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="b8dfd7fc-49fb-4ab1-b55d-2bcee43731c1" class="cell" data-execution_count="43">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb39"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>cddm_info[<span class="st">'category'</span>].value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="43">
<pre><code>category
less known (≤14)    188
well known (≥38)     62
middle (14–38)       61
Name: count, dtype: int64</code></pre>
</div>
</div>
<div id="e1807af2-c02d-45e4-ac36-9c92bd092c23" class="cell" data-execution_count="44">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb41"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>cddm_info.hist(column<span class="op">=</span><span class="st">'accuracy'</span>, by<span class="op">=</span><span class="st">'category'</span>, bins<span class="op">=</span><span class="dv">50</span>, figsize<span class="op">=</span>(<span class="dv">8</span>,<span class="dv">6</span>))<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="19c_reactome_path_plot_cddm_files/figure-html/cell-33-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div id="86c6300d-241c-46b1-a5d1-ab4e9071a0cc" class="cell" data-execution_count="45">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb42"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>cddm_info[cddm_info.path_num<span class="op">&gt;</span>q50].accuracy.mean()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="45">
<pre><code>np.float64(0.5057438274707422)</code></pre>
</div>
</div>
<div id="c5ed00eb-f28e-40e2-ba29-8919302b6147" class="cell" data-execution_count="46">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb44"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>cddm_info[cddm_info.path_num<span class="op">&gt;</span>q50].accuracy.hist(bins<span class="op">=</span><span class="dv">50</span>,figsize<span class="op">=</span>(<span class="dv">5</span>,<span class="dv">3</span>))</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>plt.xlabel(<span class="st">'Accuracy'</span>)</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>plt.ylabel(<span class="st">'Number of kinases'</span>)</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>plt.title(<span class="st">'CDDM accuracy distribution'</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="19c_reactome_path_plot_cddm_files/figure-html/cell-35-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<section id="lowest" class="level3">
<h3 class="anchored" data-anchor-id="lowest">Lowest</h3>
<div id="5f461f60-423a-485c-8f0e-2fd12bb7d41f" class="cell" data-execution_count="47">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb45"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>cddm_info.path_lowest_num[<span class="kw">lambda</span> x: x<span class="op">!=</span><span class="dv">0</span>].quantile([<span class="fl">0.75</span>])</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="47">
<pre><code>0.75    11.5
Name: path_lowest_num, dtype: float64</code></pre>
</div>
</div>
<div id="0a7c75e6-7427-4bfd-997f-9872d2f7108a" class="cell" data-execution_count="48">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb47"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>cddm_info[cddm_info.path_lowest_num<span class="op">&gt;</span><span class="dv">12</span>].accuracy.hist(bins<span class="op">=</span><span class="dv">50</span>,figsize<span class="op">=</span>(<span class="dv">5</span>,<span class="dv">3</span>))</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>plt.xlabel(<span class="st">'Accuracy'</span>)</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>plt.ylabel(<span class="st">'Number of kinases'</span>)</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>plt.title(<span class="st">'CDDM accuracy distribution (lowest-level)'</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="19c_reactome_path_plot_cddm_files/figure-html/cell-37-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="compare-with-pspa" class="level2">
<h2 class="anchored" data-anchor-id="compare-with-pspa">Compare with PSPA</h2>
<div id="e2360d7e-3d77-4535-8802-6f859c776780" class="cell" data-execution_count="27">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb48"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>pspa_info<span class="op">=</span>pd.read_parquet(<span class="st">'raw/pspa_path_info.parquet'</span>)</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>cddm_info <span class="op">=</span> pd.read_parquet(<span class="st">'raw/cddm_path_info.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="b3afcd2c-ec1d-46cc-9848-d11f241e0053" class="cell" data-execution_count="28">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb49"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>idx <span class="op">=</span> <span class="bu">list</span>(<span class="bu">set</span>(pspa_info.index)<span class="op">&amp;</span><span class="bu">set</span>(cddm_info.index))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="240ec845-527a-4634-91ff-d83b40d9a280" class="cell" data-execution_count="33">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb50"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># make sure total path number are same among the two</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>(cddm_info.loc[idx].path_num<span class="op">==</span>pspa_info.loc[idx].path_num).value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="33">
<pre><code>path_num
True    311
Name: count, dtype: int64</code></pre>
</div>
</div>
<div id="d9f8ee5f-f390-4d41-b52a-53f6b858e996" class="cell" data-execution_count="49">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb52"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>pspa_info <span class="op">=</span> pspa_info.loc[idx][pspa_info.path_num<span class="op">&gt;</span><span class="dv">14</span>]</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>cddm_info <span class="op">=</span> cddm_info.loc[idx][cddm_info.path_num<span class="op">&gt;</span><span class="dv">14</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="1d48c489-b492-4d99-a48c-b59e51556fd6" class="cell" data-execution_count="51">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb53"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>pspa_info[<span class="st">'cddm_accuracy'</span>] <span class="op">=</span> cddm_info.accuracy</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="2e4f3974-9d51-4c63-b1a7-2be146b53a34" class="cell" data-execution_count="52">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb54"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>pspa_info[<span class="st">'cddm_accuracy'</span>].isna().<span class="bu">sum</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="52">
<pre><code>np.int64(0)</code></pre>
</div>
</div>
<div id="b7d2ee9f-0efb-4bd7-b6f6-2b78db4ec976" class="cell" data-execution_count="53">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb56"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>data <span class="op">=</span> pspa_info[[<span class="st">'accuracy'</span>,<span class="st">'cddm_accuracy'</span>]].melt()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="26f0ca6f-a9db-452a-a294-bf8c04881fdd" class="cell" data-execution_count="54">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb57"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>data.head()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="54">
<div>


<table class="dataframe caption-top table table-sm table-striped small" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">variable</th>
<th data-quarto-table-cell-role="th">value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<th data-quarto-table-cell-role="th">0</th>
<td>accuracy</td>
<td>0.580645</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">1</th>
<td>accuracy</td>
<td>0.200000</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">2</th>
<td>accuracy</td>
<td>0.345679</td>
</tr>
<tr class="even">
<th data-quarto-table-cell-role="th">3</th>
<td>accuracy</td>
<td>0.129032</td>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">4</th>
<td>accuracy</td>
<td>0.500000</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div id="d91a911a-de90-4cf3-a269-f78340a7c42a" class="cell" data-execution_count="55">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb58"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>data <span class="op">=</span> data.replace({</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">'accuracy'</span>: <span class="st">'PSPA'</span>,</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">'cddm_accuracy'</span>: <span class="st">'CDDM'</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="33fa0924-7f3b-4058-b74f-aa7a47398516" class="cell" data-execution_count="56">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb59"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_violin(</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>    data, </span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>    value<span class="op">=</span><span class="st">'value'</span>,</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>    group<span class="op">=</span><span class="st">'variable'</span>, </span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>    ylabel<span class="op">=</span><span class="va">None</span>, </span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>    dots<span class="op">=</span><span class="va">True</span>, </span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>    figsize<span class="op">=</span>(<span class="dv">5</span>,<span class="dv">3</span>),</span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>    <span class="op">**</span>kwargs</span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>):</span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>    <span class="co">"Plot violin plots (with optional strip dots) for long-form data."</span></span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>figsize)</span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a>    ax<span class="op">=</span>sns.violinplot(</span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a>        data<span class="op">=</span>data, x<span class="op">=</span>group, y<span class="op">=</span>value,</span>
<span id="cb59-16"><a href="#cb59-16" aria-hidden="true" tabindex="-1"></a>        inner<span class="op">=</span><span class="st">'box'</span>, linewidth<span class="op">=</span><span class="dv">1</span>,</span>
<span id="cb59-17"><a href="#cb59-17" aria-hidden="true" tabindex="-1"></a>        cut<span class="op">=</span><span class="dv">0</span>,         <span class="co"># prevents tails extending beyond data range</span></span>
<span id="cb59-18"><a href="#cb59-18" aria-hidden="true" tabindex="-1"></a>        bw_adjust<span class="op">=</span><span class="fl">0.7</span>,  <span class="co"># the smaller, the shape have more curves that fit to the data</span></span>
<span id="cb59-19"><a href="#cb59-19" aria-hidden="true" tabindex="-1"></a>        hue<span class="op">=</span>group,<span class="op">**</span>kwargs</span>
<span id="cb59-20"><a href="#cb59-20" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb59-21"><a href="#cb59-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-22"><a href="#cb59-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> dots:</span>
<span id="cb59-23"><a href="#cb59-23" aria-hidden="true" tabindex="-1"></a>        sns.stripplot(</span>
<span id="cb59-24"><a href="#cb59-24" aria-hidden="true" tabindex="-1"></a>            data<span class="op">=</span>data, x<span class="op">=</span>group, y<span class="op">=</span>value,</span>
<span id="cb59-25"><a href="#cb59-25" aria-hidden="true" tabindex="-1"></a>            color<span class="op">=</span><span class="st">'k'</span>, size<span class="op">=</span><span class="dv">2</span>, jitter<span class="op">=</span><span class="fl">0.1</span>, alpha<span class="op">=</span><span class="fl">0.6</span></span>
<span id="cb59-26"><a href="#cb59-26" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb59-27"><a href="#cb59-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-28"><a href="#cb59-28" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">''</span>)</span>
<span id="cb59-29"><a href="#cb59-29" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> ylabel <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>: plt.ylabel(ylabel)</span>
<span id="cb59-30"><a href="#cb59-30" aria-hidden="true" tabindex="-1"></a>    plt.tight_layout()</span>
<span id="cb59-31"><a href="#cb59-31" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> ax</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="9f25e82a-e084-4047-a11b-35b9832ec044" class="cell" data-execution_count="57">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb60"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> statannotations.Annotator <span class="im">import</span> Annotator</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> itertools</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="13a1afca-80e7-4a5b-8f3f-ba1d33152418" class="cell" data-execution_count="58">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb61"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> add_stats(ax,data,value<span class="op">=</span><span class="st">'value'</span>,group<span class="op">=</span><span class="st">'variable'</span>,pairs<span class="op">=</span><span class="va">None</span>,test<span class="op">=</span><span class="st">'t-test_ind'</span>,loc<span class="op">=</span><span class="st">'inside'</span>,text_format<span class="op">=</span><span class="st">'star'</span>,<span class="op">**</span>kwargs):</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>    group_items <span class="op">=</span> data[group].unique()</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> pairs <span class="kw">is</span> <span class="va">None</span>: pairs <span class="op">=</span> <span class="bu">list</span>(itertools.combinations(group_items, <span class="dv">2</span>))</span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>    annotator <span class="op">=</span> Annotator(ax, pairs, data<span class="op">=</span>data, x<span class="op">=</span>group, y<span class="op">=</span>value)</span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>    annotator.configure(test<span class="op">=</span>test, text_format<span class="op">=</span>text_format, loc<span class="op">=</span>loc,verbose<span class="op">=</span><span class="va">False</span>,<span class="op">**</span>kwargs)</span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>    annotator.apply_and_annotate()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="d90d43ed-9003-47c2-8495-17c4ccf8bcbf" class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb62"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>wide <span class="op">=</span> pspa_info[[<span class="st">'accuracy'</span>,<span class="st">'cddm_accuracy'</span>]]</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>wide.columns<span class="op">=</span>[<span class="st">'PSPA'</span>,<span class="st">'CDDM'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="dee31ce2-9c12-4503-9d31-4c6b55e7a884" class="cell" data-execution_count="60">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb63"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> plot_violin(data,ylabel<span class="op">=</span><span class="st">'Accuracy'</span>,figsize<span class="op">=</span>(<span class="dv">4</span>,<span class="dv">3</span>))</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>plt.title(<span class="st">'Accuracy of pathway prediction'</span>)</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>add_stats(ax,data,test<span class="op">=</span><span class="st">'t-test_paired'</span>)</span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>save_svg(<span class="st">'fig/pathway_compare.svg'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="19c_reactome_path_plot_cddm_files/figure-html/cell-51-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div id="610b6bff-01a6-4ff4-8adf-346ccda07762" class="cell" data-execution_count="61">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb64"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>(pspa_info[<span class="st">'cddm_accuracy'</span>]<span class="op">-</span>pspa_info[<span class="st">'accuracy'</span>]).sort_values(ascending<span class="op">=</span><span class="va">False</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="61">
<pre><code>kinase
MEKK1     0.789474
ABL1      0.755102
PKCG      0.629630
TBK1      0.617021
CAMKK1    0.600000
            ...   
IRAK4    -0.366667
FGFR4    -0.393939
NTRK1    -0.437500
NTRK2    -0.600000
PDGFRB   -0.733333
Length: 123, dtype: float64</code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/sky1ove\.github\.io\/katlas-raw");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/sky1ove/katlas-raw/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></div></div></footer></body></html>