<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Plot reactome pathways: PSPA – katlas-raw</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-000f0825b9c55ed21d14970d810c14a9.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
<meta property="og:title" content="Plot reactome pathways: PSPA – katlas-raw">
<meta property="og:image" content="https://sky1ove.github.io/katlas-raw/19b_reactome_path_plot_pspa_files/figure-html/cell-14-output-1.png">
<meta property="og:site_name" content="katlas-raw">
<meta property="og:image:height" content="220">
<meta property="og:image:width" content="549">
<meta name="twitter:title" content="Plot reactome pathways: PSPA – katlas-raw">
<meta name="twitter:image" content="https://sky1ove.github.io/katlas-raw/19b_reactome_path_plot_pspa_files/figure-html/cell-14-output-1.png">
<meta name="twitter:image-height" content="220">
<meta name="twitter:image-width" content="549">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="./index.html" class="navbar-brand navbar-brand-logo">
    </a>
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">katlas-raw</span>
    </a>
  </div>
        <div class="quarto-navbar-tools tools-end">
</div>
          <div id="quarto-search" class="" title="Search"></div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./reactome_path_plot_pspa.html">Plot reactome pathways: PSPA</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      </a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">katlas-raw</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./preprocess_dataset.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preprocess kinase-substrate datasets</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./eda_fig1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Exploratory data analysis</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./site_promiscuity_fig1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Site promiscuity</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssm_hierarchical.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Hierarchical clustering of PSSM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05_PSSM_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">05_PSSM_example.html</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./modi_group.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Modi group</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scale_pspa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PSPA scaled data</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./background.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Background frequencies</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./specificity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Specificity comparison</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./sty_composition.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">STY composition</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./plot_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot heatmap and logo of CDDM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./plot_pspa_figure.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PSPA figures</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./13_CDDM_umap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CDDM umap</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./get_aa_relationship.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get AA relationship</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pspa_cddm_heatmap_compare.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Heatmap/ST/JSD/cosine comparison</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_no_psp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pssms_split_test_psp_02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get kinase-specific PSSMs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring_compare_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scoring evaluation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scoring_compare_psp_02.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Scoring evaluation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Reactome pathway for each PSSM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_plot_pspa.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Plot reactome pathways: PSPA</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./reactome_path_plot_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot reactome pathways: CDDM</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./motif_clustering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Kmeans motifs</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./motif_plot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Motif plot</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./22_motif_web_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">22_motif_web_data.html</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./23_motif_kinase_similarity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cosine similarity with PSPA</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">kinase_domain</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_01_get_all_kd.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Get all kinase domain sequences</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_02_pseudo_alignment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Alignment</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_03_feature_extract.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Extract KD features</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_04_prepare_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare data for training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_05_pspa_training_model_comparison_all.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Model comparison on T5 feature</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06a_pspa_training_feature_comparison_cnn_group.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on CNN model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06b_pspa_training_feature_comparison_cnn_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on CNN model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_06c_pspa_training_feature_comparison_mlp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training: Feature comparison on MLP model</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_07a_cddm_training_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_07b_pspa_training_subfamily.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">DL training</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_08a_similarity_relationship.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KD similarity &amp; PSSM similarity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_08b_similarity_relationship_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">KD similarity &amp; PSSM similarity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_09a_predict_pspa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Predict</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_09b_predict_cddm.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Predict</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_10_plot_umap_known_unk.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Plot UMAP of known &amp; unknown KD</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_11_plot_pspa_kd_t5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">kd_11_plot_pspa_kd_t5.html</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./kinase_domain/kd_12_compare_pred_psp_known.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Comopare the two</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#data" id="toc-data" class="nav-link active" data-scroll-target="#data">Data</a></li>
  <li><a href="#site-pssms" id="toc-site-pssms" class="nav-link" data-scroll-target="#site-pssms">Site pssms</a></li>
  <li><a href="#plot-distribution" id="toc-plot-distribution" class="nav-link" data-scroll-target="#plot-distribution">Plot distribution</a></li>
  <li><a href="#plot-reactome-pathway" id="toc-plot-reactome-pathway" class="nav-link" data-scroll-target="#plot-reactome-pathway">Plot reactome pathway</a></li>
  <li><a href="#get-accuracy-for-overlap-pathways" id="toc-get-accuracy-for-overlap-pathways" class="nav-link" data-scroll-target="#get-accuracy-for-overlap-pathways">Get accuracy for overlap pathways</a></li>
  <li><a href="#pathway-hue" id="toc-pathway-hue" class="nav-link" data-scroll-target="#pathway-hue">Pathway hue</a>
  <ul class="collapse">
  <li><a href="#lowest" id="toc-lowest" class="nav-link" data-scroll-target="#lowest">Lowest</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/sky1ove/katlas-raw/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div><div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="reactome_path_plot_pspa.html.md"><i class="bi bi-file-code"></i>CommonMark</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Plot reactome pathways: PSPA</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<!-- WARNING: THIS FILE WAS AUTOGENERATED! DO NOT EDIT! -->
<div id="622fd9c3-c77c-4f40-9cb4-2ccaa1154c7b" class="cell" data-execution_count="12">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.pssm <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.score <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.data <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.plot <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.pathway <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> katlas.utils <span class="im">import</span> <span class="op">*</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd,seaborn <span class="im">as</span> sns</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> matplotlib <span class="im">import</span> pyplot <span class="im">as</span> plt</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> tqdm <span class="im">import</span> tqdm</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> time, random</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<section id="data" class="level2">
<h2 class="anchored" data-anchor-id="data">Data</h2>
<div id="70a9e22d-2c0e-4e98-ad0b-756abe87a330" class="cell" data-execution_count="13">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>pspa_info<span class="op">=</span>pd.read_parquet(<span class="st">'raw/pspa_path_info.parquet'</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>df<span class="op">=</span>pd.read_parquet(<span class="st">'raw/pspa_score_df.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="cf767a2d-4347-4542-b1c8-1883e069107d" class="cell" data-execution_count="14">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>thr_cols <span class="op">=</span> [<span class="st">'S_thr'</span>,<span class="st">'T_thr'</span>,<span class="st">'Y_thr'</span>]</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>gene_cols <span class="op">=</span> [<span class="st">'S_genes'</span>,<span class="st">'T_genes'</span>,<span class="st">'Y_genes'</span>]</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>cnt_cols <span class="op">=</span> [<span class="st">'S_cnt'</span>,<span class="st">'T_cnt'</span>,<span class="st">'Y_cnt'</span>]</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>site_cols <span class="op">=</span> [<span class="st">'S_sites'</span>,<span class="st">'T_sites'</span>,<span class="st">'Y_sites'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="site-pssms" class="level2">
<h2 class="anchored" data-anchor-id="site-pssms">Site pssms</h2>
<div id="4cd09680-dfe8-43cf-a26a-addb1321cf91" class="cell" data-execution_count="4">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>human_site<span class="op">=</span> Data.get_human_site()</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>seq_map <span class="op">=</span> human_site.set_index(<span class="st">'sub_site'</span>)[<span class="st">'site_seq'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="5e10bc92-f15b-47af-994e-7191391e034a" class="cell" data-execution_count="6">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>series<span class="op">=</span>pspa_info.iloc[<span class="dv">0</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="77609ca2-4f48-489e-9594-d03899d7171f" class="cell" data-execution_count="7">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> get_site_pssm(series):</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>    pssm_dict <span class="op">=</span> {}</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>    count_dict<span class="op">=</span>{}</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> acceptor, sites <span class="kw">in</span> <span class="bu">zip</span>([<span class="st">'S'</span>,<span class="st">'T'</span>,<span class="st">'Y'</span>],series[site_cols]):</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>        sites<span class="op">=</span><span class="bu">list</span>(sites)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> sites:</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>            pssm <span class="op">=</span> get_prob(seq_map[sites])</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>            key <span class="op">=</span> series.name<span class="op">+</span><span class="st">'_'</span><span class="op">+</span>acceptor</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>            pssm_dict[key] <span class="op">=</span> pd.Series(flatten_pssm(pssm))</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>            count_dict[key]<span class="op">=</span><span class="bu">len</span>(sites)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> pssm_dict,count_dict</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="2d7467db-68bc-4346-ad2b-b641ce588ea3" class="cell" data-execution_count="8">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>pssm_out <span class="op">=</span> {}</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>count_out <span class="op">=</span> {}</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i,r <span class="kw">in</span> pspa_info.iterrows():</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>    pssms,counts <span class="op">=</span> get_site_pssm(r)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    pssm_out.update(pssms)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    count_out.update(counts)</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="acc9df07-3efb-4897-a040-27475a2a6b83" class="cell" data-execution_count="9">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>site_pssms <span class="op">=</span> pd.DataFrame(pssm_out).T</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="abb2e858-98e2-404c-a008-cd6c2f6e3651" class="cell" data-execution_count="11">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>pspa_info.index[<span class="dv">314</span>:]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="11">
<pre><code>Index(['MATK', 'DDR1', 'DDR2', 'EGFR', 'EPHA1', 'EPHA2', 'EPHA3', 'EPHA4',
       'EPHA5', 'EPHA6', 'EPHA7', 'EPHA8', 'EPHB1', 'EPHB2', 'EPHB3', 'EPHB4',
       'BMX', 'PTK2', 'FER', 'FES', 'FGFR1', 'FGFR2', 'FGFR3', 'FGFR4', 'FGR',
       'FLT3', 'FRK', 'FYN', 'HCK', 'ERBB2', 'ERBB4', 'IGF1R', 'INSR', 'INSRR',
       'ITK', 'JAK1', 'JAK2', 'JAK3', 'KIT', 'LCK', 'LIMK1_TYR', 'LIMK2_TYR',
       'LTK', 'LYN', 'MERTK', 'MET', 'MAP2K4_TYR', 'MAP2K6_TYR', 'MAP2K7_TYR',
       'MST1R', 'MUSK', 'PKMYT1_TYR', 'NEK10_TYR', 'PDGFRA', 'PDGFRB',
       'PDHK1_TYR', 'PDHK3_TYR', 'PDHK4_TYR', 'PINK1_TYR', 'PTK2B', 'RET',
       'ROS1', 'SRC', 'SRMS', 'SYK', 'TEC', 'TESK1_TYR', 'TEK', 'TNK1',
       'TNNI3K_TYR', 'NTRK1', 'NTRK2', 'NTRK3', 'TXK', 'TYK2', 'TYRO3', 'FLT1',
       'KDR', 'FLT4', 'WEE1_TYR', 'YES1', 'ZAP70'],
      dtype='object', name='kinase')</code></pre>
</div>
</div>
<div id="7355971c-a0b5-408c-9930-68244a6a85d8" class="cell" data-execution_count="12">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> k <span class="kw">in</span> tqdm(pspa_info.index,total<span class="op">=</span><span class="bu">len</span>(pspa_info)):</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>    k_pssms <span class="op">=</span> site_pssms[site_pssms.index.<span class="bu">str</span>.contains(<span class="ss">f'</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">_'</span>)]</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>    plot_logos(k_pssms,count_dict<span class="op">=</span>count_out)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>    path <span class="op">=</span> prepare_path(<span class="ss">f'fig/pspa/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/path/logo.svg'</span>)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>    save_show(path)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>100%|█████████████████████████████████████████████████████████████| 82/82 [06:25&lt;00:00,  4.70s/it]</code></pre>
</div>
</div>
<div id="002b724d-deb2-46b0-b32b-2475f980b26b" class="cell" data-execution_count="13">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>site_pssms.to_parquet(<span class="st">'raw/pspa_site_pssm.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="plot-distribution" class="level2">
<h2 class="anchored" data-anchor-id="plot-distribution">Plot distribution</h2>
<div id="645360c6-879a-4b7c-9765-f261ebffdef4" class="cell" data-execution_count="14">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_hist(df,</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>              kinase,</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>              sty_thr<span class="op">=</span><span class="va">None</span>, <span class="co"># dict of thresholds per acceptor</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>              hue<span class="op">=</span><span class="st">'acceptor'</span>):</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>    <span class="co">"Plot histogram of a column (kinase). "</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">6</span>,<span class="dv">2</span>))</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>    hist_params <span class="op">=</span> {<span class="st">'element'</span>:<span class="st">'poly'</span>,</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>              <span class="st">'edgecolor'</span>: <span class="va">None</span>,</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>              <span class="st">'alpha'</span>:<span class="fl">0.5</span>,</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>              <span class="st">'bins'</span>:<span class="dv">100</span>,</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>              <span class="st">'kde'</span>:<span class="va">True</span>,</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>              <span class="st">'palette'</span>:sty_color}</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>    sns.histplot(data<span class="op">=</span>df,x<span class="op">=</span>kinase,hue<span class="op">=</span>hue,<span class="op">**</span>hist_params)</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">''</span>)</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'</span><span class="sc">{</span>kinase<span class="sc">}</span><span class="ss"> score distribution'</span>)</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> sty_thr:</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> acceptor,thr <span class="kw">in</span> sty_thr.items():</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> thr <span class="kw">is</span> <span class="kw">not</span> <span class="va">None</span>:</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>                plt.axvline(thr,color<span class="op">=</span>sty_color[acceptor])</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="44a07cb4-c6d4-4c0d-b0c2-56d476f8a6bb" class="cell" data-execution_count="15">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>plot_hist(df,<span class="st">'AAK1'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="19b_reactome_path_plot_pspa_files/figure-html/cell-14-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div id="b7aaf657-3795-4e2f-896c-12031cb2dacd" class="cell" data-execution_count="18">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> k,r <span class="kw">in</span> tqdm(pspa_info.iterrows(),total<span class="op">=</span><span class="bu">len</span>(pspa_info)):</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>    sty_thr<span class="op">=</span>r[thr_cols].rename(index<span class="op">=</span>{<span class="st">'S_thr'</span>:<span class="st">'S'</span>,<span class="st">'T_thr'</span>:<span class="st">'T'</span>,<span class="st">'Y_thr'</span>:<span class="st">'Y'</span>})[<span class="kw">lambda</span> x: x<span class="op">!=</span><span class="dv">0</span>].to_dict()</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>    plot_hist(df,k,sty_thr<span class="op">=</span>sty_thr)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>    save_show(<span class="ss">f'fig/pspa/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/path/hist.svg'</span>,show_only<span class="op">=</span><span class="va">False</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stderr">
<pre><code>100%|███████████████████████████████████████████████████████████| 396/396 [04:01&lt;00:00,  1.64it/s]</code></pre>
</div>
</div>
</section>
<section id="plot-reactome-pathway" class="level2">
<h2 class="anchored" data-anchor-id="plot-reactome-pathway">Plot reactome pathway</h2>
<div id="cfbed19d-89ab-432d-a203-ab2ede422a0f" class="cell" data-execution_count="4">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>info <span class="op">=</span> Data.get_kinase_info()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="e8c1347f-8e65-4bd5-97a6-eb17426bcd53" class="cell" data-execution_count="5">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>info <span class="op">=</span> info[info.pseudo<span class="op">==</span><span class="st">'0'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="8d635bdb-8a9a-4196-a06a-16a96f0d781b" class="cell" data-execution_count="6">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>info.uniprot.duplicated().<span class="bu">sum</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="6">
<pre><code>np.int64(0)</code></pre>
</div>
</div>
<div id="7c054e5d-b415-4e78-87bd-bc006c42f715" class="cell" data-execution_count="7">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>kinase_map <span class="op">=</span> info.set_index(<span class="st">'kinase'</span>)[<span class="st">'uniprot'</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="28b98855-6f9a-4e80-bf26-afa0ba1e406e" class="cell" data-execution_count="8">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> get_path(series):</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>    out <span class="op">=</span> get_reactome(series[<span class="st">'genes'</span>],p_type<span class="op">=</span><span class="st">'p'</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>    kinase<span class="op">=</span>series.name.split(<span class="st">'_'</span>)[<span class="dv">0</span>] <span class="co"># for those with _TRY, need to remove that</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>    uniprot <span class="op">=</span> kinase_map[kinase]</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>    out <span class="op">=</span> add_reactome_ref(out,uniprot)</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>    exclude<span class="op">=</span>[<span class="st">'GTPase'</span>,<span class="st">'SUMO'</span>]</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>    out <span class="op">=</span> out[<span class="op">~</span>out.name.<span class="bu">str</span>.contains(<span class="st">'|'</span>.join(exclude))]</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> out</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="061cd930-e2dd-4524-9081-a29ff7ddaf40" class="cell" data-execution_count="9">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>show_only<span class="op">=</span><span class="va">False</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="e316f1cb-61d1-47f8-b2ad-a21c70735cfb" class="cell" data-execution_count="19">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>pspa_info.index.get_loc(<span class="st">'PKG1'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="19">
<pre><code>234</code></pre>
</div>
</div>
<div id="41b45c4e-4ac5-420a-88c9-5a23d72c2305" class="cell" data-execution_count="47">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="bu">len</span>(pspa_info)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="47">
<pre><code>396</code></pre>
</div>
</div>
<div id="ddb34ba3-9681-4377-a8e0-f8367326cc57" class="cell" data-quarto-private-1="{&quot;key&quot;:&quot;jupyter&quot;,&quot;value&quot;:{&quot;outputs_hidden&quot;:true}}" data-execution_count="20">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i,(k,r) <span class="kw">in</span> <span class="bu">enumerate</span>(pspa_info.iloc[<span class="dv">234</span>:].iterrows()):</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(i,<span class="st">':'</span>,k)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>    save_dir <span class="op">=</span> <span class="ss">f'fig/pspa/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/path'</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>    path_df <span class="op">=</span> get_path(r)</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>    path_df.to_parquet(<span class="ss">f'</span><span class="sc">{</span>save_dir<span class="sc">}</span><span class="ss">/path.parquet'</span>)</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>    plot_path(path_df,p_type<span class="op">=</span><span class="st">'p'</span>,ref_col<span class="op">=</span><span class="st">'ref_path'</span>,top_n<span class="op">=</span><span class="dv">15</span>,max_label_length<span class="op">=</span><span class="dv">55</span>)</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="st">'Reactome pathways of substrates based on scores'</span>)</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>    save_show(<span class="ss">f'</span><span class="sc">{</span>save_dir<span class="sc">}</span><span class="ss">/path.svg'</span>,show_only<span class="op">=</span>show_only)</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> path_df.ref_path.<span class="bu">sum</span>()<span class="op">!=</span><span class="dv">0</span>:</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>        accuracy <span class="op">=</span> get_overlap(path_df, ref_col<span class="op">=</span><span class="st">'ref_path'</span>,plot<span class="op">=</span><span class="va">True</span>,p_type<span class="op">=</span><span class="st">'p'</span>,figsize<span class="op">=</span>(<span class="dv">5</span>,<span class="dv">3</span>))</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>        plt.title(<span class="ss">f'Distribution of –log₁₀(p) for reference pathways'</span>)</span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>        save_show(<span class="ss">f'</span><span class="sc">{</span>save_dir<span class="sc">}</span><span class="ss">/overlap.svg'</span>,show_only<span class="op">=</span>show_only)</span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> path_df.ref_path_lowest.<span class="bu">sum</span>()<span class="op">!=</span><span class="dv">0</span>:</span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>        accuracy_lo <span class="op">=</span> get_overlap(path_df, ref_col<span class="op">=</span><span class="st">'ref_path_lowest'</span>,plot<span class="op">=</span><span class="va">True</span>,p_type<span class="op">=</span><span class="st">'p'</span>,figsize<span class="op">=</span>(<span class="dv">5</span>,<span class="dv">3</span>))</span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>        plt.title(<span class="ss">f'Distribution of –log₁₀(p) for reference pathways </span><span class="ch">\n</span><span class="ss">(lowest-level)'</span>)</span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>        save_show(<span class="ss">f'</span><span class="sc">{</span>save_dir<span class="sc">}</span><span class="ss">/overlap_lo.svg'</span>,show_only<span class="op">=</span>show_only)</span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>    plt.close(<span class="st">'all'</span>)</span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># break</span></span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ✅ polite delay to avoid API throttling</span></span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a>    delay <span class="op">=</span> random.uniform(<span class="fl">1.5</span>, <span class="dv">4</span>)</span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"Sleeping </span><span class="sc">{</span>delay<span class="sc">:.2f}</span><span class="ss">s before next kinase..."</span>)</span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a>    time.sleep(delay)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>0 : PKG1
Running pathway anlysis
Done
Sleeping 1.76s before next kinase...
1 : PKG2
Running pathway anlysis
Done
Sleeping 3.58s before next kinase...
2 : PKN1
Running pathway anlysis
Done
Sleeping 3.84s before next kinase...
3 : PKN2
Running pathway anlysis
Done
Sleeping 1.53s before next kinase...
4 : PKN3
Running pathway anlysis
Done
Sleeping 3.71s before next kinase...
5 : PKR
Running pathway anlysis
Done
Sleeping 2.22s before next kinase...
6 : PLK1
Running pathway anlysis
Done
Sleeping 2.26s before next kinase...
7 : PLK2
Running pathway anlysis
Done
Sleeping 3.69s before next kinase...
8 : PLK3
Running pathway anlysis
Done
Sleeping 3.72s before next kinase...
9 : PLK4
Running pathway anlysis
Done
Sleeping 2.47s before next kinase...
10 : PRKD1
Running pathway anlysis
Done
Sleeping 3.57s before next kinase...
11 : PRKD2
Running pathway anlysis
Done
Sleeping 3.49s before next kinase...
12 : PRKD3
Running pathway anlysis
Done
Sleeping 2.62s before next kinase...
13 : PRKX
Running pathway anlysis
Done
Sleeping 1.52s before next kinase...
14 : PRP4
Running pathway anlysis
Done
Sleeping 3.73s before next kinase...
15 : PRPK
Running pathway anlysis
Done
Sleeping 3.47s before next kinase...
16 : QIK
Running pathway anlysis
Done
Sleeping 3.01s before next kinase...
17 : QSK
Running pathway anlysis
Done
Sleeping 3.31s before next kinase...
18 : RAF1
Running pathway anlysis
Done
Sleeping 3.83s before next kinase...
19 : RIPK1
Running pathway anlysis
Done
Sleeping 2.40s before next kinase...
20 : RIPK2
Running pathway anlysis
Done
Sleeping 2.06s before next kinase...
21 : RIPK3
Running pathway anlysis
Done
Sleeping 2.47s before next kinase...
22 : ROCK1
Running pathway anlysis
Done
Sleeping 1.61s before next kinase...
23 : ROCK2
Running pathway anlysis
Done
Sleeping 3.03s before next kinase...
24 : RSK2
Running pathway anlysis
Done
Sleeping 3.97s before next kinase...
25 : RSK3
Running pathway anlysis
Done
Sleeping 2.58s before next kinase...
26 : RSK4
Running pathway anlysis
Done
Sleeping 3.62s before next kinase...
27 : SBK
Running pathway anlysis
Done
Sleeping 3.00s before next kinase...
28 : SGK1
Running pathway anlysis
Done
Sleeping 3.17s before next kinase...
29 : SGK3
Running pathway anlysis
Done
Sleeping 3.48s before next kinase...
30 : SIK
Running pathway anlysis
Done
Sleeping 1.87s before next kinase...
31 : SKMLCK
Running pathway anlysis
Done
Sleeping 3.27s before next kinase...
32 : SLK
Running pathway anlysis
Done
Sleeping 2.15s before next kinase...
33 : SMG1
Running pathway anlysis
Done
Sleeping 3.51s before next kinase...
34 : SMMLCK
Running pathway anlysis
Done
Sleeping 3.48s before next kinase...
35 : SNRK
Running pathway anlysis
Done
Sleeping 3.78s before next kinase...
36 : SRPK1
Running pathway anlysis
Done
Sleeping 3.51s before next kinase...
37 : SRPK2
Running pathway anlysis
Done
Sleeping 2.69s before next kinase...
38 : SRPK3
Running pathway anlysis
Done
Sleeping 2.74s before next kinase...
39 : SSTK
Running pathway anlysis
Done
Sleeping 3.58s before next kinase...
40 : STK33
Running pathway anlysis
Done
Sleeping 2.59s before next kinase...
41 : STLK3
Running pathway anlysis
Done
Sleeping 2.64s before next kinase...
42 : TAK1
Running pathway anlysis
Done
Sleeping 2.17s before next kinase...
43 : TAO1
Running pathway anlysis
Done
Sleeping 1.69s before next kinase...
44 : TAO2
Running pathway anlysis
Done
Sleeping 2.43s before next kinase...
45 : TAO3
Running pathway anlysis
Done
Sleeping 2.88s before next kinase...
46 : TBK1
Running pathway anlysis
Done
Sleeping 2.63s before next kinase...
47 : TGFBR1
Running pathway anlysis
Done
Sleeping 1.83s before next kinase...
48 : TGFBR2
Running pathway anlysis
Done
Sleeping 2.46s before next kinase...
49 : TLK1
Running pathway anlysis
Done
Sleeping 3.56s before next kinase...
50 : TLK2
Running pathway anlysis
Done
Sleeping 3.43s before next kinase...
51 : TNIK
Running pathway anlysis
Done
Sleeping 3.14s before next kinase...
52 : TSSK1
Running pathway anlysis
Done
Sleeping 2.08s before next kinase...
53 : TSSK2
Running pathway anlysis
Done
Sleeping 3.35s before next kinase...
54 : TTBK1
Running pathway anlysis
Done
Sleeping 1.84s before next kinase...
55 : TTBK2
Running pathway anlysis
Done
Sleeping 3.96s before next kinase...
56 : TTK
Running pathway anlysis
Done
Sleeping 1.95s before next kinase...
57 : ULK1
Running pathway anlysis
Done
Sleeping 1.50s before next kinase...
58 : ULK2
Running pathway anlysis
Done
Sleeping 3.75s before next kinase...
59 : VRK1
Running pathway anlysis
Done
Sleeping 1.69s before next kinase...
60 : VRK2
Running pathway anlysis
Done
Sleeping 2.76s before next kinase...
61 : WNK1
Running pathway anlysis
Done
Sleeping 2.50s before next kinase...
62 : WNK3
Running pathway anlysis
Done
Sleeping 2.96s before next kinase...
63 : WNK4
Running pathway anlysis
Done
Sleeping 2.90s before next kinase...
64 : YANK2
Running pathway anlysis
Done
Sleeping 2.29s before next kinase...
65 : YANK3
Running pathway anlysis
Done
Sleeping 3.09s before next kinase...
66 : YSK1
Running pathway anlysis
Done
Sleeping 2.05s before next kinase...
67 : YSK4
Running pathway anlysis
Done
Sleeping 3.41s before next kinase...
68 : ZAK
Running pathway anlysis
Done
Sleeping 2.01s before next kinase...
69 : ABL1
Running pathway anlysis
Done
Sleeping 2.62s before next kinase...
70 : TNK2
Running pathway anlysis
Done
Sleeping 1.94s before next kinase...
71 : ALK
Running pathway anlysis
Done
Sleeping 2.47s before next kinase...
72 : ABL2
Running pathway anlysis
Done
Sleeping 2.07s before next kinase...
73 : AXL
Running pathway anlysis
Done
Sleeping 2.26s before next kinase...
74 : BLK
Running pathway anlysis
Done
Sleeping 3.63s before next kinase...
75 : BMPR2_TYR
Running pathway anlysis
Done
Sleeping 2.52s before next kinase...
76 : PTK6
Running pathway anlysis
Done
Sleeping 1.56s before next kinase...
77 : BTK
Running pathway anlysis
Done
Sleeping 3.41s before next kinase...
78 : CSF1R
Running pathway anlysis
Done
Sleeping 2.02s before next kinase...
79 : CSK
Running pathway anlysis
Done
Sleeping 3.02s before next kinase...
80 : MATK
Running pathway anlysis
Done
Sleeping 2.36s before next kinase...
81 : DDR1
Running pathway anlysis
Done
Sleeping 3.25s before next kinase...
82 : DDR2
Running pathway anlysis
Done
Sleeping 3.56s before next kinase...
83 : EGFR
Running pathway anlysis
Done
Sleeping 3.61s before next kinase...
84 : EPHA1
Running pathway anlysis
Done
Sleeping 2.18s before next kinase...
85 : EPHA2
Running pathway anlysis
Done
Sleeping 2.43s before next kinase...
86 : EPHA3
Running pathway anlysis
Done
Sleeping 2.56s before next kinase...
87 : EPHA4
Running pathway anlysis
Done
Sleeping 2.86s before next kinase...
88 : EPHA5
Running pathway anlysis
Done
Sleeping 1.78s before next kinase...
89 : EPHA6
Running pathway anlysis
Done
Sleeping 3.61s before next kinase...
90 : EPHA7
Running pathway anlysis
Done
Sleeping 1.55s before next kinase...
91 : EPHA8
Running pathway anlysis
Done
Sleeping 3.87s before next kinase...
92 : EPHB1
Running pathway anlysis
Done
Sleeping 3.94s before next kinase...
93 : EPHB2
Running pathway anlysis
Done
Sleeping 1.95s before next kinase...
94 : EPHB3
Running pathway anlysis
Done
Sleeping 3.85s before next kinase...
95 : EPHB4
Running pathway anlysis
Done
Sleeping 3.69s before next kinase...
96 : BMX
Running pathway anlysis
Done
Sleeping 2.42s before next kinase...
97 : PTK2
Running pathway anlysis
Done
Sleeping 2.28s before next kinase...
98 : FER
Running pathway anlysis
Done
Sleeping 3.62s before next kinase...
99 : FES
Running pathway anlysis
Done
Sleeping 2.71s before next kinase...
100 : FGFR1
Running pathway anlysis
Done
Sleeping 3.76s before next kinase...
101 : FGFR2
Running pathway anlysis
Done
Sleeping 3.21s before next kinase...
102 : FGFR3
Running pathway anlysis
Done
Sleeping 3.76s before next kinase...
103 : FGFR4
Running pathway anlysis
Done
Sleeping 1.92s before next kinase...
104 : FGR
Running pathway anlysis
Done
Sleeping 3.07s before next kinase...
105 : FLT3
Running pathway anlysis
Done
Sleeping 2.74s before next kinase...
106 : FRK
Running pathway anlysis
Done
Sleeping 2.29s before next kinase...
107 : FYN
Running pathway anlysis
Done
Sleeping 3.96s before next kinase...
108 : HCK
Running pathway anlysis
Done
Sleeping 3.79s before next kinase...
109 : ERBB2
Running pathway anlysis
Done
Sleeping 3.60s before next kinase...
110 : ERBB4
Running pathway anlysis
Done
Sleeping 2.30s before next kinase...
111 : IGF1R
Running pathway anlysis
Done
Sleeping 3.69s before next kinase...
112 : INSR
Running pathway anlysis
Done
Sleeping 1.54s before next kinase...
113 : INSRR
Running pathway anlysis
Done
Sleeping 2.27s before next kinase...
114 : ITK
Running pathway anlysis
Done
Sleeping 2.54s before next kinase...
115 : JAK1
Running pathway anlysis
Done
Sleeping 3.95s before next kinase...
116 : JAK2
Running pathway anlysis
Done
Sleeping 2.08s before next kinase...
117 : JAK3
Running pathway anlysis
Done
Sleeping 2.74s before next kinase...
118 : KIT
Running pathway anlysis
Done
Sleeping 3.46s before next kinase...
119 : LCK
Running pathway anlysis
Done
Sleeping 3.49s before next kinase...
120 : LIMK1_TYR
Running pathway anlysis
Done
Sleeping 3.55s before next kinase...
121 : LIMK2_TYR
Running pathway anlysis
Done
Sleeping 1.64s before next kinase...
122 : LTK
Running pathway anlysis
Done
Sleeping 3.52s before next kinase...
123 : LYN
Running pathway anlysis
Done
Sleeping 1.60s before next kinase...
124 : MERTK
Running pathway anlysis
Done
Sleeping 3.30s before next kinase...
125 : MET
Running pathway anlysis
Done
Sleeping 2.94s before next kinase...
126 : MAP2K4_TYR
Running pathway anlysis
Done
Sleeping 3.77s before next kinase...
127 : MAP2K6_TYR
Running pathway anlysis
Done
Sleeping 3.20s before next kinase...
128 : MAP2K7_TYR
Running pathway anlysis
Done
Sleeping 3.93s before next kinase...
129 : MST1R
Running pathway anlysis
Done
Sleeping 2.87s before next kinase...
130 : MUSK
Running pathway anlysis
Done
Sleeping 1.81s before next kinase...
131 : PKMYT1_TYR
Running pathway anlysis
Done
Sleeping 1.62s before next kinase...
132 : NEK10_TYR
Running pathway anlysis
Done
Sleeping 2.77s before next kinase...
133 : PDGFRA
Running pathway anlysis
Done
Sleeping 2.99s before next kinase...
134 : PDGFRB
Running pathway anlysis
Done
Sleeping 2.59s before next kinase...
135 : PDHK1_TYR
Running pathway anlysis
Done
Sleeping 3.13s before next kinase...
136 : PDHK3_TYR
Running pathway anlysis
Done
Sleeping 1.88s before next kinase...
137 : PDHK4_TYR
Running pathway anlysis
Done
Sleeping 2.94s before next kinase...
138 : PINK1_TYR
Running pathway anlysis
Done
Sleeping 1.71s before next kinase...
139 : PTK2B
Running pathway anlysis
Done
Sleeping 1.73s before next kinase...
140 : RET
Running pathway anlysis
Done
Sleeping 2.62s before next kinase...
141 : ROS1
Running pathway anlysis
Done
Sleeping 2.28s before next kinase...
142 : SRC
Running pathway anlysis
Done
Sleeping 2.51s before next kinase...
143 : SRMS
Running pathway anlysis
Done
Sleeping 3.57s before next kinase...
144 : SYK
Running pathway anlysis
Done
Sleeping 1.58s before next kinase...
145 : TEC
Running pathway anlysis
Done
Sleeping 3.28s before next kinase...
146 : TESK1_TYR
Running pathway anlysis
Done
Sleeping 1.95s before next kinase...
147 : TEK
Running pathway anlysis
Done
Sleeping 3.01s before next kinase...
148 : TNK1
Running pathway anlysis
Done
Sleeping 2.26s before next kinase...
149 : TNNI3K_TYR
Running pathway anlysis
Done
Sleeping 2.93s before next kinase...
150 : NTRK1
Running pathway anlysis
Done
Sleeping 1.59s before next kinase...
151 : NTRK2
Running pathway anlysis
Done
Sleeping 2.25s before next kinase...
152 : NTRK3
Running pathway anlysis
Done
Sleeping 3.27s before next kinase...
153 : TXK
Running pathway anlysis
Done
Sleeping 2.32s before next kinase...
154 : TYK2
Running pathway anlysis
Done
Sleeping 2.02s before next kinase...
155 : TYRO3
Running pathway anlysis
Done
Sleeping 3.15s before next kinase...
156 : FLT1
Running pathway anlysis
Done
Sleeping 2.19s before next kinase...
157 : KDR
Running pathway anlysis
Done
Sleeping 3.82s before next kinase...
158 : FLT4
Running pathway anlysis
Done
Sleeping 2.05s before next kinase...
159 : WEE1_TYR
Running pathway anlysis
Done
Sleeping 1.99s before next kinase...
160 : YES1
Running pathway anlysis
Done
Sleeping 3.00s before next kinase...
161 : ZAP70
Running pathway anlysis
Done
Sleeping 2.91s before next kinase...</code></pre>
</div>
</div>
</section>
<section id="get-accuracy-for-overlap-pathways" class="level2">
<h2 class="anchored" data-anchor-id="get-accuracy-for-overlap-pathways">Get accuracy for overlap pathways</h2>
<div id="f23e74f7-9214-4696-ab05-9173819b9fb1" class="cell" data-execution_count="21">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>acc_dict, num_dict <span class="op">=</span> {}, {}</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>acc_lo_dict, num_lo_dict <span class="op">=</span> {}, {}</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> k,r <span class="kw">in</span> pspa_info.iterrows():</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>    save_dir <span class="op">=</span> <span class="ss">f'fig/pspa/</span><span class="sc">{</span>k<span class="sc">}</span><span class="ss">/path'</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>    path_df<span class="op">=</span>pd.read_parquet(<span class="ss">f'</span><span class="sc">{</span>save_dir<span class="sc">}</span><span class="ss">/path.parquet'</span>)</span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> path_df.ref_path.<span class="bu">sum</span>()<span class="op">!=</span><span class="dv">0</span>:</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>        accuracy <span class="op">=</span> get_overlap(path_df, ref_col<span class="op">=</span><span class="st">'ref_path'</span>,plot<span class="op">=</span><span class="va">False</span>,p_type<span class="op">=</span><span class="st">'p'</span>)</span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>        acc_dict[k] <span class="op">=</span> accuracy</span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>        num_dict[k] <span class="op">=</span>path_df.ref_path.<span class="bu">sum</span>()</span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> path_df.ref_path_lowest.<span class="bu">sum</span>()<span class="op">!=</span><span class="dv">0</span>:</span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>        accuracy_lo <span class="op">=</span> get_overlap(path_df, ref_col<span class="op">=</span><span class="st">'ref_path_lowest'</span>,plot<span class="op">=</span><span class="va">False</span>,p_type<span class="op">=</span><span class="st">'p'</span>)</span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>        acc_lo_dict[k] <span class="op">=</span> accuracy_lo</span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>        num_lo_dict[k] <span class="op">=</span> path_df.ref_path_lowest.<span class="bu">sum</span>()</span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># break</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="360b7a10-6287-41e0-8187-080978d08405" class="cell" data-execution_count="22">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>pspa_info[<span class="st">'accuracy'</span>] <span class="op">=</span> acc_dict</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>pspa_info[<span class="st">'accuracy_lowest'</span>] <span class="op">=</span> acc_lo_dict</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>pspa_info[<span class="st">'path_num'</span>] <span class="op">=</span> num_dict</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>pspa_info[<span class="st">'path_lowest_num'</span>] <span class="op">=</span> num_lo_dict</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="7ca8d04c-94f5-4735-b327-78c5de7e9450" class="cell" data-execution_count="23">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>pspa_info[<span class="st">'path_num'</span>] <span class="op">=</span> pspa_info[<span class="st">'path_num'</span>].fillna(<span class="dv">0</span>).astype(<span class="bu">int</span>)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>pspa_info[<span class="st">'path_lowest_num'</span>] <span class="op">=</span> pspa_info[<span class="st">'path_lowest_num'</span>].fillna(<span class="dv">0</span>).astype(<span class="bu">int</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="e3aa7122-4409-48dc-bb75-9db9280fbdeb" class="cell" data-execution_count="24">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>pspa_info.to_parquet(<span class="st">'raw/pspa_path_info.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="pathway-hue" class="level2">
<h2 class="anchored" data-anchor-id="pathway-hue">Pathway hue</h2>
<div id="9a7f0e04-cfbe-48c2-9411-2ea53ad182ca" class="cell" data-execution_count="2">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb35"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>pspa_info<span class="op">=</span>pd.read_parquet(<span class="st">'raw/pspa_path_info.parquet'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="0f4a430f-d1b2-4ec4-995a-aa7e3755f22d" class="cell" data-execution_count="3">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> plot_path_hist(df,</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>              colname,</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>              <span class="op">**</span>kwargs,</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>              ):</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>    <span class="co">"Plot histogram of a column (kinase). "</span></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>    plt.figure(figsize<span class="op">=</span>(<span class="dv">6</span>,<span class="dv">2</span>))</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>    hist_params <span class="op">=</span> {<span class="st">'element'</span>:<span class="st">'poly'</span>,</span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>              <span class="st">'edgecolor'</span>: <span class="va">None</span>,</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>              <span class="st">'alpha'</span>:<span class="fl">0.5</span>,</span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>              <span class="st">'bins'</span>:<span class="dv">100</span>,</span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>              <span class="st">'kde'</span>:<span class="va">True</span>,</span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>              }</span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>    sns.histplot(data<span class="op">=</span>df,x<span class="op">=</span>colname,<span class="op">**</span>kwargs, <span class="op">**</span>hist_params)</span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>    plt.xlabel(<span class="st">''</span>)</span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>    plt.title(<span class="ss">f'Accuracy score distribution'</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="47ffda82-b1ea-4739-b082-27466827075a" class="cell" data-execution_count="4">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb37"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>q75, q50 <span class="op">=</span> pspa_info.path_num[<span class="kw">lambda</span> x: x<span class="op">!=</span><span class="dv">0</span>].quantile([<span class="fl">0.75</span>,<span class="fl">0.5</span>])</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>q75, q50</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="4">
<pre><code>(33.75, 11.0)</code></pre>
</div>
</div>
<div id="773e243c-a76e-4ada-9a03-d8cacee5bd5b" class="cell" data-execution_count="5">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb39"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> categorize_path_num(x):</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> x <span class="op">&gt;=</span> q75: <span class="cf">return</span> <span class="ss">f'well known (≥</span><span class="sc">{</span>q75<span class="sc">:.0f}</span><span class="ss">)'</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">elif</span> x <span class="op">&lt;=</span> q50: <span class="cf">return</span> <span class="ss">f'less known (≤</span><span class="sc">{</span>q50<span class="sc">:.0f}</span><span class="ss">)'</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span>: <span class="cf">return</span> <span class="ss">f'middle (</span><span class="sc">{</span>q50<span class="sc">:.0f}</span><span class="ss">–</span><span class="sc">{</span>q75<span class="sc">:.0f}</span><span class="ss">)'</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="dd4ff35b-3c78-4723-8b40-33f3e0031d17" class="cell" data-execution_count="6">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb40"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>pspa_info[<span class="st">'category'</span>] <span class="op">=</span> pspa_info[<span class="st">'path_num'</span>].<span class="bu">apply</span>(categorize_path_num)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div id="4fc7446b-6c19-4d26-980a-11f16086d069" class="cell" data-execution_count="7">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb41"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>pspa_info[<span class="st">'category'</span>].value_counts()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="7">
<pre><code>category
less known (≤11)    253
well known (≥34)     74
middle (11–34)       69
Name: count, dtype: int64</code></pre>
</div>
</div>
<div id="40aca84d-f014-4cfc-bc08-497290f52d7f" class="cell" data-execution_count="8">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb43"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>pspa_info.hist(column<span class="op">=</span><span class="st">'accuracy'</span>, by<span class="op">=</span><span class="st">'category'</span>, bins<span class="op">=</span><span class="dv">50</span>, figsize<span class="op">=</span>(<span class="dv">8</span>,<span class="dv">6</span>))<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="19b_reactome_path_plot_pspa_files/figure-html/cell-35-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<div id="9c8bb76c-4a9a-4e12-b237-aea0670ca99f" class="cell" data-execution_count="9">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb44"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>pspa_info[pspa_info.path_num<span class="op">&gt;</span><span class="dv">12</span>].accuracy.hist(bins<span class="op">=</span><span class="dv">50</span>,figsize<span class="op">=</span>(<span class="dv">5</span>,<span class="dv">3</span>))</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>plt.xlabel(<span class="st">'Accuracy'</span>)</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>plt.ylabel(<span class="st">'Number of kinases'</span>)</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>plt.title(<span class="st">'PSPA accuracy distribution'</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="19b_reactome_path_plot_pspa_files/figure-html/cell-36-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
<section id="lowest" class="level3">
<h3 class="anchored" data-anchor-id="lowest">Lowest</h3>
<div id="5abba31b-5218-4811-9526-a6626b6f4845" class="cell" data-execution_count="133">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb45"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>pspa_info.path_lowest_num[<span class="kw">lambda</span> x: x<span class="op">!=</span><span class="dv">0</span>].quantile([<span class="fl">0.75</span>])</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display" data-execution_count="133">
<pre><code>0.75    10.0
Name: path_lowest_num, dtype: float64</code></pre>
</div>
</div>
<div id="3f1b6265-e19c-45a4-a946-a8733248aa7e" class="cell" data-execution_count="135">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb47"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>pspa_info[pspa_info.path_lowest_num<span class="op">&gt;</span><span class="dv">10</span>].accuracy.hist(bins<span class="op">=</span><span class="dv">50</span>,figsize<span class="op">=</span>(<span class="dv">5</span>,<span class="dv">3</span>))</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>plt.xlabel(<span class="st">'Accuracy'</span>)</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>plt.ylabel(<span class="st">'Number of kinases'</span>)</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>plt.title(<span class="st">'PSPA accuracy distribution (lowest-level)'</span>)<span class="op">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-display">
<div>
<figure class="figure">
<p><img src="19b_reactome_path_plot_pspa_files/figure-html/cell-38-output-1.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/sky1ove\.github\.io\/katlas-raw");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/sky1ove/katlas-raw/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></div></div></footer></body></html>