# Alignment


<!-- WARNING: THIS FILE WAS AUTOGENERATED! DO NOT EDIT! -->

``` python
from Bio.Seq import Seq
from Bio.SeqRecord import SeqRecord
from Bio import SeqIO, AlignIO
import pandas as pd

from katlas.utils import *
from katlas.data import *
```

## Prepare fasta file for clustalo

``` python
# kd=pd.read_excel('out/uniprot_kd.xlsx')
```

``` python
kd=Data.get_kd_uniprot()
```

``` python
kd.columns
```

    Index(['kd_ID', 'Uniprot', 'Entry Name', 'Protein names', 'Gene Names',
           'Gene Names (primary)', 'Organism', 'kd_note', 'kd_evidence',
           'kd_start', 'kd_end', 'kd_seq', 'Domain [FT]', 'Domain [CC]', 'Region',
           'Motif', 'Protein families', 'Reactome', 'ComplexPortal',
           'Subcellular location [CC]', 'Gene Ontology (biological process)',
           'Tissue specificity', 'Interacts with', 'Subunit structure',
           'Function [CC]', 'Activity regulation', 'full_seq', 'D1', 'D2', 'D3',
           'N1', 'active_D1_D2'],
          dtype='object')

``` python
get_fasta(kd,
          seq_col='kd_seq',
          id_col='kd_ID',
          path='raw/kinase_domains.fasta')
```

    5536

For human only:

``` python
# human = kd[kd.Organism=='Homo sapiens (Human)']

# get_fasta(human,path="raw/human_kinase_domains.fasta")
```

for PI3K catalytic domain

``` python
# pi3k = kd[kd.kd_note=='PI3K/PI4K catalytic']

# get_fasta(pi3k,path="raw/pi3k_kinase_domains.fasta")
```

## Run clustalo

``` bash
sudo apt-get update
sudo apt-get install clustalo
clustalo -i kinase_domains.fasta -o kinase_domains.aln --force --outfmt=clu
```

``` python
# run_clustalo(input_fasta='raw/kinase_domains.fasta', 
#              output_aln = "raw/kinase_domains.aln")
```

## Analyze alignment

``` python
df = aln2df("raw/kinase_domains.aln")
```

``` python
df.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">1</th>
<th data-quarto-table-cell-role="th">2</th>
<th data-quarto-table-cell-role="th">3</th>
<th data-quarto-table-cell-role="th">4</th>
<th data-quarto-table-cell-role="th">5</th>
<th data-quarto-table-cell-role="th">6</th>
<th data-quarto-table-cell-role="th">7</th>
<th data-quarto-table-cell-role="th">8</th>
<th data-quarto-table-cell-role="th">9</th>
<th data-quarto-table-cell-role="th">10</th>
<th data-quarto-table-cell-role="th">...</th>
<th data-quarto-table-cell-role="th">3425</th>
<th data-quarto-table-cell-role="th">3426</th>
<th data-quarto-table-cell-role="th">3427</th>
<th data-quarto-table-cell-role="th">3428</th>
<th data-quarto-table-cell-role="th">3429</th>
<th data-quarto-table-cell-role="th">3430</th>
<th data-quarto-table-cell-role="th">3431</th>
<th data-quarto-table-cell-role="th">3432</th>
<th data-quarto-table-cell-role="th">3433</th>
<th data-quarto-table-cell-role="th">3434</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">A0A075F7E9_LERK1_ORYSI_KD1</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>...</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">A0A078BQP2_GCY25_CAEEL_KD1</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>...</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">A0A078CGE6_M3KE1_BRANA_KD1</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>...</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">A0A0G2K344_PK3CA_RAT_KD1</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>...</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">A0A0H2ZM62_HK06_STRP2_KD1</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>...</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
</tr>
</tbody>
</table>

<p>5 rows × 3434 columns</p>
</div>

``` python
# change type to str to save in parquet
df.columns = df.columns.astype(str)
```

``` python
df.to_parquet('raw/uniprot_kd_align.parquet')
```

``` python
# reverse back
df.columns = df.columns.astype(int)
```

## Frequency of aa at position 1-3434

``` python
# Compute residue frequency at each position
counts_df = df.apply(lambda col: col.value_counts(), axis=0).fillna(0)
freq_df = counts_df.div(counts_df.sum(axis=0), axis=1)
```

``` python
freq_df.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">1</th>
<th data-quarto-table-cell-role="th">2</th>
<th data-quarto-table-cell-role="th">3</th>
<th data-quarto-table-cell-role="th">4</th>
<th data-quarto-table-cell-role="th">5</th>
<th data-quarto-table-cell-role="th">6</th>
<th data-quarto-table-cell-role="th">7</th>
<th data-quarto-table-cell-role="th">8</th>
<th data-quarto-table-cell-role="th">9</th>
<th data-quarto-table-cell-role="th">10</th>
<th data-quarto-table-cell-role="th">...</th>
<th data-quarto-table-cell-role="th">3425</th>
<th data-quarto-table-cell-role="th">3426</th>
<th data-quarto-table-cell-role="th">3427</th>
<th data-quarto-table-cell-role="th">3428</th>
<th data-quarto-table-cell-role="th">3429</th>
<th data-quarto-table-cell-role="th">3430</th>
<th data-quarto-table-cell-role="th">3431</th>
<th data-quarto-table-cell-role="th">3432</th>
<th data-quarto-table-cell-role="th">3433</th>
<th data-quarto-table-cell-role="th">3434</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">-</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>...</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
<td>0.999819</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">A</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>...</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">C</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>...</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">D</td>
<td>0.000000</td>
<td>0.000181</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000181</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>...</td>
<td>0.000000</td>
<td>0.000181</td>
<td>0.000181</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000181</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">E</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>...</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000000</td>
<td>0.000181</td>
<td>0.000000</td>
<td>0.000000</td>
</tr>
</tbody>
</table>

<p>5 rows × 3434 columns</p>
</div>

``` python
# remove '-' first line
max_series=freq_df.iloc[1:,:]
```

``` python
freq_max = pd.concat([max_series.idxmax(),max_series.max()],axis=1)
```

``` python
freq_max.columns = ['aa','max_value']
```

``` python
freq_max = freq_max.sort_values('max_value',ascending=False).reset_index(names='position')
```

``` python
out = freq_max[freq_max.max_value>0.1]
```

``` python
out
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">position</th>
<th data-quarto-table-cell-role="th">aa</th>
<th data-quarto-table-cell-role="th">max_value</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>1549</td>
<td>N</td>
<td>0.815390</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>2618</td>
<td>D</td>
<td>0.809429</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>1724</td>
<td>D</td>
<td>0.800759</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>1525</td>
<td>D</td>
<td>0.791004</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>1730</td>
<td>G</td>
<td>0.775470</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">214</td>
<td>640</td>
<td>G</td>
<td>0.101879</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">215</td>
<td>193</td>
<td>E</td>
<td>0.101879</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">216</td>
<td>922</td>
<td>L</td>
<td>0.101337</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">217</td>
<td>603</td>
<td>R</td>
<td>0.101156</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">218</td>
<td>2581</td>
<td>G</td>
<td>0.100614</td>
</tr>
</tbody>
</table>

<p>219 rows × 3 columns</p>
</div>

``` python
# out.to_csv('out/align_freq_max_aa.csv',index=False)
```

## Locate active motif & save

- 1549 N is after HRD motif
- 2618 D is around D\[IV\]WS motif
- 1724 D is DFG motif
- 1525 D is HRD motif

``` python
df=df.reset_index()
```

``` python
kd['D1']=(df[1525]=='D').astype(int) # HRD
```

``` python
kd['D2'] = (df[1724]=='D').astype(int) #DFG
```

``` python
kd['D3']=(df[2618]=='D').astype(int)
```

``` python
kd['N1'] = (df[1549]=='N').astype(int)
```

``` python
active_col = ['D1','D2']
```

``` python
kd['active_D1_D2'] = (kd[active_col].sum(1)==len(active_col)).astype(int)
```

``` python
kd['active_D1_D2'].value_counts()
```

    active_D1_D2
    1    4209
    0    1327
    Name: count, dtype: int64

``` python
active_all = kd[kd.active_D1_D2==1]
```

``` python
active_all.shape
```

    (4209, 32)

``` python
# active_all.to_excel('out/uniprot_kd_active_D1_D2.xlsx',index=False)
```

``` python
# kd.to_excel('out/uniprot_kd_motif_labeled.xlsx',index=False)
```

Take a look of their identity:

``` python
kd[kd.active_D1_D2==1].kd_note.value_counts()
```

    kd_note
    Protein kinase               4005
    PI3K/PI4K catalytic            71
    Protein kinase 2               61
    Protein kinase 1               37
    Histidine kinase               34
    Alpha-type protein kinase       1
    Name: count, dtype: int64

``` python
kd[kd.active_D1_D2==0].kd_note.value_counts()
```

    kd_note
    Histidine kinase                  591
    Protein kinase                    525
    PI3K/PI4K catalytic                97
    Protein kinase 1                   36
    Alpha-type protein kinase          21
    Protein kinase; inactive           13
    Protein kinase 2                   12
    HWE histidine kinase domain        11
    Guanylate kinase-like               5
    Histidine kinase 2                  4
    Histidine kinase 1                  4
    Amino-acid kinase domain (AAK)      3
    Kinase domain                       2
    Histidine kinase; first part        1
    Histidine kinase; second part       1
    Protein kinase; truncated           1
    Name: count, dtype: int64

## Add KD info to human kinase info

``` python
kd = pd.read_excel('out/uniprot_kd_motif_labeled.xlsx')
```

``` python
human= kd[kd.Organism=='Homo sapiens (Human)']
```

``` python
kd_id_uniprot_map = human.groupby('Uniprot').agg({'kd_ID': lambda x: ','.join(x.unique()),
                                                  'active_D1_D2': lambda x: x,
                                                 })
```

``` python
kd_id_uniprot_map
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">kd_ID</th>
<th data-quarto-table-cell-role="th">active_D1_D2</th>
</tr>
<tr>
<th data-quarto-table-cell-role="th">Uniprot</th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">A4D2B8</td>
<td>A4D2B8_PM2P1_HUMAN_KD1</td>
<td>0</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">A4QPH2</td>
<td>A4QPH2_PI4P2_HUMAN_KD1</td>
<td>0</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">O00141</td>
<td>O00141_SGK1_HUMAN_KD1</td>
<td>1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">O00238</td>
<td>O00238_BMR1B_HUMAN_KD1</td>
<td>1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">O00311</td>
<td>O00311_CDC7_HUMAN_KD1</td>
<td>1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">Q9Y616</td>
<td>Q9Y616_IRAK3_HUMAN_KD1</td>
<td>0</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">Q9Y6E0</td>
<td>Q9Y6E0_STK24_HUMAN_KD1</td>
<td>1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">Q9Y6M4</td>
<td>Q9Y6M4_KC1G3_HUMAN_KD1</td>
<td>1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">Q9Y6R4</td>
<td>Q9Y6R4_M3K4_HUMAN_KD1</td>
<td>1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">Q9Y6S9</td>
<td>Q9Y6S9_RPKL1_HUMAN_KD1</td>
<td>0</td>
</tr>
</tbody>
</table>

<p>523 rows × 2 columns</p>
</div>

``` python
# kd_id_uniprot_map.to_csv('raw/kd_ID_uniprot_map.csv')
```

Get the csv merge with kinase info. Manually match KD1 and KD2 to the
kinase info by considering active_D1_D2.

The data is updated on github.

``` python
from katlas.data import *
```

``` python
kd_info =Data.get_kd_uniprot()
```

``` python
kd_info.head()
```

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }
&#10;    .dataframe tbody tr th {
        vertical-align: top;
    }
&#10;    .dataframe thead th {
        text-align: right;
    }
</style>

<table class="dataframe" data-quarto-postprocess="true" data-border="1">
<thead>
<tr style="text-align: right;">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">kd_ID</th>
<th data-quarto-table-cell-role="th">Uniprot</th>
<th data-quarto-table-cell-role="th">Entry Name</th>
<th data-quarto-table-cell-role="th">Protein names</th>
<th data-quarto-table-cell-role="th">Gene Names</th>
<th data-quarto-table-cell-role="th">Gene Names (primary)</th>
<th data-quarto-table-cell-role="th">Organism</th>
<th data-quarto-table-cell-role="th">kd_note</th>
<th data-quarto-table-cell-role="th">kd_evidence</th>
<th data-quarto-table-cell-role="th">kd_start</th>
<th data-quarto-table-cell-role="th">...</th>
<th data-quarto-table-cell-role="th">Interacts with</th>
<th data-quarto-table-cell-role="th">Subunit structure</th>
<th data-quarto-table-cell-role="th">Function [CC]</th>
<th data-quarto-table-cell-role="th">Activity regulation</th>
<th data-quarto-table-cell-role="th">full_seq</th>
<th data-quarto-table-cell-role="th">D1</th>
<th data-quarto-table-cell-role="th">D2</th>
<th data-quarto-table-cell-role="th">D3</th>
<th data-quarto-table-cell-role="th">N1</th>
<th data-quarto-table-cell-role="th">active_D1_D2</th>
</tr>
</thead>
<tbody>
<tr>
<td data-quarto-table-cell-role="th">0</td>
<td>A0A075F7E9_LERK1_ORYSI_KD1</td>
<td>A0A075F7E9</td>
<td>LERK1_ORYSI</td>
<td>G-type lectin S-receptor-like serine/threonine...</td>
<td>LECRK1 LECRK OsI_14840</td>
<td>LECRK1</td>
<td>Oryza sativa subsp. indica (Rice)</td>
<td>Protein kinase</td>
<td>ECO:0000255|PROSITE-ProRule:PRU00159</td>
<td>523</td>
<td>...</td>
<td>None</td>
<td>SUBUNIT: Interacts (via kinase domain) with AD...</td>
<td>FUNCTION: Involved in innate immunity. Require...</td>
<td>None</td>
<td>MVALLLFPMLLQLLSPTCAQTQKNITLGSTLAPQGPASSWLSPSGD...</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">1</td>
<td>A0A078BQP2_GCY25_CAEEL_KD1</td>
<td>A0A078BQP2</td>
<td>GCY25_CAEEL</td>
<td>Receptor-type guanylate cyclase gcy-25 (EC 4.6...</td>
<td>gcy-25 Y105C5B.2</td>
<td>gcy-25</td>
<td>Caenorhabditis elegans</td>
<td>Protein kinase</td>
<td>ECO:0000255|PROSITE-ProRule:PRU00159</td>
<td>464</td>
<td>...</td>
<td>None</td>
<td>None</td>
<td>FUNCTION: Guanylate cyclase involved in the pr...</td>
<td>None</td>
<td>MLLLLLLLKISTFVDSFQIGHLEFENSNETRILEICMKNAGSWRDH...</td>
<td>0</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>0</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">2</td>
<td>A0A078CGE6_M3KE1_BRANA_KD1</td>
<td>A0A078CGE6</td>
<td>M3KE1_BRANA</td>
<td>MAP3K epsilon protein kinase 1 (BnM3KE1) (EC 2...</td>
<td>M3KE1 BnaA03g30290D GSBRNA2T00111755001</td>
<td>M3KE1</td>
<td>Brassica napus (Rape)</td>
<td>Protein kinase</td>
<td>ECO:0000255|PROSITE-ProRule:PRU00159</td>
<td>20</td>
<td>...</td>
<td>None</td>
<td>None</td>
<td>FUNCTION: Serine/threonine-protein kinase invo...</td>
<td>None</td>
<td>MARQMTSSQFHKSKTLDNKYMLGDEIGKGAYGRVYIGLDLENGDFV...</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">3</td>
<td>A0A0G2K344_PK3CA_RAT_KD1</td>
<td>A0A0G2K344</td>
<td>PK3CA_RAT</td>
<td>Phosphatidylinositol 4,5-bisphosphate 3-kinase...</td>
<td>Pik3ca</td>
<td>Pik3ca</td>
<td>Rattus norvegicus (Rat)</td>
<td>PI3K/PI4K catalytic</td>
<td>ECO:0000255|PROSITE-ProRule:PRU00269</td>
<td>765</td>
<td>...</td>
<td>None</td>
<td>SUBUNIT: Heterodimer of a catalytic subunit PI...</td>
<td>FUNCTION: Phosphoinositide-3-kinase (PI3K) pho...</td>
<td>None</td>
<td>MPPRPSSGELWGIHLMPPRILVECLLPNGMIVTLECLREATLVTIK...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr>
<td data-quarto-table-cell-role="th">4</td>
<td>A0A0H2ZM62_HK06_STRP2_KD1</td>
<td>A0A0H2ZM62</td>
<td>HK06_STRP2</td>
<td>Sensor histidine protein kinase HK06 (EC 2.7.1...</td>
<td>hk06 SPD_2019</td>
<td>hk06</td>
<td>Streptococcus pneumoniae serotype 2 (strain D3...</td>
<td>Histidine kinase</td>
<td>ECO:0000255|PROSITE-ProRule:PRU00107</td>
<td>239</td>
<td>...</td>
<td>None</td>
<td>None</td>
<td>FUNCTION: Member of the two-component regulato...</td>
<td>None</td>
<td>MIKNPKLLTKSFLRSFAILGGVGLVIHIAIYLTFPFYYIQLEGEKF...</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
</tbody>
</table>

<p>5 rows × 32 columns</p>
</div>

``` python
kd_info.shape
```

    (5536, 32)

``` python
info=Data.get_kinase_info()
```

``` python
info.columns
```

    Index(['kinase', 'ID_coral', 'uniprot', 'gene', 'modi_group', 'group',
           'family', 'subfamily_coral', 'subfamily', 'in_pspa_st', 'in_pspa_tyr',
           'in_pspa', 'in_cddm', 'kd_ID', 'active_D1_D2', 'active_kd_ID',
           'pspa_ID', 'pseudo', 'pspa_category_small', 'pspa_category_big',
           'cddm_big', 'cddm_small', 'length', 'human_uniprot_sequence',
           'kinasecom_domain', 'nucleus', 'cytosol', 'cytoskeleton',
           'plasma membrane', 'mitochondrion', 'Golgi apparatus',
           'endoplasmic reticulum', 'vesicle', 'centrosome', 'aggresome',
           'main_location'],
          dtype='object')

``` python
info.kd_ID
```

    0       Q2M2I8_AAK1_HUMAN_KD1
    1      Q6ZMQ8_LMTK1_HUMAN_KD1
    2       P00519_ABL1_HUMAN_KD1
    3       P42684_ABL2_HUMAN_KD1
    4      Q8NER5_ACV1C_HUMAN_KD1
                    ...          
    518      P07947_YES_HUMAN_KD1
    519    O00506_STK25_HUMAN_KD1
    520    Q56UN5_M3K19_HUMAN_KD1
    521    Q9NYL2_M3K20_HUMAN_KD1
    522    P43403_ZAP70_HUMAN_KD1
    Name: kd_ID, Length: 523, dtype: object
